<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>PLSDA-batch: a multivariate framework to correct for batch effects in microbiome data</title>
  <meta name="description" content="Vignette for paper ‘PLSDA-batch: a multivariate framework to correct for batch effects in microbiome data’." />
  <meta name="generator" content="bookdown 0.33 and GitBook 2.6.7" />

  <meta property="og:title" content="PLSDA-batch: a multivariate framework to correct for batch effects in microbiome data" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="Vignette for paper ‘PLSDA-batch: a multivariate framework to correct for batch effects in microbiome data’." />
  <meta name="github-repo" content="EvaYiwenWang/PLSDAbatch_workflow" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="PLSDA-batch: a multivariate framework to correct for batch effects in microbiome data" />
  
  <meta name="twitter:description" content="Vignette for paper ‘PLSDA-batch: a multivariate framework to correct for batch effects in microbiome data’." />
  



<meta name="date" content="2023-05-04" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  

<link rel="next" href="supplementary-materials-for-batch-effects-management-in-case-studies.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">PLSDA-batch</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Batch Effects Management in Case Studies</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#introduction"><i class="fa fa-check"></i><b>1.1</b> Introduction</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#case-studies-description"><i class="fa fa-check"></i><b>1.2</b> Case studies description</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#packages-installation-and-loading"><i class="fa fa-check"></i><b>1.3</b> Packages installation and loading</a></li>
<li class="chapter" data-level="1.4" data-path="index.html"><a href="index.html#data-pre-processing"><i class="fa fa-check"></i><b>1.4</b> Data pre-processing</a>
<ul>
<li class="chapter" data-level="1.4.1" data-path="index.html"><a href="index.html#pre-filtering"><i class="fa fa-check"></i><b>1.4.1</b> Pre-filtering</a></li>
<li class="chapter" data-level="1.4.2" data-path="index.html"><a href="index.html#transformation"><i class="fa fa-check"></i><b>1.4.2</b> Transformation</a></li>
</ul></li>
<li class="chapter" data-level="1.5" data-path="index.html"><a href="index.html#batch-effect-detection"><i class="fa fa-check"></i><b>1.5</b> Batch effect detection</a>
<ul>
<li class="chapter" data-level="1.5.1" data-path="index.html"><a href="index.html#pca"><i class="fa fa-check"></i><b>1.5.1</b> PCA</a></li>
<li class="chapter" data-level="1.5.2" data-path="index.html"><a href="index.html#boxplots-and-density-plots"><i class="fa fa-check"></i><b>1.5.2</b> Boxplots and density plots</a></li>
<li class="chapter" data-level="1.5.3" data-path="index.html"><a href="index.html#heatmap"><i class="fa fa-check"></i><b>1.5.3</b> Heatmap</a></li>
<li class="chapter" data-level="1.5.4" data-path="index.html"><a href="index.html#prda"><i class="fa fa-check"></i><b>1.5.4</b> pRDA</a></li>
</ul></li>
<li class="chapter" data-level="1.6" data-path="index.html"><a href="index.html#managing-batch-effects"><i class="fa fa-check"></i><b>1.6</b> Managing batch effects</a>
<ul>
<li class="chapter" data-level="1.6.1" data-path="index.html"><a href="index.html#accnt"><i class="fa fa-check"></i><b>1.6.1</b> Accounting for batch effects</a></li>
<li class="chapter" data-level="1.6.2" data-path="index.html"><a href="index.html#corrct"><i class="fa fa-check"></i><b>1.6.2</b> Correcting for batch effects</a></li>
</ul></li>
<li class="chapter" data-level="1.7" data-path="index.html"><a href="index.html#assessing-batch-effect-correction"><i class="fa fa-check"></i><b>1.7</b> Assessing batch effect correction</a>
<ul>
<li class="chapter" data-level="1.7.1" data-path="index.html"><a href="index.html#methods-that-detect-batch-effects"><i class="fa fa-check"></i><b>1.7.1</b> Methods that detect batch effects</a></li>
<li class="chapter" data-level="1.7.2" data-path="index.html"><a href="index.html#other-methods-1"><i class="fa fa-check"></i><b>1.7.2</b> Other methods</a></li>
</ul></li>
<li class="chapter" data-level="1.8" data-path="index.html"><a href="index.html#variable-selection"><i class="fa fa-check"></i><b>1.8</b> Variable selection</a></li>
<li class="chapter" data-level="1.9" data-path="index.html"><a href="index.html#summary"><i class="fa fa-check"></i><b>1.9</b> Summary</a></li>
<li class="chapter" data-level="1.10" data-path="index.html"><a href="index.html#references"><i class="fa fa-check"></i><b>1.10</b> References</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="supplementary-materials-for-batch-effects-management-in-case-studies.html"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html"><i class="fa fa-check"></i><b>2</b> Supplementary Materials for Batch Effects Management in Case Studies</a>
<ul>
<li class="chapter" data-level="2.1" data-path="supplementary-materials-for-batch-effects-management-in-case-studies.html"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#sponge-data"><i class="fa fa-check"></i><b>2.1</b> Sponge data</a>
<ul>
<li class="chapter" data-level="2.1.1" data-path="supplementary-materials-for-batch-effects-management-in-case-studies.html"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#data-pre-processing-1"><i class="fa fa-check"></i><b>2.1.1</b> Data pre-processing</a></li>
<li class="chapter" data-level="2.1.2" data-path="supplementary-materials-for-batch-effects-management-in-case-studies.html"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#batch-effect-detection-1"><i class="fa fa-check"></i><b>2.1.2</b> Batch effect detection</a></li>
<li class="chapter" data-level="2.1.3" data-path="supplementary-materials-for-batch-effects-management-in-case-studies.html"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#managing-batch-effects-1"><i class="fa fa-check"></i><b>2.1.3</b> Managing batch effects</a></li>
<li class="chapter" data-level="2.1.4" data-path="supplementary-materials-for-batch-effects-management-in-case-studies.html"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#assessing-batch-effect-correction-1"><i class="fa fa-check"></i><b>2.1.4</b> Assessing batch effect correction</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="supplementary-materials-for-batch-effects-management-in-case-studies.html"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#hfhs-data"><i class="fa fa-check"></i><b>2.2</b> HFHS data</a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="supplementary-materials-for-batch-effects-management-in-case-studies.html"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#data-pre-processing-2"><i class="fa fa-check"></i><b>2.2.1</b> Data pre-processing</a></li>
<li class="chapter" data-level="2.2.2" data-path="supplementary-materials-for-batch-effects-management-in-case-studies.html"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#batch-effect-detection-2"><i class="fa fa-check"></i><b>2.2.2</b> Batch effect detection</a></li>
<li class="chapter" data-level="2.2.3" data-path="supplementary-materials-for-batch-effects-management-in-case-studies.html"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#managing-batch-effects-2"><i class="fa fa-check"></i><b>2.2.3</b> Managing batch effects</a></li>
<li class="chapter" data-level="2.2.4" data-path="supplementary-materials-for-batch-effects-management-in-case-studies.html"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#assessing-batch-effect-correction-2"><i class="fa fa-check"></i><b>2.2.4</b> Assessing batch effect correction</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="supplementary-materials-for-batch-effects-management-in-case-studies.html"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#hd-data"><i class="fa fa-check"></i><b>2.3</b> HD data</a>
<ul>
<li class="chapter" data-level="2.3.1" data-path="supplementary-materials-for-batch-effects-management-in-case-studies.html"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#data-pre-processing-3"><i class="fa fa-check"></i><b>2.3.1</b> Data pre-processing</a></li>
<li class="chapter" data-level="2.3.2" data-path="supplementary-materials-for-batch-effects-management-in-case-studies.html"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#batch-effect-detection-3"><i class="fa fa-check"></i><b>2.3.2</b> Batch effect detection</a></li>
<li class="chapter" data-level="2.3.3" data-path="supplementary-materials-for-batch-effects-management-in-case-studies.html"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#managing-batch-effects-3"><i class="fa fa-check"></i><b>2.3.3</b> Managing batch effects</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="supplementary-materials-for-batch-effects-management-in-case-studies.html"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#references-1"><i class="fa fa-check"></i><b>2.4</b> References</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="batch-effects-management-in-simulation-studies-gaussian-distribution.html"><a href="batch-effects-management-in-simulation-studies-gaussian-distribution.html"><i class="fa fa-check"></i><b>3</b> Batch Effects Management in Simulation Studies (Gaussian Distribution)</a>
<ul>
<li class="chapter" data-level="3.1" data-path="batch-effects-management-in-simulation-studies-gaussian-distribution.html"><a href="batch-effects-management-in-simulation-studies-gaussian-distribution.html#introduction-1"><i class="fa fa-check"></i><b>3.1</b> Introduction</a></li>
<li class="chapter" data-level="3.2" data-path="batch-effects-management-in-simulation-studies-gaussian-distribution.html"><a href="batch-effects-management-in-simulation-studies-gaussian-distribution.html#simulations"><i class="fa fa-check"></i><b>3.2</b> Simulations</a>
<ul>
<li class="chapter" data-level="3.2.1" data-path="batch-effects-management-in-simulation-studies-gaussian-distribution.html"><a href="batch-effects-management-in-simulation-studies-gaussian-distribution.html#balanced-batch-times-treatment-design"><i class="fa fa-check"></i><b>3.2.1</b> Balanced batch <span class="math inline">\(\times\)</span> treatment design</a></li>
<li class="chapter" data-level="3.2.2" data-path="batch-effects-management-in-simulation-studies-gaussian-distribution.html"><a href="batch-effects-management-in-simulation-studies-gaussian-distribution.html#figures"><i class="fa fa-check"></i><b>3.2.2</b> Figures</a></li>
<li class="chapter" data-level="3.2.3" data-path="batch-effects-management-in-simulation-studies-gaussian-distribution.html"><a href="batch-effects-management-in-simulation-studies-gaussian-distribution.html#unbalanced-batch-times-treatment-design"><i class="fa fa-check"></i><b>3.2.3</b> Unbalanced batch <span class="math inline">\(\times\)</span> treatment design</a></li>
<li class="chapter" data-level="3.2.4" data-path="batch-effects-management-in-simulation-studies-gaussian-distribution.html"><a href="batch-effects-management-in-simulation-studies-gaussian-distribution.html#figures-1"><i class="fa fa-check"></i><b>3.2.4</b> Figures</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="batch-effects-management-in-simulation-studies-gaussian-distribution.html"><a href="batch-effects-management-in-simulation-studies-gaussian-distribution.html#references-2"><i class="fa fa-check"></i><b>3.3</b> References</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="batch-effects-management-in-simulation-studies-negative-binomial-distribution.html"><a href="batch-effects-management-in-simulation-studies-negative-binomial-distribution.html"><i class="fa fa-check"></i><b>4</b> Batch Effects Management in Simulation Studies (Negative Binomial Distribution)</a>
<ul>
<li class="chapter" data-level="4.1" data-path="batch-effects-management-in-simulation-studies-negative-binomial-distribution.html"><a href="batch-effects-management-in-simulation-studies-negative-binomial-distribution.html#introduction-2"><i class="fa fa-check"></i><b>4.1</b> Introduction</a></li>
<li class="chapter" data-level="4.2" data-path="batch-effects-management-in-simulation-studies-negative-binomial-distribution.html"><a href="batch-effects-management-in-simulation-studies-negative-binomial-distribution.html#simulations-two-batch-groups"><i class="fa fa-check"></i><b>4.2</b> Simulations (two batch groups)</a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="batch-effects-management-in-simulation-studies-negative-binomial-distribution.html"><a href="batch-effects-management-in-simulation-studies-negative-binomial-distribution.html#balanced-batch-times-treatment-design-1"><i class="fa fa-check"></i><b>4.2.1</b> Balanced batch <span class="math inline">\(\times\)</span> treatment design</a></li>
<li class="chapter" data-level="4.2.2" data-path="batch-effects-management-in-simulation-studies-negative-binomial-distribution.html"><a href="batch-effects-management-in-simulation-studies-negative-binomial-distribution.html#figures-2"><i class="fa fa-check"></i><b>4.2.2</b> Figures</a></li>
<li class="chapter" data-level="4.2.3" data-path="batch-effects-management-in-simulation-studies-negative-binomial-distribution.html"><a href="batch-effects-management-in-simulation-studies-negative-binomial-distribution.html#unbalanced-batch-times-treatment-design-1"><i class="fa fa-check"></i><b>4.2.3</b> Unbalanced batch <span class="math inline">\(\times\)</span> treatment design</a></li>
<li class="chapter" data-level="4.2.4" data-path="batch-effects-management-in-simulation-studies-negative-binomial-distribution.html"><a href="batch-effects-management-in-simulation-studies-negative-binomial-distribution.html#figures-3"><i class="fa fa-check"></i><b>4.2.4</b> Figures</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="batch-effects-management-in-simulation-studies-negative-binomial-distribution.html"><a href="batch-effects-management-in-simulation-studies-negative-binomial-distribution.html#simulations-three-batch-groups"><i class="fa fa-check"></i><b>4.3</b> Simulations (three batch groups)</a>
<ul>
<li class="chapter" data-level="4.3.1" data-path="batch-effects-management-in-simulation-studies-negative-binomial-distribution.html"><a href="batch-effects-management-in-simulation-studies-negative-binomial-distribution.html#balanced-batch-times-treatment-design-2"><i class="fa fa-check"></i><b>4.3.1</b> Balanced batch <span class="math inline">\(\times\)</span> treatment design</a></li>
<li class="chapter" data-level="4.3.2" data-path="batch-effects-management-in-simulation-studies-negative-binomial-distribution.html"><a href="batch-effects-management-in-simulation-studies-negative-binomial-distribution.html#figures-4"><i class="fa fa-check"></i><b>4.3.2</b> Figures</a></li>
<li class="chapter" data-level="4.3.3" data-path="batch-effects-management-in-simulation-studies-negative-binomial-distribution.html"><a href="batch-effects-management-in-simulation-studies-negative-binomial-distribution.html#unbalanced-batch-times-treatment-design-2"><i class="fa fa-check"></i><b>4.3.3</b> Unbalanced batch <span class="math inline">\(\times\)</span> treatment design</a></li>
<li class="chapter" data-level="4.3.4" data-path="batch-effects-management-in-simulation-studies-negative-binomial-distribution.html"><a href="batch-effects-management-in-simulation-studies-negative-binomial-distribution.html#figures-5"><i class="fa fa-check"></i><b>4.3.4</b> Figures</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="batch-effects-management-in-simulation-studies-negative-binomial-distribution.html"><a href="batch-effects-management-in-simulation-studies-negative-binomial-distribution.html#references-3"><i class="fa fa-check"></i><b>4.4</b> References</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="references-4.html"><a href="references-4.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/EvaYiwenWang/PLSDAbatch_workflow" target="blank">Source codes on github</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">PLSDA-batch: a multivariate framework to correct for batch effects in microbiome data</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="header">
<h1 class="title">PLSDA-batch: a multivariate framework to correct for batch effects in microbiome data</h1>
<p class="author"><em><div class="line-block">Yiwen Wang<span class="math inline">\(^{12}\)</span>, Kim-Anh Lê Cao<span class="math inline">\(^2\)</span><br />
<span class="math inline">\(^1\)</span>Agricultural Genomics Institute at Shenzhen, Chinese Academy of<br />
Agricultural Sciences, Shenzhen, Guangdong, China<br />
<span class="math inline">\(^2\)</span>Melbourne Integrative Genomics, School of Mathematics<br />
and Statistics, The University of Melbourne, Parkville, VIC, Australia</div></em></p>
<p class="date"><em>2023-05-04</em></p>
</div>
<div id="batch-effects-management-in-case-studies" class="section level1 hasAnchor" number="1">
<h1><span class="header-section-number">Chapter 1</span> Batch Effects Management in Case Studies<a href="index.html#batch-effects-management-in-case-studies" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="introduction" class="section level2 hasAnchor" number="1.1">
<h2><span class="header-section-number">1.1</span> Introduction<a href="index.html#introduction" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Investigating the link between microbial composition and phenotypes has become the limelight of research in recent years, as microorganisms play a key role in extensive fields including agriculture, healthcare, food production, industry and climate change <span class="citation">(<a href="#ref-wang2020characterizing" role="doc-biblioref">C. H. Wang et al. 2020</a>; <a href="#ref-ray2020microbe" role="doc-biblioref">Ray et al. 2020</a>; <a href="#ref-fan2020gut" role="doc-biblioref">Fan and Pedersen 2020</a>; <a href="#ref-poirier2020integrating" role="doc-biblioref">Poirier et al. 2020</a>)</span>. The collection of microorganisms and their genomes within a specific environment is referred to as the <em>microbiome</em> <span class="citation">(<a href="#ref-marchesi2015vocabulary" role="doc-biblioref">Marchesi and Ravel 2015</a>)</span>. The microbiome can be profiled using 16S rRNA gene sequencing or whole-genome shotgun sequencing. The microbiome data is displayed as an abundance table of counts per sample for each taxon. This type of data have their inherent characteristics, including zero inflation, uneven library sizes, compositional structure, and multivariate nature, which limit statistical analysis.</p>
<p>Microbiome research faces the challenge of results reproducibility across studies. The potential reasons are poor experimental design and lack of rigorous operating procedures, which introduce variation in the data (batch effects) that obscure the effect of interest. Microbiome data are highly susceptible to batch effects because of the dynamic nature of microbial communities <span class="citation">(<a href="#ref-wang2020managing" role="doc-biblioref">Y. Wang and LêCao 2020</a>)</span>. Numerous studies have reported batch effects introduced by sequencing batches <span class="citation">(<a href="#ref-hieken2016microbiome" role="doc-biblioref">Hieken et al. 2016</a>)</span>, the inclusion of independent studies <span class="citation">(<a href="#ref-duvallet2017meta" role="doc-biblioref">Duvallet et al. 2017</a>)</span>, geography, age, sex, health status, stress and diet <span class="citation">(<a href="#ref-gibson2004dietary" role="doc-biblioref">Gibson et al. 2004</a>; <a href="#ref-lozupone2013meta" role="doc-biblioref">Lozupone et al. 2013</a>; <a href="#ref-haro2016intestinal" role="doc-biblioref">Haro et al. 2016</a>; <a href="#ref-kim2017optimizing" role="doc-biblioref">Kim et al. 2017</a>)</span>. Attempts have been made to alleviate batch effects through standardised designs, but batch effects remain unavoidable in practice. For example, cage effects in murine experiments, age, sex and diet effects in human experiments, sequencing batches are ubiquitous in microbiome studies. So far, methods to manage batch effects in microbiome studies have been lacking, thus limiting microbiome researchers in their ability to analyse their data.</p>
<p>Several methods to handle batch effects have been proposed. However, these methods 1/ were primarily developed for gene expression data, thus do not address the inherent characteristics of microbiome data or 2/ are limited to differential abundance analysis, thus limiting the breadth of statistical analysis that can be performed to answer biological questions for microbiome researchers.</p>
<p>I developed a new batch effect correction method based on Projection to Latent Structures Discriminant Analysis named “PLSDA-batch” to correct data prior to any downstream analysis. PLSDA-batch estimates latent components related to treatment and batch effects to remove batch variation. The method is multivariate, non-parametric and performs dimension reduction. Combined with centered log ratio transformation for addressing uneven library sizes and compositional structure, PLSDA-batch addresses all characteristics of microbiome data that existing correction methods have ignored so far. I also developed two variants for 1/ unbalanced batch x treatment designs that are commonly encountered in studies with small sample size, and for 2/ selection of discriminative variables to avoid overfitting in classification problems. These two variants have widened the scope of applicability of PLSDA-batch to different data settings <span class="citation">(<a href="#ref-wang2020multivariate" role="doc-biblioref">Y. Wang and Lê Cao 2020</a>)</span>.</p>
<p>My package includes both the new method for batch effect correction, along with a comprehensive standardised framework for batch effect management including the application of existing methods ranging from accounting for batch effects (e.g. with linear models) to correcting for batch effects (e.g removeBatchEffect from the <em>limma</em> package, and ComBat from <em>sva</em>) and my proposed method for microbiome data. My package will benefit researchers who have already generated their data - despite a poor experimental design, to analyse their data appropriately. My package will also benefit well-designed studies to detect batch effects as a quality check to ensure reliable downstream results.</p>
<p>The framework includes microbiome data pre-filtering, transformation and batch effect detection, visualisation, accounting for or correcting for batch effects and assessing batch effect removal and variable selection after batch effect correction. We illustrated our framework with data sequenced using 16S rRNA gene sequencing, but shotgun sequencing data can also be analysed. Our framework guides data analysts in choosing the appropriate analytic method to either account for or correct for batch effects. In addition, batch effects can vary in sources, designs and scale of influence on microbial variables, and different methods for batch effect management have different assumptions. We have illustrated each step with exemplar studies and provided all the functions and packages for reproducibility (see Table 2). The R packages are from the open-source CRAN, Bioconductor projects or Github repositories, therefore facilitates the analysis of any microbial studies. The visualisation can be customised to apply to different results from multiple methods.</p>
</div>
<div id="case-studies-description" class="section level2 hasAnchor" number="1.2">
<h2><span class="header-section-number">1.2</span> Case studies description<a href="index.html#case-studies-description" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We considered four case studies with different data settings. We illustrated the complete analysis of only one study, and specific analysis steps for the other studies. The complete analyses of the other studies can be found in the next chapter.</p>
<p>The datasets from four case studies are stored internally in our R package <span style="color: #FF7F00;">PLSDAbatch</span>. Their basic information can be found in Table 1.</p>
<p><strong><span style="color: #0000FF;">Anaerobic digestion</span>.</strong> This study explored the microbial indicators that could improve the efficacy of anaerobic digestion (AD) bioprocess and prevent its failure <span class="citation">(<a href="#ref-chapleur2016increasing" role="doc-biblioref">Chapleur et al. 2016</a>)</span>. The samples were treated with two different ranges of phenol concentration (effect of interest) and processed at five different dates (batch effect). This study included a clear and strong batch effect with an approx. balanced batch x treatment design. Using this study, we illustrated the complete analysis of batch effect management.</p>
<p><strong><span style="color: #964B00;">Sponge</span> <em><span style="color: #964B00;">A. aerophoba</span></em>.</strong> This study investigated the relationship between metabolite concentration and microbial abundance of specific sponge tissues <span class="citation">(<a href="#ref-sacristan2011exploring" role="doc-biblioref">Sacristán-Soriano et al. 2011</a>)</span>. The samples were collected from two types of tissues (Ectosome vs. Choanosome) and processed on two separate denaturing gradient gels in electrophoresis. This study included relative abundance data only and a completely balanced batch x treatment design.</p>
<p><strong><span style="color: #808000;">High fat high sugar diet</span>.</strong> This study aimed to investigate the effect of high fat high sugar (HFHS) diet on the mouse microbiome <span class="citation">(<a href="#ref-susin2020variable" role="doc-biblioref">Susin et al. 2020</a>)</span>. The samples were collected at different days from the mice treated with two types of diets (HFHS vs. normal) and housed in different cages. This study included a extremely unbalanced (nested) batch x treatment design between cages and diets and an approx. balanced design between days and diets. With this study we illustrated how to deal with weak batch effects.</p>
<p><strong><span style="color: #8601AF;">Mice models with Huntington’s disease</span>.</strong> This study explored differences in microbial composition between Huntington’s disease (HD) and wild-type (WT) mice <span class="citation">(<a href="#ref-kong2020microbiome" role="doc-biblioref">Kong et al. 2020</a>)</span>. The samples were collected from the mice with different genotypes and housed in different cages. We illustrated how to manage batch effects with a nested batch x treatment design.</p>
<p><img src="figures/data_description.png" />
<strong>Table 1: Overview of case studies with batch effects and their experimental designs.</strong> We considered microbial studies for Anaerobic Digestion (<span style="color: #0000FF;">AD data</span>), sponge (<span style="color: #964B00;">sponge data</span>), mice models with High Fat High Sugar diets (<span style="color: #808000;">HFHS data</span>) and Huntington’s Disease (<span style="color: #8601AF;">HD data</span>).</p>
</div>
<div id="packages-installation-and-loading" class="section level2 hasAnchor" number="1.3">
<h2><span class="header-section-number">1.3</span> Packages installation and loading<a href="index.html#packages-installation-and-loading" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>First, we load the packages necessary for analysis, and check the version of each package.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="index.html#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="co"># CRAN</span></span>
<span id="cb1-2"><a href="index.html#cb1-2" aria-hidden="true" tabindex="-1"></a>cran.pkgs <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&#39;pheatmap&#39;</span>, <span class="st">&#39;vegan&#39;</span>, <span class="st">&#39;ruv&#39;</span>, <span class="st">&#39;UpSetR&#39;</span>, <span class="st">&#39;gplots&#39;</span>, </span>
<span id="cb1-3"><a href="index.html#cb1-3" aria-hidden="true" tabindex="-1"></a>               <span class="st">&#39;ggplot2&#39;</span>, <span class="st">&#39;gridExtra&#39;</span>, <span class="st">&#39;performance&#39;</span>, <span class="st">&#39;Rdpack&#39;</span>, <span class="st">&#39;pROC&#39;</span>)</span>
<span id="cb1-4"><a href="index.html#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Bioconductor</span></span>
<span id="cb1-5"><a href="index.html#cb1-5" aria-hidden="true" tabindex="-1"></a>bioc.pkgs <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&#39;mixOmics&#39;</span>, <span class="st">&#39;sva&#39;</span>, <span class="st">&#39;limma&#39;</span>, <span class="st">&#39;Biobase&#39;</span>, <span class="st">&#39;metagenomeSeq&#39;</span>)</span>
<span id="cb1-6"><a href="index.html#cb1-6" aria-hidden="true" tabindex="-1"></a><span class="co"># GitHub</span></span>
<span id="cb1-7"><a href="index.html#cb1-7" aria-hidden="true" tabindex="-1"></a>github.pkg <span class="ot">&lt;-</span> <span class="st">&#39;PLSDAbatch&#39;</span> </span>
<span id="cb1-8"><a href="index.html#cb1-8" aria-hidden="true" tabindex="-1"></a><span class="co"># devtools::install_github(&quot;https://github.com/EvaYiwenWang/PLSDAbatch&quot;)</span></span>
<span id="cb1-9"><a href="index.html#cb1-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-10"><a href="index.html#cb1-10" aria-hidden="true" tabindex="-1"></a><span class="co"># built-in functions</span></span>
<span id="cb1-11"><a href="index.html#cb1-11" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="at">file =</span> <span class="st">&#39;./built_in_functions/data_simulation.R&#39;</span>)</span>
<span id="cb1-12"><a href="index.html#cb1-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-13"><a href="index.html#cb1-13" aria-hidden="true" tabindex="-1"></a><span class="co"># load packages </span></span>
<span id="cb1-14"><a href="index.html#cb1-14" aria-hidden="true" tabindex="-1"></a><span class="fu">sapply</span>(<span class="fu">c</span>(cran.pkgs, bioc.pkgs, github.pkg), require, <span class="at">character.only =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<pre><code>##      pheatmap         vegan           ruv        UpSetR        gplots 
##          TRUE          TRUE          TRUE          TRUE          TRUE 
##       ggplot2     gridExtra   performance        Rdpack          pROC 
##          TRUE          TRUE          TRUE          TRUE          TRUE 
##      mixOmics           sva         limma       Biobase metagenomeSeq 
##          TRUE          TRUE          TRUE          TRUE          TRUE 
##    PLSDAbatch 
##          TRUE</code></pre>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="index.html#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="co"># print package versions</span></span>
<span id="cb3-2"><a href="index.html#cb3-2" aria-hidden="true" tabindex="-1"></a><span class="fu">sapply</span>(<span class="fu">c</span>(cran.pkgs, bioc.pkgs, github.pkg), package.version)</span></code></pre></div>
<pre><code>##      pheatmap         vegan           ruv        UpSetR        gplots 
##      &quot;1.0.12&quot;       &quot;2.6-4&quot;     &quot;0.9.7.1&quot;       &quot;1.4.0&quot;       &quot;3.1.3&quot; 
##       ggplot2     gridExtra   performance        Rdpack          pROC 
##       &quot;3.4.2&quot;         &quot;2.3&quot;      &quot;0.10.3&quot;         &quot;2.4&quot;      &quot;1.18.0&quot; 
##      mixOmics           sva         limma       Biobase metagenomeSeq 
##      &quot;6.24.0&quot;      &quot;3.48.0&quot;      &quot;3.56.0&quot;      &quot;2.60.0&quot;      &quot;1.41.0&quot; 
##    PLSDAbatch 
##      &quot;0.99.0&quot;</code></pre>
</div>
<div id="data-pre-processing" class="section level2 hasAnchor" number="1.4">
<h2><span class="header-section-number">1.4</span> Data pre-processing<a href="index.html#data-pre-processing" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="pre-filtering" class="section level3 hasAnchor" number="1.4.1">
<h3><span class="header-section-number">1.4.1</span> Pre-filtering<a href="index.html#pre-filtering" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>We load the <span style="color: #0000FF;">AD data</span> stored internally with function <code>data()</code>.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="index.html#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="co"># AD data</span></span>
<span id="cb5-2"><a href="index.html#cb5-2" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&#39;AD_data&#39;</span>) </span>
<span id="cb5-3"><a href="index.html#cb5-3" aria-hidden="true" tabindex="-1"></a>ad.count <span class="ot">&lt;-</span> AD_data<span class="sc">$</span>FullData<span class="sc">$</span>X.count</span>
<span id="cb5-4"><a href="index.html#cb5-4" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(ad.count)</span></code></pre></div>
<pre><code>## [1]  75 567</code></pre>
<p>The raw <span style="color: #0000FF;">AD data</span> include 567 OTUs and 75 samples. We then use the function <code>PreFL()</code> from our <span style="color: #FF7F00;">PLSDAbatch</span> R package to filter the data.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="index.html#cb7-1" aria-hidden="true" tabindex="-1"></a>ad.filter.res <span class="ot">&lt;-</span> <span class="fu">PreFL</span>(<span class="at">data =</span> ad.count)</span>
<span id="cb7-2"><a href="index.html#cb7-2" aria-hidden="true" tabindex="-1"></a>ad.filter <span class="ot">&lt;-</span> ad.filter.res<span class="sc">$</span>data.filter</span>
<span id="cb7-3"><a href="index.html#cb7-3" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(ad.filter)</span></code></pre></div>
<pre><code>## [1]  75 231</code></pre>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="index.html#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="co"># zero proportion before filtering</span></span>
<span id="cb9-2"><a href="index.html#cb9-2" aria-hidden="true" tabindex="-1"></a>ad.filter.res<span class="sc">$</span>zero.prob</span></code></pre></div>
<pre><code>## [1] 0.6328042</code></pre>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="index.html#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="co"># zero proportion after filtering</span></span>
<span id="cb11-2"><a href="index.html#cb11-2" aria-hidden="true" tabindex="-1"></a><span class="fu">sum</span>(ad.filter <span class="sc">==</span> <span class="dv">0</span>)<span class="sc">/</span>(<span class="fu">nrow</span>(ad.filter) <span class="sc">*</span> <span class="fu">ncol</span>(ad.filter))</span></code></pre></div>
<pre><code>## [1] 0.3806638</code></pre>
<p>After filtering, 231 OTUs remained, and the proportion of zeroes decreased from 63% to 38%.</p>
<p>We can also load the other data, such as the <span style="color: #964B00;">sponge data</span>.</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="index.html#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Sponge data</span></span>
<span id="cb13-2"><a href="index.html#cb13-2" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&#39;sponge_data&#39;</span>) </span>
<span id="cb13-3"><a href="index.html#cb13-3" aria-hidden="true" tabindex="-1"></a>sponge.tss <span class="ot">&lt;-</span> sponge_data<span class="sc">$</span>X.tss</span>
<span id="cb13-4"><a href="index.html#cb13-4" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(sponge.tss)</span></code></pre></div>
<pre><code>## [1] 32 24</code></pre>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="index.html#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="co"># zero proportion</span></span>
<span id="cb15-2"><a href="index.html#cb15-2" aria-hidden="true" tabindex="-1"></a><span class="fu">sum</span>(sponge.tss <span class="sc">==</span> <span class="dv">0</span>)<span class="sc">/</span>(<span class="fu">nrow</span>(sponge.tss) <span class="sc">*</span> <span class="fu">ncol</span>(sponge.tss))</span></code></pre></div>
<pre><code>## [1] 0.5455729</code></pre>
<p>The <span style="color: #964B00;">sponge data</span> include the relative abundance of 24 OTUs and 32 samples. Given the small number of OTUs, we advise not to pre-filter the data.</p>
<p>Note: The <code>PreFL()</code> function is only dedicated to raw counts, rather than relative abundance data. We also recommend to start the pre-filtering on raw counts, rather than relative abundance data to mitigate the compositionality issue.</p>
</div>
<div id="transformation" class="section level3 hasAnchor" number="1.4.2">
<h3><span class="header-section-number">1.4.2</span> Transformation<a href="index.html#transformation" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Prior to CLR transformation, we recommend adding 1 as the offset for the <span style="color: #0000FF;">AD data</span> - that are raw count data, and 0.01 as the offset for the <span style="color: #964B00;">sponge data</span> - that are relative abundance data. We use <code>logratio.transfo()</code> function in <span style="color: #FF7F00;">mixOmics</span> package to CLR transform the data.</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="index.html#cb17-1" aria-hidden="true" tabindex="-1"></a>ad.clr <span class="ot">&lt;-</span> <span class="fu">logratio.transfo</span>(<span class="at">X =</span> ad.filter, <span class="at">logratio =</span> <span class="st">&#39;CLR&#39;</span>, <span class="at">offset =</span> <span class="dv">1</span>) </span>
<span id="cb17-2"><a href="index.html#cb17-2" aria-hidden="true" tabindex="-1"></a><span class="fu">class</span>(ad.clr) <span class="ot">=</span> <span class="st">&#39;matrix&#39;</span></span>
<span id="cb17-3"><a href="index.html#cb17-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-4"><a href="index.html#cb17-4" aria-hidden="true" tabindex="-1"></a>sponge.clr <span class="ot">&lt;-</span> <span class="fu">logratio.transfo</span>(<span class="at">X =</span> sponge.tss, <span class="at">logratio =</span> <span class="st">&#39;CLR&#39;</span>, <span class="at">offset =</span> <span class="fl">0.01</span>)</span>
<span id="cb17-5"><a href="index.html#cb17-5" aria-hidden="true" tabindex="-1"></a><span class="fu">class</span>(sponge.clr) <span class="ot">=</span> <span class="st">&#39;matrix&#39;</span></span></code></pre></div>
</div>
</div>
<div id="batch-effect-detection" class="section level2 hasAnchor" number="1.5">
<h2><span class="header-section-number">1.5</span> Batch effect detection<a href="index.html#batch-effect-detection" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="pca" class="section level3 hasAnchor" number="1.5.1">
<h3><span class="header-section-number">1.5.1</span> PCA<a href="index.html#pca" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>We apply <code>pca()</code> function from <span style="color: #FF7F00;">mixOmics</span> package to the <span style="color: #0000FF;">AD data</span> and <code>Scatter_Density()</code> function from <span style="color: #FF7F00;">PLSDAbatch</span> to represent the PCA sample plot with densities.</p>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="index.html#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="co"># AD data</span></span>
<span id="cb18-2"><a href="index.html#cb18-2" aria-hidden="true" tabindex="-1"></a>ad.pca.before <span class="ot">&lt;-</span> <span class="fu">pca</span>(ad.clr, <span class="at">ncomp =</span> <span class="dv">3</span>, <span class="at">scale =</span> <span class="cn">TRUE</span>)</span>
<span id="cb18-3"><a href="index.html#cb18-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-4"><a href="index.html#cb18-4" aria-hidden="true" tabindex="-1"></a>ad.metadata <span class="ot">&lt;-</span> AD_data<span class="sc">$</span>FullData<span class="sc">$</span>metadata</span>
<span id="cb18-5"><a href="index.html#cb18-5" aria-hidden="true" tabindex="-1"></a>ad.batch <span class="ot">=</span> <span class="fu">factor</span>(ad.metadata<span class="sc">$</span>sequencing_run_date, </span>
<span id="cb18-6"><a href="index.html#cb18-6" aria-hidden="true" tabindex="-1"></a>                  <span class="at">levels =</span> <span class="fu">unique</span>(ad.metadata<span class="sc">$</span>sequencing_run_date))</span>
<span id="cb18-7"><a href="index.html#cb18-7" aria-hidden="true" tabindex="-1"></a>ad.trt <span class="ot">=</span> <span class="fu">as.factor</span>(ad.metadata<span class="sc">$</span>initial_phenol_concentration.regroup)</span>
<span id="cb18-8"><a href="index.html#cb18-8" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(ad.batch) <span class="ot">&lt;-</span> <span class="fu">names</span>(ad.trt) <span class="ot">&lt;-</span> <span class="fu">rownames</span>(ad.metadata)</span>
<span id="cb18-9"><a href="index.html#cb18-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-10"><a href="index.html#cb18-10" aria-hidden="true" tabindex="-1"></a><span class="fu">Scatter_Density</span>(<span class="at">object =</span> ad.pca.before, <span class="at">batch =</span> ad.batch, <span class="at">trt =</span> ad.trt, </span>
<span id="cb18-11"><a href="index.html#cb18-11" aria-hidden="true" tabindex="-1"></a>                <span class="at">title =</span> <span class="st">&#39;AD data&#39;</span>, <span class="at">trt.legend.title =</span> <span class="st">&#39;Phenol conc.&#39;</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:ADpcaBefore"></span>
<img src="PLSDAbatch_workflow_files/figure-html/ADpcaBefore-1.png" alt="The PCA sample plot with densities in the AD data." width="60%" />
<p class="caption">
Figure 1.1: The PCA sample plot with densities in the AD data.
</p>
</div>
<p>In the above figure, we observed 1) the distinction between samples treated with different phenol concentrations and 2) the differences between samples sequenced at “14/04/2016”, “21/09/2017” and the other dates. Therefore, the batch effect related to dates needs to be removed.</p>
</div>
<div id="boxplots-and-density-plots" class="section level3 hasAnchor" number="1.5.2">
<h3><span class="header-section-number">1.5.2</span> Boxplots and density plots<a href="index.html#boxplots-and-density-plots" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>We first identify the top OTU driving the major variance in PCA using <code>selectVar()</code> in <span style="color: #FF7F00;">mixOmics</span> package. Each identified OTU can then be plotted as boxplots and density plots using <code>box_plot()</code> and <code>density_plot()</code> in <span style="color: #FF7F00;">PLSDAbatch</span>.</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="index.html#cb19-1" aria-hidden="true" tabindex="-1"></a>ad.OTU.name <span class="ot">&lt;-</span> <span class="fu">selectVar</span>(ad.pca.before, <span class="at">comp =</span> <span class="dv">1</span>)<span class="sc">$</span>name[<span class="dv">1</span>]</span>
<span id="cb19-2"><a href="index.html#cb19-2" aria-hidden="true" tabindex="-1"></a>ad.OTU_batch <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">value =</span> ad.clr[,ad.OTU.name], <span class="at">batch =</span> ad.batch)</span>
<span id="cb19-3"><a href="index.html#cb19-3" aria-hidden="true" tabindex="-1"></a><span class="fu">box_plot</span>(<span class="at">df =</span> ad.OTU_batch, <span class="at">title =</span> <span class="fu">paste</span>(ad.OTU.name, <span class="st">&#39;(AD data)&#39;</span>), </span>
<span id="cb19-4"><a href="index.html#cb19-4" aria-hidden="true" tabindex="-1"></a>         <span class="at">x.angle =</span> <span class="dv">30</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:ADboxBefore"></span>
<img src="PLSDAbatch_workflow_files/figure-html/ADboxBefore-1.png" alt="Boxplots of sample values in &quot;OTU28&quot; before batch effect correction in the AD data." width="60%" />
<p class="caption">
Figure 1.2: Boxplots of sample values in “OTU28” before batch effect correction in the AD data.
</p>
</div>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="index.html#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="fu">density_plot</span>(<span class="at">df =</span> ad.OTU_batch, <span class="at">title =</span> <span class="fu">paste</span>(ad.OTU.name, <span class="st">&#39;(AD data)&#39;</span>))</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:ADdensityBefore"></span>
<img src="PLSDAbatch_workflow_files/figure-html/ADdensityBefore-1.png" alt="Density plots of sample values in &quot;OTU28&quot; before batch effect correction in the AD data." width="60%" />
<p class="caption">
Figure 1.3: Density plots of sample values in “OTU28” before batch effect correction in the AD data.
</p>
</div>
<p>The boxplot and density plot indicated a strong date batch effect because of the differences between “14/04/2016”, “21/09/2017” and the other dates in the “OTU28”.</p>
<p>We also apply a linear regression model to the “OTU28” using <code>linear_regres()</code> from <span style="color: #FF7F00;">PLSDAbatch</span> with batch and treatment effects as covariates. We set “14/04/2016” and “21/09/2017” as the reference batch respectively with <code>relevel()</code> from <span style="color: #FF7F00;">stats</span>.</p>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="index.html#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="co"># reference batch: 14/04/2016</span></span>
<span id="cb21-2"><a href="index.html#cb21-2" aria-hidden="true" tabindex="-1"></a>ad.batch <span class="ot">&lt;-</span> <span class="fu">relevel</span>(<span class="at">x =</span> ad.batch, <span class="at">ref =</span> <span class="st">&#39;14/04/2016&#39;</span>)</span>
<span id="cb21-3"><a href="index.html#cb21-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-4"><a href="index.html#cb21-4" aria-hidden="true" tabindex="-1"></a>ad.OTU.lm <span class="ot">&lt;-</span> <span class="fu">linear_regres</span>(<span class="at">data =</span> ad.clr[,ad.OTU.name], </span>
<span id="cb21-5"><a href="index.html#cb21-5" aria-hidden="true" tabindex="-1"></a>                           <span class="at">trt =</span> ad.trt, <span class="at">batch.fix =</span> ad.batch, </span>
<span id="cb21-6"><a href="index.html#cb21-6" aria-hidden="true" tabindex="-1"></a>                           <span class="at">type =</span> <span class="st">&#39;linear model&#39;</span>)</span>
<span id="cb21-7"><a href="index.html#cb21-7" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(ad.OTU.lm<span class="sc">$</span>model<span class="sc">$</span>data)</span></code></pre></div>
<pre><code>## 
## Call:
## lm(formula = data[, i] ~ trt + batch.fix)
## 
## Residuals:
##     Min      1Q  Median      3Q     Max 
## -1.9384 -0.3279  0.1635  0.3849  0.9887 
## 
## Coefficients:
##                     Estimate Std. Error t value Pr(&gt;|t|)    
## (Intercept)           0.8501     0.2196   3.871 0.000243 ***
## trt1-2               -1.6871     0.1754  -9.617 2.27e-14 ***
## batch.fix09/04/2015   1.5963     0.2950   5.410 8.55e-07 ***
## batch.fix01/07/2016   2.0839     0.2345   8.886 4.82e-13 ***
## batch.fix14/11/2016   1.7405     0.2480   7.018 1.24e-09 ***
## batch.fix21/09/2017   1.2646     0.2690   4.701 1.28e-05 ***
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
## 
## Residual standard error: 0.7033 on 69 degrees of freedom
## Multiple R-squared:  0.7546, Adjusted R-squared:  0.7368 
## F-statistic: 42.44 on 5 and 69 DF,  p-value: &lt; 2.2e-16</code></pre>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb23-1"><a href="index.html#cb23-1" aria-hidden="true" tabindex="-1"></a><span class="co"># reference batch: 21/09/2017</span></span>
<span id="cb23-2"><a href="index.html#cb23-2" aria-hidden="true" tabindex="-1"></a>ad.batch <span class="ot">&lt;-</span> <span class="fu">relevel</span>(<span class="at">x =</span> ad.batch, <span class="at">ref =</span> <span class="st">&#39;21/09/2017&#39;</span>)</span>
<span id="cb23-3"><a href="index.html#cb23-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-4"><a href="index.html#cb23-4" aria-hidden="true" tabindex="-1"></a>ad.OTU.lm <span class="ot">&lt;-</span> <span class="fu">linear_regres</span>(<span class="at">data =</span> ad.clr[,ad.OTU.name], </span>
<span id="cb23-5"><a href="index.html#cb23-5" aria-hidden="true" tabindex="-1"></a>                           <span class="at">trt =</span> ad.trt, <span class="at">batch.fix =</span> ad.batch, </span>
<span id="cb23-6"><a href="index.html#cb23-6" aria-hidden="true" tabindex="-1"></a>                           <span class="at">type =</span> <span class="st">&#39;linear model&#39;</span>)</span>
<span id="cb23-7"><a href="index.html#cb23-7" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(ad.OTU.lm<span class="sc">$</span>model<span class="sc">$</span>data)</span></code></pre></div>
<pre><code>## 
## Call:
## lm(formula = data[, i] ~ trt + batch.fix)
## 
## Residuals:
##     Min      1Q  Median      3Q     Max 
## -1.9384 -0.3279  0.1635  0.3849  0.9887 
## 
## Coefficients:
##                     Estimate Std. Error t value Pr(&gt;|t|)    
## (Intercept)           2.1147     0.2502   8.453 2.97e-12 ***
## trt1-2               -1.6871     0.1754  -9.617 2.27e-14 ***
## batch.fix14/04/2016  -1.2646     0.2690  -4.701 1.28e-05 ***
## batch.fix09/04/2015   0.3317     0.3139   1.056  0.29446    
## batch.fix01/07/2016   0.8193     0.2573   3.185  0.00218 ** 
## batch.fix14/11/2016   0.4759     0.2705   1.760  0.08292 .  
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
## 
## Residual standard error: 0.7033 on 69 degrees of freedom
## Multiple R-squared:  0.7546, Adjusted R-squared:  0.7368 
## F-statistic: 42.44 on 5 and 69 DF,  p-value: &lt; 2.2e-16</code></pre>
<p>From the results of linear regression, we observed P &lt; 0.001 for the regression coefficients associated with all the other batches when the reference batch was “14/04/2016”, which confirmed the difference between the samples from batch “14/04/2016” and the other samples as observed from previous plots. When the reference batch was “21/09/2017”, we also observed significant differences between batch “21/09/2017” and “14/04/2016”, between “21/09/2017” and “01/07/2016”. Therefore, the batch effect because of “21/09/2017” also exists.</p>
</div>
<div id="heatmap" class="section level3 hasAnchor" number="1.5.3">
<h3><span class="header-section-number">1.5.3</span> Heatmap<a href="index.html#heatmap" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>We produce a heatmap using <span style="color: #FF7F00;">pheatmap</span> package. The data first need to be scaled on both OTUs and samples.</p>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb25-1"><a href="index.html#cb25-1" aria-hidden="true" tabindex="-1"></a><span class="co"># scale the clr data on both OTUs and samples</span></span>
<span id="cb25-2"><a href="index.html#cb25-2" aria-hidden="true" tabindex="-1"></a>ad.clr.s <span class="ot">&lt;-</span> <span class="fu">scale</span>(ad.clr, <span class="at">center =</span> <span class="cn">TRUE</span>, <span class="at">scale =</span> <span class="cn">TRUE</span>)</span>
<span id="cb25-3"><a href="index.html#cb25-3" aria-hidden="true" tabindex="-1"></a>ad.clr.ss <span class="ot">&lt;-</span> <span class="fu">scale</span>(<span class="fu">t</span>(ad.clr.s), <span class="at">center =</span> <span class="cn">TRUE</span>, <span class="at">scale =</span> <span class="cn">TRUE</span>)</span>
<span id="cb25-4"><a href="index.html#cb25-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-5"><a href="index.html#cb25-5" aria-hidden="true" tabindex="-1"></a>ad.anno_col <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">Batch =</span> ad.batch, <span class="at">Treatment =</span> ad.trt)</span>
<span id="cb25-6"><a href="index.html#cb25-6" aria-hidden="true" tabindex="-1"></a>ad.anno_colors <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="at">Batch =</span> <span class="fu">color.mixo</span>(<span class="fu">seq_len</span>(<span class="dv">5</span>)), </span>
<span id="cb25-7"><a href="index.html#cb25-7" aria-hidden="true" tabindex="-1"></a>                       <span class="at">Treatment =</span> <span class="fu">pb_color</span>(<span class="fu">seq_len</span>(<span class="dv">2</span>)))</span>
<span id="cb25-8"><a href="index.html#cb25-8" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(ad.anno_colors<span class="sc">$</span>Batch) <span class="ot">=</span> <span class="fu">levels</span>(ad.batch)</span>
<span id="cb25-9"><a href="index.html#cb25-9" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(ad.anno_colors<span class="sc">$</span>Treatment) <span class="ot">=</span> <span class="fu">levels</span>(ad.trt)</span>
<span id="cb25-10"><a href="index.html#cb25-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-11"><a href="index.html#cb25-11" aria-hidden="true" tabindex="-1"></a><span class="fu">pheatmap</span>(ad.clr.ss, </span>
<span id="cb25-12"><a href="index.html#cb25-12" aria-hidden="true" tabindex="-1"></a>         <span class="at">cluster_rows =</span> <span class="cn">FALSE</span>, </span>
<span id="cb25-13"><a href="index.html#cb25-13" aria-hidden="true" tabindex="-1"></a>         <span class="at">fontsize_row =</span> <span class="dv">4</span>, </span>
<span id="cb25-14"><a href="index.html#cb25-14" aria-hidden="true" tabindex="-1"></a>         <span class="at">fontsize_col =</span> <span class="dv">6</span>,</span>
<span id="cb25-15"><a href="index.html#cb25-15" aria-hidden="true" tabindex="-1"></a>         <span class="at">fontsize =</span> <span class="dv">8</span>,</span>
<span id="cb25-16"><a href="index.html#cb25-16" aria-hidden="true" tabindex="-1"></a>         <span class="at">clustering_distance_rows =</span> <span class="st">&#39;euclidean&#39;</span>,</span>
<span id="cb25-17"><a href="index.html#cb25-17" aria-hidden="true" tabindex="-1"></a>         <span class="at">clustering_method =</span> <span class="st">&#39;ward.D&#39;</span>,</span>
<span id="cb25-18"><a href="index.html#cb25-18" aria-hidden="true" tabindex="-1"></a>         <span class="at">treeheight_row =</span> <span class="dv">30</span>,</span>
<span id="cb25-19"><a href="index.html#cb25-19" aria-hidden="true" tabindex="-1"></a>         <span class="at">annotation_col =</span> ad.anno_col,</span>
<span id="cb25-20"><a href="index.html#cb25-20" aria-hidden="true" tabindex="-1"></a>         <span class="at">annotation_colors =</span> ad.anno_colors,</span>
<span id="cb25-21"><a href="index.html#cb25-21" aria-hidden="true" tabindex="-1"></a>         <span class="at">border_color =</span> <span class="st">&#39;NA&#39;</span>,</span>
<span id="cb25-22"><a href="index.html#cb25-22" aria-hidden="true" tabindex="-1"></a>         <span class="at">main =</span> <span class="st">&#39;AD data - Scaled&#39;</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:ADheatmap"></span>
<img src="PLSDAbatch_workflow_files/figure-html/ADheatmap-1.png" alt="Hierarchical clustering for samples in the AD data." width="90%" />
<p class="caption">
Figure 1.4: Hierarchical clustering for samples in the AD data.
</p>
</div>
<p>In the heatmap, samples in the <span style="color: #0000FF;">AD data</span> from batch dated “14/04/2016” were clustered and distinct from other samples, indicating a batch effect.</p>
</div>
<div id="prda" class="section level3 hasAnchor" number="1.5.4">
<h3><span class="header-section-number">1.5.4</span> pRDA<a href="index.html#prda" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>We apply pRDA with <code>varpart()</code> function from <span style="color: #FF7F00;">vegan</span> R package.</p>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb26-1"><a href="index.html#cb26-1" aria-hidden="true" tabindex="-1"></a><span class="co"># AD data</span></span>
<span id="cb26-2"><a href="index.html#cb26-2" aria-hidden="true" tabindex="-1"></a>ad.factors.df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">trt =</span> ad.trt, <span class="at">batch =</span> ad.batch)</span>
<span id="cb26-3"><a href="index.html#cb26-3" aria-hidden="true" tabindex="-1"></a><span class="fu">class</span>(ad.clr) <span class="ot">&lt;-</span> <span class="st">&#39;matrix&#39;</span></span>
<span id="cb26-4"><a href="index.html#cb26-4" aria-hidden="true" tabindex="-1"></a>ad.rda.before <span class="ot">&lt;-</span> <span class="fu">varpart</span>(ad.clr, <span class="sc">~</span> trt, <span class="sc">~</span> batch, </span>
<span id="cb26-5"><a href="index.html#cb26-5" aria-hidden="true" tabindex="-1"></a>                         <span class="at">data =</span> ad.factors.df, <span class="at">scale =</span> <span class="cn">TRUE</span>)</span>
<span id="cb26-6"><a href="index.html#cb26-6" aria-hidden="true" tabindex="-1"></a>ad.rda.before<span class="sc">$</span>part<span class="sc">$</span>indfract</span></code></pre></div>
<pre><code>##                 Df R.squared Adj.R.squared Testable
## [a] = X1|X2      1        NA    0.08943682     TRUE
## [b] = X2|X1      4        NA    0.26604420     TRUE
## [c]              0        NA    0.01296248    FALSE
## [d] = Residuals NA        NA    0.63155651    FALSE</code></pre>
<p>In the result, <code>X1</code> and <code>X2</code> represent the first and second covariates fitted in the model. <code>[a]</code>, <code>[b]</code> represent the independent proportion of variance explained by <code>X1</code> and <code>X2</code> respectively, and <code>[c]</code> represents the intersection variance shared between <code>X1</code> and <code>X2</code>. In the <span style="color: #0000FF;">AD data</span>, batch variance (<code>X2</code>) was larger than treatment variance (<code>X1</code>) with some interaction proportion (indicated in line <code>[c]</code>, Adj.R.squared = 0.013). The greater the intersection variance, the more unbalanced batch x treatment design is. In this study, we considered the design as approx. balanced.</p>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb28-1"><a href="index.html#cb28-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Sponge data</span></span>
<span id="cb28-2"><a href="index.html#cb28-2" aria-hidden="true" tabindex="-1"></a>sponge.batch <span class="ot">&lt;-</span> sponge_data<span class="sc">$</span>Y.bat</span>
<span id="cb28-3"><a href="index.html#cb28-3" aria-hidden="true" tabindex="-1"></a>sponge.trt <span class="ot">&lt;-</span> sponge_data<span class="sc">$</span>Y.trt</span>
<span id="cb28-4"><a href="index.html#cb28-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-5"><a href="index.html#cb28-5" aria-hidden="true" tabindex="-1"></a>sponge.factors.df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">trt =</span> sponge.trt, <span class="at">batch =</span> sponge.batch)</span>
<span id="cb28-6"><a href="index.html#cb28-6" aria-hidden="true" tabindex="-1"></a><span class="fu">class</span>(sponge.clr) <span class="ot">&lt;-</span> <span class="st">&#39;matrix&#39;</span></span>
<span id="cb28-7"><a href="index.html#cb28-7" aria-hidden="true" tabindex="-1"></a>sponge.rda.before <span class="ot">&lt;-</span> <span class="fu">varpart</span>(sponge.clr, <span class="sc">~</span> trt, <span class="sc">~</span> batch, </span>
<span id="cb28-8"><a href="index.html#cb28-8" aria-hidden="true" tabindex="-1"></a>                             <span class="at">data =</span> sponge.factors.df, <span class="at">scale =</span> <span class="cn">TRUE</span>)</span>
<span id="cb28-9"><a href="index.html#cb28-9" aria-hidden="true" tabindex="-1"></a>sponge.rda.before<span class="sc">$</span>part<span class="sc">$</span>indfract</span></code></pre></div>
<pre><code>##                 Df R.squared Adj.R.squared Testable
## [a] = X1|X2      1        NA    0.16572246     TRUE
## [b] = X2|X1      1        NA    0.16396277     TRUE
## [c]              0        NA   -0.01063501    FALSE
## [d] = Residuals NA        NA    0.68094977    FALSE</code></pre>
<p>The <span style="color: #964B00;">sponge data</span> have a completely balanced batch x treatment design, so there was no intersection variance (indicated in line <code>[c]</code>, Adj.R.squared = -0.01) detected. The proportion of treatment and batch variance is nearly equal as indicated in lines <code>[a]</code> and <code>[b]</code>.</p>
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb30-1"><a href="index.html#cb30-1" aria-hidden="true" tabindex="-1"></a><span class="co"># HD data</span></span>
<span id="cb30-2"><a href="index.html#cb30-2" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="at">file =</span> <span class="st">&#39;./data/HD_data.rda&#39;</span>)</span>
<span id="cb30-3"><a href="index.html#cb30-3" aria-hidden="true" tabindex="-1"></a>hd.clr <span class="ot">&lt;-</span> HD_data<span class="sc">$</span>EgData<span class="sc">$</span>X.clr</span>
<span id="cb30-4"><a href="index.html#cb30-4" aria-hidden="true" tabindex="-1"></a>hd.trt <span class="ot">&lt;-</span> HD_data<span class="sc">$</span>EgData<span class="sc">$</span>Y.trt</span>
<span id="cb30-5"><a href="index.html#cb30-5" aria-hidden="true" tabindex="-1"></a>hd.batch <span class="ot">&lt;-</span> HD_data<span class="sc">$</span>EgData<span class="sc">$</span>Y.bat</span>
<span id="cb30-6"><a href="index.html#cb30-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-7"><a href="index.html#cb30-7" aria-hidden="true" tabindex="-1"></a>hd.factors.df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">trt =</span> hd.trt, <span class="at">batch =</span> hd.batch)</span>
<span id="cb30-8"><a href="index.html#cb30-8" aria-hidden="true" tabindex="-1"></a><span class="fu">class</span>(hd.clr) <span class="ot">&lt;-</span> <span class="st">&#39;matrix&#39;</span></span>
<span id="cb30-9"><a href="index.html#cb30-9" aria-hidden="true" tabindex="-1"></a>hd.rda.before <span class="ot">&lt;-</span> <span class="fu">varpart</span>(hd.clr, <span class="sc">~</span> trt, <span class="sc">~</span> batch, </span>
<span id="cb30-10"><a href="index.html#cb30-10" aria-hidden="true" tabindex="-1"></a>                         <span class="at">data =</span> hd.factors.df, <span class="at">scale =</span> <span class="cn">TRUE</span>)</span>
<span id="cb30-11"><a href="index.html#cb30-11" aria-hidden="true" tabindex="-1"></a>hd.rda.before<span class="sc">$</span>part<span class="sc">$</span>indfract</span></code></pre></div>
<pre><code>##                 Df R.squared Adj.R.squared Testable
## [a] = X1|X2      0        NA  1.110223e-16    FALSE
## [b] = X2|X1      8        NA  1.608205e-01     TRUE
## [c]              0        NA  9.730583e-02    FALSE
## [d] = Residuals NA        NA  7.418737e-01    FALSE</code></pre>
<p>Collinearity was detected in the <span style="color: #8601AF;">HD data</span> between treatment and batch as indicated by lines <code>[a]</code> (Adj.R.squared = 0) and <code>[c]</code> (Adj.R.squared = 0.097) that all treatment variance was explained as intersection variance, because the batch x treatment design is nested. The intersection variance in such a design is usually considerable. As the intersection is shared between batch and treatment effects, the batch variance in the <span style="color: #8601AF;">HD data</span> was larger than the treatment variance.</p>
</div>
</div>
<div id="managing-batch-effects" class="section level2 hasAnchor" number="1.6">
<h2><span class="header-section-number">1.6</span> Managing batch effects<a href="index.html#managing-batch-effects" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="accnt" class="section level3 hasAnchor" number="1.6.1">
<h3><span class="header-section-number">1.6.1</span> Accounting for batch effects<a href="index.html#accnt" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The methods that we use to account for batch effects include the methods designed for microbiome data: zero-inflated Gaussian (ZIG) mixture model and the methods adapted for microbiome data: linear regression, SVA and RUV4. Among them, SVA and RUV4 were designed for unknown batch effects.</p>
<div id="methods-designed-for-microbiome-data" class="section level4 hasAnchor" number="1.6.1.1">
<h4><span class="header-section-number">1.6.1.1</span> Methods designed for microbiome data<a href="index.html#methods-designed-for-microbiome-data" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p><strong>Zero-inflated Gaussian mixture model</strong> To use the ZIG model, we first create a <code>MRexperiment</code> object applying <code>newMRexperiment()</code> (from <span style="color: #FF7F00;">metagenomeSeq</span> package) to microbiome counts and annotated data frames with metadata and taxonomic information generated with <code>AnnotatedDataFrame()</code> from <span style="color: #FF7F00;">Biobase</span> package.</p>
<div class="sourceCode" id="cb32"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb32-1"><a href="index.html#cb32-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Creating a MRexperiment object (make sure no NA in metadata)</span></span>
<span id="cb32-2"><a href="index.html#cb32-2" aria-hidden="true" tabindex="-1"></a>AD.phenotypeData <span class="ot">=</span> <span class="fu">AnnotatedDataFrame</span>(<span class="at">data =</span> AD_data<span class="sc">$</span>FullData<span class="sc">$</span>metadata)</span>
<span id="cb32-3"><a href="index.html#cb32-3" aria-hidden="true" tabindex="-1"></a>AD.taxaData <span class="ot">=</span> <span class="fu">AnnotatedDataFrame</span>(<span class="at">data =</span> AD_data<span class="sc">$</span>FullData<span class="sc">$</span>taxa)</span>
<span id="cb32-4"><a href="index.html#cb32-4" aria-hidden="true" tabindex="-1"></a>AD.obj <span class="ot">=</span> <span class="fu">newMRexperiment</span>(<span class="at">counts =</span> <span class="fu">t</span>(AD_data<span class="sc">$</span>FullData<span class="sc">$</span>X.count), </span>
<span id="cb32-5"><a href="index.html#cb32-5" aria-hidden="true" tabindex="-1"></a>                         <span class="at">phenoData =</span> AD.phenotypeData, </span>
<span id="cb32-6"><a href="index.html#cb32-6" aria-hidden="true" tabindex="-1"></a>                         <span class="at">featureData =</span> AD.taxaData)</span>
<span id="cb32-7"><a href="index.html#cb32-7" aria-hidden="true" tabindex="-1"></a>AD.obj</span></code></pre></div>
<pre><code>## MRexperiment (storageMode: environment)
## assayData: 567 features, 75 samples 
##   element names: counts 
## protocolData: none
## phenoData
##   sampleNames: E1aJ16_A E5aJ16_A ... E8cJ96_E (75 total)
##   varLabels: sample_name.data.extraction analysis_name ...
##     initial_phenol_concentration.regroup (7 total)
##   varMetadata: labelDescription
## featureData
##   featureNames: OTU12 OTU29 ... OTU17710 (567 total)
##   fvarLabels: Kingdom Phylum ... Species (7 total)
##   fvarMetadata: labelDescription
## experimentData: use &#39;experimentData(object)&#39;
## Annotation:</code></pre>
<p>The <span style="color: #0000FF;">AD count data</span> are then filtered with <code>filterData()</code> function (from <span style="color: #FF7F00;">metagenomeSeq</span>). We can use <code>MRcounts()</code> to extract the count data from the <code>MRexperiment</code> object.</p>
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb34-1"><a href="index.html#cb34-1" aria-hidden="true" tabindex="-1"></a><span class="co"># filtering data to maintain a threshold of minimum depth or OTU presence</span></span>
<span id="cb34-2"><a href="index.html#cb34-2" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(<span class="fu">MRcounts</span>(AD.obj))</span></code></pre></div>
<pre><code>## [1] 567  75</code></pre>
<div class="sourceCode" id="cb36"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb36-1"><a href="index.html#cb36-1" aria-hidden="true" tabindex="-1"></a>AD.obj <span class="ot">=</span> <span class="fu">filterData</span>(<span class="at">obj =</span> AD.obj, <span class="at">present =</span> <span class="dv">20</span>, <span class="at">depth =</span> <span class="dv">5</span>)</span>
<span id="cb36-2"><a href="index.html#cb36-2" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(<span class="fu">MRcounts</span>(AD.obj))</span></code></pre></div>
<pre><code>## [1] 289  75</code></pre>
<p>After filtering, the <span style="color: #0000FF;">AD count data</span> were reduced to 289 OTUs and 75 samples.</p>
<p>We calculate the percentile for CSS normalisation with <code>cumNormStatFast()</code> function (from <span style="color: #FF7F00;">metagenomeSeq</span> package). The CSS normalisation is applied with <code>cumNorm()</code> and the normalised data can be exported using <code>MRcounts()</code> with <code>norm = TRUE</code>. The normalisation scaling factors for each sample, which are the sum of counts up to the calculated percentile, can be accessed through <code>normFactors()</code>. We calculate the log transfomed scaling factors by diving them with their median, which are better than the default scaling factors that are divided by 1000 (<code>log2(normFactors(obj)/1000 + 1)</code>).</p>
<div class="sourceCode" id="cb38"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb38-1"><a href="index.html#cb38-1" aria-hidden="true" tabindex="-1"></a><span class="co"># calculate the percentile for CSS normalisation</span></span>
<span id="cb38-2"><a href="index.html#cb38-2" aria-hidden="true" tabindex="-1"></a>AD.pctl <span class="ot">=</span> <span class="fu">cumNormStatFast</span>(<span class="at">obj =</span> AD.obj)</span>
<span id="cb38-3"><a href="index.html#cb38-3" aria-hidden="true" tabindex="-1"></a><span class="co"># CSS normalisation</span></span>
<span id="cb38-4"><a href="index.html#cb38-4" aria-hidden="true" tabindex="-1"></a>AD.obj <span class="ot">&lt;-</span> <span class="fu">cumNorm</span>(<span class="at">obj =</span> AD.obj, <span class="at">p =</span> AD.pctl)</span>
<span id="cb38-5"><a href="index.html#cb38-5" aria-hidden="true" tabindex="-1"></a><span class="co"># export normalised data</span></span>
<span id="cb38-6"><a href="index.html#cb38-6" aria-hidden="true" tabindex="-1"></a>AD.norm.data <span class="ot">&lt;-</span> <span class="fu">MRcounts</span>(<span class="at">obj =</span> AD.obj, <span class="at">norm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb38-7"><a href="index.html#cb38-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-8"><a href="index.html#cb38-8" aria-hidden="true" tabindex="-1"></a><span class="co"># normalisation scaling factors for each sample </span></span>
<span id="cb38-9"><a href="index.html#cb38-9" aria-hidden="true" tabindex="-1"></a>AD.normFactor <span class="ot">=</span> <span class="fu">normFactors</span>(<span class="at">object =</span> AD.obj)</span>
<span id="cb38-10"><a href="index.html#cb38-10" aria-hidden="true" tabindex="-1"></a>AD.normFactor <span class="ot">=</span> <span class="fu">log2</span>(AD.normFactor<span class="sc">/</span><span class="fu">median</span>(AD.normFactor) <span class="sc">+</span> <span class="dv">1</span>)</span></code></pre></div>
<p>We create a design matrix with treatment variable (<code>phenol_conc</code>), batch variable (<code>seq_run</code>) and the log transformed scaling factors using <code>model.matrix()</code>, and then apply the ZIG model by <code>fitZig()</code> function. We set <code>useCSSoffset = FALSE</code> to avoid using the default scaling factors as we have already included our customised scaling factor (<code>AD.normFactor</code>) in the design matrix.</p>
<div class="sourceCode" id="cb39"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb39-1"><a href="index.html#cb39-1" aria-hidden="true" tabindex="-1"></a><span class="co"># treatment variable</span></span>
<span id="cb39-2"><a href="index.html#cb39-2" aria-hidden="true" tabindex="-1"></a>phenol_conc <span class="ot">=</span> <span class="fu">pData</span>(<span class="at">object =</span> AD.obj)<span class="sc">$</span>initial_phenol_concentration.regroup</span>
<span id="cb39-3"><a href="index.html#cb39-3" aria-hidden="true" tabindex="-1"></a><span class="co"># batch variable</span></span>
<span id="cb39-4"><a href="index.html#cb39-4" aria-hidden="true" tabindex="-1"></a>seq_run <span class="ot">=</span> <span class="fu">pData</span>(<span class="at">object =</span> AD.obj)<span class="sc">$</span>sequencing_run_date</span>
<span id="cb39-5"><a href="index.html#cb39-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-6"><a href="index.html#cb39-6" aria-hidden="true" tabindex="-1"></a><span class="co"># build a design matrix</span></span>
<span id="cb39-7"><a href="index.html#cb39-7" aria-hidden="true" tabindex="-1"></a>AD.mod.full <span class="ot">=</span> <span class="fu">model.matrix</span>(<span class="sc">~</span> phenol_conc <span class="sc">+</span> seq_run <span class="sc">+</span> AD.normFactor)</span>
<span id="cb39-8"><a href="index.html#cb39-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-9"><a href="index.html#cb39-9" aria-hidden="true" tabindex="-1"></a><span class="co"># settings for the fitZig() function</span></span>
<span id="cb39-10"><a href="index.html#cb39-10" aria-hidden="true" tabindex="-1"></a>AD.settings <span class="ot">&lt;-</span> <span class="fu">zigControl</span>(<span class="at">maxit =</span> <span class="dv">10</span>, <span class="at">verbose =</span> <span class="cn">TRUE</span>)</span>
<span id="cb39-11"><a href="index.html#cb39-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-12"><a href="index.html#cb39-12" aria-hidden="true" tabindex="-1"></a><span class="co"># apply the ZIG model</span></span>
<span id="cb39-13"><a href="index.html#cb39-13" aria-hidden="true" tabindex="-1"></a>ADfit <span class="ot">&lt;-</span> <span class="fu">fitZig</span>(<span class="at">obj =</span> AD.obj, <span class="at">mod =</span> AD.mod.full, </span>
<span id="cb39-14"><a href="index.html#cb39-14" aria-hidden="true" tabindex="-1"></a>                <span class="at">useCSSoffset =</span> <span class="cn">FALSE</span>, <span class="at">control =</span> AD.settings)</span></code></pre></div>
<pre><code>## it= 0, nll=123.44, log10(eps+1)=Inf, stillActive=289
## it= 1, nll=134.33, log10(eps+1)=0.04, stillActive=10
## it= 2, nll=134.64, log10(eps+1)=0.01, stillActive=1
## it= 3, nll=134.80, log10(eps+1)=0.00, stillActive=0</code></pre>
<p>The OTUs with the top 50 smallest p values are extracted using <code>MRcoefs()</code>. We set <code>eff = 0.5</code>, so only the OTUs with at least “0.5” quantile (50%) number of effective samples (positive samples + estimated undersampling zeroes) are extracted.</p>
<div class="sourceCode" id="cb41"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb41-1"><a href="index.html#cb41-1" aria-hidden="true" tabindex="-1"></a>ADcoefs <span class="ot">&lt;-</span> <span class="fu">MRcoefs</span>(ADfit, <span class="at">coef =</span> <span class="dv">2</span>, <span class="at">group =</span> <span class="dv">3</span>, <span class="at">number =</span> <span class="dv">50</span>, <span class="at">eff =</span> <span class="fl">0.5</span>)</span>
<span id="cb41-2"><a href="index.html#cb41-2" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(ADcoefs)</span></code></pre></div>
<pre><code>##       phenol_conc1-2      pvalues   adjPvalues
## OTU68      -2.901041 2.467451e-15 4.081959e-13
## OTU46      -2.847212 2.824885e-15 4.081959e-13
## OTU28      -2.449290 4.012023e-14 3.864915e-12
## OTU24      -2.846452 6.397959e-14 4.622526e-12
## OTU59      -1.815250 3.784889e-13 2.187666e-11
## OTU65      -2.141183 3.342621e-11 1.610029e-09</code></pre>
</div>
<div id="other-methods-adapted-for-microbiome-data" class="section level4 hasAnchor" number="1.6.1.2">
<h4><span class="header-section-number">1.6.1.2</span> Other methods adapted for microbiome data<a href="index.html#other-methods-adapted-for-microbiome-data" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p><strong>Linear regression</strong> Linear regression is conducted with <code>linear_regres()</code> function in <span style="color: #FF7F00;">PLSDAbatch</span>. We integrated the <span style="color: #FF7F00;">performance</span> package that assesses performance of regression models into our function <code>linear_regres()</code>. Therefore, we can apply <code>check_model()</code> from <span style="color: #FF7F00;">performance</span> to the outputs from <code>linear_regres()</code> to diagnose the validity of the model fitted with treatment and batch effects for each variable <span class="citation">(<a href="#ref-daniel2020performance" role="doc-biblioref">LÃŒdecke et al. 2020</a>)</span>. We can extract performance measurements such as adjusted R2, RMSE, RSE, AIC and BIC for the models fitted with and without batch effects, which are also the outputs of <code>linear_regres()</code>.</p>
<p>We apply <code>type = "linear model"</code> to the <span style="color: #0000FF;">AD data</span> because of the balanced batch x treatment design.</p>
<div class="sourceCode" id="cb43"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb43-1"><a href="index.html#cb43-1" aria-hidden="true" tabindex="-1"></a><span class="co"># AD data</span></span>
<span id="cb43-2"><a href="index.html#cb43-2" aria-hidden="true" tabindex="-1"></a>ad.clr <span class="ot">&lt;-</span> ad.clr[<span class="fu">seq_len</span>(<span class="fu">nrow</span>(ad.clr)), <span class="fu">seq_len</span>(<span class="fu">ncol</span>(ad.clr))]</span>
<span id="cb43-3"><a href="index.html#cb43-3" aria-hidden="true" tabindex="-1"></a>ad.lm <span class="ot">&lt;-</span> <span class="fu">linear_regres</span>(<span class="at">data =</span> ad.clr, <span class="at">trt =</span> ad.trt, </span>
<span id="cb43-4"><a href="index.html#cb43-4" aria-hidden="true" tabindex="-1"></a>                       <span class="at">batch.fix =</span> ad.batch, <span class="at">type =</span> <span class="st">&#39;linear model&#39;</span>)</span>
<span id="cb43-5"><a href="index.html#cb43-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-6"><a href="index.html#cb43-6" aria-hidden="true" tabindex="-1"></a>ad.p <span class="ot">&lt;-</span> <span class="fu">sapply</span>(ad.lm<span class="sc">$</span>lm.table, <span class="cf">function</span>(x){x<span class="sc">$</span>coefficients[<span class="dv">2</span>,<span class="dv">4</span>]})</span>
<span id="cb43-7"><a href="index.html#cb43-7" aria-hidden="true" tabindex="-1"></a>ad.p.adj <span class="ot">&lt;-</span> <span class="fu">p.adjust</span>(<span class="at">p =</span> ad.p, <span class="at">method =</span> <span class="st">&#39;fdr&#39;</span>)</span>
<span id="cb43-8"><a href="index.html#cb43-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-9"><a href="index.html#cb43-9" aria-hidden="true" tabindex="-1"></a><span class="fu">check_model</span>(ad.lm<span class="sc">$</span>model<span class="sc">$</span>OTU12)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:ADlm"></span>
<img src="PLSDAbatch_workflow_files/figure-html/ADlm-1.png" alt="Diagnostic plots for the model fitted with batch effects of &quot;OTU12&quot; in the AD data." width="100%" />
<p class="caption">
Figure 1.5: Diagnostic plots for the model fitted with batch effects of “OTU12” in the AD data.
</p>
</div>
<p>To diagnose the validity of the model fitted with both treatment and batch effects, we use different plots to check the assumptions of each microbial variable. For example, the diagnostic plots of “OTU12” are shown in the above figure panel. The simulated data under the fitted model were close to the real data (shown as green line), indicating good model fitness (top panel). The linearity (or homoscedasticity) and homogeneity of variance were not satisfied. The correlation between batch (<code>batch.fix</code>) and treatment (<code>trt</code>) effects was very low, indicating a good model with low collinearity. Some samples could be classified as outliers with a Cook’s distance larger than or equal to 0.5, for example, “57”, “39”, “47”, “44” and “16” (middle panel). The distribution of residuals was very close to normal (bottom panel). For the microbial variables with some assumptions not met, we should be careful about their results.</p>
<p>For performance measurements of models fitted with or without batch effects, We show an example of the results for some variables.</p>
<div class="sourceCode" id="cb44"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb44-1"><a href="index.html#cb44-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(ad.lm<span class="sc">$</span>adj.R2)</span></code></pre></div>
<pre><code>##            trt.only trt.batch
## OTU12   0.008445545 0.4785532
## OTU29  -0.012966020 0.7808945
## OTU77  -0.013343146 0.4779754
## OTU86   0.183743803 0.2850153
## OTU97  -0.008989284 0.4502563
## OTU106  0.072963581 0.8745676</code></pre>
<p>The adjusted <span class="math inline">\(R^2\)</span> of the model with both treatment and batch effects for all the OTUs listed was larger than the model with treatment effects only, suggesting that the model fitted with batch effects explained more data variance, and was thus better than the model without batch effects.</p>
<p>We next look at the AIC of models fitted with or without batch effects.</p>
<div class="sourceCode" id="cb46"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb46-1"><a href="index.html#cb46-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(ad.lm<span class="sc">$</span>AIC)</span></code></pre></div>
<pre><code>##        trt.only trt.batch
## OTU12  208.5626 164.13617
## OTU29  266.0485 154.99072
## OTU77  180.7800 134.80638
## OTU86  212.2987 206.13722
## OTU97  161.6715 119.90108
## OTU106 205.4135  59.17005</code></pre>
<p>A lower AIC indicates a better fit, here for a model fitted with batch effects for all the OTUs.</p>
<p>Both results strongly indicated that a batch effect should be fitted in the linear model.</p>
<p>We apply a <code>"linear mixed model"</code> to the <span style="color: #8601AF;">HD data</span> because of the nested batch x treatment design.</p>
<div class="sourceCode" id="cb48"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb48-1"><a href="index.html#cb48-1" aria-hidden="true" tabindex="-1"></a><span class="co"># HD data</span></span>
<span id="cb48-2"><a href="index.html#cb48-2" aria-hidden="true" tabindex="-1"></a>hd.lmm <span class="ot">&lt;-</span> <span class="fu">linear_regres</span>(<span class="at">data =</span> hd.clr, <span class="at">trt =</span> hd.trt, </span>
<span id="cb48-3"><a href="index.html#cb48-3" aria-hidden="true" tabindex="-1"></a>                        <span class="at">batch.random =</span> hd.batch, </span>
<span id="cb48-4"><a href="index.html#cb48-4" aria-hidden="true" tabindex="-1"></a>                        <span class="at">type =</span> <span class="st">&#39;linear mixed model&#39;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb49"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb49-1"><a href="index.html#cb49-1" aria-hidden="true" tabindex="-1"></a>hd.p <span class="ot">&lt;-</span> <span class="fu">sapply</span>(hd.lmm<span class="sc">$</span>lmm.table, <span class="cf">function</span>(x){x<span class="sc">$</span>coefficients[<span class="dv">2</span>,<span class="dv">5</span>]})</span>
<span id="cb49-2"><a href="index.html#cb49-2" aria-hidden="true" tabindex="-1"></a>hd.p.adj <span class="ot">&lt;-</span> <span class="fu">p.adjust</span>(<span class="at">p =</span> hd.p, <span class="at">method =</span> <span class="st">&#39;fdr&#39;</span>)</span>
<span id="cb49-3"><a href="index.html#cb49-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-4"><a href="index.html#cb49-4" aria-hidden="true" tabindex="-1"></a><span class="fu">check_model</span>(hd.lmm<span class="sc">$</span>model<span class="sc">$</span>OTU1)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:HDlmm"></span>
<img src="PLSDAbatch_workflow_files/figure-html/HDlmm-1.png" alt="Diagnostic plots for the model fitted with batch effects of &quot;OTU1&quot; in the HD data." width="100%" />
<p class="caption">
Figure 1.6: Diagnostic plots for the model fitted with batch effects of “OTU1” in the HD data.
</p>
</div>
<p>According to the diagnostic plots of “OTU1” as shown in the above figure panel, the simulated data under the fitted model were not very close to the real data (shown as green line), indicating an imperfect fitness (top panel). The linearity (or homoscedasticity) and homogeneity of variance were not satisfied. Some samples could be classified as outliers with a Cook’s distance larger than or equal to 0.5, for example, “12”, “21”, “20”, “11” and “6” (middle panel). The distribution of residuals was not normal, while the one of random effects was very close to normal (bottom panel).</p>
<p>We then look at the AIC of models fitted with or without batch effects.</p>
<div class="sourceCode" id="cb50"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb50-1"><a href="index.html#cb50-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(hd.lmm<span class="sc">$</span>AIC)</span></code></pre></div>
<pre><code>##       trt.only trt.batch
## OTU1  60.00637  65.46410
## OTU2 130.14167 130.96344
## OTU3  71.76985  75.58915
## OTU4 134.59992 134.04812
## OTU5 150.25059 148.30644
## OTU6  99.92864  99.05391</code></pre>
<p>For some OTUs, the model fitted without batch effects was better with a lower AIC, e.g., “OTU1”, “OTU2” and “OTU3”. “OTU4”, “OTU5” and “OTU6” were more appropriate within a model with batch effects.</p>
<p>Since we apply a <code>"linear mixed model"</code> to the <span style="color: #8601AF;">HD data</span>, this type of model can only output conditional <span class="math inline">\(R^2\)</span> that includes the variance of both fixed and random effects (treatment, fixed and random batch effects) and marginal <span class="math inline">\(R^2\)</span> that includes only the variance of fixed effects (treatment and fixed batch effects) <span class="citation">(<a href="#ref-nakagawa2013general" role="doc-biblioref">Nakagawa and Schielzeth 2013</a>)</span>. The marginal <span class="math inline">\(R^2\)</span>s of the models with and without the batch effect in the <span style="color: #8601AF;">HD data</span> should be the same theoretically, as marginal <span class="math inline">\(R^2\)</span> only includes fixed effects and the treatment effect was the only fixed effect here. In the actual calculation, the results of these two models were not the same, because part of the variance was explained by the random effect in the model fitted with the batch effect.</p>
<div class="sourceCode" id="cb52"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb52-1"><a href="index.html#cb52-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(hd.lmm<span class="sc">$</span>cond.R2)</span></code></pre></div>
<pre><code>##        trt.only trt.batch
## OTU1 0.47917775 0.5161317
## OTU2 0.39223824 0.4729848
## OTU3 0.02091376 0.2636432
## OTU4 0.37200835 0.5672392
## OTU5 0.07858373 0.2938212
## OTU6 0.41388250 0.6143134</code></pre>
<div class="sourceCode" id="cb54"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb54-1"><a href="index.html#cb54-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(hd.lmm<span class="sc">$</span>marg.R2)</span></code></pre></div>
<pre><code>##        trt.only  trt.batch
## OTU1 0.47917775 0.46735273
## OTU2 0.39223824 0.38459672
## OTU3 0.02091376 0.01489608
## OTU4 0.37200835 0.36122754
## OTU5 0.07858373 0.06052938
## OTU6 0.41388250 0.40749149</code></pre>
<p>We observed that some variables resulted in singular fits (<code>error message: Can't compute random effect variances. Some variance components equal zero. Your model may suffer from singulariy.</code>), which are expected to happen in linear mixed models when covariates are nested. We recommend noting the variables for which this error occurs, as it may lead to unreliable results.</p>
<p><strong>SVA</strong> accounts for unknown batch effects. Here we assume that the batch grouping information in the <span style="color: #0000FF;">AD data</span> is unknown. We first build two design matrices with (<code>ad.mod</code>) and without (<code>ad.mod0</code>) treatment grouping information generated with <code>model.matrix()</code> function from <span style="color: #FF7F00;">stats</span>. We then use <code>num.sv()</code> from <span style="color: #FF7F00;">sva</span> package to determine the number of batch variables <code>n.sv</code> that is used to estimate batch effects in function <code>sva()</code>.</p>
<div class="sourceCode" id="cb56"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb56-1"><a href="index.html#cb56-1" aria-hidden="true" tabindex="-1"></a><span class="co"># estimate batch effects</span></span>
<span id="cb56-2"><a href="index.html#cb56-2" aria-hidden="true" tabindex="-1"></a>ad.mod <span class="ot">&lt;-</span> <span class="fu">model.matrix</span>( <span class="sc">~</span> ad.trt)</span>
<span id="cb56-3"><a href="index.html#cb56-3" aria-hidden="true" tabindex="-1"></a>ad.mod0 <span class="ot">&lt;-</span> <span class="fu">model.matrix</span>( <span class="sc">~</span> <span class="dv">1</span>, <span class="at">data =</span> ad.trt)</span>
<span id="cb56-4"><a href="index.html#cb56-4" aria-hidden="true" tabindex="-1"></a>ad.sva.n <span class="ot">&lt;-</span> <span class="fu">num.sv</span>(<span class="at">dat =</span> <span class="fu">t</span>(ad.clr), <span class="at">mod =</span> ad.mod, <span class="at">method =</span> <span class="st">&#39;leek&#39;</span>)</span>
<span id="cb56-5"><a href="index.html#cb56-5" aria-hidden="true" tabindex="-1"></a>ad.sva <span class="ot">&lt;-</span> <span class="fu">sva</span>(<span class="fu">t</span>(ad.clr), ad.mod, ad.mod0, <span class="at">n.sv =</span> ad.sva.n)</span></code></pre></div>
<pre><code>## Number of significant surrogate variables is:  4 
## Iteration (out of 5 ):1  2  3  4  5</code></pre>
<p>The estimated batch effects are then applied to <code>f.pvalue()</code> to calculate the P-values of treatment effects. The estimated batch effects in SVA are assumed to be independent of the treatment effects. However, SVA considers some correlation between batch and treatment effects <span class="citation">(<a href="#ref-wang2020managing" role="doc-biblioref">Y. Wang and LêCao 2020</a>)</span>.</p>
<div class="sourceCode" id="cb58"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb58-1"><a href="index.html#cb58-1" aria-hidden="true" tabindex="-1"></a><span class="co"># include estimated batch effects in the linear model</span></span>
<span id="cb58-2"><a href="index.html#cb58-2" aria-hidden="true" tabindex="-1"></a>ad.mod.batch <span class="ot">&lt;-</span> <span class="fu">cbind</span>(ad.mod, ad.sva<span class="sc">$</span>sv)</span>
<span id="cb58-3"><a href="index.html#cb58-3" aria-hidden="true" tabindex="-1"></a>ad.mod0.batch <span class="ot">&lt;-</span> <span class="fu">cbind</span>(ad.mod0, ad.sva<span class="sc">$</span>sv)</span>
<span id="cb58-4"><a href="index.html#cb58-4" aria-hidden="true" tabindex="-1"></a>ad.sva.p <span class="ot">&lt;-</span> <span class="fu">f.pvalue</span>(<span class="fu">t</span>(ad.clr), ad.mod.batch, ad.mod0.batch)</span>
<span id="cb58-5"><a href="index.html#cb58-5" aria-hidden="true" tabindex="-1"></a>ad.sva.p.adj <span class="ot">&lt;-</span> <span class="fu">p.adjust</span>(ad.sva.p, <span class="at">method =</span> <span class="st">&#39;fdr&#39;</span>)</span></code></pre></div>
<p><strong>RUV4</strong> Before applying RUV4 (<code>RUV4()</code> from <span style="color: #FF7F00;">ruv</span> package), we need to specify negative control variables and the number of batch variables to estimate. We can use the empirical negative controls that are not significantly differentially abundant (adjusted P &gt; 0.05) from a linear regression with the treatment information as the only covariate.</p>
<p>We use a loop to fit a linear regression for each microbial variable and adjust P values of treatment effects for multiple comparisons with <code>p.adjust()</code> from <span style="color: #FF7F00;">stats</span>. The empirical negative controls are then extracted according to the adjusted P values.</p>
<div class="sourceCode" id="cb59"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb59-1"><a href="index.html#cb59-1" aria-hidden="true" tabindex="-1"></a><span class="co"># empirical negative controls</span></span>
<span id="cb59-2"><a href="index.html#cb59-2" aria-hidden="true" tabindex="-1"></a>ad.empir.p <span class="ot">&lt;-</span> <span class="fu">c</span>()</span>
<span id="cb59-3"><a href="index.html#cb59-3" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(e <span class="cf">in</span> <span class="fu">seq_len</span>(<span class="fu">ncol</span>(ad.clr))){</span>
<span id="cb59-4"><a href="index.html#cb59-4" aria-hidden="true" tabindex="-1"></a>  ad.empir.lm <span class="ot">&lt;-</span> <span class="fu">lm</span>(ad.clr[,e] <span class="sc">~</span> ad.trt)</span>
<span id="cb59-5"><a href="index.html#cb59-5" aria-hidden="true" tabindex="-1"></a>  ad.empir.p[e] <span class="ot">&lt;-</span> <span class="fu">summary</span>(ad.empir.lm)<span class="sc">$</span>coefficients[<span class="dv">2</span>,<span class="dv">4</span>]</span>
<span id="cb59-6"><a href="index.html#cb59-6" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb59-7"><a href="index.html#cb59-7" aria-hidden="true" tabindex="-1"></a>ad.empir.p.adj <span class="ot">&lt;-</span> <span class="fu">p.adjust</span>(<span class="at">p =</span> ad.empir.p, <span class="at">method =</span> <span class="st">&#39;fdr&#39;</span>)</span>
<span id="cb59-8"><a href="index.html#cb59-8" aria-hidden="true" tabindex="-1"></a>ad.nc <span class="ot">&lt;-</span> ad.empir.p.adj <span class="sc">&gt;</span> <span class="fl">0.05</span></span></code></pre></div>
<p>The number of batch variables <code>k</code> can be determined using <code>getK()</code> function.</p>
<div class="sourceCode" id="cb60"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb60-1"><a href="index.html#cb60-1" aria-hidden="true" tabindex="-1"></a><span class="co"># estimate k</span></span>
<span id="cb60-2"><a href="index.html#cb60-2" aria-hidden="true" tabindex="-1"></a>ad.k.res <span class="ot">&lt;-</span> <span class="fu">getK</span>(<span class="at">Y =</span> ad.clr, <span class="at">X =</span> ad.trt, <span class="at">ctl =</span> ad.nc)</span>
<span id="cb60-3"><a href="index.html#cb60-3" aria-hidden="true" tabindex="-1"></a>ad.k <span class="ot">&lt;-</span> ad.k.res<span class="sc">$</span>k</span></code></pre></div>
<p>We then apply <code>RUV4()</code> with known treatment variables, estimated negative control variables and <code>k</code> batch variables. The calculated P values also need to be adjusted for multiple comparisons.</p>
<div class="sourceCode" id="cb61"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb61-1"><a href="index.html#cb61-1" aria-hidden="true" tabindex="-1"></a><span class="co"># RUV4</span></span>
<span id="cb61-2"><a href="index.html#cb61-2" aria-hidden="true" tabindex="-1"></a>ad.ruv4 <span class="ot">&lt;-</span> <span class="fu">RUV4</span>(<span class="at">Y =</span> ad.clr, <span class="at">X =</span> ad.trt, <span class="at">ctl =</span> ad.nc, <span class="at">k =</span> ad.k) </span>
<span id="cb61-3"><a href="index.html#cb61-3" aria-hidden="true" tabindex="-1"></a>ad.ruv4.p <span class="ot">&lt;-</span> ad.ruv4<span class="sc">$</span>p</span>
<span id="cb61-4"><a href="index.html#cb61-4" aria-hidden="true" tabindex="-1"></a>ad.ruv4.p.adj <span class="ot">&lt;-</span> <span class="fu">p.adjust</span>(ad.ruv4.p, <span class="at">method =</span> <span class="st">&quot;fdr&quot;</span>)</span></code></pre></div>
<p>Note: A package named <span style="color: #FF7F00;">RUVSeq</span> has been developed for count data. It provides <code>RUVg()</code> using negative control variables, and also other functions <code>RUVs()</code> and <code>RUVr()</code> using sample replicates <span class="citation">(<a href="#ref-moskovicz2020skin" role="doc-biblioref">Moskovicz et al. 2020</a>)</span> or residuals from the regression on treatment effects to estimate and then account for latent batch effects. However, for CLR-transformed data, we still recommend the standard <span style="color: #FF7F00;">ruv</span> package.</p>
</div>
</div>
<div id="corrct" class="section level3 hasAnchor" number="1.6.2">
<h3><span class="header-section-number">1.6.2</span> Correcting for batch effects<a href="index.html#corrct" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The methods that we use to correct for batch effects include removeBatchEffect, ComBat, PLSDA-batch, sPLSDA-batch, Percentile Normalisation and RUVIII. Among them, RUVIII was designed for unknown batch effects. They were all applied to and illustrated with the <span style="color: #0000FF;">AD data</span>.</p>
<p><strong>removeBatchEffect</strong></p>
<p>The <code>removeBatchEffect()</code> function is implemented in <span style="color: #FF7F00;">limma</span> package. The design matrix (<code>design</code>) with treatment grouping information can be generated with <code>model.matrix()</code> function from <span style="color: #FF7F00;">stats</span> as shown in section <strong>accounting for batch effects</strong> method “SVA”.</p>
<p>Here we use <code>removeBatchEffect()</code> function with batch grouping information (<code>batch</code>) and treatment design matrix (<code>design</code>) to calculate batch effect corrected data <code>ad.rBE</code>.</p>
<div class="sourceCode" id="cb62"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb62-1"><a href="index.html#cb62-1" aria-hidden="true" tabindex="-1"></a>ad.rBE <span class="ot">&lt;-</span> <span class="fu">t</span>(<span class="fu">removeBatchEffect</span>(<span class="fu">t</span>(ad.clr), <span class="at">batch =</span> ad.batch, </span>
<span id="cb62-2"><a href="index.html#cb62-2" aria-hidden="true" tabindex="-1"></a>                              <span class="at">design =</span> ad.mod))</span></code></pre></div>
<p><strong>ComBat</strong></p>
<p>The <code>ComBat()</code> function (from <span style="color: #FF7F00;">sva</span> package) is implemented as parametric or non-parametric correction with option <code>par.prior</code>. Under a parametric adjustment, we can assess the model’s validity with <code>prior.plots = T</code> <span class="citation">(<a href="#ref-leek2012sva" role="doc-biblioref">Leek et al. 2012</a>)</span>.</p>
<p>Here we use a non-parametric correction (<code>par.prior = FALSE</code>) with batch grouping information (<code>batch</code>) and treatment design matrix (<code>mod</code>) to calculate batch effect corrected data <code>ad.ComBat</code>.</p>
<div class="sourceCode" id="cb63"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb63-1"><a href="index.html#cb63-1" aria-hidden="true" tabindex="-1"></a>ad.ComBat <span class="ot">&lt;-</span> <span class="fu">t</span>(<span class="fu">ComBat</span>(<span class="fu">t</span>(ad.clr), <span class="at">batch =</span> ad.batch, </span>
<span id="cb63-2"><a href="index.html#cb63-2" aria-hidden="true" tabindex="-1"></a>                      <span class="at">mod =</span> ad.mod, <span class="at">par.prior =</span> <span class="cn">FALSE</span>))</span></code></pre></div>
<p><strong>PLSDA-batch</strong></p>
<p>The <code>PLSDA_batch()</code> function is implemented in <span style="color: #FF7F00;">PLSDAbatch</span> package. To use this function, we need to specify the optimal number of components related to treatment (<code>ncomp.trt</code>) or batch effects (<code>ncomp.bat</code>).</p>
<p>Here in the <span style="color: #0000FF;">AD data</span>, we use <code>plsda()</code> from <span style="color: #FF7F00;">mixOmics</span> with only treatment grouping information to estimate the optimal number of treatment components to preserve.</p>
<div class="sourceCode" id="cb64"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb64-1"><a href="index.html#cb64-1" aria-hidden="true" tabindex="-1"></a><span class="co"># estimate the number of treatment components</span></span>
<span id="cb64-2"><a href="index.html#cb64-2" aria-hidden="true" tabindex="-1"></a>ad.trt.tune <span class="ot">&lt;-</span> <span class="fu">plsda</span>(<span class="at">X =</span> ad.clr, <span class="at">Y =</span> ad.trt, <span class="at">ncomp =</span> <span class="dv">5</span>)</span>
<span id="cb64-3"><a href="index.html#cb64-3" aria-hidden="true" tabindex="-1"></a>ad.trt.tune<span class="sc">$</span>prop_expl_var <span class="co">#1</span></span></code></pre></div>
<pre><code>## $X
##      comp1      comp2      comp3      comp4      comp5 
## 0.18619506 0.07873817 0.08257396 0.09263497 0.06594757 
## 
## $Y
##      comp1      comp2      comp3      comp4      comp5 
## 1.00000000 0.33857374 0.17315267 0.10551296 0.08185822</code></pre>
<p>We choose the number that explains 100% variance in the outcome matrix <code>Y</code>, thus from the result, 1 component was enough to preserve the treatment information.</p>
<p>We then use <code>PLSDA_batch()</code> function with both treatment and batch grouping information to estimate the optimal number of batch components to remove.</p>
<div class="sourceCode" id="cb66"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb66-1"><a href="index.html#cb66-1" aria-hidden="true" tabindex="-1"></a><span class="co"># estimate the number of batch components</span></span>
<span id="cb66-2"><a href="index.html#cb66-2" aria-hidden="true" tabindex="-1"></a>ad.batch.tune <span class="ot">&lt;-</span> <span class="fu">PLSDA_batch</span>(<span class="at">X =</span> ad.clr, </span>
<span id="cb66-3"><a href="index.html#cb66-3" aria-hidden="true" tabindex="-1"></a>                             <span class="at">Y.trt =</span> ad.trt, <span class="at">Y.bat =</span> ad.batch,</span>
<span id="cb66-4"><a href="index.html#cb66-4" aria-hidden="true" tabindex="-1"></a>                             <span class="at">ncomp.trt =</span> <span class="dv">1</span>, <span class="at">ncomp.bat =</span> <span class="dv">10</span>)</span>
<span id="cb66-5"><a href="index.html#cb66-5" aria-hidden="true" tabindex="-1"></a>ad.batch.tune<span class="sc">$</span>explained_variance.bat <span class="co">#4</span></span></code></pre></div>
<pre><code>## $X
##      comp1      comp2      comp3      comp4      comp5      comp6      comp7 
## 0.17470922 0.11481264 0.10122717 0.07507395 0.03988220 0.03636321 0.03540296 
##      comp8      comp9     comp10 
## 0.03024209 0.01751481 0.01158604 
## 
## $Y
##      comp1      comp2      comp3      comp4      comp5      comp6      comp7 
## 0.24746537 0.26157408 0.23013824 0.26082231 0.23017979 0.26074379 0.09299897 
##      comp8      comp9     comp10 
## 0.15539874 0.26067870 0.23017211</code></pre>
<div class="sourceCode" id="cb68"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb68-1"><a href="index.html#cb68-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sum</span>(ad.batch.tune<span class="sc">$</span>explained_variance.bat<span class="sc">$</span>Y[<span class="fu">seq_len</span>(<span class="dv">4</span>)])</span></code></pre></div>
<pre><code>## [1] 1</code></pre>
<p>Using the same criterion as choosing treatment components, we choose the number of batch components that explains 100% variance in the outcome matrix of batch. According to the result, 4 components were required to remove batch effects.</p>
<p>We then can correct for batch effects applying <code>PLSDA_batch()</code> with treatment, batch grouping information and corresponding optimal number of related components.</p>
<div class="sourceCode" id="cb70"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb70-1"><a href="index.html#cb70-1" aria-hidden="true" tabindex="-1"></a>ad.PLSDA_batch.res <span class="ot">&lt;-</span> <span class="fu">PLSDA_batch</span>(<span class="at">X =</span> ad.clr, </span>
<span id="cb70-2"><a href="index.html#cb70-2" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">Y.trt =</span> ad.trt, <span class="at">Y.bat =</span> ad.batch,</span>
<span id="cb70-3"><a href="index.html#cb70-3" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">ncomp.trt =</span> <span class="dv">1</span>, <span class="at">ncomp.bat =</span> <span class="dv">4</span>)</span>
<span id="cb70-4"><a href="index.html#cb70-4" aria-hidden="true" tabindex="-1"></a>ad.PLSDA_batch <span class="ot">&lt;-</span> ad.PLSDA_batch.res<span class="sc">$</span>X.nobatch</span></code></pre></div>
<p><strong>sPLSDA-batch</strong></p>
<p>We apply sPLSDA-batch using the same function <code>PLSDA_batch()</code>, but we specify the number of variables to select on each component (usually only treatment-related components <code>keepX.trt</code>). To determine the optimal number of variables to select, we use <code>tune.splsda()</code> function from <span style="color: #FF7F00;">mixOmics</span> package <span class="citation">(<a href="#ref-rohart2017mixomics" role="doc-biblioref">Rohart et al. 2017</a>)</span> with all possible numbers of variables to select for each component (<code>test.keepX</code>).</p>
<div class="sourceCode" id="cb71"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb71-1"><a href="index.html#cb71-1" aria-hidden="true" tabindex="-1"></a><span class="co"># estimate the number of variables to select per treatment component</span></span>
<span id="cb71-2"><a href="index.html#cb71-2" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">777</span>)</span>
<span id="cb71-3"><a href="index.html#cb71-3" aria-hidden="true" tabindex="-1"></a>ad.test.keepX <span class="ot">=</span> <span class="fu">c</span>(<span class="fu">seq</span>(<span class="dv">1</span>, <span class="dv">10</span>, <span class="dv">1</span>), <span class="fu">seq</span>(<span class="dv">20</span>, <span class="dv">100</span>, <span class="dv">10</span>), </span>
<span id="cb71-4"><a href="index.html#cb71-4" aria-hidden="true" tabindex="-1"></a>                  <span class="fu">seq</span>(<span class="dv">150</span>, <span class="dv">231</span>, <span class="dv">50</span>), <span class="dv">231</span>)</span>
<span id="cb71-5"><a href="index.html#cb71-5" aria-hidden="true" tabindex="-1"></a>ad.trt.tune.v <span class="ot">&lt;-</span> <span class="fu">tune.splsda</span>(<span class="at">X =</span> ad.clr, <span class="at">Y =</span> ad.trt, </span>
<span id="cb71-6"><a href="index.html#cb71-6" aria-hidden="true" tabindex="-1"></a>                             <span class="at">ncomp =</span> <span class="dv">1</span>, <span class="at">test.keepX =</span> ad.test.keepX, </span>
<span id="cb71-7"><a href="index.html#cb71-7" aria-hidden="true" tabindex="-1"></a>                             <span class="at">validation =</span> <span class="st">&#39;Mfold&#39;</span>, <span class="at">folds =</span> <span class="dv">4</span>, </span>
<span id="cb71-8"><a href="index.html#cb71-8" aria-hidden="true" tabindex="-1"></a>                             <span class="at">nrepeat =</span> <span class="dv">50</span>)</span>
<span id="cb71-9"><a href="index.html#cb71-9" aria-hidden="true" tabindex="-1"></a>ad.trt.tune.v<span class="sc">$</span>choice.keepX <span class="co">#100</span></span></code></pre></div>
<p>Here the optimal number of variables to select for the treatment component was 100. Since we have adjusted the amount of treatment variation to preserve, we need to re-choose the optimal number of components related to batch effects using the same criterion mentioned in section <strong>correcting for batch effects</strong> method “PLSDA-batch”.</p>
<div class="sourceCode" id="cb72"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb72-1"><a href="index.html#cb72-1" aria-hidden="true" tabindex="-1"></a><span class="co"># estimate the number of batch components</span></span>
<span id="cb72-2"><a href="index.html#cb72-2" aria-hidden="true" tabindex="-1"></a>ad.batch.tune <span class="ot">&lt;-</span> <span class="fu">PLSDA_batch</span>(<span class="at">X =</span> ad.clr, </span>
<span id="cb72-3"><a href="index.html#cb72-3" aria-hidden="true" tabindex="-1"></a>                             <span class="at">Y.trt =</span> ad.trt, <span class="at">Y.bat =</span> ad.batch,</span>
<span id="cb72-4"><a href="index.html#cb72-4" aria-hidden="true" tabindex="-1"></a>                             <span class="at">ncomp.trt =</span> <span class="dv">1</span>, <span class="at">keepX.trt =</span> <span class="dv">100</span>,</span>
<span id="cb72-5"><a href="index.html#cb72-5" aria-hidden="true" tabindex="-1"></a>                             <span class="at">ncomp.bat =</span> <span class="dv">10</span>)</span>
<span id="cb72-6"><a href="index.html#cb72-6" aria-hidden="true" tabindex="-1"></a>ad.batch.tune<span class="sc">$</span>explained_variance.bat <span class="co">#4</span></span></code></pre></div>
<pre><code>## $X
##      comp1      comp2      comp3      comp4      comp5      comp6      comp7 
## 0.17420018 0.11477097 0.09813477 0.07894965 0.03532822 0.03830648 0.03525225 
##      comp8      comp9     comp10 
## 0.02862004 0.02061328 0.01738354 
## 
## $Y
##      comp1      comp2      comp3      comp4      comp5      comp6      comp7 
## 0.24747749 0.26067155 0.23010809 0.26174286 0.25429273 0.23772308 0.24766631 
##      comp8      comp9     comp10 
## 0.18179443 0.07852346 0.24608017</code></pre>
<div class="sourceCode" id="cb74"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb74-1"><a href="index.html#cb74-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sum</span>(ad.batch.tune<span class="sc">$</span>explained_variance.bat<span class="sc">$</span>Y[<span class="fu">seq_len</span>(<span class="dv">4</span>)])</span></code></pre></div>
<pre><code>## [1] 1</code></pre>
<p>According to the result, we needed 4 batch related components to remove batch variance from the data with function <code>PLSDA_batch()</code>.</p>
<div class="sourceCode" id="cb76"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb76-1"><a href="index.html#cb76-1" aria-hidden="true" tabindex="-1"></a>ad.sPLSDA_batch.res <span class="ot">&lt;-</span> <span class="fu">PLSDA_batch</span>(<span class="at">X =</span> ad.clr, </span>
<span id="cb76-2"><a href="index.html#cb76-2" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">Y.trt =</span> ad.trt, <span class="at">Y.bat =</span> ad.batch,</span>
<span id="cb76-3"><a href="index.html#cb76-3" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">ncomp.trt =</span> <span class="dv">1</span>, <span class="at">keepX.trt =</span> <span class="dv">100</span>,</span>
<span id="cb76-4"><a href="index.html#cb76-4" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">ncomp.bat =</span> <span class="dv">4</span>)</span>
<span id="cb76-5"><a href="index.html#cb76-5" aria-hidden="true" tabindex="-1"></a>ad.sPLSDA_batch <span class="ot">&lt;-</span> ad.sPLSDA_batch.res<span class="sc">$</span>X.nobatch</span></code></pre></div>
<p>Note: for unbalanced batch x treatment design (with the exception of the nested design), we can specify <code>balance = FALSE</code> in <code>PLSDA_batch()</code> function to apply weighted PLSDA-batch.</p>
<p><strong>Percentile Normalisation</strong></p>
<p>To apply <code>percentile_norm()</code> function from <span style="color: #FF7F00;">PLSDAbatch</span> package, we need to indicate a control group (<code>ctrl.grp</code>).</p>
<div class="sourceCode" id="cb77"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb77-1"><a href="index.html#cb77-1" aria-hidden="true" tabindex="-1"></a>ad.PN <span class="ot">&lt;-</span> <span class="fu">percentile_norm</span>(<span class="at">data =</span> ad.clr, <span class="at">batch =</span> ad.batch, </span>
<span id="cb77-2"><a href="index.html#cb77-2" aria-hidden="true" tabindex="-1"></a>                         <span class="at">trt =</span> ad.trt, <span class="at">ctrl.grp =</span> <span class="st">&#39;0-0.5&#39;</span>)</span></code></pre></div>
<p><strong>RUVIII</strong></p>
<p>The <code>RUVIII()</code> function is from <span style="color: #FF7F00;">ruv</span> package. Similar to <code>RUV4()</code>, we use empirical negative control variables and <code>getK()</code> to determine the number of batch variables (<code>k</code>) to estimate. We also need sample replicates which should be structured into a mapping matrix using <code>replicate.matrix()</code>. We can then obtain the batch effect corrected data applying <code>RUVIII()</code> with above elements.</p>
<div class="sourceCode" id="cb78"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb78-1"><a href="index.html#cb78-1" aria-hidden="true" tabindex="-1"></a>ad.replicates <span class="ot">&lt;-</span> ad.metadata<span class="sc">$</span>sample_name.data.extraction</span>
<span id="cb78-2"><a href="index.html#cb78-2" aria-hidden="true" tabindex="-1"></a>ad.replicates.matrix <span class="ot">&lt;-</span> <span class="fu">replicate.matrix</span>(ad.replicates)</span>
<span id="cb78-3"><a href="index.html#cb78-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-4"><a href="index.html#cb78-4" aria-hidden="true" tabindex="-1"></a>ad.RUVIII <span class="ot">&lt;-</span> <span class="fu">RUVIII</span>(<span class="at">Y =</span> ad.clr, <span class="at">M =</span> ad.replicates.matrix, </span>
<span id="cb78-5"><a href="index.html#cb78-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">ctl =</span> ad.nc, <span class="at">k =</span> ad.k)</span>
<span id="cb78-6"><a href="index.html#cb78-6" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(ad.RUVIII) <span class="ot">&lt;-</span> <span class="fu">rownames</span>(ad.clr)</span></code></pre></div>
</div>
</div>
<div id="assessing-batch-effect-correction" class="section level2 hasAnchor" number="1.7">
<h2><span class="header-section-number">1.7</span> Assessing batch effect correction<a href="index.html#assessing-batch-effect-correction" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We apply different visualisation and quantitative methods to assessing batch effect correction.</p>
<div id="methods-that-detect-batch-effects" class="section level3 hasAnchor" number="1.7.1">
<h3><span class="header-section-number">1.7.1</span> Methods that detect batch effects<a href="index.html#methods-that-detect-batch-effects" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><strong>PCA</strong></p>
<p>In the <span style="color: #0000FF;">AD data</span>, we compared the PCA sample plots before and after batch effect correction with different methods.</p>
<div class="sourceCode" id="cb79"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb79-1"><a href="index.html#cb79-1" aria-hidden="true" tabindex="-1"></a>ad.pca.before <span class="ot">&lt;-</span> <span class="fu">pca</span>(ad.clr, <span class="at">ncomp =</span> <span class="dv">3</span>, <span class="at">scale =</span> <span class="cn">TRUE</span>)</span>
<span id="cb79-2"><a href="index.html#cb79-2" aria-hidden="true" tabindex="-1"></a>ad.pca.rBE <span class="ot">&lt;-</span> <span class="fu">pca</span>(ad.rBE, <span class="at">ncomp =</span> <span class="dv">3</span>, <span class="at">scale =</span> <span class="cn">TRUE</span>)</span>
<span id="cb79-3"><a href="index.html#cb79-3" aria-hidden="true" tabindex="-1"></a>ad.pca.ComBat <span class="ot">&lt;-</span> <span class="fu">pca</span>(ad.ComBat, <span class="at">ncomp =</span> <span class="dv">3</span>, <span class="at">scale =</span> <span class="cn">TRUE</span>)</span>
<span id="cb79-4"><a href="index.html#cb79-4" aria-hidden="true" tabindex="-1"></a>ad.pca.PLSDA_batch <span class="ot">&lt;-</span> <span class="fu">pca</span>(ad.PLSDA_batch, <span class="at">ncomp =</span> <span class="dv">3</span>, <span class="at">scale =</span> <span class="cn">TRUE</span>)</span>
<span id="cb79-5"><a href="index.html#cb79-5" aria-hidden="true" tabindex="-1"></a>ad.pca.sPLSDA_batch <span class="ot">&lt;-</span> <span class="fu">pca</span>(ad.sPLSDA_batch, <span class="at">ncomp =</span> <span class="dv">3</span>, <span class="at">scale =</span> <span class="cn">TRUE</span>)</span>
<span id="cb79-6"><a href="index.html#cb79-6" aria-hidden="true" tabindex="-1"></a>ad.pca.PN <span class="ot">&lt;-</span> <span class="fu">pca</span>(ad.PN, <span class="at">ncomp =</span> <span class="dv">3</span>, <span class="at">scale =</span> <span class="cn">TRUE</span>)</span>
<span id="cb79-7"><a href="index.html#cb79-7" aria-hidden="true" tabindex="-1"></a>ad.pca.RUVIII <span class="ot">&lt;-</span> <span class="fu">pca</span>(ad.RUVIII, <span class="at">ncomp =</span> <span class="dv">3</span>, <span class="at">scale =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<div class="sourceCode" id="cb80"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb80-1"><a href="index.html#cb80-1" aria-hidden="true" tabindex="-1"></a><span class="co"># order batches</span></span>
<span id="cb80-2"><a href="index.html#cb80-2" aria-hidden="true" tabindex="-1"></a>ad.batch <span class="ot">=</span> <span class="fu">factor</span>(ad.metadata<span class="sc">$</span>sequencing_run_date, </span>
<span id="cb80-3"><a href="index.html#cb80-3" aria-hidden="true" tabindex="-1"></a>                  <span class="at">levels =</span> <span class="fu">unique</span>(ad.metadata<span class="sc">$</span>sequencing_run_date))</span>
<span id="cb80-4"><a href="index.html#cb80-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb80-5"><a href="index.html#cb80-5" aria-hidden="true" tabindex="-1"></a>ad.pca.before.plot <span class="ot">&lt;-</span> <span class="fu">Scatter_Density</span>(<span class="at">object =</span> ad.pca.before, </span>
<span id="cb80-6"><a href="index.html#cb80-6" aria-hidden="true" tabindex="-1"></a>                                      <span class="at">batch =</span> ad.batch, </span>
<span id="cb80-7"><a href="index.html#cb80-7" aria-hidden="true" tabindex="-1"></a>                                      <span class="at">trt =</span> ad.trt, </span>
<span id="cb80-8"><a href="index.html#cb80-8" aria-hidden="true" tabindex="-1"></a>                                      <span class="at">title =</span> <span class="st">&#39;Before correction&#39;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb81"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb81-1"><a href="index.html#cb81-1" aria-hidden="true" tabindex="-1"></a>ad.pca.rBE.plot <span class="ot">&lt;-</span> <span class="fu">Scatter_Density</span>(<span class="at">object =</span> ad.pca.rBE, </span>
<span id="cb81-2"><a href="index.html#cb81-2" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">batch =</span> ad.batch, </span>
<span id="cb81-3"><a href="index.html#cb81-3" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">trt =</span> ad.trt, </span>
<span id="cb81-4"><a href="index.html#cb81-4" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">title =</span> <span class="st">&#39;removeBatchEffect&#39;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb82"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb82-1"><a href="index.html#cb82-1" aria-hidden="true" tabindex="-1"></a>ad.pca.ComBat.plot <span class="ot">&lt;-</span> <span class="fu">Scatter_Density</span>(<span class="at">object =</span> ad.pca.ComBat, </span>
<span id="cb82-2"><a href="index.html#cb82-2" aria-hidden="true" tabindex="-1"></a>                                      <span class="at">batch =</span> ad.batch, </span>
<span id="cb82-3"><a href="index.html#cb82-3" aria-hidden="true" tabindex="-1"></a>                                      <span class="at">trt =</span> ad.trt, </span>
<span id="cb82-4"><a href="index.html#cb82-4" aria-hidden="true" tabindex="-1"></a>                                      <span class="at">title =</span> <span class="st">&#39;ComBat&#39;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb83"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb83-1"><a href="index.html#cb83-1" aria-hidden="true" tabindex="-1"></a>ad.pca.PLSDA_batch.plot <span class="ot">&lt;-</span> <span class="fu">Scatter_Density</span>(<span class="at">object =</span> ad.pca.PLSDA_batch, </span>
<span id="cb83-2"><a href="index.html#cb83-2" aria-hidden="true" tabindex="-1"></a>                                           <span class="at">batch =</span> ad.batch, </span>
<span id="cb83-3"><a href="index.html#cb83-3" aria-hidden="true" tabindex="-1"></a>                                           <span class="at">trt =</span> ad.trt, </span>
<span id="cb83-4"><a href="index.html#cb83-4" aria-hidden="true" tabindex="-1"></a>                                           <span class="at">title =</span> <span class="st">&#39;PLSDA-batch&#39;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb84"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb84-1"><a href="index.html#cb84-1" aria-hidden="true" tabindex="-1"></a>ad.pca.sPLSDA_batch.plot <span class="ot">&lt;-</span> <span class="fu">Scatter_Density</span>(<span class="at">object =</span> ad.pca.sPLSDA_batch, </span>
<span id="cb84-2"><a href="index.html#cb84-2" aria-hidden="true" tabindex="-1"></a>                                            <span class="at">batch =</span> ad.batch, </span>
<span id="cb84-3"><a href="index.html#cb84-3" aria-hidden="true" tabindex="-1"></a>                                            <span class="at">trt =</span> ad.trt, </span>
<span id="cb84-4"><a href="index.html#cb84-4" aria-hidden="true" tabindex="-1"></a>                                            <span class="at">title =</span> <span class="st">&#39;sPLSDA-batch&#39;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb85"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb85-1"><a href="index.html#cb85-1" aria-hidden="true" tabindex="-1"></a>ad.pca.PN.plot <span class="ot">&lt;-</span> <span class="fu">Scatter_Density</span>(<span class="at">object =</span> ad.pca.PN, </span>
<span id="cb85-2"><a href="index.html#cb85-2" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">batch =</span> ad.batch, </span>
<span id="cb85-3"><a href="index.html#cb85-3" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">trt =</span> ad.trt, </span>
<span id="cb85-4"><a href="index.html#cb85-4" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">title =</span> <span class="st">&#39;Percentile Normalisation&#39;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb86"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb86-1"><a href="index.html#cb86-1" aria-hidden="true" tabindex="-1"></a>ad.pca.RUVIII.plot <span class="ot">&lt;-</span> <span class="fu">Scatter_Density</span>(<span class="at">object =</span> ad.pca.RUVIII, </span>
<span id="cb86-2"><a href="index.html#cb86-2" aria-hidden="true" tabindex="-1"></a>                                      <span class="at">batch =</span> ad.batch, </span>
<span id="cb86-3"><a href="index.html#cb86-3" aria-hidden="true" tabindex="-1"></a>                                      <span class="at">trt =</span> ad.trt, </span>
<span id="cb86-4"><a href="index.html#cb86-4" aria-hidden="true" tabindex="-1"></a>                                      <span class="at">title =</span> <span class="st">&#39;RUVIII&#39;</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:ADpca"></span>
<img src="PLSDAbatch_workflow_files/figure-html/ADpca-1.png" alt="The PCA sample plots with densities before and after batch effect correction in the AD data." width="100%" />
<p class="caption">
Figure 1.7: The PCA sample plots with densities before and after batch effect correction in the AD data.
</p>
</div>
<p>As shown in the PCA sample plots, the differences between the samples sequenced at “14/04/2016”, “21/09/2017” and the other dates were removed after batch effect correction with most methods except percentile normalisation. The data corrected with PLSDA-batch included more treatment variation mostly on the first PC than other method-corrected data, as indicated on the x-axis label (26%). We can also compare the boxplots and density plots for key variables identified in PCA driving the major variance or heatmaps showing obvious patterns before and after batch effect correction (results not shown).</p>
<p><strong>pRDA</strong></p>
<p>We calculate the global explained variance across all microbial variables using pRDA. To achieve this, we create a loop for each variable from the original (uncorrected) and batch effect-corrected data. The final results are then displayed with <code>partVar_plot()</code> from <span style="color: #FF7F00;">PLSDAbatch</span> package.</p>
<div class="sourceCode" id="cb87"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb87-1"><a href="index.html#cb87-1" aria-hidden="true" tabindex="-1"></a><span class="co"># AD data</span></span>
<span id="cb87-2"><a href="index.html#cb87-2" aria-hidden="true" tabindex="-1"></a>ad.corrected.list <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="st">`</span><span class="at">Before correction</span><span class="st">`</span> <span class="ot">=</span> ad.clr, </span>
<span id="cb87-3"><a href="index.html#cb87-3" aria-hidden="true" tabindex="-1"></a>                          <span class="at">removeBatchEffect =</span> ad.rBE, </span>
<span id="cb87-4"><a href="index.html#cb87-4" aria-hidden="true" tabindex="-1"></a>                          <span class="at">ComBat =</span> ad.ComBat, </span>
<span id="cb87-5"><a href="index.html#cb87-5" aria-hidden="true" tabindex="-1"></a>                          <span class="st">`</span><span class="at">PLSDA-batch</span><span class="st">`</span> <span class="ot">=</span> ad.PLSDA_batch, </span>
<span id="cb87-6"><a href="index.html#cb87-6" aria-hidden="true" tabindex="-1"></a>                          <span class="st">`</span><span class="at">sPLSDA-batch</span><span class="st">`</span> <span class="ot">=</span> ad.sPLSDA_batch, </span>
<span id="cb87-7"><a href="index.html#cb87-7" aria-hidden="true" tabindex="-1"></a>                          <span class="st">`</span><span class="at">Percentile Normalisation</span><span class="st">`</span> <span class="ot">=</span> ad.PN,</span>
<span id="cb87-8"><a href="index.html#cb87-8" aria-hidden="true" tabindex="-1"></a>                          <span class="at">RUVIII =</span> ad.RUVIII)</span>
<span id="cb87-9"><a href="index.html#cb87-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb87-10"><a href="index.html#cb87-10" aria-hidden="true" tabindex="-1"></a>ad.prop.df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">Treatment =</span> <span class="cn">NA</span>, <span class="at">Batch =</span> <span class="cn">NA</span>, </span>
<span id="cb87-11"><a href="index.html#cb87-11" aria-hidden="true" tabindex="-1"></a>                         <span class="at">Intersection =</span> <span class="cn">NA</span>, </span>
<span id="cb87-12"><a href="index.html#cb87-12" aria-hidden="true" tabindex="-1"></a>                         <span class="at">Residuals =</span> <span class="cn">NA</span>) </span>
<span id="cb87-13"><a href="index.html#cb87-13" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="fu">seq_len</span>(<span class="fu">length</span>(ad.corrected.list))){</span>
<span id="cb87-14"><a href="index.html#cb87-14" aria-hidden="true" tabindex="-1"></a>  rda.res <span class="ot">=</span> <span class="fu">varpart</span>(ad.corrected.list[[i]], <span class="sc">~</span> trt, <span class="sc">~</span> batch,</span>
<span id="cb87-15"><a href="index.html#cb87-15" aria-hidden="true" tabindex="-1"></a>                    <span class="at">data =</span> ad.factors.df, <span class="at">scale =</span> <span class="cn">TRUE</span>)</span>
<span id="cb87-16"><a href="index.html#cb87-16" aria-hidden="true" tabindex="-1"></a>  ad.prop.df[i, ] <span class="ot">&lt;-</span> rda.res<span class="sc">$</span>part<span class="sc">$</span>indfract<span class="sc">$</span>Adj.R.squared}</span>
<span id="cb87-17"><a href="index.html#cb87-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb87-18"><a href="index.html#cb87-18" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(ad.prop.df) <span class="ot">=</span> <span class="fu">names</span>(ad.corrected.list)</span>
<span id="cb87-19"><a href="index.html#cb87-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb87-20"><a href="index.html#cb87-20" aria-hidden="true" tabindex="-1"></a>ad.prop.df <span class="ot">&lt;-</span> ad.prop.df[, <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">3</span>,<span class="dv">2</span>,<span class="dv">4</span>)]</span>
<span id="cb87-21"><a href="index.html#cb87-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb87-22"><a href="index.html#cb87-22" aria-hidden="true" tabindex="-1"></a>ad.prop.df[ad.prop.df <span class="sc">&lt;</span> <span class="dv">0</span>] <span class="ot">=</span> <span class="dv">0</span></span>
<span id="cb87-23"><a href="index.html#cb87-23" aria-hidden="true" tabindex="-1"></a>ad.prop.df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">t</span>(<span class="fu">apply</span>(ad.prop.df, <span class="dv">1</span>, </span>
<span id="cb87-24"><a href="index.html#cb87-24" aria-hidden="true" tabindex="-1"></a>                                    <span class="cf">function</span>(x){x<span class="sc">/</span><span class="fu">sum</span>(x)})))</span>
<span id="cb87-25"><a href="index.html#cb87-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb87-26"><a href="index.html#cb87-26" aria-hidden="true" tabindex="-1"></a><span class="fu">partVar_plot</span>(<span class="at">prop.df =</span> ad.prop.df)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:ADprda"></span>
<img src="PLSDAbatch_workflow_files/figure-html/ADprda-1.png" alt="Global explained variance before and after batch effect correction for the AD data." width="60%" />
<p class="caption">
Figure 1.8: Global explained variance before and after batch effect correction for the AD data.
</p>
</div>
<p>As shown in the above figure, the intersection between batch and treatment variance was small (1.3%) for the <span style="color: #0000FF;">AD data</span>, which implies that the batch x treatment design is not highly unbalanced. Thus the unweighted PLSDA-batch and sPLSDA-batch were still applicable, and thus the weighted versions were not used. sPLSDA-batch corrected data led to the best performance with a slightly higher proportion of treatment variance explained and undetectable batch and intersection variance compared to the other methods.</p>
<div class="sourceCode" id="cb88"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb88-1"><a href="index.html#cb88-1" aria-hidden="true" tabindex="-1"></a><span class="co"># HFHS data</span></span>
<span id="cb88-2"><a href="index.html#cb88-2" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="at">file =</span> <span class="st">&#39;./data/HFHS_data.rda&#39;</span>)</span>
<span id="cb88-3"><a href="index.html#cb88-3" aria-hidden="true" tabindex="-1"></a>hfhs.corrected.list <span class="ot">&lt;-</span> HFHS_data<span class="sc">$</span>CorrectData<span class="sc">$</span>data</span>
<span id="cb88-4"><a href="index.html#cb88-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb88-5"><a href="index.html#cb88-5" aria-hidden="true" tabindex="-1"></a>hfhs.trt <span class="ot">&lt;-</span> HFHS_data<span class="sc">$</span>CorrectData<span class="sc">$</span>Y.trt</span>
<span id="cb88-6"><a href="index.html#cb88-6" aria-hidden="true" tabindex="-1"></a>hfhs.batch <span class="ot">&lt;-</span> HFHS_data<span class="sc">$</span>CorrectData<span class="sc">$</span>Y.bat</span>
<span id="cb88-7"><a href="index.html#cb88-7" aria-hidden="true" tabindex="-1"></a>hfhs.factors.df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">trt =</span> hfhs.trt, <span class="at">batch =</span> hfhs.batch)</span>
<span id="cb88-8"><a href="index.html#cb88-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb88-9"><a href="index.html#cb88-9" aria-hidden="true" tabindex="-1"></a>hfhs.prop.df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">Treatment =</span> <span class="cn">NA</span>, <span class="at">Batch =</span> <span class="cn">NA</span>, </span>
<span id="cb88-10"><a href="index.html#cb88-10" aria-hidden="true" tabindex="-1"></a>                           <span class="at">Intersection =</span> <span class="cn">NA</span>, </span>
<span id="cb88-11"><a href="index.html#cb88-11" aria-hidden="true" tabindex="-1"></a>                           <span class="at">Residuals =</span> <span class="cn">NA</span>) </span>
<span id="cb88-12"><a href="index.html#cb88-12" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="fu">seq_len</span>(<span class="fu">length</span>(hfhs.corrected.list))){</span>
<span id="cb88-13"><a href="index.html#cb88-13" aria-hidden="true" tabindex="-1"></a>  rda.res <span class="ot">=</span> <span class="fu">varpart</span>(hfhs.corrected.list[[i]], <span class="sc">~</span> trt, <span class="sc">~</span> batch,</span>
<span id="cb88-14"><a href="index.html#cb88-14" aria-hidden="true" tabindex="-1"></a>                    <span class="at">data =</span> hfhs.factors.df, <span class="at">scale =</span> <span class="cn">TRUE</span>)</span>
<span id="cb88-15"><a href="index.html#cb88-15" aria-hidden="true" tabindex="-1"></a>  hfhs.prop.df[i, ] <span class="ot">&lt;-</span> rda.res<span class="sc">$</span>part<span class="sc">$</span>indfract<span class="sc">$</span>Adj.R.squared}</span>
<span id="cb88-16"><a href="index.html#cb88-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb88-17"><a href="index.html#cb88-17" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(hfhs.prop.df) <span class="ot">=</span> <span class="fu">names</span>(hfhs.corrected.list)</span>
<span id="cb88-18"><a href="index.html#cb88-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb88-19"><a href="index.html#cb88-19" aria-hidden="true" tabindex="-1"></a>hfhs.prop.df <span class="ot">&lt;-</span> hfhs.prop.df[, <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">3</span>,<span class="dv">2</span>,<span class="dv">4</span>)]</span>
<span id="cb88-20"><a href="index.html#cb88-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb88-21"><a href="index.html#cb88-21" aria-hidden="true" tabindex="-1"></a>hfhs.prop.df[hfhs.prop.df <span class="sc">&lt;</span> <span class="dv">0</span>] <span class="ot">=</span> <span class="dv">0</span></span>
<span id="cb88-22"><a href="index.html#cb88-22" aria-hidden="true" tabindex="-1"></a>hfhs.prop.df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">t</span>(<span class="fu">apply</span>(hfhs.prop.df, <span class="dv">1</span>, </span>
<span id="cb88-23"><a href="index.html#cb88-23" aria-hidden="true" tabindex="-1"></a>                                      <span class="cf">function</span>(x){x<span class="sc">/</span><span class="fu">sum</span>(x)})))</span>
<span id="cb88-24"><a href="index.html#cb88-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb88-25"><a href="index.html#cb88-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb88-26"><a href="index.html#cb88-26" aria-hidden="true" tabindex="-1"></a><span class="fu">partVar_plot</span>(<span class="at">prop.df =</span> hfhs.prop.df)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:HFHSprda"></span>
<img src="PLSDAbatch_workflow_files/figure-html/HFHSprda-1.png" alt="Global explained variance before and after batch effect correction for the HFHS data." width="60%" />
<p class="caption">
Figure 1.9: Global explained variance before and after batch effect correction for the HFHS data.
</p>
</div>
<p>As shown in the above figure, a small amount of batch variance was observed (3.6%) for the <span style="color: #808000;">HFHS data</span>. PLSDA-batch achieved the best performance for preserving the largest treatment variance and completely removing batch variance compared to the other methods. The results also indicate that PLSDA-batch is more appropriate for weak batch effects, while sPLSDA-batch is more appropriate for strong batch effects. The RUVIII performed better in the <span style="color: #808000;">HFHS data</span> than the <span style="color: #0000FF;">AD data</span> because the sample replicates might capture more batch variation than in the <span style="color: #0000FF;">AD data</span>. Indeed, the sample replicates in <span style="color: #808000;">HFHS data</span> are across different day batches, while the replicates in <span style="color: #0000FF;">AD data</span> do not exist in all batches. Therefore, sample replicates play a critical role in RUVIII.</p>
</div>
<div id="other-methods-1" class="section level3 hasAnchor" number="1.7.2">
<h3><span class="header-section-number">1.7.2</span> Other methods<a href="index.html#other-methods-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><span class="math inline">\(\mathbf{R^2}\)</span></p>
<p>The <span class="math inline">\(R^2\)</span> values for each variable are calculated with <code>lm()</code> from <span style="color: #FF7F00;">stats</span> package. To compare the <span class="math inline">\(R^2\)</span> values among variables, we scale the corrected data before <span class="math inline">\(R^2\)</span> calculation. The results are displayed with <code>ggplot()</code> from <span style="color: #FF7F00;">ggplot2</span> R package.</p>
<div class="sourceCode" id="cb89"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb89-1"><a href="index.html#cb89-1" aria-hidden="true" tabindex="-1"></a><span class="co"># AD data</span></span>
<span id="cb89-2"><a href="index.html#cb89-2" aria-hidden="true" tabindex="-1"></a><span class="co"># scale</span></span>
<span id="cb89-3"><a href="index.html#cb89-3" aria-hidden="true" tabindex="-1"></a>ad.corr_scale.list <span class="ot">&lt;-</span> <span class="fu">lapply</span>(ad.corrected.list, </span>
<span id="cb89-4"><a href="index.html#cb89-4" aria-hidden="true" tabindex="-1"></a>                             <span class="cf">function</span>(x){<span class="fu">apply</span>(x, <span class="dv">2</span>, scale)})</span>
<span id="cb89-5"><a href="index.html#cb89-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb89-6"><a href="index.html#cb89-6" aria-hidden="true" tabindex="-1"></a>ad.r_values.list <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb89-7"><a href="index.html#cb89-7" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="fu">seq_len</span>(<span class="fu">length</span>(ad.corr_scale.list))){</span>
<span id="cb89-8"><a href="index.html#cb89-8" aria-hidden="true" tabindex="-1"></a>  ad.r_values <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">trt =</span> <span class="cn">NA</span>, <span class="at">batch =</span> <span class="cn">NA</span>)</span>
<span id="cb89-9"><a href="index.html#cb89-9" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span>(c <span class="cf">in</span> <span class="fu">seq_len</span>(<span class="fu">ncol</span>(ad.corr_scale.list[[i]]))){</span>
<span id="cb89-10"><a href="index.html#cb89-10" aria-hidden="true" tabindex="-1"></a>    ad.fit.res.trt <span class="ot">&lt;-</span> <span class="fu">lm</span>(ad.corr_scale.list[[i]][,c] <span class="sc">~</span> ad.trt)</span>
<span id="cb89-11"><a href="index.html#cb89-11" aria-hidden="true" tabindex="-1"></a>    ad.r_values[c,<span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="fu">summary</span>(ad.fit.res.trt)<span class="sc">$</span>r.squared</span>
<span id="cb89-12"><a href="index.html#cb89-12" aria-hidden="true" tabindex="-1"></a>    ad.fit.res.batch <span class="ot">&lt;-</span> <span class="fu">lm</span>(ad.corr_scale.list[[i]][,c] <span class="sc">~</span> ad.batch)</span>
<span id="cb89-13"><a href="index.html#cb89-13" aria-hidden="true" tabindex="-1"></a>    ad.r_values[c,<span class="dv">2</span>] <span class="ot">&lt;-</span> <span class="fu">summary</span>(ad.fit.res.batch)<span class="sc">$</span>r.squared</span>
<span id="cb89-14"><a href="index.html#cb89-14" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb89-15"><a href="index.html#cb89-15" aria-hidden="true" tabindex="-1"></a>  ad.r_values.list[[i]] <span class="ot">&lt;-</span> ad.r_values</span>
<span id="cb89-16"><a href="index.html#cb89-16" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb89-17"><a href="index.html#cb89-17" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(ad.r_values.list) <span class="ot">&lt;-</span> <span class="fu">names</span>(ad.corr_scale.list)</span>
<span id="cb89-18"><a href="index.html#cb89-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb89-19"><a href="index.html#cb89-19" aria-hidden="true" tabindex="-1"></a>ad.boxp.list <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb89-20"><a href="index.html#cb89-20" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="fu">seq_len</span>(<span class="fu">length</span>(ad.r_values.list))){</span>
<span id="cb89-21"><a href="index.html#cb89-21" aria-hidden="true" tabindex="-1"></a>  ad.boxp.list[[i]] <span class="ot">&lt;-</span> </span>
<span id="cb89-22"><a href="index.html#cb89-22" aria-hidden="true" tabindex="-1"></a>    <span class="fu">data.frame</span>(<span class="at">r2 =</span> <span class="fu">c</span>(ad.r_values.list[[i]][ ,<span class="st">&#39;trt&#39;</span>],</span>
<span id="cb89-23"><a href="index.html#cb89-23" aria-hidden="true" tabindex="-1"></a>                      ad.r_values.list[[i]][ ,<span class="st">&#39;batch&#39;</span>]), </span>
<span id="cb89-24"><a href="index.html#cb89-24" aria-hidden="true" tabindex="-1"></a>               <span class="at">Effects =</span> <span class="fu">as.factor</span>(<span class="fu">rep</span>(<span class="fu">c</span>(<span class="st">&#39;Treatment&#39;</span>,<span class="st">&#39;Batch&#39;</span>), </span>
<span id="cb89-25"><a href="index.html#cb89-25" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">each =</span> <span class="dv">231</span>)))</span>
<span id="cb89-26"><a href="index.html#cb89-26" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb89-27"><a href="index.html#cb89-27" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(ad.boxp.list) <span class="ot">&lt;-</span> <span class="fu">names</span>(ad.r_values.list)</span>
<span id="cb89-28"><a href="index.html#cb89-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb89-29"><a href="index.html#cb89-29" aria-hidden="true" tabindex="-1"></a>ad.r2.boxp <span class="ot">&lt;-</span> <span class="fu">rbind</span>(ad.boxp.list<span class="sc">$</span><span class="st">`</span><span class="at">Before correction</span><span class="st">`</span>,</span>
<span id="cb89-30"><a href="index.html#cb89-30" aria-hidden="true" tabindex="-1"></a>                    ad.boxp.list<span class="sc">$</span>removeBatchEffect,</span>
<span id="cb89-31"><a href="index.html#cb89-31" aria-hidden="true" tabindex="-1"></a>                    ad.boxp.list<span class="sc">$</span>ComBat,</span>
<span id="cb89-32"><a href="index.html#cb89-32" aria-hidden="true" tabindex="-1"></a>                    ad.boxp.list<span class="sc">$</span><span class="st">`</span><span class="at">PLSDA-batch</span><span class="st">`</span>,</span>
<span id="cb89-33"><a href="index.html#cb89-33" aria-hidden="true" tabindex="-1"></a>                    ad.boxp.list<span class="sc">$</span><span class="st">`</span><span class="at">sPLSDA-batch</span><span class="st">`</span>,</span>
<span id="cb89-34"><a href="index.html#cb89-34" aria-hidden="true" tabindex="-1"></a>                    ad.boxp.list<span class="sc">$</span><span class="st">`</span><span class="at">Percentile Normalisation</span><span class="st">`</span>,</span>
<span id="cb89-35"><a href="index.html#cb89-35" aria-hidden="true" tabindex="-1"></a>                    ad.boxp.list<span class="sc">$</span>RUVIII)</span>
<span id="cb89-36"><a href="index.html#cb89-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb89-37"><a href="index.html#cb89-37" aria-hidden="true" tabindex="-1"></a>ad.r2.boxp<span class="sc">$</span>methods <span class="ot">&lt;-</span> <span class="fu">rep</span>(<span class="fu">c</span>(<span class="st">&#39;Before correction&#39;</span>, <span class="st">&#39; removeBatchEffect&#39;</span>, </span>
<span id="cb89-38"><a href="index.html#cb89-38" aria-hidden="true" tabindex="-1"></a>                            <span class="st">&#39;ComBat&#39;</span>,<span class="st">&#39;PLSDA-batch&#39;</span>, <span class="st">&#39;sPLSDA-batch&#39;</span>,</span>
<span id="cb89-39"><a href="index.html#cb89-39" aria-hidden="true" tabindex="-1"></a>                            <span class="st">&#39;Percentile Normalisation&#39;</span>, <span class="st">&#39;RUVIII&#39;</span>), <span class="at">each =</span> <span class="dv">462</span>)</span>
<span id="cb89-40"><a href="index.html#cb89-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb89-41"><a href="index.html#cb89-41" aria-hidden="true" tabindex="-1"></a>ad.r2.boxp<span class="sc">$</span>methods <span class="ot">&lt;-</span> <span class="fu">factor</span>(ad.r2.boxp<span class="sc">$</span>methods, </span>
<span id="cb89-42"><a href="index.html#cb89-42" aria-hidden="true" tabindex="-1"></a>                             <span class="at">levels =</span> <span class="fu">unique</span>(ad.r2.boxp<span class="sc">$</span>methods))</span>
<span id="cb89-43"><a href="index.html#cb89-43" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb89-44"><a href="index.html#cb89-44" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(ad.r2.boxp, <span class="fu">aes</span>(<span class="at">x =</span> Effects, <span class="at">y =</span> r2, <span class="at">fill =</span> Effects)) <span class="sc">+</span></span>
<span id="cb89-45"><a href="index.html#cb89-45" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>(<span class="at">alpha =</span> <span class="fl">0.80</span>) <span class="sc">+</span></span>
<span id="cb89-46"><a href="index.html#cb89-46" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span> </span>
<span id="cb89-47"><a href="index.html#cb89-47" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">18</span>),</span>
<span id="cb89-48"><a href="index.html#cb89-48" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.title.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb89-49"><a href="index.html#cb89-49" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.title.y =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb89-50"><a href="index.html#cb89-50" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">60</span>, <span class="at">hjust =</span> <span class="dv">1</span>, <span class="at">size =</span> <span class="dv">18</span>),</span>
<span id="cb89-51"><a href="index.html#cb89-51" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.text.y =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">18</span>),</span>
<span id="cb89-52"><a href="index.html#cb89-52" aria-hidden="true" tabindex="-1"></a>        <span class="at">panel.grid.minor.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb89-53"><a href="index.html#cb89-53" aria-hidden="true" tabindex="-1"></a>        <span class="at">panel.grid.major.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb89-54"><a href="index.html#cb89-54" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.position =</span> <span class="st">&quot;right&quot;</span>) <span class="sc">+</span> <span class="fu">facet_grid</span>( <span class="sc">~</span> methods) <span class="sc">+</span> </span>
<span id="cb89-55"><a href="index.html#cb89-55" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values=</span><span class="fu">pb_color</span>(<span class="fu">c</span>(<span class="dv">12</span>,<span class="dv">14</span>))) </span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:ADr21"></span>
<img src="PLSDAbatch_workflow_files/figure-html/ADr21-1.png" alt="AD study: $R^2$ values for each microbial variable before and after batch effect correction." width="100%" />
<p class="caption">
Figure 1.10: AD study: <span class="math inline">\(R^2\)</span> values for each microbial variable before and after batch effect correction.
</p>
</div>
<p>The corrected data from ComBat still included a few variables with a large proportion of batch variance. A large number of variables from the data corrected by percentile normalisation and RUVIII still included considerable batch variance, especially RUVIII, which resulting data contained more proportion of batch variance than treatment variance.</p>
<div class="sourceCode" id="cb90"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb90-1"><a href="index.html#cb90-1" aria-hidden="true" tabindex="-1"></a><span class="do">##################################</span></span>
<span id="cb90-2"><a href="index.html#cb90-2" aria-hidden="true" tabindex="-1"></a>ad.barp.list <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb90-3"><a href="index.html#cb90-3" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="fu">seq_len</span>(<span class="fu">length</span>(ad.r_values.list))){</span>
<span id="cb90-4"><a href="index.html#cb90-4" aria-hidden="true" tabindex="-1"></a>  ad.barp.list[[i]] <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">r2 =</span> <span class="fu">c</span>(<span class="fu">sum</span>(ad.r_values.list[[i]][ ,<span class="st">&#39;trt&#39;</span>]),</span>
<span id="cb90-5"><a href="index.html#cb90-5" aria-hidden="true" tabindex="-1"></a>                                         <span class="fu">sum</span>(ad.r_values.list[[i]][ ,<span class="st">&#39;batch&#39;</span>])), </span>
<span id="cb90-6"><a href="index.html#cb90-6" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">Effects =</span> <span class="fu">c</span>(<span class="st">&#39;Treatment&#39;</span>,<span class="st">&#39;Batch&#39;</span>))</span>
<span id="cb90-7"><a href="index.html#cb90-7" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb90-8"><a href="index.html#cb90-8" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(ad.barp.list) <span class="ot">&lt;-</span> <span class="fu">names</span>(ad.r_values.list)</span>
<span id="cb90-9"><a href="index.html#cb90-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-10"><a href="index.html#cb90-10" aria-hidden="true" tabindex="-1"></a>ad.r2.barp <span class="ot">&lt;-</span> <span class="fu">rbind</span>(ad.barp.list<span class="sc">$</span><span class="st">`</span><span class="at">Before correction</span><span class="st">`</span>,</span>
<span id="cb90-11"><a href="index.html#cb90-11" aria-hidden="true" tabindex="-1"></a>                    ad.barp.list<span class="sc">$</span>removeBatchEffect,</span>
<span id="cb90-12"><a href="index.html#cb90-12" aria-hidden="true" tabindex="-1"></a>                    ad.barp.list<span class="sc">$</span>ComBat,</span>
<span id="cb90-13"><a href="index.html#cb90-13" aria-hidden="true" tabindex="-1"></a>                    ad.barp.list<span class="sc">$</span><span class="st">`</span><span class="at">PLSDA-batch</span><span class="st">`</span>,</span>
<span id="cb90-14"><a href="index.html#cb90-14" aria-hidden="true" tabindex="-1"></a>                    ad.barp.list<span class="sc">$</span><span class="st">`</span><span class="at">sPLSDA-batch</span><span class="st">`</span>,</span>
<span id="cb90-15"><a href="index.html#cb90-15" aria-hidden="true" tabindex="-1"></a>                    ad.barp.list<span class="sc">$</span><span class="st">`</span><span class="at">Percentile Normalisation</span><span class="st">`</span>,</span>
<span id="cb90-16"><a href="index.html#cb90-16" aria-hidden="true" tabindex="-1"></a>                    ad.barp.list<span class="sc">$</span>RUVIII)</span>
<span id="cb90-17"><a href="index.html#cb90-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-18"><a href="index.html#cb90-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-19"><a href="index.html#cb90-19" aria-hidden="true" tabindex="-1"></a>ad.r2.barp<span class="sc">$</span>methods <span class="ot">&lt;-</span> <span class="fu">rep</span>(<span class="fu">c</span>(<span class="st">&#39;Before correction&#39;</span>, <span class="st">&#39; removeBatchEffect&#39;</span>, </span>
<span id="cb90-20"><a href="index.html#cb90-20" aria-hidden="true" tabindex="-1"></a>                            <span class="st">&#39;ComBat&#39;</span>,<span class="st">&#39;PLSDA-batch&#39;</span>, <span class="st">&#39;sPLSDA-batch&#39;</span>,</span>
<span id="cb90-21"><a href="index.html#cb90-21" aria-hidden="true" tabindex="-1"></a>                            <span class="st">&#39;Percentile Normalisation&#39;</span>, <span class="st">&#39;RUVIII&#39;</span>), <span class="at">each =</span> <span class="dv">2</span>)</span>
<span id="cb90-22"><a href="index.html#cb90-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-23"><a href="index.html#cb90-23" aria-hidden="true" tabindex="-1"></a>ad.r2.barp<span class="sc">$</span>methods <span class="ot">&lt;-</span> <span class="fu">factor</span>(ad.r2.barp<span class="sc">$</span>methods, </span>
<span id="cb90-24"><a href="index.html#cb90-24" aria-hidden="true" tabindex="-1"></a>                             <span class="at">levels =</span> <span class="fu">unique</span>(ad.r2.barp<span class="sc">$</span>methods))</span>
<span id="cb90-25"><a href="index.html#cb90-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-26"><a href="index.html#cb90-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-27"><a href="index.html#cb90-27" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(ad.r2.barp, <span class="fu">aes</span>(<span class="at">x =</span> Effects, <span class="at">y =</span> r2, <span class="at">fill =</span> Effects)) <span class="sc">+</span></span>
<span id="cb90-28"><a href="index.html#cb90-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">stat=</span><span class="st">&quot;identity&quot;</span>) <span class="sc">+</span> </span>
<span id="cb90-29"><a href="index.html#cb90-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span> </span>
<span id="cb90-30"><a href="index.html#cb90-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">18</span>),</span>
<span id="cb90-31"><a href="index.html#cb90-31" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.title.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb90-32"><a href="index.html#cb90-32" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.title.y =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb90-33"><a href="index.html#cb90-33" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">60</span>, <span class="at">hjust =</span> <span class="dv">1</span>, <span class="at">size =</span> <span class="dv">18</span>),</span>
<span id="cb90-34"><a href="index.html#cb90-34" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.text.y =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">18</span>),</span>
<span id="cb90-35"><a href="index.html#cb90-35" aria-hidden="true" tabindex="-1"></a>        <span class="at">panel.grid.minor.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb90-36"><a href="index.html#cb90-36" aria-hidden="true" tabindex="-1"></a>        <span class="at">panel.grid.major.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb90-37"><a href="index.html#cb90-37" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.position =</span> <span class="st">&quot;right&quot;</span>) <span class="sc">+</span> <span class="fu">facet_grid</span>( <span class="sc">~</span> methods) <span class="sc">+</span> </span>
<span id="cb90-38"><a href="index.html#cb90-38" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values=</span><span class="fu">pb_color</span>(<span class="fu">c</span>(<span class="dv">12</span>,<span class="dv">14</span>)))</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:ADr22"></span>
<img src="PLSDAbatch_workflow_files/figure-html/ADr22-1.png" alt="AD study: the sum of $R^2$ values for each microbial variable before and after batch effect correction." width="100%" />
<p class="caption">
Figure 1.11: AD study: the sum of <span class="math inline">\(R^2\)</span> values for each microbial variable before and after batch effect correction.
</p>
</div>
<p>The overall sum of <span class="math inline">\(R^2\)</span> values indicated that removeBatchEffect removed slightly more batch variance (removeBatchEffect: 1.70, PLSDA-batch: 12.40, sPLSDA-batch: 9.25) but preserved less treatment variance (removeBatchEffect: 31.75, PLSDA-batch: 40.00, sPLSDA-batch: 36.22) than our proposed approaches</p>
<p><strong>Alignment scores</strong></p>
<p>To use the <code>alignment_score()</code> function from <span style="color: #FF7F00;">PLSDAbatch</span>, we need to specify the proportion of data variance to explain (<code>var</code>), the number of nearest neighbours (<code>k</code>) and the number of principal components to estimate (<code>ncomp</code>). We then use <code>ggplot()</code> function from <span style="color: #FF7F00;">ggplot2</span> to visualise the results.</p>
<div class="sourceCode" id="cb91"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb91-1"><a href="index.html#cb91-1" aria-hidden="true" tabindex="-1"></a><span class="co"># AD data</span></span>
<span id="cb91-2"><a href="index.html#cb91-2" aria-hidden="true" tabindex="-1"></a>ad.scores <span class="ot">&lt;-</span> <span class="fu">c</span>()</span>
<span id="cb91-3"><a href="index.html#cb91-3" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(ad.batch) <span class="ot">&lt;-</span> <span class="fu">rownames</span>(ad.clr)</span>
<span id="cb91-4"><a href="index.html#cb91-4" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="fu">seq_len</span>(<span class="fu">length</span>(ad.corrected.list))){</span>
<span id="cb91-5"><a href="index.html#cb91-5" aria-hidden="true" tabindex="-1"></a>  res <span class="ot">&lt;-</span> <span class="fu">alignment_score</span>(<span class="at">data =</span> ad.corrected.list[[i]], </span>
<span id="cb91-6"><a href="index.html#cb91-6" aria-hidden="true" tabindex="-1"></a>                         <span class="at">batch =</span> ad.batch, </span>
<span id="cb91-7"><a href="index.html#cb91-7" aria-hidden="true" tabindex="-1"></a>                         <span class="at">var =</span> <span class="fl">0.95</span>, </span>
<span id="cb91-8"><a href="index.html#cb91-8" aria-hidden="true" tabindex="-1"></a>                         <span class="at">k =</span> <span class="dv">8</span>, </span>
<span id="cb91-9"><a href="index.html#cb91-9" aria-hidden="true" tabindex="-1"></a>                         <span class="at">ncomp =</span> <span class="dv">50</span>)</span>
<span id="cb91-10"><a href="index.html#cb91-10" aria-hidden="true" tabindex="-1"></a>  ad.scores <span class="ot">&lt;-</span> <span class="fu">c</span>(ad.scores, res)</span>
<span id="cb91-11"><a href="index.html#cb91-11" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb91-12"><a href="index.html#cb91-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb91-13"><a href="index.html#cb91-13" aria-hidden="true" tabindex="-1"></a>ad.scores.df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">scores =</span> ad.scores, </span>
<span id="cb91-14"><a href="index.html#cb91-14" aria-hidden="true" tabindex="-1"></a>                           <span class="at">methods =</span> <span class="fu">names</span>(ad.corrected.list))</span>
<span id="cb91-15"><a href="index.html#cb91-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb91-16"><a href="index.html#cb91-16" aria-hidden="true" tabindex="-1"></a>ad.scores.df<span class="sc">$</span>methods <span class="ot">&lt;-</span> <span class="fu">factor</span>(ad.scores.df<span class="sc">$</span>methods, </span>
<span id="cb91-17"><a href="index.html#cb91-17" aria-hidden="true" tabindex="-1"></a>                               <span class="at">levels =</span> <span class="fu">rev</span>(<span class="fu">names</span>(ad.corrected.list)))</span>
<span id="cb91-18"><a href="index.html#cb91-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb91-19"><a href="index.html#cb91-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb91-20"><a href="index.html#cb91-20" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>() <span class="sc">+</span> <span class="fu">geom_col</span>(<span class="fu">aes</span>(<span class="at">x =</span> ad.scores.df<span class="sc">$</span>methods, </span>
<span id="cb91-21"><a href="index.html#cb91-21" aria-hidden="true" tabindex="-1"></a>                        <span class="at">y =</span> ad.scores.df<span class="sc">$</span>scores)) <span class="sc">+</span> </span>
<span id="cb91-22"><a href="index.html#cb91-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_text</span>(<span class="fu">aes</span>(<span class="at">x =</span> ad.scores.df<span class="sc">$</span>methods, </span>
<span id="cb91-23"><a href="index.html#cb91-23" aria-hidden="true" tabindex="-1"></a>                <span class="at">y =</span> ad.scores.df<span class="sc">$</span>scores<span class="sc">/</span><span class="dv">2</span>, </span>
<span id="cb91-24"><a href="index.html#cb91-24" aria-hidden="true" tabindex="-1"></a>                <span class="at">label =</span> <span class="fu">round</span>(ad.scores.df<span class="sc">$</span>scores, <span class="dv">3</span>)), </span>
<span id="cb91-25"><a href="index.html#cb91-25" aria-hidden="true" tabindex="-1"></a>            <span class="at">size =</span> <span class="dv">3</span>, <span class="at">col =</span> <span class="st">&#39;white&#39;</span>) <span class="sc">+</span> </span>
<span id="cb91-26"><a href="index.html#cb91-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_flip</span>() <span class="sc">+</span> <span class="fu">theme_bw</span>() <span class="sc">+</span> <span class="fu">ylab</span>(<span class="st">&#39;Alignment Scores&#39;</span>) <span class="sc">+</span> </span>
<span id="cb91-27"><a href="index.html#cb91-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">&#39;&#39;</span>) <span class="sc">+</span> <span class="fu">ylim</span>(<span class="dv">0</span>,<span class="fl">0.85</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:ADalignment"></span>
<img src="PLSDAbatch_workflow_files/figure-html/ADalignment-1.png" alt="Comparison of alignment scores before and after batch effect correction using different methods for the AD data." width="60%" />
<p class="caption">
Figure 1.12: Comparison of alignment scores before and after batch effect correction using different methods for the AD data.
</p>
</div>
<p>The alignment scores complement the PCA results, especially when batch effect removal is difficult to assess on PCA sample plots. For example in previous PCA sample plots (<strong>Figure 7</strong>), we observed that the samples across different batches were better mixed after batch effect correction with different methods than before, whereas the performance of difference methods was difficult to compare. Since a higher alignment score indicates that samples are better mixed, as shown in the above bar plot, Combat gave a superior performance compared to the other methods. However, with the <span class="math inline">\(R^2\)</span> values the ComBat corrected data still included a few variables with a large proportion of batch variance (see section <strong>other methods</strong> method “<span class="math inline">\(R^2\)</span>”). Therefore, it is important to compare different techniques and outputs for an unbiased assessment. In this example, the lower alignment scores of PLSDA-batch and sPLSDA-batch corrected data might result from the difference in the PCA sample projections of the batch effect corrected matrices. The data corrected with removeBatchEffect and ComBat had a large variance in their PCA projection, while PLSDA-batch and sPLSDA-batch corrected data had a small variance. A small variance projection results in a small alignment score, as it is easy to locate the samples from the same batch as nearest neighbours. In fact, pRDA presented above (<strong>Figure 8</strong>) quantitatively confirmed that both PLSDA-batch and sPLSDA-batch entirely removed the batch variance <span class="citation">(<a href="#ref-wang2020multivariate" role="doc-biblioref">Y. Wang and Lê Cao 2020</a>)</span>.</p>
</div>
</div>
<div id="variable-selection" class="section level2 hasAnchor" number="1.8">
<h2><span class="header-section-number">1.8</span> Variable selection<a href="index.html#variable-selection" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We use <code>splsda()</code> from <span style="color: #FF7F00;">mixOmics</span> to select the top 50 microbial variables that, in combination, discriminate the different treatment groups in the <span style="color: #0000FF;">AD data</span>. We apply <code>splsda()</code> to the different batch effect corrected data from all methods. Then we use <code>upset()</code> from <span style="color: #FF7F00;">UpSetR</span> package <span class="citation">(<a href="#ref-lex2014upset" role="doc-biblioref">Lex et al. 2014</a>)</span> to visualise the concordance of variables selected.</p>
<p>In the code below, we first need to convert the list of variables selected from different method-corrected data into a data frame compatible with <code>upset()</code> using <code>fromList()</code>. We then assign different colour schemes for each variable selection.</p>
<div class="sourceCode" id="cb92"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb92-1"><a href="index.html#cb92-1" aria-hidden="true" tabindex="-1"></a>ad.splsda.select <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb92-2"><a href="index.html#cb92-2" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="fu">seq_len</span>(<span class="fu">length</span>(ad.corrected.list))){</span>
<span id="cb92-3"><a href="index.html#cb92-3" aria-hidden="true" tabindex="-1"></a>  splsda.res <span class="ot">&lt;-</span> <span class="fu">splsda</span>(<span class="at">X =</span> ad.corrected.list[[i]], <span class="at">Y =</span> ad.trt, </span>
<span id="cb92-4"><a href="index.html#cb92-4" aria-hidden="true" tabindex="-1"></a>                       <span class="at">ncomp =</span> <span class="dv">3</span>, <span class="at">keepX =</span> <span class="fu">rep</span>(<span class="dv">50</span>,<span class="dv">3</span>))</span>
<span id="cb92-5"><a href="index.html#cb92-5" aria-hidden="true" tabindex="-1"></a>  select.res <span class="ot">&lt;-</span> <span class="fu">selectVar</span>(splsda.res, <span class="at">comp =</span> <span class="dv">1</span>)<span class="sc">$</span>name</span>
<span id="cb92-6"><a href="index.html#cb92-6" aria-hidden="true" tabindex="-1"></a>  ad.splsda.select[[i]] <span class="ot">&lt;-</span> select.res</span>
<span id="cb92-7"><a href="index.html#cb92-7" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb92-8"><a href="index.html#cb92-8" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(ad.splsda.select) <span class="ot">&lt;-</span> <span class="fu">names</span>(ad.corrected.list)</span>
<span id="cb92-9"><a href="index.html#cb92-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb92-10"><a href="index.html#cb92-10" aria-hidden="true" tabindex="-1"></a><span class="co"># can only visualise 5 methods</span></span>
<span id="cb92-11"><a href="index.html#cb92-11" aria-hidden="true" tabindex="-1"></a>ad.splsda.select <span class="ot">&lt;-</span> ad.splsda.select[<span class="fu">seq_len</span>(<span class="dv">5</span>)]</span>
<span id="cb92-12"><a href="index.html#cb92-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb92-13"><a href="index.html#cb92-13" aria-hidden="true" tabindex="-1"></a>ad.splsda.upsetR <span class="ot">&lt;-</span> <span class="fu">fromList</span>(ad.splsda.select)</span>
<span id="cb92-14"><a href="index.html#cb92-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb92-15"><a href="index.html#cb92-15" aria-hidden="true" tabindex="-1"></a><span class="fu">upset</span>(ad.splsda.upsetR, <span class="at">main.bar.color =</span> <span class="st">&#39;gray36&#39;</span>,</span>
<span id="cb92-16"><a href="index.html#cb92-16" aria-hidden="true" tabindex="-1"></a>      <span class="at">sets.bar.color =</span> <span class="fu">pb_color</span>(<span class="fu">c</span>(<span class="dv">25</span><span class="sc">:</span><span class="dv">22</span>,<span class="dv">20</span>)), <span class="at">matrix.color =</span> <span class="st">&#39;gray36&#39;</span>,</span>
<span id="cb92-17"><a href="index.html#cb92-17" aria-hidden="true" tabindex="-1"></a>      <span class="at">order.by =</span> <span class="st">&#39;freq&#39;</span>, <span class="at">empty.intersections =</span> <span class="st">&#39;on&#39;</span>,</span>
<span id="cb92-18"><a href="index.html#cb92-18" aria-hidden="true" tabindex="-1"></a>      <span class="at">queries =</span> <span class="fu">list</span>(<span class="fu">list</span>(<span class="at">query =</span> intersects, </span>
<span id="cb92-19"><a href="index.html#cb92-19" aria-hidden="true" tabindex="-1"></a>                          <span class="at">params =</span> <span class="fu">list</span>(<span class="st">&#39;Before correction&#39;</span>), </span>
<span id="cb92-20"><a href="index.html#cb92-20" aria-hidden="true" tabindex="-1"></a>                          <span class="at">color =</span> <span class="fu">pb_color</span>(<span class="dv">20</span>), <span class="at">active =</span> <span class="cn">TRUE</span>),</span>
<span id="cb92-21"><a href="index.html#cb92-21" aria-hidden="true" tabindex="-1"></a>                     <span class="fu">list</span>(<span class="at">query =</span> intersects, </span>
<span id="cb92-22"><a href="index.html#cb92-22" aria-hidden="true" tabindex="-1"></a>                          <span class="at">params =</span> <span class="fu">list</span>(<span class="st">&#39;removeBatchEffect&#39;</span>), </span>
<span id="cb92-23"><a href="index.html#cb92-23" aria-hidden="true" tabindex="-1"></a>                          <span class="at">color =</span> <span class="fu">pb_color</span>(<span class="dv">22</span>), <span class="at">active =</span> <span class="cn">TRUE</span>),</span>
<span id="cb92-24"><a href="index.html#cb92-24" aria-hidden="true" tabindex="-1"></a>                     <span class="fu">list</span>(<span class="at">query =</span> intersects, </span>
<span id="cb92-25"><a href="index.html#cb92-25" aria-hidden="true" tabindex="-1"></a>                          <span class="at">params =</span> <span class="fu">list</span>(<span class="st">&#39;ComBat&#39;</span>), </span>
<span id="cb92-26"><a href="index.html#cb92-26" aria-hidden="true" tabindex="-1"></a>                          <span class="at">color =</span> <span class="fu">pb_color</span>(<span class="dv">23</span>), <span class="at">active =</span> <span class="cn">TRUE</span>),</span>
<span id="cb92-27"><a href="index.html#cb92-27" aria-hidden="true" tabindex="-1"></a>                     <span class="fu">list</span>(<span class="at">query =</span> intersects, </span>
<span id="cb92-28"><a href="index.html#cb92-28" aria-hidden="true" tabindex="-1"></a>                          <span class="at">params =</span> <span class="fu">list</span>(<span class="st">&#39;PLSDA-batch&#39;</span>), </span>
<span id="cb92-29"><a href="index.html#cb92-29" aria-hidden="true" tabindex="-1"></a>                          <span class="at">color =</span> <span class="fu">pb_color</span>(<span class="dv">24</span>), <span class="at">active =</span> <span class="cn">TRUE</span>),</span>
<span id="cb92-30"><a href="index.html#cb92-30" aria-hidden="true" tabindex="-1"></a>                     <span class="fu">list</span>(<span class="at">query =</span> intersects, </span>
<span id="cb92-31"><a href="index.html#cb92-31" aria-hidden="true" tabindex="-1"></a>                          <span class="at">params =</span> <span class="fu">list</span>(<span class="st">&#39;sPLSDA-batch&#39;</span>), </span>
<span id="cb92-32"><a href="index.html#cb92-32" aria-hidden="true" tabindex="-1"></a>                          <span class="at">color =</span> <span class="fu">pb_color</span>(<span class="dv">25</span>), <span class="at">active =</span> <span class="cn">TRUE</span>)))</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:ADupsetR"></span>
<img src="PLSDAbatch_workflow_files/figure-html/ADupsetR-1.png" alt="UpSet plot showing overlap between variables selected from different corrected data for the AD study." width="100%" />
<p class="caption">
Figure 1.13: UpSet plot showing overlap between variables selected from different corrected data for the AD study.
</p>
</div>
<p>In the above UpSet plot, the left bars indicate the number of variables selected from each data corrected with different methods. The right bar plot combined with the scatterplot show different intersection and their aggregates. We obtained a high overlap of 36 out of 50 selected variables between different corrected and uncorrected data. However, the data from each method still included unique variables that were not selected in the other corrected data, e.g., PLSDA-batch and sPLSDA-batch. As <code>upset()</code> can only include five datasets at once, we only displayed the uncorrected data and four corrected data that had been more efficiently corrected for batch effects from our previous assessments compared to the other datasets.</p>
<p>We then extract each intersection of selected variables between these five corrected data with <code>venn()</code> function from <span style="color: #FF7F00;">gplots</span>. We can also save the taxonomic information of these OTUs into a text file for further interpretation.</p>
<div class="sourceCode" id="cb93"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb93-1"><a href="index.html#cb93-1" aria-hidden="true" tabindex="-1"></a>ad.splsda.select.overlap <span class="ot">&lt;-</span> <span class="fu">venn</span>(ad.splsda.select, <span class="at">show.plot =</span> <span class="cn">FALSE</span>)</span>
<span id="cb93-2"><a href="index.html#cb93-2" aria-hidden="true" tabindex="-1"></a>ad.inters.splsda <span class="ot">&lt;-</span> <span class="fu">attr</span>(ad.splsda.select.overlap, <span class="st">&#39;intersections&#39;</span>)</span>
<span id="cb93-3"><a href="index.html#cb93-3" aria-hidden="true" tabindex="-1"></a>ad.inters.splsda.taxa <span class="ot">&lt;-</span> </span>
<span id="cb93-4"><a href="index.html#cb93-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">lapply</span>(ad.inters.splsda, </span>
<span id="cb93-5"><a href="index.html#cb93-5" aria-hidden="true" tabindex="-1"></a>         <span class="at">FUN =</span> <span class="cf">function</span>(x){<span class="fu">as.data.frame</span>(AD_data<span class="sc">$</span>FullData<span class="sc">$</span>taxa[x, ])})</span>
<span id="cb93-6"><a href="index.html#cb93-6" aria-hidden="true" tabindex="-1"></a><span class="fu">capture.output</span>(ad.inters.splsda.taxa, </span>
<span id="cb93-7"><a href="index.html#cb93-7" aria-hidden="true" tabindex="-1"></a>               <span class="at">file =</span> <span class="st">&quot;GeneratedData/ADselected_50_splsda.txt&quot;</span>)</span></code></pre></div>
</div>
<div id="summary" class="section level2 hasAnchor" number="1.9">
<h2><span class="header-section-number">1.9</span> Summary<a href="index.html#summary" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>This vignette presents a complete framework for batch effect management in microbiome data. The first step includes pre-processing, with pre-filtering to remove low-count samples and variables, and data transformation to handle compositional data characteristics. For this step, we advise against using relative abundance data that are compositional, if possible. Visual tools, such as PCA, boxplots, density plots and heatmap, and quantitative approach pRDA are then applied to detect batch effects. When the batch effect is very weak, for example as informed by a very small proportion of variance explained by batch with pRDA, or difficult to visualise with PCA, then there may be no need to manage batch effects. However, when the batch effect is strong, two solutions are to either account for batch effects, or to correct for batch effects from the original data. For either solution, the batch x treatment design matters. If nested, we can only account for batch effects with a linear mixed model. If unbalanced but not nested, we can account for batch effects with any linear models or remove them using weighted PLSDA-batch. Regarding different batch effects, our proposed method PLSDA-batch is more appropriate for a weak batch effect, while sPLSDA-batch for a strong batch effect.</p>
<p>We usually assume batch grouping information is known. When the batch information is unknown, methods such as SVA estimate batch effects from the variables least affected by treatment effects, while RUV4 and RUVIII estimate batch effects from negative control variables or/and sample replicates. For the latter, negative control variables and sample replicates should capture the complete batch variation, if not, there may be a risk that batch effects cannot be completely considered or removed. We emphasised on this point in the <span style="color: #0000FF;">AD</span> and the <span style="color: #808000;">HFHS</span> studies. For these methods that can estimate batch effects, the estimated batch effects are independent of treatment effects. If a correlation between batch and treatment exists, this correlation cannot be estimated. Spurious correlation between batch and treatment effects may also be present, for example SVA, which estimates batch effects from the variables that may still include more or less treatment effects.</p>
<p>In addition, most methods assume systematic batch effects across the whole dataset. This is the case of ComBat, where we recommend assessing the model’s validity first.</p>
<p>The next critical step is to assess the efficacy of batch effect management for a given method. Such assessment is often implicit in methods that account for batch effects. In contrast, the methods that remove batch effects can be assessed by comparing the data before and after correction. All methods we have presented for batch effect detection can be used for assessing batch effect correction. We can also calculate the explained variance <span class="math inline">\(R^2\)</span> of each microbial variable for batch and treatment covariate respectively, as well as alignment scores measuring the performance of mixing samples across batches. We also have noted that some methods may not be very sensitive. For example, PCA plots rely on somewhat subjective evaluation of batch and treatment variation visualisation, and alignment scores only measure the performance of mixing samples. Therefore, a reliable conclusion should be made based on multiple assessment methods.</p>
<p>Once batch effects are removed, downstream analysis such as multivariate discriminant analysis, or univariate differential analysis can be performed. Multivariate methods may be more suitable for microbiome data analysis since microbial variables are naturally correlated because of mutual interactions.</p>
<p><img src="figures/function_description.png" />
<strong>Table 2: A summary of all functions in each section.</strong></p>
</div>
<div id="references" class="section level2 hasAnchor" number="1.10">
<h2><span class="header-section-number">1.10</span> References<a href="index.html#references" class="anchor-section" aria-label="Anchor link to header"></a></h2>

</div>
</div>
<h3>References<a href="references-4.html#references-4" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-chapleur2016increasing" class="csl-entry">
Chapleur, Olivier, Céline Madigou, Raphaël Civade, Yohan Rodolphe, Laurent Mazéas, and Théodore Bouchez. 2016. <span>“Increasing Concentrations of Phenol Progressively Affect Anaerobic Digestion of Cellulose and Associated Microbial Communities.”</span> <em>Biodegradation</em> 27 (1): 15–27.
</div>
<div id="ref-duvallet2017meta" class="csl-entry">
Duvallet, Claire, Sean M Gibbons, Thomas Gurry, Rafael A Irizarry, and Eric J Alm. 2017. <span>“Meta-Analysis of Gut Microbiome Studies Identifies Disease-Specific and Shared Responses.”</span> <em>Nature Communications</em> 8 (1): 1–10.
</div>
<div id="ref-fan2020gut" class="csl-entry">
Fan, Yong, and Oluf Pedersen. 2020. <span>“Gut Microbiota in Human Metabolic Health and Disease.”</span> <em>Nature Reviews Microbiology</em>, 1–17.
</div>
<div id="ref-gibson2004dietary" class="csl-entry">
Gibson, Glenn R, Hollie M Probert, Jan Van Loo, Robert A Rastall, and Marcel B Roberfroid. 2004. <span>“Dietary Modulation of the Human Colonic Microbiota: Updating the Concept of Prebiotics.”</span> <em>Nutrition Research Reviews</em> 17 (2): 259–75.
</div>
<div id="ref-haro2016intestinal" class="csl-entry">
Haro, Carmen, Oriol A Rangel-Zúñiga, Juan F Alcalá-Dı́az, Francisco Gómez-Delgado, Pablo Pérez-Martı́nez, Javier Delgado-Lista, Gracia M Quintana-Navarro, et al. 2016. <span>“Intestinal Microbiota Is Influenced by Gender and Body Mass Index.”</span> <em>PloS One</em> 11 (5): e0154090.
</div>
<div id="ref-hieken2016microbiome" class="csl-entry">
Hieken, Tina J, Jun Chen, Tanya L Hoskin, Marina Walther-Antonio, Stephen Johnson, Sheri Ramaker, Jian Xiao, et al. 2016. <span>“The Microbiome of Aseptically Collected Human Breast Tissue in Benign and Malignant Disease.”</span> <em>Scientific Reports</em> 6: 30751.
</div>
<div id="ref-kim2017optimizing" class="csl-entry">
Kim, Dorothy, Casey E Hofstaedter, Chunyu Zhao, Lisa Mattei, Ceylan Tanes, Erik Clarke, Abigail Lauder, et al. 2017. <span>“Optimizing Methods and Dodging Pitfalls in Microbiome Research.”</span> <em>Microbiome</em> 5 (1): 52.
</div>
<div id="ref-kong2020microbiome" class="csl-entry">
Kong, Geraldine, Kim-Anh Lê Cao, Louise M Judd, ShanShan Li, Thibault Renoir, and Anthony J Hannan. 2020. <span>“Microbiome Profiling Reveals Gut Dysbiosis in a Transgenic Mouse Model of Huntington’s Disease.”</span> <em>Neurobiology of Disease</em> 135: 104268.
</div>
<div id="ref-daniel2020performance" class="csl-entry">
LÃŒdecke, Daniel, Dominique Makowski, Philip Waggoner, and Indrajeet Patil. 2020. <span>“Performance: Assessment of Regression Models Performance.”</span> <em>CRAN</em>. <a href="https://doi.org/10.5281/zenodo.3952174">https://doi.org/10.5281/zenodo.3952174</a>.
</div>
<div id="ref-leek2012sva" class="csl-entry">
Leek, Jeffrey T, W Evan Johnson, Hilary S Parker, Andrew E Jaffe, and John D Storey. 2012. <span>“The Sva Package for Removing Batch Effects and Other Unwanted Variation in High-Throughput Experiments.”</span> <em>Bioinformatics</em> 28 (6): 882–83.
</div>
<div id="ref-lex2014upset" class="csl-entry">
Lex, Alexander, Nils Gehlenborg, Hendrik Strobelt, Romain Vuillemot, and Hanspeter Pfister. 2014. <span>“UpSet: Visualization of Intersecting Sets.”</span> <em>IEEE Transactions on Visualization and Computer Graphics</em> 20 (12): 1983–92.
</div>
<div id="ref-lozupone2013meta" class="csl-entry">
Lozupone, Catherine, Jesse Stombaugh, Antonio Gonzalez, Gail Ackermann, Doug Wendel, Yoshiki Vázquez-Baeza, Janet K Jansson, Jeffrey I Gordon, and Rob Knight. 2013. <span>“Meta-Analyses of Studies of the Human Microbiota.”</span> <em>Genome Research</em>, gr–151803.
</div>
<div id="ref-marchesi2015vocabulary" class="csl-entry">
Marchesi, Julian R, and Jacques Ravel. 2015. <span>“The Vocabulary of Microbiome Research: A Proposal.”</span> Springer.
</div>
<div id="ref-moskovicz2020skin" class="csl-entry">
Moskovicz, Veronica, Rina Ben-El, Guy Horev, and Boaz Mizrahi. 2020. <span>“Skin Microbiota Dynamics Following b. Subtilis Formulation Challenge.”</span>
</div>
<div id="ref-nakagawa2013general" class="csl-entry">
Nakagawa, Shinichi, and Holger Schielzeth. 2013. <span>“A General and Simple Method for Obtaining R2 from Generalized Linear Mixed-Effects Models.”</span> <em>Methods in Ecology and Evolution</em> 4 (2): 133–42.
</div>
<div id="ref-poirier2020integrating" class="csl-entry">
Poirier, Simon, Sébastien Déjean, Cédric Midoux, Kim-Anh Lê Cao, and Olivier Chapleur. 2020. <span>“Integrating Independent Microbial Studies to Build Predictive Models of Anaerobic Digestion Inhibition by Ammonia and Phenol.”</span> <em>Bioresource Technology</em> 316: 123952.
</div>
<div id="ref-ray2020microbe" class="csl-entry">
Ray, Prasun, Venkatachalam Lakshmanan, Jessy L Labbé, and Kelly D Craven. 2020. <span>“Microbe to Microbiome: A Paradigm Shift in the Application of Microorganisms for Sustainable Agriculture.”</span> <em>Frontiers in Microbiology</em> 11: 3323.
</div>
<div id="ref-rohart2017mixomics" class="csl-entry">
Rohart, Florian, Benoı̂t Gautier, Amrit Singh, and Kim-Anh Lê Cao. 2017. <span>“mixOmics: An r Package for ‘Omics Feature Selection and Multiple Data Integration.”</span> <em>PLoS Computational Biology</em> 13 (11): e1005752.
</div>
<div id="ref-sacristan2011exploring" class="csl-entry">
Sacristán-Soriano, Oriol, Bernard Banaigs, Emilio O Casamayor, and Mikel A Becerro. 2011. <span>“Exploring the Links Between Natural Products and Bacterial Assemblages in the Sponge Aplysina Aerophoba.”</span> <em>Appl. Environ. Microbiol.</em> 77 (3): 862–70.
</div>
<div id="ref-susin2020variable" class="csl-entry">
Susin, Antoni, Yiwen Wang, Kim-Anh Lê Cao, and M Luz Calle. 2020. <span>“Variable Selection in Microbiome Compositional Data Analysis.”</span> <em>NAR Genomics and Bioinformatics</em> 2 (2): lqaa029.
</div>
<div id="ref-wang2020characterizing" class="csl-entry">
Wang, Charlotte H, Linda Wu, Zengyan Wang, Magdy S Alabady, Daniel Parson, Zainab Molumo, and Sarah C Fankhauser. 2020. <span>“Characterizing Changes in Soil Microbiome Abundance and Diversity Due to Different Cover Crop Techniques.”</span> <em>PloS One</em> 15 (5): e0232453.
</div>
<div id="ref-wang2020multivariate" class="csl-entry">
Wang, Yiwen, and Kim-Anh Lê Cao. 2020. <span>“A Multivariate Method to Correct for Batch Effects in Microbiome Data.”</span> <em>bioRxiv</em>.
</div>
<div id="ref-wang2020managing" class="csl-entry">
Wang, Yiwen, and Kim-Anh LêCao. 2020. <span>“Managing Batch Effects in Microbiome Data.”</span> <em>Briefings in Bioinformatics</em> 21 (6): 1954–70.
</div>
</div>
            </section>

          </div>
        </div>
      </div>

<a href="supplementary-materials-for-batch-effects-management-in-case-studies.html" class="navigation navigation-next navigation-unique" aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["PLSDAbatch_workflow.pdf", "PLSDAbatch_workflow.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
