<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>References | PLSDA-batch: a multivariate framework to correct for batch effects in microbiome data</title>
  <meta name="description" content="Vignette for paper ‘PLSDA-batch: a multivariate framework to correct for batch effects in microbiome data’." />
  <meta name="generator" content="bookdown 0.33 and GitBook 2.6.7" />

  <meta property="og:title" content="References | PLSDA-batch: a multivariate framework to correct for batch effects in microbiome data" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="Vignette for paper ‘PLSDA-batch: a multivariate framework to correct for batch effects in microbiome data’." />
  <meta name="github-repo" content="EvaYiwenWang/PLSDAbatch_workflow" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="References | PLSDA-batch: a multivariate framework to correct for batch effects in microbiome data" />
  
  <meta name="twitter:description" content="Vignette for paper ‘PLSDA-batch: a multivariate framework to correct for batch effects in microbiome data’." />
  



<meta name="date" content="2023-05-04" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="batch-effects-management-in-simulation-studies-negative-binomial-distribution.html"/>

<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">PLSDA-batch</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Batch Effects Management in Case Studies</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#introduction"><i class="fa fa-check"></i><b>1.1</b> Introduction</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#case-studies-description"><i class="fa fa-check"></i><b>1.2</b> Case studies description</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#packages-installation-and-loading"><i class="fa fa-check"></i><b>1.3</b> Packages installation and loading</a></li>
<li class="chapter" data-level="1.4" data-path="index.html"><a href="index.html#data-pre-processing"><i class="fa fa-check"></i><b>1.4</b> Data pre-processing</a>
<ul>
<li class="chapter" data-level="1.4.1" data-path="index.html"><a href="index.html#pre-filtering"><i class="fa fa-check"></i><b>1.4.1</b> Pre-filtering</a></li>
<li class="chapter" data-level="1.4.2" data-path="index.html"><a href="index.html#transformation"><i class="fa fa-check"></i><b>1.4.2</b> Transformation</a></li>
</ul></li>
<li class="chapter" data-level="1.5" data-path="index.html"><a href="index.html#batch-effect-detection"><i class="fa fa-check"></i><b>1.5</b> Batch effect detection</a>
<ul>
<li class="chapter" data-level="1.5.1" data-path="index.html"><a href="index.html#pca"><i class="fa fa-check"></i><b>1.5.1</b> PCA</a></li>
<li class="chapter" data-level="1.5.2" data-path="index.html"><a href="index.html#boxplots-and-density-plots"><i class="fa fa-check"></i><b>1.5.2</b> Boxplots and density plots</a></li>
<li class="chapter" data-level="1.5.3" data-path="index.html"><a href="index.html#heatmap"><i class="fa fa-check"></i><b>1.5.3</b> Heatmap</a></li>
<li class="chapter" data-level="1.5.4" data-path="index.html"><a href="index.html#prda"><i class="fa fa-check"></i><b>1.5.4</b> pRDA</a></li>
</ul></li>
<li class="chapter" data-level="1.6" data-path="index.html"><a href="index.html#managing-batch-effects"><i class="fa fa-check"></i><b>1.6</b> Managing batch effects</a>
<ul>
<li class="chapter" data-level="1.6.1" data-path="index.html"><a href="index.html#accnt"><i class="fa fa-check"></i><b>1.6.1</b> Accounting for batch effects</a></li>
<li class="chapter" data-level="1.6.2" data-path="index.html"><a href="index.html#corrct"><i class="fa fa-check"></i><b>1.6.2</b> Correcting for batch effects</a></li>
</ul></li>
<li class="chapter" data-level="1.7" data-path="index.html"><a href="index.html#assessing-batch-effect-correction"><i class="fa fa-check"></i><b>1.7</b> Assessing batch effect correction</a>
<ul>
<li class="chapter" data-level="1.7.1" data-path="index.html"><a href="index.html#methods-that-detect-batch-effects"><i class="fa fa-check"></i><b>1.7.1</b> Methods that detect batch effects</a></li>
<li class="chapter" data-level="1.7.2" data-path="index.html"><a href="index.html#other-methods-1"><i class="fa fa-check"></i><b>1.7.2</b> Other methods</a></li>
</ul></li>
<li class="chapter" data-level="1.8" data-path="index.html"><a href="index.html#variable-selection"><i class="fa fa-check"></i><b>1.8</b> Variable selection</a></li>
<li class="chapter" data-level="1.9" data-path="index.html"><a href="index.html#summary"><i class="fa fa-check"></i><b>1.9</b> Summary</a></li>
<li class="chapter" data-level="1.10" data-path="index.html"><a href="index.html#references"><i class="fa fa-check"></i><b>1.10</b> References</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="supplementary-materials-for-batch-effects-management-in-case-studies.html"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html"><i class="fa fa-check"></i><b>2</b> Supplementary Materials for Batch Effects Management in Case Studies</a>
<ul>
<li class="chapter" data-level="2.1" data-path="supplementary-materials-for-batch-effects-management-in-case-studies.html"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#sponge-data"><i class="fa fa-check"></i><b>2.1</b> Sponge data</a>
<ul>
<li class="chapter" data-level="2.1.1" data-path="supplementary-materials-for-batch-effects-management-in-case-studies.html"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#data-pre-processing-1"><i class="fa fa-check"></i><b>2.1.1</b> Data pre-processing</a></li>
<li class="chapter" data-level="2.1.2" data-path="supplementary-materials-for-batch-effects-management-in-case-studies.html"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#batch-effect-detection-1"><i class="fa fa-check"></i><b>2.1.2</b> Batch effect detection</a></li>
<li class="chapter" data-level="2.1.3" data-path="supplementary-materials-for-batch-effects-management-in-case-studies.html"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#managing-batch-effects-1"><i class="fa fa-check"></i><b>2.1.3</b> Managing batch effects</a></li>
<li class="chapter" data-level="2.1.4" data-path="supplementary-materials-for-batch-effects-management-in-case-studies.html"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#assessing-batch-effect-correction-1"><i class="fa fa-check"></i><b>2.1.4</b> Assessing batch effect correction</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="supplementary-materials-for-batch-effects-management-in-case-studies.html"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#hfhs-data"><i class="fa fa-check"></i><b>2.2</b> HFHS data</a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="supplementary-materials-for-batch-effects-management-in-case-studies.html"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#data-pre-processing-2"><i class="fa fa-check"></i><b>2.2.1</b> Data pre-processing</a></li>
<li class="chapter" data-level="2.2.2" data-path="supplementary-materials-for-batch-effects-management-in-case-studies.html"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#batch-effect-detection-2"><i class="fa fa-check"></i><b>2.2.2</b> Batch effect detection</a></li>
<li class="chapter" data-level="2.2.3" data-path="supplementary-materials-for-batch-effects-management-in-case-studies.html"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#managing-batch-effects-2"><i class="fa fa-check"></i><b>2.2.3</b> Managing batch effects</a></li>
<li class="chapter" data-level="2.2.4" data-path="supplementary-materials-for-batch-effects-management-in-case-studies.html"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#assessing-batch-effect-correction-2"><i class="fa fa-check"></i><b>2.2.4</b> Assessing batch effect correction</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="supplementary-materials-for-batch-effects-management-in-case-studies.html"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#hd-data"><i class="fa fa-check"></i><b>2.3</b> HD data</a>
<ul>
<li class="chapter" data-level="2.3.1" data-path="supplementary-materials-for-batch-effects-management-in-case-studies.html"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#data-pre-processing-3"><i class="fa fa-check"></i><b>2.3.1</b> Data pre-processing</a></li>
<li class="chapter" data-level="2.3.2" data-path="supplementary-materials-for-batch-effects-management-in-case-studies.html"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#batch-effect-detection-3"><i class="fa fa-check"></i><b>2.3.2</b> Batch effect detection</a></li>
<li class="chapter" data-level="2.3.3" data-path="supplementary-materials-for-batch-effects-management-in-case-studies.html"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#managing-batch-effects-3"><i class="fa fa-check"></i><b>2.3.3</b> Managing batch effects</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="supplementary-materials-for-batch-effects-management-in-case-studies.html"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#references-1"><i class="fa fa-check"></i><b>2.4</b> References</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="batch-effects-management-in-simulation-studies-gaussian-distribution.html"><a href="batch-effects-management-in-simulation-studies-gaussian-distribution.html"><i class="fa fa-check"></i><b>3</b> Batch Effects Management in Simulation Studies (Gaussian Distribution)</a>
<ul>
<li class="chapter" data-level="3.1" data-path="batch-effects-management-in-simulation-studies-gaussian-distribution.html"><a href="batch-effects-management-in-simulation-studies-gaussian-distribution.html#introduction-1"><i class="fa fa-check"></i><b>3.1</b> Introduction</a></li>
<li class="chapter" data-level="3.2" data-path="batch-effects-management-in-simulation-studies-gaussian-distribution.html"><a href="batch-effects-management-in-simulation-studies-gaussian-distribution.html#simulations"><i class="fa fa-check"></i><b>3.2</b> Simulations</a>
<ul>
<li class="chapter" data-level="3.2.1" data-path="batch-effects-management-in-simulation-studies-gaussian-distribution.html"><a href="batch-effects-management-in-simulation-studies-gaussian-distribution.html#balanced-batch-times-treatment-design"><i class="fa fa-check"></i><b>3.2.1</b> Balanced batch <span class="math inline">\(\times\)</span> treatment design</a></li>
<li class="chapter" data-level="3.2.2" data-path="batch-effects-management-in-simulation-studies-gaussian-distribution.html"><a href="batch-effects-management-in-simulation-studies-gaussian-distribution.html#figures"><i class="fa fa-check"></i><b>3.2.2</b> Figures</a></li>
<li class="chapter" data-level="3.2.3" data-path="batch-effects-management-in-simulation-studies-gaussian-distribution.html"><a href="batch-effects-management-in-simulation-studies-gaussian-distribution.html#unbalanced-batch-times-treatment-design"><i class="fa fa-check"></i><b>3.2.3</b> Unbalanced batch <span class="math inline">\(\times\)</span> treatment design</a></li>
<li class="chapter" data-level="3.2.4" data-path="batch-effects-management-in-simulation-studies-gaussian-distribution.html"><a href="batch-effects-management-in-simulation-studies-gaussian-distribution.html#figures-1"><i class="fa fa-check"></i><b>3.2.4</b> Figures</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="batch-effects-management-in-simulation-studies-gaussian-distribution.html"><a href="batch-effects-management-in-simulation-studies-gaussian-distribution.html#references-2"><i class="fa fa-check"></i><b>3.3</b> References</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="batch-effects-management-in-simulation-studies-negative-binomial-distribution.html"><a href="batch-effects-management-in-simulation-studies-negative-binomial-distribution.html"><i class="fa fa-check"></i><b>4</b> Batch Effects Management in Simulation Studies (Negative Binomial Distribution)</a>
<ul>
<li class="chapter" data-level="4.1" data-path="batch-effects-management-in-simulation-studies-negative-binomial-distribution.html"><a href="batch-effects-management-in-simulation-studies-negative-binomial-distribution.html#introduction-2"><i class="fa fa-check"></i><b>4.1</b> Introduction</a></li>
<li class="chapter" data-level="4.2" data-path="batch-effects-management-in-simulation-studies-negative-binomial-distribution.html"><a href="batch-effects-management-in-simulation-studies-negative-binomial-distribution.html#simulations-two-batch-groups"><i class="fa fa-check"></i><b>4.2</b> Simulations (two batch groups)</a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="batch-effects-management-in-simulation-studies-negative-binomial-distribution.html"><a href="batch-effects-management-in-simulation-studies-negative-binomial-distribution.html#balanced-batch-times-treatment-design-1"><i class="fa fa-check"></i><b>4.2.1</b> Balanced batch <span class="math inline">\(\times\)</span> treatment design</a></li>
<li class="chapter" data-level="4.2.2" data-path="batch-effects-management-in-simulation-studies-negative-binomial-distribution.html"><a href="batch-effects-management-in-simulation-studies-negative-binomial-distribution.html#figures-2"><i class="fa fa-check"></i><b>4.2.2</b> Figures</a></li>
<li class="chapter" data-level="4.2.3" data-path="batch-effects-management-in-simulation-studies-negative-binomial-distribution.html"><a href="batch-effects-management-in-simulation-studies-negative-binomial-distribution.html#unbalanced-batch-times-treatment-design-1"><i class="fa fa-check"></i><b>4.2.3</b> Unbalanced batch <span class="math inline">\(\times\)</span> treatment design</a></li>
<li class="chapter" data-level="4.2.4" data-path="batch-effects-management-in-simulation-studies-negative-binomial-distribution.html"><a href="batch-effects-management-in-simulation-studies-negative-binomial-distribution.html#figures-3"><i class="fa fa-check"></i><b>4.2.4</b> Figures</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="batch-effects-management-in-simulation-studies-negative-binomial-distribution.html"><a href="batch-effects-management-in-simulation-studies-negative-binomial-distribution.html#simulations-three-batch-groups"><i class="fa fa-check"></i><b>4.3</b> Simulations (three batch groups)</a>
<ul>
<li class="chapter" data-level="4.3.1" data-path="batch-effects-management-in-simulation-studies-negative-binomial-distribution.html"><a href="batch-effects-management-in-simulation-studies-negative-binomial-distribution.html#balanced-batch-times-treatment-design-2"><i class="fa fa-check"></i><b>4.3.1</b> Balanced batch <span class="math inline">\(\times\)</span> treatment design</a></li>
<li class="chapter" data-level="4.3.2" data-path="batch-effects-management-in-simulation-studies-negative-binomial-distribution.html"><a href="batch-effects-management-in-simulation-studies-negative-binomial-distribution.html#figures-4"><i class="fa fa-check"></i><b>4.3.2</b> Figures</a></li>
<li class="chapter" data-level="4.3.3" data-path="batch-effects-management-in-simulation-studies-negative-binomial-distribution.html"><a href="batch-effects-management-in-simulation-studies-negative-binomial-distribution.html#unbalanced-batch-times-treatment-design-2"><i class="fa fa-check"></i><b>4.3.3</b> Unbalanced batch <span class="math inline">\(\times\)</span> treatment design</a></li>
<li class="chapter" data-level="4.3.4" data-path="batch-effects-management-in-simulation-studies-negative-binomial-distribution.html"><a href="batch-effects-management-in-simulation-studies-negative-binomial-distribution.html#figures-5"><i class="fa fa-check"></i><b>4.3.4</b> Figures</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="batch-effects-management-in-simulation-studies-negative-binomial-distribution.html"><a href="batch-effects-management-in-simulation-studies-negative-binomial-distribution.html#references-3"><i class="fa fa-check"></i><b>4.4</b> References</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="references-4.html"><a href="references-4.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/EvaYiwenWang/PLSDAbatch_workflow" target="blank">Source codes on github</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">PLSDA-batch: a multivariate framework to correct for batch effects in microbiome data</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="references-4" class="section level1 unnumbered hasAnchor">
<h1>References<a href="references-4.html#references-4" class="anchor-section" aria-label="Anchor link to header"></a></h1>

<div id="refs" class="references csl-bib-body hanging-indent">
<div class="csl-entry">
Chapleur, Olivier, Céline Madigou, Raphaël Civade, Yohan Rodolphe, Laurent Mazéas, and Théodore Bouchez. 2016. <span>“Increasing Concentrations of Phenol Progressively Affect Anaerobic Digestion of Cellulose and Associated Microbial Communities.”</span> <em>Biodegradation</em> 27 (1): 15–27.
</div>
<div class="csl-entry">
Duvallet, Claire, Sean M Gibbons, Thomas Gurry, Rafael A Irizarry, and Eric J Alm. 2017. <span>“Meta-Analysis of Gut Microbiome Studies Identifies Disease-Specific and Shared Responses.”</span> <em>Nature Communications</em> 8 (1): 1–10.
</div>
<div class="csl-entry">
Fan, Yong, and Oluf Pedersen. 2020. <span>“Gut Microbiota in Human Metabolic Health and Disease.”</span> <em>Nature Reviews Microbiology</em>, 1–17.
</div>
<div class="csl-entry">
Gibson, Glenn R, Hollie M Probert, Jan Van Loo, Robert A Rastall, and Marcel B Roberfroid. 2004. <span>“Dietary Modulation of the Human Colonic Microbiota: Updating the Concept of Prebiotics.”</span> <em>Nutrition Research Reviews</em> 17 (2): 259–75.
</div>
<div class="csl-entry">
Haro, Carmen, Oriol A Rangel-Zúñiga, Juan F Alcalá-Dı́az, Francisco Gómez-Delgado, Pablo Pérez-Martı́nez, Javier Delgado-Lista, Gracia M Quintana-Navarro, et al. 2016. <span>“Intestinal Microbiota Is Influenced by Gender and Body Mass Index.”</span> <em>PloS One</em> 11 (5): e0154090.
</div>
<div class="csl-entry">
Hawinkel, Stijn, Federico Mattiello, Luc Bijnens, and Olivier Thas. 2019. <span>“A Broken Promise: Microbiome Differential Abundance Methods Do Not Control the False Discovery Rate.”</span> <em>Briefings in Bioinformatics</em> 20 (1): 210–21.
</div>
<div class="csl-entry">
Hieken, Tina J, Jun Chen, Tanya L Hoskin, Marina Walther-Antonio, Stephen Johnson, Sheri Ramaker, Jian Xiao, et al. 2016. <span>“The Microbiome of Aseptically Collected Human Breast Tissue in Benign and Malignant Disease.”</span> <em>Scientific Reports</em> 6: 30751.
</div>
<div class="csl-entry">
Kim, Dorothy, Casey E Hofstaedter, Chunyu Zhao, Lisa Mattei, Ceylan Tanes, Erik Clarke, Abigail Lauder, et al. 2017. <span>“Optimizing Methods and Dodging Pitfalls in Microbiome Research.”</span> <em>Microbiome</em> 5 (1): 52.
</div>
<div class="csl-entry">
Kong, Geraldine, Kim-Anh Lê Cao, Louise M Judd, ShanShan Li, Thibault Renoir, and Anthony J Hannan. 2020. <span>“Microbiome Profiling Reveals Gut Dysbiosis in a Transgenic Mouse Model of Huntington’s Disease.”</span> <em>Neurobiology of Disease</em> 135: 104268.
</div>
<div class="csl-entry">
LÃŒdecke, Daniel, Dominique Makowski, Philip Waggoner, and Indrajeet Patil. 2020. <span>“Performance: Assessment of Regression Models Performance.”</span> <em>CRAN</em>. <a href="https://doi.org/10.5281/zenodo.3952174">https://doi.org/10.5281/zenodo.3952174</a>.
</div>
<div class="csl-entry">
Leek, Jeffrey T, W Evan Johnson, Hilary S Parker, Andrew E Jaffe, and John D Storey. 2012. <span>“The Sva Package for Removing Batch Effects and Other Unwanted Variation in High-Throughput Experiments.”</span> <em>Bioinformatics</em> 28 (6): 882–83.
</div>
<div class="csl-entry">
Lex, Alexander, Nils Gehlenborg, Hendrik Strobelt, Romain Vuillemot, and Hanspeter Pfister. 2014. <span>“UpSet: Visualization of Intersecting Sets.”</span> <em>IEEE Transactions on Visualization and Computer Graphics</em> 20 (12): 1983–92.
</div>
<div class="csl-entry">
Lozupone, Catherine, Jesse Stombaugh, Antonio Gonzalez, Gail Ackermann, Doug Wendel, Yoshiki Vázquez-Baeza, Janet K Jansson, Jeffrey I Gordon, and Rob Knight. 2013. <span>“Meta-Analyses of Studies of the Human Microbiota.”</span> <em>Genome Research</em>, gr–151803.
</div>
<div class="csl-entry">
Marchesi, Julian R, and Jacques Ravel. 2015. <span>“The Vocabulary of Microbiome Research: A Proposal.”</span> Springer.
</div>
<div class="csl-entry">
McGregor, Kevin, Aurélie Labbe, and Celia MT Greenwood. 2020. <span>“MDiNE: A Model to Estimate Differential Co-Occurrence Networks in Microbiome Studies.”</span> <em>Bioinformatics</em> 36 (6): 1840–47.
</div>
<div class="csl-entry">
McMurdie, Paul J, and Susan Holmes. 2014. <span>“Waste Not, Want Not: Why Rarefying Microbiome Data Is Inadmissible.”</span> <em>PLoS Computational Biology</em> 10 (4): e1003531.
</div>
<div class="csl-entry">
Moskovicz, Veronica, Rina Ben-El, Guy Horev, and Boaz Mizrahi. 2020. <span>“Skin Microbiota Dynamics Following b. Subtilis Formulation Challenge.”</span>
</div>
<div class="csl-entry">
Nakagawa, Shinichi, and Holger Schielzeth. 2013. <span>“A General and Simple Method for Obtaining R2 from Generalized Linear Mixed-Effects Models.”</span> <em>Methods in Ecology and Evolution</em> 4 (2): 133–42.
</div>
<div class="csl-entry">
Poirier, Simon, Sébastien Déjean, Cédric Midoux, Kim-Anh Lê Cao, and Olivier Chapleur. 2020. <span>“Integrating Independent Microbial Studies to Build Predictive Models of Anaerobic Digestion Inhibition by Ammonia and Phenol.”</span> <em>Bioresource Technology</em> 316: 123952.
</div>
<div class="csl-entry">
Quinn, Thomas P, Ionas Erb, Mark F Richardson, and Tamsyn M Crowley. 2018. <span>“Understanding Sequencing Data as Compositions: An Outlook and Review.”</span> <em>Bioinformatics</em> 34 (16): 2870–78.
</div>
<div class="csl-entry">
Ray, Prasun, Venkatachalam Lakshmanan, Jessy L Labbé, and Kelly D Craven. 2020. <span>“Microbe to Microbiome: A Paradigm Shift in the Application of Microorganisms for Sustainable Agriculture.”</span> <em>Frontiers in Microbiology</em> 11: 3323.
</div>
<div class="csl-entry">
Rohart, Florian, Benoı̂t Gautier, Amrit Singh, and Kim-Anh Lê Cao. 2017. <span>“mixOmics: An r Package for ‘Omics Feature Selection and Multiple Data Integration.”</span> <em>PLoS Computational Biology</em> 13 (11): e1005752.
</div>
<div class="csl-entry">
Sacristán-Soriano, Oriol, Bernard Banaigs, Emilio O Casamayor, and Mikel A Becerro. 2011. <span>“Exploring the Links Between Natural Products and Bacterial Assemblages in the Sponge Aplysina Aerophoba.”</span> <em>Appl. Environ. Microbiol.</em> 77 (3): 862–70.
</div>
<div class="csl-entry">
Singh, Amrit, Casey P Shannon, Benoı̂t Gautier, Florian Rohart, Michaël Vacher, Scott J Tebbutt, and Kim-Anh Lê Cao. 2019. <span>“DIABLO: An Integrative Approach for Identifying Key Molecular Drivers from Multi-Omics Assays.”</span> <em>Bioinformatics</em> 35 (17): 3055–62.
</div>
<div class="csl-entry">
Susin, Antoni, Yiwen Wang, Kim-Anh Lê Cao, and M Luz Calle. 2020. <span>“Variable Selection in Microbiome Compositional Data Analysis.”</span> <em>NAR Genomics and Bioinformatics</em> 2 (2): lqaa029.
</div>
<div class="csl-entry">
Wang, Charlotte H, Linda Wu, Zengyan Wang, Magdy S Alabady, Daniel Parson, Zainab Molumo, and Sarah C Fankhauser. 2020. <span>“Characterizing Changes in Soil Microbiome Abundance and Diversity Due to Different Cover Crop Techniques.”</span> <em>PloS One</em> 15 (5): e0232453.
</div>
<div class="csl-entry">
Wang, Yiwen, and Kim-Anh Lê Cao. 2020. <span>“A Multivariate Method to Correct for Batch Effects in Microbiome Data.”</span> <em>bioRxiv</em>.
</div>
<div class="csl-entry">
Wang, Yiwen, and Kim-Anh LêCao. 2020. <span>“Managing Batch Effects in Microbiome Data.”</span> <em>Briefings in Bioinformatics</em> 21 (6): 1954–70.
</div>
<div class="csl-entry">
Weiss, Sophie, Zhenjiang Zech Xu, Shyamal Peddada, Amnon Amir, Kyle Bittinger, Antonio Gonzalez, Catherine Lozupone, et al. 2017. <span>“Normalization and Microbial Differential Abundance Strategies Depend Upon Data Characteristics.”</span> <em>Microbiome</em> 5 (1): 1–18.
</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="batch-effects-management-in-simulation-studies-negative-binomial-distribution.html" class="navigation navigation-prev navigation-unique" aria-label="Previous page"><i class="fa fa-angle-left"></i></a>

    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["PLSDAbatch_workflow.pdf", "PLSDAbatch_workflow.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
