<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 2 Supplementary Materials for Batch Effects Management in Case Studies | PLSDA-batch: a multivariate framework to correct for batch effects in microbiome data</title>
  <meta name="description" content="Vignette for paper ‘PLSDA-batch: a multivariate framework to correct for batch effects in microbiome data’." />
  <meta name="generator" content="bookdown 0.33 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 2 Supplementary Materials for Batch Effects Management in Case Studies | PLSDA-batch: a multivariate framework to correct for batch effects in microbiome data" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="Vignette for paper ‘PLSDA-batch: a multivariate framework to correct for batch effects in microbiome data’." />
  <meta name="github-repo" content="EvaYiwenWang/PLSDAbatch_workflow" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 2 Supplementary Materials for Batch Effects Management in Case Studies | PLSDA-batch: a multivariate framework to correct for batch effects in microbiome data" />
  
  <meta name="twitter:description" content="Vignette for paper ‘PLSDA-batch: a multivariate framework to correct for batch effects in microbiome data’." />
  



<meta name="date" content="2023-05-04" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="index.html"/>
<link rel="next" href="batch-effects-management-in-simulation-studies-gaussian-distribution.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">PLSDA-batch</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Batch Effects Management in Case Studies</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#introduction"><i class="fa fa-check"></i><b>1.1</b> Introduction</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#case-studies-description"><i class="fa fa-check"></i><b>1.2</b> Case studies description</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#packages-installation-and-loading"><i class="fa fa-check"></i><b>1.3</b> Packages installation and loading</a></li>
<li class="chapter" data-level="1.4" data-path="index.html"><a href="index.html#data-pre-processing"><i class="fa fa-check"></i><b>1.4</b> Data pre-processing</a>
<ul>
<li class="chapter" data-level="1.4.1" data-path="index.html"><a href="index.html#pre-filtering"><i class="fa fa-check"></i><b>1.4.1</b> Pre-filtering</a></li>
<li class="chapter" data-level="1.4.2" data-path="index.html"><a href="index.html#transformation"><i class="fa fa-check"></i><b>1.4.2</b> Transformation</a></li>
</ul></li>
<li class="chapter" data-level="1.5" data-path="index.html"><a href="index.html#batch-effect-detection"><i class="fa fa-check"></i><b>1.5</b> Batch effect detection</a>
<ul>
<li class="chapter" data-level="1.5.1" data-path="index.html"><a href="index.html#pca"><i class="fa fa-check"></i><b>1.5.1</b> PCA</a></li>
<li class="chapter" data-level="1.5.2" data-path="index.html"><a href="index.html#boxplots-and-density-plots"><i class="fa fa-check"></i><b>1.5.2</b> Boxplots and density plots</a></li>
<li class="chapter" data-level="1.5.3" data-path="index.html"><a href="index.html#heatmap"><i class="fa fa-check"></i><b>1.5.3</b> Heatmap</a></li>
<li class="chapter" data-level="1.5.4" data-path="index.html"><a href="index.html#prda"><i class="fa fa-check"></i><b>1.5.4</b> pRDA</a></li>
</ul></li>
<li class="chapter" data-level="1.6" data-path="index.html"><a href="index.html#managing-batch-effects"><i class="fa fa-check"></i><b>1.6</b> Managing batch effects</a>
<ul>
<li class="chapter" data-level="1.6.1" data-path="index.html"><a href="index.html#accnt"><i class="fa fa-check"></i><b>1.6.1</b> Accounting for batch effects</a></li>
<li class="chapter" data-level="1.6.2" data-path="index.html"><a href="index.html#corrct"><i class="fa fa-check"></i><b>1.6.2</b> Correcting for batch effects</a></li>
</ul></li>
<li class="chapter" data-level="1.7" data-path="index.html"><a href="index.html#assessing-batch-effect-correction"><i class="fa fa-check"></i><b>1.7</b> Assessing batch effect correction</a>
<ul>
<li class="chapter" data-level="1.7.1" data-path="index.html"><a href="index.html#methods-that-detect-batch-effects"><i class="fa fa-check"></i><b>1.7.1</b> Methods that detect batch effects</a></li>
<li class="chapter" data-level="1.7.2" data-path="index.html"><a href="index.html#other-methods-1"><i class="fa fa-check"></i><b>1.7.2</b> Other methods</a></li>
</ul></li>
<li class="chapter" data-level="1.8" data-path="index.html"><a href="index.html#variable-selection"><i class="fa fa-check"></i><b>1.8</b> Variable selection</a></li>
<li class="chapter" data-level="1.9" data-path="index.html"><a href="index.html#summary"><i class="fa fa-check"></i><b>1.9</b> Summary</a></li>
<li class="chapter" data-level="1.10" data-path="index.html"><a href="index.html#references"><i class="fa fa-check"></i><b>1.10</b> References</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="supplementary-materials-for-batch-effects-management-in-case-studies.html"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html"><i class="fa fa-check"></i><b>2</b> Supplementary Materials for Batch Effects Management in Case Studies</a>
<ul>
<li class="chapter" data-level="2.1" data-path="supplementary-materials-for-batch-effects-management-in-case-studies.html"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#sponge-data"><i class="fa fa-check"></i><b>2.1</b> Sponge data</a>
<ul>
<li class="chapter" data-level="2.1.1" data-path="supplementary-materials-for-batch-effects-management-in-case-studies.html"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#data-pre-processing-1"><i class="fa fa-check"></i><b>2.1.1</b> Data pre-processing</a></li>
<li class="chapter" data-level="2.1.2" data-path="supplementary-materials-for-batch-effects-management-in-case-studies.html"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#batch-effect-detection-1"><i class="fa fa-check"></i><b>2.1.2</b> Batch effect detection</a></li>
<li class="chapter" data-level="2.1.3" data-path="supplementary-materials-for-batch-effects-management-in-case-studies.html"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#managing-batch-effects-1"><i class="fa fa-check"></i><b>2.1.3</b> Managing batch effects</a></li>
<li class="chapter" data-level="2.1.4" data-path="supplementary-materials-for-batch-effects-management-in-case-studies.html"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#assessing-batch-effect-correction-1"><i class="fa fa-check"></i><b>2.1.4</b> Assessing batch effect correction</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="supplementary-materials-for-batch-effects-management-in-case-studies.html"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#hfhs-data"><i class="fa fa-check"></i><b>2.2</b> HFHS data</a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="supplementary-materials-for-batch-effects-management-in-case-studies.html"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#data-pre-processing-2"><i class="fa fa-check"></i><b>2.2.1</b> Data pre-processing</a></li>
<li class="chapter" data-level="2.2.2" data-path="supplementary-materials-for-batch-effects-management-in-case-studies.html"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#batch-effect-detection-2"><i class="fa fa-check"></i><b>2.2.2</b> Batch effect detection</a></li>
<li class="chapter" data-level="2.2.3" data-path="supplementary-materials-for-batch-effects-management-in-case-studies.html"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#managing-batch-effects-2"><i class="fa fa-check"></i><b>2.2.3</b> Managing batch effects</a></li>
<li class="chapter" data-level="2.2.4" data-path="supplementary-materials-for-batch-effects-management-in-case-studies.html"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#assessing-batch-effect-correction-2"><i class="fa fa-check"></i><b>2.2.4</b> Assessing batch effect correction</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="supplementary-materials-for-batch-effects-management-in-case-studies.html"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#hd-data"><i class="fa fa-check"></i><b>2.3</b> HD data</a>
<ul>
<li class="chapter" data-level="2.3.1" data-path="supplementary-materials-for-batch-effects-management-in-case-studies.html"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#data-pre-processing-3"><i class="fa fa-check"></i><b>2.3.1</b> Data pre-processing</a></li>
<li class="chapter" data-level="2.3.2" data-path="supplementary-materials-for-batch-effects-management-in-case-studies.html"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#batch-effect-detection-3"><i class="fa fa-check"></i><b>2.3.2</b> Batch effect detection</a></li>
<li class="chapter" data-level="2.3.3" data-path="supplementary-materials-for-batch-effects-management-in-case-studies.html"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#managing-batch-effects-3"><i class="fa fa-check"></i><b>2.3.3</b> Managing batch effects</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="supplementary-materials-for-batch-effects-management-in-case-studies.html"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#references-1"><i class="fa fa-check"></i><b>2.4</b> References</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="batch-effects-management-in-simulation-studies-gaussian-distribution.html"><a href="batch-effects-management-in-simulation-studies-gaussian-distribution.html"><i class="fa fa-check"></i><b>3</b> Batch Effects Management in Simulation Studies (Gaussian Distribution)</a>
<ul>
<li class="chapter" data-level="3.1" data-path="batch-effects-management-in-simulation-studies-gaussian-distribution.html"><a href="batch-effects-management-in-simulation-studies-gaussian-distribution.html#introduction-1"><i class="fa fa-check"></i><b>3.1</b> Introduction</a></li>
<li class="chapter" data-level="3.2" data-path="batch-effects-management-in-simulation-studies-gaussian-distribution.html"><a href="batch-effects-management-in-simulation-studies-gaussian-distribution.html#simulations"><i class="fa fa-check"></i><b>3.2</b> Simulations</a>
<ul>
<li class="chapter" data-level="3.2.1" data-path="batch-effects-management-in-simulation-studies-gaussian-distribution.html"><a href="batch-effects-management-in-simulation-studies-gaussian-distribution.html#balanced-batch-times-treatment-design"><i class="fa fa-check"></i><b>3.2.1</b> Balanced batch <span class="math inline">\(\times\)</span> treatment design</a></li>
<li class="chapter" data-level="3.2.2" data-path="batch-effects-management-in-simulation-studies-gaussian-distribution.html"><a href="batch-effects-management-in-simulation-studies-gaussian-distribution.html#figures"><i class="fa fa-check"></i><b>3.2.2</b> Figures</a></li>
<li class="chapter" data-level="3.2.3" data-path="batch-effects-management-in-simulation-studies-gaussian-distribution.html"><a href="batch-effects-management-in-simulation-studies-gaussian-distribution.html#unbalanced-batch-times-treatment-design"><i class="fa fa-check"></i><b>3.2.3</b> Unbalanced batch <span class="math inline">\(\times\)</span> treatment design</a></li>
<li class="chapter" data-level="3.2.4" data-path="batch-effects-management-in-simulation-studies-gaussian-distribution.html"><a href="batch-effects-management-in-simulation-studies-gaussian-distribution.html#figures-1"><i class="fa fa-check"></i><b>3.2.4</b> Figures</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="batch-effects-management-in-simulation-studies-gaussian-distribution.html"><a href="batch-effects-management-in-simulation-studies-gaussian-distribution.html#references-2"><i class="fa fa-check"></i><b>3.3</b> References</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="batch-effects-management-in-simulation-studies-negative-binomial-distribution.html"><a href="batch-effects-management-in-simulation-studies-negative-binomial-distribution.html"><i class="fa fa-check"></i><b>4</b> Batch Effects Management in Simulation Studies (Negative Binomial Distribution)</a>
<ul>
<li class="chapter" data-level="4.1" data-path="batch-effects-management-in-simulation-studies-negative-binomial-distribution.html"><a href="batch-effects-management-in-simulation-studies-negative-binomial-distribution.html#introduction-2"><i class="fa fa-check"></i><b>4.1</b> Introduction</a></li>
<li class="chapter" data-level="4.2" data-path="batch-effects-management-in-simulation-studies-negative-binomial-distribution.html"><a href="batch-effects-management-in-simulation-studies-negative-binomial-distribution.html#simulations-two-batch-groups"><i class="fa fa-check"></i><b>4.2</b> Simulations (two batch groups)</a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="batch-effects-management-in-simulation-studies-negative-binomial-distribution.html"><a href="batch-effects-management-in-simulation-studies-negative-binomial-distribution.html#balanced-batch-times-treatment-design-1"><i class="fa fa-check"></i><b>4.2.1</b> Balanced batch <span class="math inline">\(\times\)</span> treatment design</a></li>
<li class="chapter" data-level="4.2.2" data-path="batch-effects-management-in-simulation-studies-negative-binomial-distribution.html"><a href="batch-effects-management-in-simulation-studies-negative-binomial-distribution.html#figures-2"><i class="fa fa-check"></i><b>4.2.2</b> Figures</a></li>
<li class="chapter" data-level="4.2.3" data-path="batch-effects-management-in-simulation-studies-negative-binomial-distribution.html"><a href="batch-effects-management-in-simulation-studies-negative-binomial-distribution.html#unbalanced-batch-times-treatment-design-1"><i class="fa fa-check"></i><b>4.2.3</b> Unbalanced batch <span class="math inline">\(\times\)</span> treatment design</a></li>
<li class="chapter" data-level="4.2.4" data-path="batch-effects-management-in-simulation-studies-negative-binomial-distribution.html"><a href="batch-effects-management-in-simulation-studies-negative-binomial-distribution.html#figures-3"><i class="fa fa-check"></i><b>4.2.4</b> Figures</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="batch-effects-management-in-simulation-studies-negative-binomial-distribution.html"><a href="batch-effects-management-in-simulation-studies-negative-binomial-distribution.html#simulations-three-batch-groups"><i class="fa fa-check"></i><b>4.3</b> Simulations (three batch groups)</a>
<ul>
<li class="chapter" data-level="4.3.1" data-path="batch-effects-management-in-simulation-studies-negative-binomial-distribution.html"><a href="batch-effects-management-in-simulation-studies-negative-binomial-distribution.html#balanced-batch-times-treatment-design-2"><i class="fa fa-check"></i><b>4.3.1</b> Balanced batch <span class="math inline">\(\times\)</span> treatment design</a></li>
<li class="chapter" data-level="4.3.2" data-path="batch-effects-management-in-simulation-studies-negative-binomial-distribution.html"><a href="batch-effects-management-in-simulation-studies-negative-binomial-distribution.html#figures-4"><i class="fa fa-check"></i><b>4.3.2</b> Figures</a></li>
<li class="chapter" data-level="4.3.3" data-path="batch-effects-management-in-simulation-studies-negative-binomial-distribution.html"><a href="batch-effects-management-in-simulation-studies-negative-binomial-distribution.html#unbalanced-batch-times-treatment-design-2"><i class="fa fa-check"></i><b>4.3.3</b> Unbalanced batch <span class="math inline">\(\times\)</span> treatment design</a></li>
<li class="chapter" data-level="4.3.4" data-path="batch-effects-management-in-simulation-studies-negative-binomial-distribution.html"><a href="batch-effects-management-in-simulation-studies-negative-binomial-distribution.html#figures-5"><i class="fa fa-check"></i><b>4.3.4</b> Figures</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="batch-effects-management-in-simulation-studies-negative-binomial-distribution.html"><a href="batch-effects-management-in-simulation-studies-negative-binomial-distribution.html#references-3"><i class="fa fa-check"></i><b>4.4</b> References</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="references-4.html"><a href="references-4.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/EvaYiwenWang/PLSDAbatch_workflow" target="blank">Source codes on github</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">PLSDA-batch: a multivariate framework to correct for batch effects in microbiome data</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="supplementary-materials-for-batch-effects-management-in-case-studies" class="section level1 hasAnchor" number="2">
<h1><span class="header-section-number">Chapter 2</span> Supplementary Materials for Batch Effects Management in Case Studies<a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#supplementary-materials-for-batch-effects-management-in-case-studies" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="sponge-data" class="section level2 hasAnchor" number="2.1">
<h2><span class="header-section-number">2.1</span> Sponge data<a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#sponge-data" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="data-pre-processing-1" class="section level3 hasAnchor" number="2.1.1">
<h3><span class="header-section-number">2.1.1</span> Data pre-processing<a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#data-pre-processing-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="pre-filtering-1" class="section level4 hasAnchor" number="2.1.1.1">
<h4><span class="header-section-number">2.1.1.1</span> Pre-filtering<a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#pre-filtering-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>We load the <span style="color: #964B00;">sponge data</span> stored internally with function <code>data()</code>.</p>
<div class="sourceCode" id="cb94"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb94-1"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb94-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Sponge data</span></span>
<span id="cb94-2"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb94-2" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&#39;sponge_data&#39;</span>) </span>
<span id="cb94-3"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb94-3" aria-hidden="true" tabindex="-1"></a>sponge.tss <span class="ot">&lt;-</span> sponge_data<span class="sc">$</span>X.tss</span>
<span id="cb94-4"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb94-4" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(sponge.tss)</span></code></pre></div>
<pre><code>## [1] 32 24</code></pre>
<div class="sourceCode" id="cb96"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb96-1"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb96-1" aria-hidden="true" tabindex="-1"></a><span class="co"># zero proportion</span></span>
<span id="cb96-2"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb96-2" aria-hidden="true" tabindex="-1"></a><span class="fu">mean</span>(sponge.tss <span class="sc">==</span> <span class="dv">0</span>)</span></code></pre></div>
<pre><code>## [1] 0.5455729</code></pre>
<p>The <span style="color: #964B00;">sponge data</span> include the relative abundance of 24 OTUs and 32 samples. Given the small number of OTUs we advise not to pre-filter the data.</p>
</div>
<div id="transformation-1" class="section level4 hasAnchor" number="2.1.1.2">
<h4><span class="header-section-number">2.1.1.2</span> Transformation<a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#transformation-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Prior to CLR transformation, we recommend adding 0.01 as the offset for the <span style="color: #964B00;">sponge data</span> - that are relative abundance data. We use <code>logratio.transfo()</code> function in <span style="color: #FF7F00;">mixOmics</span> package to CLR transform the data.</p>
<div class="sourceCode" id="cb98"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb98-1"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb98-1" aria-hidden="true" tabindex="-1"></a>sponge.clr <span class="ot">&lt;-</span> <span class="fu">logratio.transfo</span>(<span class="at">X =</span> sponge.tss, <span class="at">logratio =</span> <span class="st">&#39;CLR&#39;</span>, <span class="at">offset =</span> <span class="fl">0.01</span>)</span>
<span id="cb98-2"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb98-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb98-3"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb98-3" aria-hidden="true" tabindex="-1"></a><span class="fu">class</span>(sponge.clr) <span class="ot">&lt;-</span> <span class="st">&#39;matrix&#39;</span></span></code></pre></div>
</div>
</div>
<div id="batch-effect-detection-1" class="section level3 hasAnchor" number="2.1.2">
<h3><span class="header-section-number">2.1.2</span> Batch effect detection<a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#batch-effect-detection-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="pca-1" class="section level4 hasAnchor" number="2.1.2.1">
<h4><span class="header-section-number">2.1.2.1</span> PCA<a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#pca-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>We apply <code>pca()</code> function from <span style="color: #FF7F00;">mixOmics</span> package to the <span style="color: #964B00;">sponge data</span> and <code>Scatter_Density()</code> function from <span style="color: #FF7F00;">PLSDAbatch</span> to represent the PCA sample plot with densities.</p>
<div class="sourceCode" id="cb99"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb99-1"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb99-1" aria-hidden="true" tabindex="-1"></a>sponge.pca.before <span class="ot">&lt;-</span> <span class="fu">pca</span>(sponge.clr, <span class="at">ncomp =</span> <span class="dv">3</span>, <span class="at">scale =</span> <span class="cn">TRUE</span>)</span>
<span id="cb99-2"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb99-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb99-3"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb99-3" aria-hidden="true" tabindex="-1"></a>sponge.batch <span class="ot">&lt;-</span> sponge_data<span class="sc">$</span>Y.bat</span>
<span id="cb99-4"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb99-4" aria-hidden="true" tabindex="-1"></a>sponge.trt <span class="ot">&lt;-</span> sponge_data<span class="sc">$</span>Y.trt</span>
<span id="cb99-5"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb99-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb99-6"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb99-6" aria-hidden="true" tabindex="-1"></a><span class="fu">Scatter_Density</span>(<span class="at">object =</span> sponge.pca.before, </span>
<span id="cb99-7"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb99-7" aria-hidden="true" tabindex="-1"></a>                <span class="at">batch =</span> sponge.batch, </span>
<span id="cb99-8"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb99-8" aria-hidden="true" tabindex="-1"></a>                <span class="at">trt =</span> sponge.trt, <span class="at">title =</span> <span class="st">&#39;Sponge data&#39;</span>, </span>
<span id="cb99-9"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb99-9" aria-hidden="true" tabindex="-1"></a>                <span class="at">trt.legend.title =</span> <span class="st">&#39;Tissue types&#39;</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:SpongepcaBefore"></span>
<img src="PLSDAbatch_workflow_files/figure-html/SpongepcaBefore-1.png" alt="The PCA sample plot with densities in the sponge data." width="60%" />
<p class="caption">
Figure 2.1: The PCA sample plot with densities in the sponge data.
</p>
</div>
<p>In the above figure, the tissue variation (effects of interest) accounted for 24% of data variance on component 1, while the gel variation (batch effects) for 21% on component 2. Therefore, the batch effect is slightly weaker than the treatment effect in the <span style="color: #964B00;">sponge data</span>.</p>
</div>
<div id="heatmap-1" class="section level4 hasAnchor" number="2.1.2.2">
<h4><span class="header-section-number">2.1.2.2</span> Heatmap<a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#heatmap-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>We produce a heatmap using <span style="color: #FF7F00;">pheatmap</span> package. The data first need to be scaled on both OTUs and samples.</p>
<div class="sourceCode" id="cb100"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb100-1"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb100-1" aria-hidden="true" tabindex="-1"></a><span class="co"># scale the clr data on both OTUs and samples</span></span>
<span id="cb100-2"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb100-2" aria-hidden="true" tabindex="-1"></a>sponge.clr.s <span class="ot">&lt;-</span> <span class="fu">scale</span>(sponge.clr, <span class="at">center =</span> T, <span class="at">scale =</span> T)</span>
<span id="cb100-3"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb100-3" aria-hidden="true" tabindex="-1"></a>sponge.clr.ss <span class="ot">&lt;-</span> <span class="fu">scale</span>(<span class="fu">t</span>(sponge.clr.s), <span class="at">center =</span> T, <span class="at">scale =</span> T)</span>
<span id="cb100-4"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb100-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb100-5"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb100-5" aria-hidden="true" tabindex="-1"></a>sponge.anno_col <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">Batch =</span> sponge.batch, <span class="at">Tissue =</span> sponge.trt)</span>
<span id="cb100-6"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb100-6" aria-hidden="true" tabindex="-1"></a>sponge.anno_colors <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="at">Batch =</span> <span class="fu">color.mixo</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>), </span>
<span id="cb100-7"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb100-7" aria-hidden="true" tabindex="-1"></a>                           <span class="at">Tissue =</span> <span class="fu">pb_color</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>))</span>
<span id="cb100-8"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb100-8" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(sponge.anno_colors<span class="sc">$</span>Batch) <span class="ot">=</span> <span class="fu">levels</span>(sponge.batch)</span>
<span id="cb100-9"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb100-9" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(sponge.anno_colors<span class="sc">$</span>Tissue) <span class="ot">=</span> <span class="fu">levels</span>(sponge.trt)</span>
<span id="cb100-10"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb100-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb100-11"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb100-11" aria-hidden="true" tabindex="-1"></a><span class="fu">pheatmap</span>(sponge.clr.ss, </span>
<span id="cb100-12"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb100-12" aria-hidden="true" tabindex="-1"></a>         <span class="at">cluster_rows =</span> F, </span>
<span id="cb100-13"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb100-13" aria-hidden="true" tabindex="-1"></a>         <span class="at">fontsize_row =</span> <span class="dv">4</span>, </span>
<span id="cb100-14"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb100-14" aria-hidden="true" tabindex="-1"></a>         <span class="at">fontsize_col =</span> <span class="dv">6</span>,</span>
<span id="cb100-15"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb100-15" aria-hidden="true" tabindex="-1"></a>         <span class="at">fontsize =</span> <span class="dv">8</span>,</span>
<span id="cb100-16"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb100-16" aria-hidden="true" tabindex="-1"></a>         <span class="at">clustering_distance_rows =</span> <span class="st">&#39;euclidean&#39;</span>,</span>
<span id="cb100-17"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb100-17" aria-hidden="true" tabindex="-1"></a>         <span class="at">clustering_method =</span> <span class="st">&#39;ward.D&#39;</span>,</span>
<span id="cb100-18"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb100-18" aria-hidden="true" tabindex="-1"></a>         <span class="at">treeheight_row =</span> <span class="dv">30</span>,</span>
<span id="cb100-19"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb100-19" aria-hidden="true" tabindex="-1"></a>         <span class="at">annotation_col =</span> sponge.anno_col,</span>
<span id="cb100-20"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb100-20" aria-hidden="true" tabindex="-1"></a>         <span class="at">annotation_colors =</span> sponge.anno_colors,</span>
<span id="cb100-21"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb100-21" aria-hidden="true" tabindex="-1"></a>         <span class="at">border_color =</span> <span class="st">&#39;NA&#39;</span>,</span>
<span id="cb100-22"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb100-22" aria-hidden="true" tabindex="-1"></a>         <span class="at">main =</span> <span class="st">&#39;Sponge data - Scaled&#39;</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:Spongeheatmap"></span>
<img src="PLSDAbatch_workflow_files/figure-html/Spongeheatmap-1.png" alt="Hierarchical clustering for samples in the sponge data." width="90%" />
<p class="caption">
Figure 2.2: Hierarchical clustering for samples in the sponge data.
</p>
</div>
<p>In the heatmap, samples in the <span style="color: #964B00;">sponge data</span> were clustered or grouped by batches instead of tissues, indicating a batch effect.</p>
</div>
<div id="prda-1" class="section level4 hasAnchor" number="2.1.2.3">
<h4><span class="header-section-number">2.1.2.3</span> pRDA<a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#prda-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>We apply pRDA with <code>varpart()</code> function from <span style="color: #FF7F00;">vegan</span> R package.</p>
<div class="sourceCode" id="cb101"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb101-1"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb101-1" aria-hidden="true" tabindex="-1"></a>sponge.factors.df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">trt =</span> sponge.trt, </span>
<span id="cb101-2"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb101-2" aria-hidden="true" tabindex="-1"></a>                                <span class="at">batch =</span> sponge.batch)</span>
<span id="cb101-3"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb101-3" aria-hidden="true" tabindex="-1"></a>sponge.rda.before <span class="ot">&lt;-</span> <span class="fu">varpart</span>(sponge.clr, <span class="sc">~</span> trt, <span class="sc">~</span> batch, </span>
<span id="cb101-4"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb101-4" aria-hidden="true" tabindex="-1"></a>                             <span class="at">data =</span> sponge.factors.df, </span>
<span id="cb101-5"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb101-5" aria-hidden="true" tabindex="-1"></a>                             <span class="at">scale =</span> T)</span>
<span id="cb101-6"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb101-6" aria-hidden="true" tabindex="-1"></a>sponge.rda.before<span class="sc">$</span>part<span class="sc">$</span>indfract</span></code></pre></div>
<pre><code>##                 Df R.squared Adj.R.squared Testable
## [a] = X1|X2      1        NA    0.16572246     TRUE
## [b] = X2|X1      1        NA    0.16396277     TRUE
## [c]              0        NA   -0.01063501    FALSE
## [d] = Residuals NA        NA    0.68094977    FALSE</code></pre>
<p>In the result, <code>X1</code> and <code>X2</code> represent the first and second covariates fitted in the model. <code>[a]</code>, <code>[b]</code> represent the independent proportion of variance explained by <code>X1</code> and <code>X2</code> respectively, and <code>[c]</code> represents the intersection variance shared between <code>X1</code> and <code>X2</code>. The <span style="color: #964B00;">sponge data</span> have a completely balanced batch x treatment design, so there was no intersection variance (indicated in line <code>[c]</code>, Adj.R.squared = -0.01) detected. The proportion of treatment and batch variance was nearly equal as indicated in lines <code>[a]</code> and <code>[b]</code>.</p>
</div>
</div>
<div id="managing-batch-effects-1" class="section level3 hasAnchor" number="2.1.3">
<h3><span class="header-section-number">2.1.3</span> Managing batch effects<a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#managing-batch-effects-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="accounting-for-batch-effects" class="section level4 hasAnchor" number="2.1.3.1">
<h4><span class="header-section-number">2.1.3.1</span> Accounting for batch effects<a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#accounting-for-batch-effects" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>The methods that we use to account for batch effects include the method designed for microbiome data: zero-inflated Gaussian (ZIG) mixture model and the methods adapted for microbiome data: linear regression, SVA and RUV4. Among them, SVA and RUV4 are designed for unknown batch effects.</p>
<p>As the ZIG model is applicable to counts data, the <span style="color: #964B00;">sponge data</span> are not qualified for this model. We therefore only apply the methods that can be adapted for microbiome data.</p>
<p><strong>Linear regression</strong></p>
<p>Linear regression is conducted with <code>linear_regres()</code> function in <span style="color: #FF7F00;">PLSDAbatch</span>. We integrated the <span style="color: #FF7F00;">performance</span> package that assesses performance of regression models into our function <code>linear_regres()</code>. Therefore, we can apply <code>check_model()</code> from <span style="color: #FF7F00;">performance</span> to the outputs from <code>linear_regres()</code> to diagnose the validity of the model fitted with treatment and batch effects for each variable <span class="citation">(<a href="#ref-daniel2020performance" role="doc-biblioref">LÃŒdecke et al. 2020</a>)</span>. We can extract performance measurements such as adjusted R2, RMSE, RSE, AIC and BIC for the models fitted with and without batch effects, which are also the outputs of <code>linear_regres()</code>.</p>
<p>We apply <code>type = "linear model"</code> to the <span style="color: #964B00;">sponge data</span> because of the balanced batch x treatment design.</p>
<div class="sourceCode" id="cb103"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb103-1"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb103-1" aria-hidden="true" tabindex="-1"></a>sponge.clr <span class="ot">&lt;-</span> sponge.clr[<span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(sponge.clr), <span class="dv">1</span><span class="sc">:</span><span class="fu">ncol</span>(sponge.clr)]</span>
<span id="cb103-2"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb103-2" aria-hidden="true" tabindex="-1"></a>sponge.lm <span class="ot">&lt;-</span> <span class="fu">linear_regres</span>(<span class="at">data =</span> sponge.clr, </span>
<span id="cb103-3"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb103-3" aria-hidden="true" tabindex="-1"></a>                           <span class="at">trt =</span> sponge.trt, </span>
<span id="cb103-4"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb103-4" aria-hidden="true" tabindex="-1"></a>                           <span class="at">batch.fix =</span> sponge.batch, </span>
<span id="cb103-5"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb103-5" aria-hidden="true" tabindex="-1"></a>                           <span class="at">type =</span> <span class="st">&#39;linear model&#39;</span>)</span>
<span id="cb103-6"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb103-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb103-7"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb103-7" aria-hidden="true" tabindex="-1"></a>sponge.p <span class="ot">&lt;-</span> <span class="fu">sapply</span>(sponge.lm<span class="sc">$</span>lm.table, <span class="cf">function</span>(x){x<span class="sc">$</span>coefficients[<span class="dv">2</span>,<span class="dv">4</span>]})</span>
<span id="cb103-8"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb103-8" aria-hidden="true" tabindex="-1"></a>sponge.p.adj <span class="ot">&lt;-</span> <span class="fu">p.adjust</span>(<span class="at">p =</span> sponge.p, <span class="at">method =</span> <span class="st">&#39;fdr&#39;</span>)</span>
<span id="cb103-9"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb103-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb103-10"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb103-10" aria-hidden="true" tabindex="-1"></a><span class="fu">check_model</span>(sponge.lm<span class="sc">$</span>model<span class="sc">$</span>OTU2)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:Spongelm"></span>
<img src="PLSDAbatch_workflow_files/figure-html/Spongelm-1.png" alt="Diagnostic plots for the model fitted with batch effects of &quot;OTU2&quot; in the sponge data." width="100%" />
<p class="caption">
Figure 2.3: Diagnostic plots for the model fitted with batch effects of “OTU2” in the sponge data.
</p>
</div>
<p>To diagnose the validity of the model fitted with both treatment and batch effects, we use different plots to check the assumptions of each microbial variable. For example, the diagnostic plots of “OTU2” are shown in the above figure panel. The simulated data under the fitted model were not very close to the real data (shown as green line), indicating an imperfect fitness (top panel). The linearity (or homoscedasticity) and homogeneity of variance were not satisfied. The correlation between batch (<code>batch.fix</code>) and treatment (<code>trt</code>) effects was very low, indicating a good model with low collinearity. Some samples could be classified as outliers with a Cook’s distance larger than or equal to 0.5, for example, “21”, “15”, “7”, “1” and “13” (middle panel). The distribution of residuals was not close to normal (bottom panel). For the microbial variables with some assumptions not met, we should be careful about their results.</p>
<p>For performance measurements of models fitted with or without batch effects, we show an example of the results for some variables.</p>
<div class="sourceCode" id="cb104"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb104-1"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb104-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(sponge.lm<span class="sc">$</span>adj.R2)</span></code></pre></div>
<pre><code>##         trt.only  trt.batch
## OTU1  0.06423228 0.05701977
## OTU2  0.60492094 0.67117014
## OTU3  0.19003076 0.18125188
## OTU4 -0.03327738 0.23731780
## OTU5  0.97483298 0.97643741
## OTU6  0.97496003 0.97634939</code></pre>
<p>If the adjusted <span class="math inline">\(R^2\)</span> of the model with both treatment and batch effects is larger than the model with treatment effects only, e.g., OTU2, OTU4, OTU5 and OTU6, the model fitted with batch effects explains more data variance, and is thus better than the model without batch effects. Otherwise, the batch effect is not necessary to be added into the linear model.</p>
<p>We next look at the AIC of models fitted with or without batch effects.</p>
<div class="sourceCode" id="cb106"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb106-1"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb106-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(sponge.lm<span class="sc">$</span>AIC)</span></code></pre></div>
<pre><code>##       trt.only  trt.batch
## OTU1 47.901506 49.0623531
## OTU2 74.001621 69.0433180
## OTU3 -5.630456 -4.3703385
## OTU4 90.699895 81.8982606
## OTU5 21.727333 20.5345128
## OTU6  1.196912  0.2853681</code></pre>
<p>A lower AIC indicates a better fit. Both results of the adjusted <span class="math inline">\(R^2\)</span> and AIC were consistent on model selection for the listed OTUs.</p>
<p><strong>SVA</strong></p>
<p>SVA accounts for unknown batch effects. Here we assume that the batch grouping information in the <span style="color: #964B00;">sponge data</span> is unknown. We first build two design matrices with (<code>sponge.mod</code>) and without (<code>sponge.mod0</code>) treatment grouping information generated with <code>model.matrix()</code> function from <span style="color: #FF7F00;">stats</span>. We then use <code>num.sv()</code> from <span style="color: #FF7F00;">sva</span> package to determine the number of batch variables <code>n.sv</code> that is used to estimate batch effects in function <code>sva()</code>.</p>
<div class="sourceCode" id="cb108"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb108-1"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb108-1" aria-hidden="true" tabindex="-1"></a><span class="co"># estimate batch matrix</span></span>
<span id="cb108-2"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb108-2" aria-hidden="true" tabindex="-1"></a>sponge.mod <span class="ot">&lt;-</span> <span class="fu">model.matrix</span>( <span class="sc">~</span> sponge.trt)</span>
<span id="cb108-3"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb108-3" aria-hidden="true" tabindex="-1"></a>sponge.mod0 <span class="ot">&lt;-</span> <span class="fu">model.matrix</span>( <span class="sc">~</span> <span class="dv">1</span>, <span class="at">data =</span> sponge.trt)</span>
<span id="cb108-4"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb108-4" aria-hidden="true" tabindex="-1"></a>sponge.sva.n <span class="ot">&lt;-</span> <span class="fu">num.sv</span>(<span class="at">dat =</span> <span class="fu">t</span>(sponge.clr), <span class="at">mod =</span> sponge.mod, <span class="at">method =</span> <span class="st">&#39;leek&#39;</span>)</span>
<span id="cb108-5"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb108-5" aria-hidden="true" tabindex="-1"></a>sponge.sva <span class="ot">&lt;-</span> <span class="fu">sva</span>(<span class="fu">t</span>(sponge.clr), sponge.mod, sponge.mod0, <span class="at">n.sv =</span> sponge.sva.n)</span></code></pre></div>
<pre><code>## Number of significant surrogate variables is:  1 
## Iteration (out of 5 ):1  2  3  4  5</code></pre>
<p>The estimated batch effects are then applied to <code>f.pvalue()</code> to calculate the P-values of treatment effects. The estimated batch effects in SVA are assumed to be independent of the treatment effects. However, SVA considers some correlation between batch and treatment effects <span class="citation">(<a href="#ref-wang2020managing" role="doc-biblioref">Y. Wang and LêCao 2020</a>)</span>.</p>
<div class="sourceCode" id="cb110"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb110-1"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb110-1" aria-hidden="true" tabindex="-1"></a><span class="co"># include estimated batch effects in the linear model</span></span>
<span id="cb110-2"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb110-2" aria-hidden="true" tabindex="-1"></a>sponge.mod.batch <span class="ot">&lt;-</span> <span class="fu">cbind</span>(sponge.mod, sponge.sva<span class="sc">$</span>sv)</span>
<span id="cb110-3"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb110-3" aria-hidden="true" tabindex="-1"></a>sponge.mod0.batch <span class="ot">&lt;-</span> <span class="fu">cbind</span>(sponge.mod0, sponge.sva<span class="sc">$</span>sv)</span>
<span id="cb110-4"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb110-4" aria-hidden="true" tabindex="-1"></a>sponge.sva.p <span class="ot">&lt;-</span> <span class="fu">f.pvalue</span>(<span class="fu">t</span>(sponge.clr), sponge.mod.batch, sponge.mod0.batch)</span>
<span id="cb110-5"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb110-5" aria-hidden="true" tabindex="-1"></a>sponge.sva.p.adj <span class="ot">&lt;-</span> <span class="fu">p.adjust</span>(sponge.sva.p, <span class="at">method =</span> <span class="st">&#39;fdr&#39;</span>)</span></code></pre></div>
<p><strong>RUV4</strong></p>
<p>Before applying RUV4 (<code>RUV4()</code> from <span style="color: #FF7F00;">ruv</span> package), we need to specify negative control variables and the number of batch variables to estimate. We can use the empirical negative controls that are not significantly differentially abundant (adjusted P &gt; 0.05) from a linear regression with the treatment information as the only covariate.</p>
<p>We use a loop to fit a linear regression for each microbial variable and adjust P values of treatment effects for multiple comparisons with <code>p.adjust()</code> from <span style="color: #FF7F00;">stats</span>. The empirical negative controls are then extracted according to the adjusted P values.</p>
<div class="sourceCode" id="cb111"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb111-1"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb111-1" aria-hidden="true" tabindex="-1"></a><span class="co"># empirical negative controls</span></span>
<span id="cb111-2"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb111-2" aria-hidden="true" tabindex="-1"></a>sponge.empir.p <span class="ot">&lt;-</span> <span class="fu">c</span>()</span>
<span id="cb111-3"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb111-3" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(e <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">ncol</span>(sponge.clr)){</span>
<span id="cb111-4"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb111-4" aria-hidden="true" tabindex="-1"></a>  sponge.empir.lm <span class="ot">&lt;-</span> <span class="fu">lm</span>(sponge.clr[,e] <span class="sc">~</span> sponge.trt)</span>
<span id="cb111-5"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb111-5" aria-hidden="true" tabindex="-1"></a>  sponge.empir.p[e] <span class="ot">&lt;-</span> <span class="fu">summary</span>(sponge.empir.lm)<span class="sc">$</span>coefficients[<span class="dv">2</span>,<span class="dv">4</span>]</span>
<span id="cb111-6"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb111-6" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb111-7"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb111-7" aria-hidden="true" tabindex="-1"></a>sponge.empir.p.adj <span class="ot">&lt;-</span> <span class="fu">p.adjust</span>(<span class="at">p =</span> sponge.empir.p, <span class="at">method =</span> <span class="st">&#39;fdr&#39;</span>)</span>
<span id="cb111-8"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb111-8" aria-hidden="true" tabindex="-1"></a>sponge.nc <span class="ot">&lt;-</span> sponge.empir.p.adj <span class="sc">&gt;</span> <span class="fl">0.05</span></span></code></pre></div>
<p>The number of batch variables <code>k</code> can be determined using <code>getK()</code> function.</p>
<div class="sourceCode" id="cb112"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb112-1"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb112-1" aria-hidden="true" tabindex="-1"></a><span class="co"># estimate k</span></span>
<span id="cb112-2"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb112-2" aria-hidden="true" tabindex="-1"></a>sponge.k.res <span class="ot">&lt;-</span> <span class="fu">getK</span>(<span class="at">Y =</span> sponge.clr, <span class="at">X =</span> sponge.trt, <span class="at">ctl =</span> sponge.nc)</span>
<span id="cb112-3"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb112-3" aria-hidden="true" tabindex="-1"></a>sponge.k <span class="ot">&lt;-</span> sponge.k.res<span class="sc">$</span>k</span>
<span id="cb112-4"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb112-4" aria-hidden="true" tabindex="-1"></a>sponge.k <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(sponge.k <span class="sc">!=</span> <span class="dv">0</span>, sponge.k, <span class="dv">1</span>)</span></code></pre></div>
<p>We then apply <code>RUV4()</code> with known treatment variables, estimated negative control variables and <code>k</code> batch variables. The calculated P values also need to be adjusted for multiple comparisons.</p>
<div class="sourceCode" id="cb113"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb113-1"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb113-1" aria-hidden="true" tabindex="-1"></a>sponge.ruv4 <span class="ot">&lt;-</span> <span class="fu">RUV4</span>(<span class="at">Y =</span> sponge.clr, <span class="at">X =</span> sponge.trt, </span>
<span id="cb113-2"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb113-2" aria-hidden="true" tabindex="-1"></a>                    <span class="at">ctl =</span> sponge.nc, <span class="at">k =</span> sponge.k) </span>
<span id="cb113-3"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb113-3" aria-hidden="true" tabindex="-1"></a>sponge.ruv4.p <span class="ot">&lt;-</span> sponge.ruv4<span class="sc">$</span>p</span>
<span id="cb113-4"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb113-4" aria-hidden="true" tabindex="-1"></a>sponge.ruv4.p.adj <span class="ot">&lt;-</span> <span class="fu">p.adjust</span>(sponge.ruv4.p, <span class="at">method =</span> <span class="st">&quot;fdr&quot;</span>)</span></code></pre></div>
<p>Note: A package named <span style="color: #FF7F00;">RUVSeq</span> has been developed for count data. It provides <code>RUVg()</code> using negative control variables, and also other functions <code>RUVs()</code> and <code>RUVr()</code> using sample replicates <span class="citation">(<a href="#ref-moskovicz2020skin" role="doc-biblioref">Moskovicz et al. 2020</a>)</span> or residuals from the regression on treatment effects to estimate and then account for latent batch effects. However, for CLR-transformed data, we still recommend the standard <span style="color: #FF7F00;">ruv</span> package.</p>
</div>
<div id="correcting-for-batch-effects" class="section level4 hasAnchor" number="2.1.3.2">
<h4><span class="header-section-number">2.1.3.2</span> Correcting for batch effects<a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#correcting-for-batch-effects" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>The methods that we use to correct for batch effects include removeBatchEffect, ComBat, PLSDA-batch, sPLSDA-batch, Percentile Normalisation and RUVIII. Among them, RUVIII is designed for unknown batch effects. Except RUVIII that requires sample replicates which the <span style="color: #964B00;">sponge data</span> do not have, these methods were all applied to and illustrated with the <span style="color: #964B00;">sponge data</span>.</p>
<p><strong>removeBatchEffect</strong></p>
<p>The <code>removeBatchEffect()</code> function is implemented in <span style="color: #FF7F00;">limma</span> package. The design matrix (<code>design</code>) with treatment grouping information can be generated with <code>model.matrix()</code> function from <span style="color: #FF7F00;">stats</span> as shown in section <strong>accounting for batch effects</strong> method “SVA”.</p>
<p>Here we use <code>removeBatchEffect()</code> function with batch grouping information (<code>batch</code>) and treatment design matrix (<code>design</code>) to calculate batch effect corrected data <code>sponge.rBE</code>.</p>
<div class="sourceCode" id="cb114"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb114-1"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb114-1" aria-hidden="true" tabindex="-1"></a>sponge.rBE <span class="ot">&lt;-</span> <span class="fu">t</span>(<span class="fu">removeBatchEffect</span>(<span class="fu">t</span>(sponge.clr), <span class="at">batch =</span> sponge.batch, </span>
<span id="cb114-2"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb114-2" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">design =</span> sponge.mod))</span></code></pre></div>
<p><strong>ComBat</strong></p>
<p>The <code>ComBat()</code> function (from <span style="color: #FF7F00;">sva</span> package) is implemented as parametric or non-parametric correction with option <code>par.prior</code>. Under a parametric adjustment, we can assess the model’s validity with <code>prior.plots = T</code> <span class="citation">(<a href="#ref-leek2012sva" role="doc-biblioref">Leek et al. 2012</a>)</span>.</p>
<p>Here we use a non-parametric correction (<code>par.prior = FALSE</code>) with batch grouping information (<code>batch</code>) and treatment design matrix (<code>mod</code>) to calculate batch effect corrected data <code>sponge.ComBat</code>.</p>
<div class="sourceCode" id="cb115"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb115-1"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb115-1" aria-hidden="true" tabindex="-1"></a>sponge.ComBat <span class="ot">&lt;-</span> <span class="fu">t</span>(<span class="fu">ComBat</span>(<span class="fu">t</span>(sponge.clr), <span class="at">batch =</span> sponge.batch, </span>
<span id="cb115-2"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb115-2" aria-hidden="true" tabindex="-1"></a>                          <span class="at">mod =</span> sponge.mod, <span class="at">par.prior =</span> F))</span></code></pre></div>
<p><strong>PLSDA-batch</strong></p>
<p>The <code>PLSDA_batch()</code> function is implemented in <span style="color: #FF7F00;">PLSDAbatch</span> package. To use this function, we need to specify the optimal number of components related to treatment (<code>ncomp.trt</code>) or batch effects (<code>ncomp.bat</code>).</p>
<p>Here in the <span style="color: #964B00;">sponge data</span>, we use <code>plsda()</code> from <span style="color: #FF7F00;">mixOmics</span> with only treatment grouping information to estimate the optimal number of treatment components to preserve.</p>
<div class="sourceCode" id="cb116"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb116-1"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb116-1" aria-hidden="true" tabindex="-1"></a><span class="co"># estimate the number of treatment components</span></span>
<span id="cb116-2"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb116-2" aria-hidden="true" tabindex="-1"></a>sponge.trt.tune <span class="ot">&lt;-</span> <span class="fu">plsda</span>(<span class="at">X =</span> sponge.clr, <span class="at">Y =</span> sponge.trt, <span class="at">ncomp =</span> <span class="dv">5</span>)</span>
<span id="cb116-3"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb116-3" aria-hidden="true" tabindex="-1"></a>sponge.trt.tune<span class="sc">$</span>prop_expl_var <span class="co">#1</span></span></code></pre></div>
<pre><code>## $X
##      comp1      comp2      comp3      comp4      comp5 
## 0.22802218 0.10478658 0.16376238 0.06861581 0.04524292 
## 
## $Y
##      comp1      comp2      comp3      comp4      comp5 
## 1.00000000 0.16584332 0.06024061 0.03278432 0.01144732</code></pre>
<p>We choose the number that explains 100% variance in the outcome matrix <code>Y</code>, thus from the result, 1 component was enough to preserve the treatment information.</p>
<p>We then use <code>PLSDA_batch()</code> function with both treatment and batch grouping information to estimate the optimal number of batch components to remove.</p>
<div class="sourceCode" id="cb118"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb118-1"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb118-1" aria-hidden="true" tabindex="-1"></a><span class="co"># estimate the number of batch components</span></span>
<span id="cb118-2"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb118-2" aria-hidden="true" tabindex="-1"></a>sponge.batch.tune <span class="ot">&lt;-</span> <span class="fu">PLSDA_batch</span>(<span class="at">X =</span> sponge.clr, </span>
<span id="cb118-3"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb118-3" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">Y.trt =</span> sponge.trt, </span>
<span id="cb118-4"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb118-4" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">Y.bat =</span> sponge.batch,</span>
<span id="cb118-5"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb118-5" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">ncomp.trt =</span> <span class="dv">1</span>, </span>
<span id="cb118-6"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb118-6" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">ncomp.bat =</span> <span class="dv">5</span>)</span>
<span id="cb118-7"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb118-7" aria-hidden="true" tabindex="-1"></a>sponge.batch.tune<span class="sc">$</span>explained_variance.bat <span class="co">#1</span></span></code></pre></div>
<pre><code>## $X
##      comp1      comp2      comp3      comp4      comp5 
## 0.27660372 0.08286158 0.08187289 0.07026693 0.07081271 
## 
## $Y
## comp1 comp2 comp3 comp4 comp5 
##     1     1     1     1     1</code></pre>
<p>Using the same criterion as choosing treatment components, we choose the number of batch components that explains 100% variance in the outcome matrix of batch. According to the result, 1 component was required to remove batch effects.</p>
<p>We then can correct for batch effects applying <code>PLSDA_batch()</code> with treatment, batch grouping information and corresponding optimal number of related components.</p>
<div class="sourceCode" id="cb120"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb120-1"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb120-1" aria-hidden="true" tabindex="-1"></a>sponge.PLSDA_batch.res <span class="ot">&lt;-</span> <span class="fu">PLSDA_batch</span>(<span class="at">X =</span> sponge.clr, </span>
<span id="cb120-2"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb120-2" aria-hidden="true" tabindex="-1"></a>                                      <span class="at">Y.trt =</span> sponge.trt, </span>
<span id="cb120-3"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb120-3" aria-hidden="true" tabindex="-1"></a>                                      <span class="at">Y.bat =</span> sponge.batch,</span>
<span id="cb120-4"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb120-4" aria-hidden="true" tabindex="-1"></a>                                      <span class="at">ncomp.trt =</span> <span class="dv">1</span>, </span>
<span id="cb120-5"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb120-5" aria-hidden="true" tabindex="-1"></a>                                      <span class="at">ncomp.bat =</span> <span class="dv">1</span>)</span>
<span id="cb120-6"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb120-6" aria-hidden="true" tabindex="-1"></a>sponge.PLSDA_batch <span class="ot">&lt;-</span> sponge.PLSDA_batch.res<span class="sc">$</span>X.nobatch</span></code></pre></div>
<p><strong>sPLSDA-batch</strong></p>
<p>We apply sPLSDA-batch using the same function <code>PLSDA_batch()</code> but we specify the number of variables to select on each component (usually only treatment-related components <code>keepX.trt</code>). To determine the optimal number of variables to select, we use <code>tune.splsda()</code> function from <span style="color: #FF7F00;">mixOmics</span> package <span class="citation">(<a href="#ref-rohart2017mixomics" role="doc-biblioref">Rohart et al. 2017</a>)</span> with all possible numbers of variables to select for each component (<code>test.keepX</code>).</p>
<div class="sourceCode" id="cb121"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb121-1"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb121-1" aria-hidden="true" tabindex="-1"></a><span class="co"># estimate the number of variables to select per treatment component</span></span>
<span id="cb121-2"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb121-2" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">777</span>)</span>
<span id="cb121-3"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb121-3" aria-hidden="true" tabindex="-1"></a>sponge.test.keepX <span class="ot">=</span> <span class="fu">seq</span>(<span class="dv">1</span>, <span class="dv">24</span>, <span class="dv">1</span>)</span>
<span id="cb121-4"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb121-4" aria-hidden="true" tabindex="-1"></a>sponge.trt.tune.v <span class="ot">&lt;-</span> <span class="fu">tune.splsda</span>(<span class="at">X =</span> sponge.clr, </span>
<span id="cb121-5"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb121-5" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">Y =</span> sponge.trt, </span>
<span id="cb121-6"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb121-6" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">ncomp =</span> <span class="dv">1</span>, </span>
<span id="cb121-7"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb121-7" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">test.keepX =</span> sponge.test.keepX, </span>
<span id="cb121-8"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb121-8" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">validation =</span> <span class="st">&#39;Mfold&#39;</span>, </span>
<span id="cb121-9"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb121-9" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">folds =</span> <span class="dv">4</span>, <span class="at">nrepeat =</span> <span class="dv">50</span>)</span>
<span id="cb121-10"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb121-10" aria-hidden="true" tabindex="-1"></a>sponge.trt.tune.v<span class="sc">$</span>choice.keepX <span class="co">#1</span></span></code></pre></div>
<pre><code>## comp1 
##     1</code></pre>
<p>Here the optimal number of variables to select for the treatment component was 1. Since we have adjusted the amount of treatment variation to preserve, we need to re-choose the optimal number of components related to batch effects using the same criterion mentioned in section <strong>correcting for batch effects</strong> method “PLSDA-batch”.</p>
<div class="sourceCode" id="cb123"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb123-1"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb123-1" aria-hidden="true" tabindex="-1"></a><span class="co"># estimate the number of batch components</span></span>
<span id="cb123-2"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb123-2" aria-hidden="true" tabindex="-1"></a>sponge.batch.tune <span class="ot">&lt;-</span> <span class="fu">PLSDA_batch</span>(<span class="at">X =</span> sponge.clr, </span>
<span id="cb123-3"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb123-3" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">Y.trt =</span> sponge.trt, </span>
<span id="cb123-4"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb123-4" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">Y.bat =</span> sponge.batch,</span>
<span id="cb123-5"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb123-5" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">ncomp.trt =</span> <span class="dv">1</span>, </span>
<span id="cb123-6"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb123-6" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">keepX.trt =</span> <span class="dv">1</span>,</span>
<span id="cb123-7"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb123-7" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">ncomp.bat =</span> <span class="dv">5</span>)</span>
<span id="cb123-8"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb123-8" aria-hidden="true" tabindex="-1"></a>sponge.batch.tune<span class="sc">$</span>explained_variance.bat <span class="co">#1</span></span></code></pre></div>
<pre><code>## $X
##      comp1      comp2      comp3      comp4      comp5 
## 0.25672979 0.07716171 0.08936628 0.08434746 0.09422519 
## 
## $Y
## comp1 comp2 comp3 comp4 comp5 
##     1     1     1     1     1</code></pre>
<p>According to the result, we needed only 1 batch related component to remove batch variance from the data with function <code>PLSDA_batch()</code>.</p>
<div class="sourceCode" id="cb125"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb125-1"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb125-1" aria-hidden="true" tabindex="-1"></a>sponge.sPLSDA_batch.res <span class="ot">&lt;-</span> <span class="fu">PLSDA_batch</span>(<span class="at">X =</span> sponge.clr, </span>
<span id="cb125-2"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb125-2" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">Y.trt =</span> sponge.trt, </span>
<span id="cb125-3"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb125-3" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">Y.bat =</span> sponge.batch,</span>
<span id="cb125-4"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb125-4" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">ncomp.trt =</span> <span class="dv">1</span>, </span>
<span id="cb125-5"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb125-5" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">keepX.trt =</span> <span class="dv">1</span>,</span>
<span id="cb125-6"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb125-6" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">ncomp.bat =</span> <span class="dv">1</span>)</span>
<span id="cb125-7"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb125-7" aria-hidden="true" tabindex="-1"></a>sponge.sPLSDA_batch <span class="ot">&lt;-</span> sponge.sPLSDA_batch.res<span class="sc">$</span>X.nobatch</span></code></pre></div>
<p>Note: for unbalanced batch x treatment design (with the exception of the nested design), we can specify <code>balance = FALSE</code> in <code>PLSDA_batch()</code> function to apply weighted PLSDA-batch.</p>
<p><strong>Percentile Normalisation</strong></p>
<p>To apply <code>percentile_norm()</code> function from <span style="color: #FF7F00;">PLSDAbatch</span> package, we need to indicate a control group (<code>ctrl.grp</code>).</p>
<div class="sourceCode" id="cb126"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb126-1"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb126-1" aria-hidden="true" tabindex="-1"></a>sponge.PN <span class="ot">&lt;-</span> <span class="fu">percentile_norm</span>(<span class="at">data =</span> sponge.clr, </span>
<span id="cb126-2"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb126-2" aria-hidden="true" tabindex="-1"></a>                             <span class="at">batch =</span> sponge.batch, </span>
<span id="cb126-3"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb126-3" aria-hidden="true" tabindex="-1"></a>                             <span class="at">trt =</span> sponge.trt, </span>
<span id="cb126-4"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb126-4" aria-hidden="true" tabindex="-1"></a>                             <span class="at">ctrl.grp =</span> <span class="st">&#39;C&#39;</span>)</span></code></pre></div>
</div>
</div>
<div id="assessing-batch-effect-correction-1" class="section level3 hasAnchor" number="2.1.4">
<h3><span class="header-section-number">2.1.4</span> Assessing batch effect correction<a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#assessing-batch-effect-correction-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>We apply different visualisation and quantitative methods to assessing batch effect correction.</p>
<div id="methods-that-detect-batch-effects-1" class="section level4 hasAnchor" number="2.1.4.1">
<h4><span class="header-section-number">2.1.4.1</span> Methods that detect batch effects<a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#methods-that-detect-batch-effects-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p><strong>PCA</strong></p>
<p>In the <span style="color: #964B00;">sponge data</span>, we compared the PCA sample plots before and after batch effect correction with different methods.</p>
<div class="sourceCode" id="cb127"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb127-1"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb127-1" aria-hidden="true" tabindex="-1"></a>sponge.pca.before <span class="ot">&lt;-</span> <span class="fu">pca</span>(sponge.clr, <span class="at">ncomp =</span> <span class="dv">3</span>, </span>
<span id="cb127-2"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb127-2" aria-hidden="true" tabindex="-1"></a>                         <span class="at">scale =</span> <span class="cn">TRUE</span>)</span>
<span id="cb127-3"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb127-3" aria-hidden="true" tabindex="-1"></a>sponge.pca.rBE <span class="ot">&lt;-</span> <span class="fu">pca</span>(sponge.rBE, <span class="at">ncomp =</span> <span class="dv">3</span>, </span>
<span id="cb127-4"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb127-4" aria-hidden="true" tabindex="-1"></a>                      <span class="at">scale =</span> <span class="cn">TRUE</span>)</span>
<span id="cb127-5"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb127-5" aria-hidden="true" tabindex="-1"></a>sponge.pca.ComBat <span class="ot">&lt;-</span> <span class="fu">pca</span>(sponge.ComBat, <span class="at">ncomp =</span> <span class="dv">3</span>, </span>
<span id="cb127-6"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb127-6" aria-hidden="true" tabindex="-1"></a>                         <span class="at">scale =</span> <span class="cn">TRUE</span>)</span>
<span id="cb127-7"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb127-7" aria-hidden="true" tabindex="-1"></a>sponge.pca.PLSDA_batch <span class="ot">&lt;-</span> <span class="fu">pca</span>(sponge.PLSDA_batch, <span class="at">ncomp =</span> <span class="dv">3</span>, </span>
<span id="cb127-8"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb127-8" aria-hidden="true" tabindex="-1"></a>                              <span class="at">scale =</span> <span class="cn">TRUE</span>)</span>
<span id="cb127-9"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb127-9" aria-hidden="true" tabindex="-1"></a>sponge.pca.sPLSDA_batch <span class="ot">&lt;-</span> <span class="fu">pca</span>(sponge.sPLSDA_batch, <span class="at">ncomp =</span> <span class="dv">3</span>, </span>
<span id="cb127-10"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb127-10" aria-hidden="true" tabindex="-1"></a>                               <span class="at">scale =</span> <span class="cn">TRUE</span>)</span>
<span id="cb127-11"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb127-11" aria-hidden="true" tabindex="-1"></a>sponge.pca.PN <span class="ot">&lt;-</span> <span class="fu">pca</span>(sponge.PN, <span class="at">ncomp =</span> <span class="dv">3</span>, </span>
<span id="cb127-12"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb127-12" aria-hidden="true" tabindex="-1"></a>                     <span class="at">scale =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<div class="sourceCode" id="cb128"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb128-1"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb128-1" aria-hidden="true" tabindex="-1"></a>sponge.pca.before.plot <span class="ot">&lt;-</span> </span>
<span id="cb128-2"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb128-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">Scatter_Density</span>(<span class="at">object =</span> sponge.pca.before, </span>
<span id="cb128-3"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb128-3" aria-hidden="true" tabindex="-1"></a>                  <span class="at">batch =</span> sponge.batch, </span>
<span id="cb128-4"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb128-4" aria-hidden="true" tabindex="-1"></a>                  <span class="at">trt =</span> sponge.trt, </span>
<span id="cb128-5"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb128-5" aria-hidden="true" tabindex="-1"></a>                  <span class="at">title =</span> <span class="st">&#39;Before&#39;</span>, </span>
<span id="cb128-6"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb128-6" aria-hidden="true" tabindex="-1"></a>                  <span class="at">trt.legend.title =</span> <span class="st">&#39;Tissue&#39;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb129"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb129-1"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb129-1" aria-hidden="true" tabindex="-1"></a>sponge.pca.rBE.plot <span class="ot">&lt;-</span> </span>
<span id="cb129-2"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb129-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">Scatter_Density</span>(<span class="at">object =</span> sponge.pca.rBE, </span>
<span id="cb129-3"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb129-3" aria-hidden="true" tabindex="-1"></a>                  <span class="at">batch =</span> sponge.batch, </span>
<span id="cb129-4"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb129-4" aria-hidden="true" tabindex="-1"></a>                  <span class="at">trt =</span> sponge.trt, </span>
<span id="cb129-5"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb129-5" aria-hidden="true" tabindex="-1"></a>                  <span class="at">title =</span> <span class="st">&#39;removeBatchEffect&#39;</span>, </span>
<span id="cb129-6"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb129-6" aria-hidden="true" tabindex="-1"></a>                  <span class="at">trt.legend.title =</span> <span class="st">&#39;Tissue&#39;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb130"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb130-1"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb130-1" aria-hidden="true" tabindex="-1"></a>sponge.pca.ComBat.plot <span class="ot">&lt;-</span> </span>
<span id="cb130-2"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb130-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">Scatter_Density</span>(<span class="at">object =</span> sponge.pca.ComBat, </span>
<span id="cb130-3"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb130-3" aria-hidden="true" tabindex="-1"></a>                  <span class="at">batch =</span> sponge.batch, </span>
<span id="cb130-4"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb130-4" aria-hidden="true" tabindex="-1"></a>                  <span class="at">trt =</span> sponge.trt, </span>
<span id="cb130-5"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb130-5" aria-hidden="true" tabindex="-1"></a>                  <span class="at">title =</span> <span class="st">&#39;ComBat&#39;</span>, </span>
<span id="cb130-6"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb130-6" aria-hidden="true" tabindex="-1"></a>                  <span class="at">trt.legend.title =</span> <span class="st">&#39;Tissue&#39;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb131"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb131-1"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb131-1" aria-hidden="true" tabindex="-1"></a>sponge.pca.PLSDA_batch.plot <span class="ot">&lt;-</span> </span>
<span id="cb131-2"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb131-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">Scatter_Density</span>(<span class="at">object =</span> sponge.pca.PLSDA_batch, </span>
<span id="cb131-3"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb131-3" aria-hidden="true" tabindex="-1"></a>                  <span class="at">batch =</span> sponge.batch, </span>
<span id="cb131-4"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb131-4" aria-hidden="true" tabindex="-1"></a>                  <span class="at">trt =</span> sponge.trt, </span>
<span id="cb131-5"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb131-5" aria-hidden="true" tabindex="-1"></a>                  <span class="at">title =</span> <span class="st">&#39;PLSDA-batch&#39;</span>, </span>
<span id="cb131-6"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb131-6" aria-hidden="true" tabindex="-1"></a>                  <span class="at">trt.legend.title =</span> <span class="st">&#39;Tissue&#39;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb132"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb132-1"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb132-1" aria-hidden="true" tabindex="-1"></a>sponge.pca.sPLSDA_batch.plot <span class="ot">&lt;-</span> </span>
<span id="cb132-2"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb132-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">Scatter_Density</span>(<span class="at">object =</span> sponge.pca.sPLSDA_batch, </span>
<span id="cb132-3"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb132-3" aria-hidden="true" tabindex="-1"></a>                  <span class="at">batch =</span> sponge.batch, </span>
<span id="cb132-4"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb132-4" aria-hidden="true" tabindex="-1"></a>                  <span class="at">trt =</span> sponge.trt, </span>
<span id="cb132-5"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb132-5" aria-hidden="true" tabindex="-1"></a>                  <span class="at">title =</span> <span class="st">&#39;sPLSDA-batch&#39;</span>, </span>
<span id="cb132-6"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb132-6" aria-hidden="true" tabindex="-1"></a>                  <span class="at">trt.legend.title =</span> <span class="st">&#39;Tissue&#39;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb133"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb133-1"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb133-1" aria-hidden="true" tabindex="-1"></a>sponge.pca.PN.plot <span class="ot">&lt;-</span> </span>
<span id="cb133-2"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb133-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">Scatter_Density</span>(<span class="at">object =</span> sponge.pca.PN, </span>
<span id="cb133-3"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb133-3" aria-hidden="true" tabindex="-1"></a>                  <span class="at">batch =</span> sponge.batch, </span>
<span id="cb133-4"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb133-4" aria-hidden="true" tabindex="-1"></a>                  <span class="at">trt =</span> sponge.trt, </span>
<span id="cb133-5"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb133-5" aria-hidden="true" tabindex="-1"></a>                  <span class="at">title =</span> <span class="st">&#39;Percentile Normalisation&#39;</span>, </span>
<span id="cb133-6"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb133-6" aria-hidden="true" tabindex="-1"></a>                  <span class="at">trt.legend.title =</span> <span class="st">&#39;Tissue&#39;</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:Spongepca"></span>
<img src="PLSDAbatch_workflow_files/figure-html/Spongepca-1.png" alt="The PCA sample plots with densities before and after batch effect correction in the sponge data." width="100%" />
<p class="caption">
Figure 2.4: The PCA sample plots with densities before and after batch effect correction in the sponge data.
</p>
</div>
<p>As shown in the PCA sample plots, the difference between the samples from two different batches was removed after batch effect correction with most methods except percentile normalisation. In the data corrected with percentile normalisation, the samples from tissue E still included a distinction between samples from two batches. We can also compare the boxplots and density plots for key variables identified in PCA driving the major variance or heatmaps showing obvious patterns before and after batch effect correction (results not shown).</p>
<p><strong>pRDA</strong></p>
<p>We calculate the global explained variance across all microbial variables using pRDA. To achieve this, we create a loop for each variable from the original (uncorrected) and batch effect-corrected data. The final results are then displayed with <code>partVar_plot()</code> from <span style="color: #FF7F00;">PLSDAbatch</span> package.</p>
<div class="sourceCode" id="cb134"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb134-1"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb134-1" aria-hidden="true" tabindex="-1"></a>sponge.corrected.list <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="st">`</span><span class="at">Before correction</span><span class="st">`</span> <span class="ot">=</span> sponge.clr, </span>
<span id="cb134-2"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb134-2" aria-hidden="true" tabindex="-1"></a>                              <span class="at">removeBatchEffect =</span> sponge.rBE, </span>
<span id="cb134-3"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb134-3" aria-hidden="true" tabindex="-1"></a>                              <span class="at">ComBat =</span> sponge.ComBat, </span>
<span id="cb134-4"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb134-4" aria-hidden="true" tabindex="-1"></a>                              <span class="st">`</span><span class="at">PLSDA-batch</span><span class="st">`</span> <span class="ot">=</span> sponge.PLSDA_batch, </span>
<span id="cb134-5"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb134-5" aria-hidden="true" tabindex="-1"></a>                              <span class="st">`</span><span class="at">sPLSDA-batch</span><span class="st">`</span> <span class="ot">=</span> sponge.sPLSDA_batch, </span>
<span id="cb134-6"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb134-6" aria-hidden="true" tabindex="-1"></a>                              <span class="st">`</span><span class="at">Percentile Normalisation</span><span class="st">`</span> <span class="ot">=</span> sponge.PN)</span>
<span id="cb134-7"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb134-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb134-8"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb134-8" aria-hidden="true" tabindex="-1"></a>sponge.prop.df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">Treatment =</span> <span class="cn">NA</span>, <span class="at">Batch =</span> <span class="cn">NA</span>, </span>
<span id="cb134-9"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb134-9" aria-hidden="true" tabindex="-1"></a>                             <span class="at">Intersection =</span> <span class="cn">NA</span>, </span>
<span id="cb134-10"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb134-10" aria-hidden="true" tabindex="-1"></a>                             <span class="at">Residuals =</span> <span class="cn">NA</span>) </span>
<span id="cb134-11"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb134-11" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(sponge.corrected.list)){</span>
<span id="cb134-12"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb134-12" aria-hidden="true" tabindex="-1"></a>  rda.res <span class="ot">=</span> <span class="fu">varpart</span>(sponge.corrected.list[[i]], <span class="sc">~</span> trt, <span class="sc">~</span> batch,</span>
<span id="cb134-13"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb134-13" aria-hidden="true" tabindex="-1"></a>                    <span class="at">data =</span> sponge.factors.df, <span class="at">scale =</span> T)</span>
<span id="cb134-14"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb134-14" aria-hidden="true" tabindex="-1"></a>  sponge.prop.df[i, ] <span class="ot">&lt;-</span> rda.res<span class="sc">$</span>part<span class="sc">$</span>indfract<span class="sc">$</span>Adj.R.squared}</span>
<span id="cb134-15"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb134-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb134-16"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb134-16" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(sponge.prop.df) <span class="ot">=</span> <span class="fu">names</span>(sponge.corrected.list)</span>
<span id="cb134-17"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb134-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb134-18"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb134-18" aria-hidden="true" tabindex="-1"></a>sponge.prop.df <span class="ot">&lt;-</span> sponge.prop.df[, <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">3</span>,<span class="dv">2</span>,<span class="dv">4</span>)]</span>
<span id="cb134-19"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb134-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb134-20"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb134-20" aria-hidden="true" tabindex="-1"></a>sponge.prop.df[sponge.prop.df <span class="sc">&lt;</span> <span class="dv">0</span>] <span class="ot">=</span> <span class="dv">0</span></span>
<span id="cb134-21"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb134-21" aria-hidden="true" tabindex="-1"></a>sponge.prop.df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">t</span>(<span class="fu">apply</span>(sponge.prop.df, <span class="dv">1</span>, </span>
<span id="cb134-22"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb134-22" aria-hidden="true" tabindex="-1"></a>                                        <span class="cf">function</span>(x){x<span class="sc">/</span><span class="fu">sum</span>(x)})))</span>
<span id="cb134-23"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb134-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb134-24"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb134-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb134-25"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb134-25" aria-hidden="true" tabindex="-1"></a><span class="fu">partVar_plot</span>(<span class="at">prop.df =</span> sponge.prop.df)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:Spongeprda"></span>
<img src="PLSDAbatch_workflow_files/figure-html/Spongeprda-1.png" alt="Global explained variance before and after batch effect correction for the sponge data." width="60%" />
<p class="caption">
Figure 2.5: Global explained variance before and after batch effect correction for the sponge data.
</p>
</div>
<p>As shown in the above figure, the intersection between batch and treatment variance was none for the <span style="color: #964B00;">sponge data</span>, because the batch x treatment design is balanced. PLSDA-batch and sPLSDA-batch corrected data led to the best performance with a slightly higher proportion of explained treatment variance and no batch variance compared to the other methods.</p>
</div>
<div id="other-methods" class="section level4 hasAnchor" number="2.1.4.2">
<h4><span class="header-section-number">2.1.4.2</span> Other methods<a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#other-methods" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p><span class="math inline">\(\mathbf{R^2}\)</span></p>
<p>The <span class="math inline">\(R^2\)</span> values for each variable are calculated with <code>lm()</code> from <span style="color: #FF7F00;">stats</span> package. To compare the <span class="math inline">\(R^2\)</span> values among variables, we scale the corrected data before <span class="math inline">\(R^2\)</span> calculation. The results are displayed with <code>ggplot()</code> from <span style="color: #FF7F00;">ggplot2</span> R package.</p>
<div class="sourceCode" id="cb135"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb135-1"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb135-1" aria-hidden="true" tabindex="-1"></a><span class="co"># scale</span></span>
<span id="cb135-2"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb135-2" aria-hidden="true" tabindex="-1"></a>sponge.corr_scale.list <span class="ot">&lt;-</span> <span class="fu">lapply</span>(sponge.corrected.list, </span>
<span id="cb135-3"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb135-3" aria-hidden="true" tabindex="-1"></a>                                 <span class="cf">function</span>(x){<span class="fu">apply</span>(x, <span class="dv">2</span>, scale)})</span>
<span id="cb135-4"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb135-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb135-5"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb135-5" aria-hidden="true" tabindex="-1"></a>sponge.r_values.list <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb135-6"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb135-6" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(sponge.corr_scale.list)){</span>
<span id="cb135-7"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb135-7" aria-hidden="true" tabindex="-1"></a>  sponge.r_values <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">trt =</span> <span class="cn">NA</span>, <span class="at">batch =</span> <span class="cn">NA</span>)</span>
<span id="cb135-8"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb135-8" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span>(c <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">ncol</span>(sponge.corr_scale.list[[i]])){</span>
<span id="cb135-9"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb135-9" aria-hidden="true" tabindex="-1"></a>    sponge.fit.res.trt <span class="ot">&lt;-</span> <span class="fu">lm</span>(sponge.corr_scale.list[[i]][,c] <span class="sc">~</span> sponge.trt)</span>
<span id="cb135-10"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb135-10" aria-hidden="true" tabindex="-1"></a>    sponge.r_values[c,<span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="fu">summary</span>(sponge.fit.res.trt)<span class="sc">$</span>r.squared</span>
<span id="cb135-11"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb135-11" aria-hidden="true" tabindex="-1"></a>    sponge.fit.res.batch <span class="ot">&lt;-</span> <span class="fu">lm</span>(sponge.corr_scale.list[[i]][,c] <span class="sc">~</span> sponge.batch)</span>
<span id="cb135-12"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb135-12" aria-hidden="true" tabindex="-1"></a>    sponge.r_values[c,<span class="dv">2</span>] <span class="ot">&lt;-</span> <span class="fu">summary</span>(sponge.fit.res.batch)<span class="sc">$</span>r.squared</span>
<span id="cb135-13"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb135-13" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb135-14"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb135-14" aria-hidden="true" tabindex="-1"></a>  sponge.r_values.list[[i]] <span class="ot">&lt;-</span> sponge.r_values</span>
<span id="cb135-15"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb135-15" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb135-16"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb135-16" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(sponge.r_values.list) <span class="ot">&lt;-</span> <span class="fu">names</span>(sponge.corr_scale.list)</span>
<span id="cb135-17"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb135-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb135-18"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb135-18" aria-hidden="true" tabindex="-1"></a>sponge.boxp.list <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb135-19"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb135-19" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="fu">seq_len</span>(<span class="fu">length</span>(sponge.r_values.list))){</span>
<span id="cb135-20"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb135-20" aria-hidden="true" tabindex="-1"></a>  sponge.boxp.list[[i]] <span class="ot">&lt;-</span> </span>
<span id="cb135-21"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb135-21" aria-hidden="true" tabindex="-1"></a>    <span class="fu">data.frame</span>(<span class="at">r2 =</span> <span class="fu">c</span>(sponge.r_values.list[[i]][ ,<span class="st">&#39;trt&#39;</span>],</span>
<span id="cb135-22"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb135-22" aria-hidden="true" tabindex="-1"></a>                      sponge.r_values.list[[i]][ ,<span class="st">&#39;batch&#39;</span>]), </span>
<span id="cb135-23"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb135-23" aria-hidden="true" tabindex="-1"></a>               <span class="at">Effects =</span> <span class="fu">as.factor</span>(<span class="fu">rep</span>(<span class="fu">c</span>(<span class="st">&#39;Treatment&#39;</span>,<span class="st">&#39;Batch&#39;</span>), </span>
<span id="cb135-24"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb135-24" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">each =</span> <span class="dv">24</span>)))</span>
<span id="cb135-25"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb135-25" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb135-26"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb135-26" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(sponge.boxp.list) <span class="ot">&lt;-</span> <span class="fu">names</span>(sponge.r_values.list)</span>
<span id="cb135-27"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb135-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb135-28"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb135-28" aria-hidden="true" tabindex="-1"></a>sponge.r2.boxp <span class="ot">&lt;-</span> <span class="fu">rbind</span>(sponge.boxp.list<span class="sc">$</span><span class="st">`</span><span class="at">Before correction</span><span class="st">`</span>,</span>
<span id="cb135-29"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb135-29" aria-hidden="true" tabindex="-1"></a>                        sponge.boxp.list<span class="sc">$</span>removeBatchEffect,</span>
<span id="cb135-30"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb135-30" aria-hidden="true" tabindex="-1"></a>                        sponge.boxp.list<span class="sc">$</span>ComBat,</span>
<span id="cb135-31"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb135-31" aria-hidden="true" tabindex="-1"></a>                        sponge.boxp.list<span class="sc">$</span><span class="st">`</span><span class="at">PLSDA-batch</span><span class="st">`</span>,</span>
<span id="cb135-32"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb135-32" aria-hidden="true" tabindex="-1"></a>                        sponge.boxp.list<span class="sc">$</span><span class="st">`</span><span class="at">sPLSDA-batch</span><span class="st">`</span>,</span>
<span id="cb135-33"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb135-33" aria-hidden="true" tabindex="-1"></a>                        sponge.boxp.list<span class="sc">$</span><span class="st">`</span><span class="at">Percentile Normalisation</span><span class="st">`</span>,</span>
<span id="cb135-34"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb135-34" aria-hidden="true" tabindex="-1"></a>                        sponge.boxp.list<span class="sc">$</span>RUVIII)</span>
<span id="cb135-35"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb135-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb135-36"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb135-36" aria-hidden="true" tabindex="-1"></a>sponge.r2.boxp<span class="sc">$</span>methods <span class="ot">&lt;-</span> <span class="fu">rep</span>(<span class="fu">c</span>(<span class="st">&#39;Before correction&#39;</span>, <span class="st">&#39; removeBatchEffect&#39;</span>, </span>
<span id="cb135-37"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb135-37" aria-hidden="true" tabindex="-1"></a>                                <span class="st">&#39;ComBat&#39;</span>,<span class="st">&#39;PLSDA-batch&#39;</span>, <span class="st">&#39;sPLSDA-batch&#39;</span>,</span>
<span id="cb135-38"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb135-38" aria-hidden="true" tabindex="-1"></a>                                <span class="st">&#39;Percentile Normalisation&#39;</span>), <span class="at">each =</span> <span class="dv">48</span>)</span>
<span id="cb135-39"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb135-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb135-40"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb135-40" aria-hidden="true" tabindex="-1"></a>sponge.r2.boxp<span class="sc">$</span>methods <span class="ot">&lt;-</span> <span class="fu">factor</span>(sponge.r2.boxp<span class="sc">$</span>methods, </span>
<span id="cb135-41"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb135-41" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">levels =</span> <span class="fu">unique</span>(sponge.r2.boxp<span class="sc">$</span>methods))</span>
<span id="cb135-42"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb135-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb135-43"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb135-43" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(sponge.r2.boxp, <span class="fu">aes</span>(<span class="at">x =</span> Effects, <span class="at">y =</span> r2, <span class="at">fill =</span> Effects)) <span class="sc">+</span></span>
<span id="cb135-44"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb135-44" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>(<span class="at">alpha =</span> <span class="fl">0.80</span>) <span class="sc">+</span></span>
<span id="cb135-45"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb135-45" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span> </span>
<span id="cb135-46"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb135-46" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">18</span>),</span>
<span id="cb135-47"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb135-47" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.title.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb135-48"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb135-48" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.title.y =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb135-49"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb135-49" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">60</span>, <span class="at">hjust =</span> <span class="dv">1</span>, <span class="at">size =</span> <span class="dv">18</span>),</span>
<span id="cb135-50"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb135-50" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.text.y =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">18</span>),</span>
<span id="cb135-51"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb135-51" aria-hidden="true" tabindex="-1"></a>        <span class="at">panel.grid.minor.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb135-52"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb135-52" aria-hidden="true" tabindex="-1"></a>        <span class="at">panel.grid.major.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb135-53"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb135-53" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.position =</span> <span class="st">&quot;right&quot;</span>) <span class="sc">+</span> <span class="fu">facet_grid</span>( <span class="sc">~</span> methods) <span class="sc">+</span> </span>
<span id="cb135-54"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb135-54" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values=</span><span class="fu">pb_color</span>(<span class="fu">c</span>(<span class="dv">12</span>,<span class="dv">14</span>))) </span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:Sponger21"></span>
<img src="PLSDAbatch_workflow_files/figure-html/Sponger21-1.png" alt="Sponge study: $R^2$ values for each microbial variable before and after batch effect correction." width="100%" />
<p class="caption">
Figure 2.6: Sponge study: <span class="math inline">\(R^2\)</span> values for each microbial variable before and after batch effect correction.
</p>
</div>
<p>We observed that the data corrected by ComBat, percentile normalisation still included a few variables with a large proportion of batch variance as shown in the above figures.</p>
<div class="sourceCode" id="cb136"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb136-1"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb136-1" aria-hidden="true" tabindex="-1"></a><span class="do">##################################</span></span>
<span id="cb136-2"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb136-2" aria-hidden="true" tabindex="-1"></a>sponge.barp.list <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb136-3"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb136-3" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="fu">seq_len</span>(<span class="fu">length</span>(sponge.r_values.list))){</span>
<span id="cb136-4"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb136-4" aria-hidden="true" tabindex="-1"></a>  sponge.barp.list[[i]] <span class="ot">&lt;-</span> </span>
<span id="cb136-5"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb136-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">data.frame</span>(<span class="at">r2 =</span> <span class="fu">c</span>(<span class="fu">sum</span>(sponge.r_values.list[[i]][ ,<span class="st">&#39;trt&#39;</span>]),</span>
<span id="cb136-6"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb136-6" aria-hidden="true" tabindex="-1"></a>                      <span class="fu">sum</span>(sponge.r_values.list[[i]][ ,<span class="st">&#39;batch&#39;</span>])), </span>
<span id="cb136-7"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb136-7" aria-hidden="true" tabindex="-1"></a>               <span class="at">Effects =</span> <span class="fu">c</span>(<span class="st">&#39;Treatment&#39;</span>,<span class="st">&#39;Batch&#39;</span>))</span>
<span id="cb136-8"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb136-8" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb136-9"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb136-9" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(sponge.barp.list) <span class="ot">&lt;-</span> <span class="fu">names</span>(sponge.r_values.list)</span>
<span id="cb136-10"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb136-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb136-11"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb136-11" aria-hidden="true" tabindex="-1"></a>sponge.r2.barp <span class="ot">&lt;-</span> <span class="fu">rbind</span>(sponge.barp.list<span class="sc">$</span><span class="st">`</span><span class="at">Before correction</span><span class="st">`</span>,</span>
<span id="cb136-12"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb136-12" aria-hidden="true" tabindex="-1"></a>                        sponge.barp.list<span class="sc">$</span>removeBatchEffect,</span>
<span id="cb136-13"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb136-13" aria-hidden="true" tabindex="-1"></a>                        sponge.barp.list<span class="sc">$</span>ComBat,</span>
<span id="cb136-14"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb136-14" aria-hidden="true" tabindex="-1"></a>                        sponge.barp.list<span class="sc">$</span><span class="st">`</span><span class="at">PLSDA-batch</span><span class="st">`</span>,</span>
<span id="cb136-15"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb136-15" aria-hidden="true" tabindex="-1"></a>                        sponge.barp.list<span class="sc">$</span><span class="st">`</span><span class="at">sPLSDA-batch</span><span class="st">`</span>,</span>
<span id="cb136-16"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb136-16" aria-hidden="true" tabindex="-1"></a>                        sponge.barp.list<span class="sc">$</span><span class="st">`</span><span class="at">Percentile Normalisation</span><span class="st">`</span>,</span>
<span id="cb136-17"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb136-17" aria-hidden="true" tabindex="-1"></a>                        sponge.barp.list<span class="sc">$</span>RUVIII)</span>
<span id="cb136-18"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb136-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb136-19"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb136-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb136-20"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb136-20" aria-hidden="true" tabindex="-1"></a>sponge.r2.barp<span class="sc">$</span>methods <span class="ot">&lt;-</span> <span class="fu">rep</span>(<span class="fu">c</span>(<span class="st">&#39;Before correction&#39;</span>, <span class="st">&#39; removeBatchEffect&#39;</span>, </span>
<span id="cb136-21"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb136-21" aria-hidden="true" tabindex="-1"></a>                                <span class="st">&#39;ComBat&#39;</span>,<span class="st">&#39;PLSDA-batch&#39;</span>, <span class="st">&#39;sPLSDA-batch&#39;</span>,</span>
<span id="cb136-22"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb136-22" aria-hidden="true" tabindex="-1"></a>                                <span class="st">&#39;Percentile Normalisation&#39;</span>), <span class="at">each =</span> <span class="dv">2</span>)</span>
<span id="cb136-23"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb136-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb136-24"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb136-24" aria-hidden="true" tabindex="-1"></a>sponge.r2.barp<span class="sc">$</span>methods <span class="ot">&lt;-</span> <span class="fu">factor</span>(sponge.r2.barp<span class="sc">$</span>methods, </span>
<span id="cb136-25"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb136-25" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">levels =</span> <span class="fu">unique</span>(sponge.r2.barp<span class="sc">$</span>methods))</span>
<span id="cb136-26"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb136-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb136-27"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb136-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb136-28"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb136-28" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(sponge.r2.barp, <span class="fu">aes</span>(<span class="at">x =</span> Effects, <span class="at">y =</span> r2, <span class="at">fill =</span> Effects)) <span class="sc">+</span></span>
<span id="cb136-29"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb136-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">stat=</span><span class="st">&quot;identity&quot;</span>) <span class="sc">+</span> </span>
<span id="cb136-30"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb136-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span> </span>
<span id="cb136-31"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb136-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">18</span>),</span>
<span id="cb136-32"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb136-32" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.title.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb136-33"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb136-33" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.title.y =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb136-34"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb136-34" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">60</span>, <span class="at">hjust =</span> <span class="dv">1</span>, <span class="at">size =</span> <span class="dv">18</span>),</span>
<span id="cb136-35"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb136-35" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.text.y =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">18</span>),</span>
<span id="cb136-36"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb136-36" aria-hidden="true" tabindex="-1"></a>        <span class="at">panel.grid.minor.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb136-37"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb136-37" aria-hidden="true" tabindex="-1"></a>        <span class="at">panel.grid.major.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb136-38"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb136-38" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.position =</span> <span class="st">&quot;right&quot;</span>) <span class="sc">+</span> <span class="fu">facet_grid</span>( <span class="sc">~</span> methods) <span class="sc">+</span> </span>
<span id="cb136-39"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb136-39" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values=</span><span class="fu">pb_color</span>(<span class="fu">c</span>(<span class="dv">12</span>,<span class="dv">14</span>)))</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:Sponger22"></span>
<img src="PLSDAbatch_workflow_files/figure-html/Sponger22-1.png" alt="Sponge study: the sum of $R^2$ values for each microbial variable before and after batch effect correction." width="100%" />
<p class="caption">
Figure 2.7: Sponge study: the sum of <span class="math inline">\(R^2\)</span> values for each microbial variable before and after batch effect correction.
</p>
</div>
<p>When considering the sum of all variables, the remaining batch variance of corrected data from sPLSDA-batch was similar as PLSDA-batch, which was greater than removeBatchEffect. The preserved treatment variance of corrected data from sPLSDA-batch was also similar as PLSDA-batch which was greater than removeBatchEffect.</p>
<p><strong>Alignment scores</strong></p>
<p>To use the <code>alignment_score()</code> function from <span style="color: #FF7F00;">PLSDAbatch</span>, we need to specify the proportion of data variance to explain (<code>var</code>), the number of nearest neighbours (<code>k</code>) and the number of principal components to estimate (<code>ncomp</code>). We then use <code>ggplot()</code> function from <span style="color: #FF7F00;">ggplot2</span> to visualise the results.</p>
<div class="sourceCode" id="cb137"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb137-1"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb137-1" aria-hidden="true" tabindex="-1"></a>sponge.var <span class="ot">=</span> <span class="fl">0.95</span></span>
<span id="cb137-2"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb137-2" aria-hidden="true" tabindex="-1"></a>sponge.k <span class="ot">=</span> <span class="dv">3</span></span>
<span id="cb137-3"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb137-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb137-4"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb137-4" aria-hidden="true" tabindex="-1"></a>sponge.scores <span class="ot">&lt;-</span> <span class="fu">c</span>()</span>
<span id="cb137-5"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb137-5" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(sponge.corrected.list)){</span>
<span id="cb137-6"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb137-6" aria-hidden="true" tabindex="-1"></a>  res <span class="ot">&lt;-</span> <span class="fu">alignment_score</span>(<span class="at">data =</span> sponge.corrected.list[[i]], </span>
<span id="cb137-7"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb137-7" aria-hidden="true" tabindex="-1"></a>                         <span class="at">batch =</span> sponge.batch, </span>
<span id="cb137-8"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb137-8" aria-hidden="true" tabindex="-1"></a>                         <span class="at">var =</span> sponge.var, </span>
<span id="cb137-9"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb137-9" aria-hidden="true" tabindex="-1"></a>                         <span class="at">k =</span> sponge.k, </span>
<span id="cb137-10"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb137-10" aria-hidden="true" tabindex="-1"></a>                         <span class="at">ncomp =</span> <span class="dv">20</span>)</span>
<span id="cb137-11"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb137-11" aria-hidden="true" tabindex="-1"></a>  sponge.scores <span class="ot">&lt;-</span> <span class="fu">c</span>(sponge.scores, res)</span>
<span id="cb137-12"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb137-12" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb137-13"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb137-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb137-14"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb137-14" aria-hidden="true" tabindex="-1"></a>sponge.scores.df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">scores =</span> sponge.scores, </span>
<span id="cb137-15"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb137-15" aria-hidden="true" tabindex="-1"></a>                               <span class="at">methods =</span> <span class="fu">names</span>(sponge.corrected.list))</span>
<span id="cb137-16"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb137-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb137-17"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb137-17" aria-hidden="true" tabindex="-1"></a>sponge.scores.df<span class="sc">$</span>methods <span class="ot">&lt;-</span> <span class="fu">factor</span>(sponge.scores.df<span class="sc">$</span>methods, </span>
<span id="cb137-18"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb137-18" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">levels =</span> <span class="fu">rev</span>(<span class="fu">names</span>(sponge.corrected.list)))</span>
<span id="cb137-19"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb137-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb137-20"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb137-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb137-21"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb137-21" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>() <span class="sc">+</span> <span class="fu">geom_col</span>(<span class="fu">aes</span>(<span class="at">x =</span> sponge.scores.df<span class="sc">$</span>methods, </span>
<span id="cb137-22"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb137-22" aria-hidden="true" tabindex="-1"></a>                        <span class="at">y =</span> sponge.scores.df<span class="sc">$</span>scores)) <span class="sc">+</span> </span>
<span id="cb137-23"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb137-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_text</span>(<span class="fu">aes</span>(<span class="at">x =</span> sponge.scores.df<span class="sc">$</span>methods, </span>
<span id="cb137-24"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb137-24" aria-hidden="true" tabindex="-1"></a>                <span class="at">y =</span> sponge.scores.df<span class="sc">$</span>scores<span class="sc">/</span><span class="dv">2</span>, </span>
<span id="cb137-25"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb137-25" aria-hidden="true" tabindex="-1"></a>                <span class="at">label =</span> <span class="fu">round</span>(sponge.scores.df<span class="sc">$</span>scores, <span class="dv">3</span>)), </span>
<span id="cb137-26"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb137-26" aria-hidden="true" tabindex="-1"></a>            <span class="at">size =</span> <span class="dv">3</span>, <span class="at">col =</span> <span class="st">&#39;white&#39;</span>) <span class="sc">+</span> </span>
<span id="cb137-27"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb137-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_flip</span>() <span class="sc">+</span> <span class="fu">theme_bw</span>() <span class="sc">+</span> <span class="fu">ylab</span>(<span class="st">&#39;Alignment Scores&#39;</span>) <span class="sc">+</span> </span>
<span id="cb137-28"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb137-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">&#39;&#39;</span>) <span class="sc">+</span> <span class="fu">ylim</span>(<span class="dv">0</span>,<span class="fl">0.4</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:Spongealignment"></span>
<img src="PLSDAbatch_workflow_files/figure-html/Spongealignment-1.png" alt="Comparison of alignment scores before and after batch effect correction using different methods for the sponge data." width="60%" />
<p class="caption">
Figure 2.8: Comparison of alignment scores before and after batch effect correction using different methods for the sponge data.
</p>
</div>
<p>The alignment scores complement the PCA results, especially when batch effect removal is difficult to assess on PCA sample plots. Since a higher alignment score indicates that samples are better mixed, as shown in the above figure, sPLSDA-batch gave a superior performance compared to the other methods.</p>
<p><strong>Variable selection</strong></p>
<p>We use <code>splsda()</code> from <span style="color: #FF7F00;">mixOmics</span> to select the top 5 microbial variables that, in combination, discriminate the different treatment groups in the <span style="color: #964B00;">sponge data</span>. We apply <code>splsda()</code> to the different batch effect corrected data from all methods. Then we use <code>upset()</code> from <span style="color: #FF7F00;">UpSetR</span> package <span class="citation">(<a href="#ref-lex2014upset" role="doc-biblioref">Lex et al. 2014</a>)</span> to visualise the concordance of variables selected.</p>
<p>In the code below, we first need to convert the list of variables selected from different method-corrected data into a data frame compatible with <code>upset()</code> using <code>fromList()</code>. We then assign different colour schemes for each variable selection.</p>
<div class="sourceCode" id="cb138"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb138-1"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb138-1" aria-hidden="true" tabindex="-1"></a>sponge.splsda.select <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb138-2"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb138-2" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(sponge.corrected.list)){</span>
<span id="cb138-3"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb138-3" aria-hidden="true" tabindex="-1"></a>  splsda.res <span class="ot">&lt;-</span> <span class="fu">splsda</span>(<span class="at">X =</span> sponge.corrected.list[[i]], </span>
<span id="cb138-4"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb138-4" aria-hidden="true" tabindex="-1"></a>                       <span class="at">Y =</span> sponge.trt, </span>
<span id="cb138-5"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb138-5" aria-hidden="true" tabindex="-1"></a>                       <span class="at">ncomp =</span> <span class="dv">3</span>, <span class="at">keepX =</span> <span class="fu">rep</span>(<span class="dv">5</span>,<span class="dv">3</span>))</span>
<span id="cb138-6"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb138-6" aria-hidden="true" tabindex="-1"></a>  select.res <span class="ot">&lt;-</span> <span class="fu">selectVar</span>(splsda.res, <span class="at">comp =</span> <span class="dv">1</span>)<span class="sc">$</span>name</span>
<span id="cb138-7"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb138-7" aria-hidden="true" tabindex="-1"></a>  sponge.splsda.select[[i]] <span class="ot">&lt;-</span> select.res</span>
<span id="cb138-8"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb138-8" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb138-9"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb138-9" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(sponge.splsda.select) <span class="ot">&lt;-</span> <span class="fu">names</span>(sponge.corrected.list)</span>
<span id="cb138-10"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb138-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb138-11"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb138-11" aria-hidden="true" tabindex="-1"></a><span class="co"># can only visualise 5 methods</span></span>
<span id="cb138-12"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb138-12" aria-hidden="true" tabindex="-1"></a>sponge.splsda.select <span class="ot">&lt;-</span> sponge.splsda.select[<span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>)]</span>
<span id="cb138-13"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb138-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb138-14"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb138-14" aria-hidden="true" tabindex="-1"></a>sponge.splsda.upsetR <span class="ot">&lt;-</span> <span class="fu">fromList</span>(sponge.splsda.select)</span>
<span id="cb138-15"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb138-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb138-16"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb138-16" aria-hidden="true" tabindex="-1"></a><span class="fu">upset</span>(sponge.splsda.upsetR, <span class="at">main.bar.color =</span> <span class="st">&#39;gray36&#39;</span>,</span>
<span id="cb138-17"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb138-17" aria-hidden="true" tabindex="-1"></a>      <span class="at">sets.bar.color =</span> <span class="fu">pb_color</span>(<span class="fu">c</span>(<span class="dv">25</span><span class="sc">:</span><span class="dv">22</span>,<span class="dv">20</span>)), <span class="at">matrix.color =</span> <span class="st">&#39;gray36&#39;</span>,</span>
<span id="cb138-18"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb138-18" aria-hidden="true" tabindex="-1"></a>      <span class="at">order.by =</span> <span class="st">&#39;freq&#39;</span>, <span class="at">empty.intersections =</span> <span class="st">&#39;on&#39;</span>,</span>
<span id="cb138-19"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb138-19" aria-hidden="true" tabindex="-1"></a>      <span class="at">queries =</span> </span>
<span id="cb138-20"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb138-20" aria-hidden="true" tabindex="-1"></a>        <span class="fu">list</span>(<span class="fu">list</span>(<span class="at">query =</span> intersects, <span class="at">params =</span> <span class="fu">list</span>(<span class="st">&#39;Before correction&#39;</span>), </span>
<span id="cb138-21"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb138-21" aria-hidden="true" tabindex="-1"></a>                  <span class="at">color =</span> <span class="fu">pb_color</span>(<span class="dv">20</span>), <span class="at">active =</span> T),</span>
<span id="cb138-22"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb138-22" aria-hidden="true" tabindex="-1"></a>             <span class="fu">list</span>(<span class="at">query =</span> intersects, <span class="at">params =</span> <span class="fu">list</span>(<span class="st">&#39;removeBatchEffect&#39;</span>), </span>
<span id="cb138-23"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb138-23" aria-hidden="true" tabindex="-1"></a>                  <span class="at">color =</span> <span class="fu">pb_color</span>(<span class="dv">22</span>), <span class="at">active =</span> T),</span>
<span id="cb138-24"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb138-24" aria-hidden="true" tabindex="-1"></a>             <span class="fu">list</span>(<span class="at">query =</span> intersects, <span class="at">params =</span> <span class="fu">list</span>(<span class="st">&#39;ComBat&#39;</span>), </span>
<span id="cb138-25"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb138-25" aria-hidden="true" tabindex="-1"></a>                  <span class="at">color =</span> <span class="fu">pb_color</span>(<span class="dv">23</span>), <span class="at">active =</span> T),</span>
<span id="cb138-26"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb138-26" aria-hidden="true" tabindex="-1"></a>             <span class="fu">list</span>(<span class="at">query =</span> intersects, <span class="at">params =</span> <span class="fu">list</span>(<span class="st">&#39;PLSDA-batch&#39;</span>), </span>
<span id="cb138-27"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb138-27" aria-hidden="true" tabindex="-1"></a>                  <span class="at">color =</span> <span class="fu">pb_color</span>(<span class="dv">24</span>), <span class="at">active =</span> T),</span>
<span id="cb138-28"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb138-28" aria-hidden="true" tabindex="-1"></a>             <span class="fu">list</span>(<span class="at">query =</span> intersects, <span class="at">params =</span> <span class="fu">list</span>(<span class="st">&#39;sPLSDA-batch&#39;</span>), </span>
<span id="cb138-29"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb138-29" aria-hidden="true" tabindex="-1"></a>                  <span class="at">color =</span> <span class="fu">pb_color</span>(<span class="dv">25</span>), <span class="at">active =</span> T)))</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:SpongeupsetR"></span>
<img src="PLSDAbatch_workflow_files/figure-html/SpongeupsetR-1.png" alt="UpSet plot showing overlap between variables selected from different corrected data for the sponge study." width="100%" />
<p class="caption">
Figure 2.9: UpSet plot showing overlap between variables selected from different corrected data for the sponge study.
</p>
</div>
<p>In the above figure, the left bars indicate the number of variables selected from each data corrected with different methods. The right bar plot combined with the scatterplot show different intersection and their aggregates. We obtained an overlap of 4 out of 5 selected variables between different corrected and uncorrected data. However, the data from each method still included unique variables that were not selected in the other corrected data, e.g., ComBat and PLSDA-batch. As <code>upset()</code> can only include five datasets at once, we only displayed the uncorrected data and four corrected data that had been more efficiently corrected for batch effects from our previous assessments compared to the other datasets.</p>
</div>
</div>
</div>
<div id="hfhs-data" class="section level2 hasAnchor" number="2.2">
<h2><span class="header-section-number">2.2</span> HFHS data<a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#hfhs-data" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="data-pre-processing-2" class="section level3 hasAnchor" number="2.2.1">
<h3><span class="header-section-number">2.2.1</span> Data pre-processing<a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#data-pre-processing-2" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="prefiltering" class="section level4 hasAnchor" number="2.2.1.1">
<h4><span class="header-section-number">2.2.1.1</span> Prefiltering<a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#prefiltering" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>We load the <span style="color: #808000;">HFHS data</span> stored internally with function <code>data()</code>.</p>
<div class="sourceCode" id="cb139"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb139-1"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb139-1" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="at">file =</span> <span class="st">&#39;./data/HFHS_data.rda&#39;</span>)</span>
<span id="cb139-2"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb139-2" aria-hidden="true" tabindex="-1"></a>hfhs.count <span class="ot">&lt;-</span> HFHS_data<span class="sc">$</span>FullData<span class="sc">$</span>X.count</span>
<span id="cb139-3"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb139-3" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(hfhs.count)</span></code></pre></div>
<pre><code>## [1]  250 4524</code></pre>
<div class="sourceCode" id="cb141"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb141-1"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb141-1" aria-hidden="true" tabindex="-1"></a>hfhs.filter.res <span class="ot">&lt;-</span> <span class="fu">PreFL</span>(<span class="at">data =</span> hfhs.count)</span>
<span id="cb141-2"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb141-2" aria-hidden="true" tabindex="-1"></a>hfhs.filter <span class="ot">&lt;-</span> hfhs.filter.res<span class="sc">$</span>data.filter</span>
<span id="cb141-3"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb141-3" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(hfhs.filter)</span></code></pre></div>
<pre><code>## [1] 250 515</code></pre>
<div class="sourceCode" id="cb143"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb143-1"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb143-1" aria-hidden="true" tabindex="-1"></a><span class="co"># zero proportion before filtering</span></span>
<span id="cb143-2"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb143-2" aria-hidden="true" tabindex="-1"></a>hfhs.filter.res<span class="sc">$</span>zero.prob</span></code></pre></div>
<pre><code>## [1] 0.8490805</code></pre>
<div class="sourceCode" id="cb145"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb145-1"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb145-1" aria-hidden="true" tabindex="-1"></a><span class="co"># zero proportion after filtering</span></span>
<span id="cb145-2"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb145-2" aria-hidden="true" tabindex="-1"></a><span class="fu">sum</span>(hfhs.filter <span class="sc">==</span> <span class="dv">0</span>)<span class="sc">/</span>(<span class="fu">nrow</span>(hfhs.filter) <span class="sc">*</span> <span class="fu">ncol</span>(hfhs.filter))</span></code></pre></div>
<pre><code>## [1] 0.2955184</code></pre>
<p>The raw <span style="color: #808000;">HFHS data</span> include 4524 OTUs and 250 samples. We use the function <code>PreFL()</code> from our <span style="color: #FF7F00;">PLSDAbatch</span> R package to filter the data. After filtering, the <span style="color: #808000;">HFHS data</span> were reduced to 515 OTUs and 250 samples. The proportion of zeroes was reduced from 85% to 30%.</p>
</div>
<div id="transformation-2" class="section level4 hasAnchor" number="2.2.1.2">
<h4><span class="header-section-number">2.2.1.2</span> Transformation<a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#transformation-2" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Prior to CLR transformation, we recommend adding 1 as the offset for the <span style="color: #808000;">HFHS data</span> - that are raw count data. We use <code>logratio.transfo()</code> function in <span style="color: #FF7F00;">mixOmics</span> package to CLR transform the data.</p>
<div class="sourceCode" id="cb147"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb147-1"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb147-1" aria-hidden="true" tabindex="-1"></a>hfhs.clr <span class="ot">&lt;-</span> <span class="fu">logratio.transfo</span>(<span class="at">X =</span> hfhs.filter, <span class="at">logratio =</span> <span class="st">&#39;CLR&#39;</span>, <span class="at">offset =</span> <span class="dv">1</span>)</span>
<span id="cb147-2"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb147-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb147-3"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb147-3" aria-hidden="true" tabindex="-1"></a><span class="fu">class</span>(hfhs.clr) <span class="ot">&lt;-</span> <span class="st">&#39;matrix&#39;</span></span></code></pre></div>
</div>
</div>
<div id="batch-effect-detection-2" class="section level3 hasAnchor" number="2.2.2">
<h3><span class="header-section-number">2.2.2</span> Batch effect detection<a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#batch-effect-detection-2" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="pca-2" class="section level4 hasAnchor" number="2.2.2.1">
<h4><span class="header-section-number">2.2.2.1</span> PCA<a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#pca-2" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>We apply <code>pca()</code> function from <span style="color: #FF7F00;">mixOmics</span> package to the <span style="color: #808000;">HFHS data</span> and use <code>Scatter_Density()</code> function from <span style="color: #FF7F00;">PLSDAbatch</span> to represent the PCA sample plot with densities.</p>
<div class="sourceCode" id="cb148"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb148-1"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb148-1" aria-hidden="true" tabindex="-1"></a>hfhs.pca.before <span class="ot">&lt;-</span> <span class="fu">pca</span>(hfhs.clr, <span class="at">ncomp =</span> <span class="dv">3</span>, <span class="at">scale =</span> <span class="cn">TRUE</span>)</span>
<span id="cb148-2"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb148-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb148-3"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb148-3" aria-hidden="true" tabindex="-1"></a>hfhs.metadata <span class="ot">&lt;-</span> HFHS_data<span class="sc">$</span>FullData<span class="sc">$</span>metadata</span>
<span id="cb148-4"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb148-4" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(hfhs.metadata) <span class="ot">&lt;-</span> hfhs.metadata<span class="sc">$</span>SampleID</span>
<span id="cb148-5"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb148-5" aria-hidden="true" tabindex="-1"></a>hfhs.metadata <span class="ot">&lt;-</span> hfhs.metadata[<span class="fu">rownames</span>(hfhs.clr),]</span>
<span id="cb148-6"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb148-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb148-7"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb148-7" aria-hidden="true" tabindex="-1"></a>hfhs.cage <span class="ot">&lt;-</span> <span class="fu">as.factor</span>(hfhs.metadata<span class="sc">$</span>DietCage)</span>
<span id="cb148-8"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb148-8" aria-hidden="true" tabindex="-1"></a>hfhs.day <span class="ot">&lt;-</span> <span class="fu">as.factor</span>(hfhs.metadata<span class="sc">$</span>Day)</span>
<span id="cb148-9"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb148-9" aria-hidden="true" tabindex="-1"></a>hfhs.trt <span class="ot">&lt;-</span> <span class="fu">as.factor</span>(hfhs.metadata<span class="sc">$</span>Diet)</span>
<span id="cb148-10"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb148-10" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(hfhs.cage) <span class="ot">=</span> <span class="fu">names</span>(hfhs.day) <span class="ot">=</span> <span class="fu">names</span>(hfhs.trt) <span class="ot">=</span> hfhs.metadata<span class="sc">$</span>SampleID</span>
<span id="cb148-11"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb148-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb148-12"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb148-12" aria-hidden="true" tabindex="-1"></a>hfhs.pca.before.cage.plot <span class="ot">&lt;-</span> </span>
<span id="cb148-13"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb148-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">Scatter_Density</span>(<span class="at">object =</span> hfhs.pca.before, </span>
<span id="cb148-14"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb148-14" aria-hidden="true" tabindex="-1"></a>                  <span class="at">batch =</span> hfhs.cage, </span>
<span id="cb148-15"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb148-15" aria-hidden="true" tabindex="-1"></a>                  <span class="at">trt =</span> hfhs.trt, </span>
<span id="cb148-16"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb148-16" aria-hidden="true" tabindex="-1"></a>                  <span class="at">title =</span> <span class="st">&#39;HFHS data&#39;</span>, </span>
<span id="cb148-17"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb148-17" aria-hidden="true" tabindex="-1"></a>                  <span class="at">trt.legend.title =</span> <span class="st">&#39;Diet&#39;</span>,</span>
<span id="cb148-18"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb148-18" aria-hidden="true" tabindex="-1"></a>                  <span class="at">batch.legend.title =</span> <span class="st">&#39;Cage&#39;</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:HFHSpcaBefore1"></span>
<img src="PLSDAbatch_workflow_files/figure-html/HFHSpcaBefore1-1.png" alt="The PCA sample plot with densities coloured by cages in the HFHS data." width="60%" />
<p class="caption">
Figure 2.10: The PCA sample plot with densities coloured by cages in the HFHS data.
</p>
</div>
<p>In the above figure, part of the samples with different diets were mixed together, while all the samples from different batches (i.e., cages) were well mixed and not distinct. This result indicates no cage effect for the whole data and no treatment effect for part of samples.</p>
<div class="sourceCode" id="cb149"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb149-1"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb149-1" aria-hidden="true" tabindex="-1"></a>hfhs.pca.before.day.plot <span class="ot">&lt;-</span> </span>
<span id="cb149-2"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb149-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">Scatter_Density</span>(<span class="at">object =</span> hfhs.pca.before, </span>
<span id="cb149-3"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb149-3" aria-hidden="true" tabindex="-1"></a>                  <span class="at">batch =</span> hfhs.day, </span>
<span id="cb149-4"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb149-4" aria-hidden="true" tabindex="-1"></a>                  <span class="at">trt =</span> hfhs.trt, </span>
<span id="cb149-5"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb149-5" aria-hidden="true" tabindex="-1"></a>                  <span class="at">title =</span> <span class="st">&#39;HFHS data&#39;</span>, </span>
<span id="cb149-6"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb149-6" aria-hidden="true" tabindex="-1"></a>                  <span class="at">trt.legend.title =</span> <span class="st">&#39;Diet&#39;</span>,</span>
<span id="cb149-7"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb149-7" aria-hidden="true" tabindex="-1"></a>                  <span class="at">batch.legend.title =</span> <span class="st">&#39;Day&#39;</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:HFHSpcaBefore2"></span>
<img src="PLSDAbatch_workflow_files/figure-html/HFHSpcaBefore2-1.png" alt="Figure 11: The PCA sample plot with densities coloured by days in the HFHS data." width="60%" />
<p class="caption">
Figure 2.11: Figure 11: The PCA sample plot with densities coloured by days in the HFHS data.
</p>
</div>
<p>As shown in the above figure, we did not observe a difference between samples with different diets on arrival day “A” and “Day0”. This result was consistent with the experimental design, as mice were not treated with different diets on these two days. We thus removed the samples from these two days.</p>
<div class="sourceCode" id="cb150"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb150-1"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb150-1" aria-hidden="true" tabindex="-1"></a><span class="co"># remove samples from arrival day and day0</span></span>
<span id="cb150-2"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb150-2" aria-hidden="true" tabindex="-1"></a>hfhs.dA0.idx <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fu">which</span>(hfhs.day <span class="sc">==</span> <span class="st">&#39;A&#39;</span>), <span class="fu">which</span>(hfhs.day <span class="sc">==</span> <span class="st">&#39;Day0&#39;</span>))</span>
<span id="cb150-3"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb150-3" aria-hidden="true" tabindex="-1"></a>hfhs.clr.less <span class="ot">&lt;-</span> hfhs.clr[<span class="sc">-</span>hfhs.dA0.idx,]</span>
<span id="cb150-4"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb150-4" aria-hidden="true" tabindex="-1"></a>hfhs.metadata.less <span class="ot">&lt;-</span> hfhs.metadata[<span class="sc">-</span>hfhs.dA0.idx,]</span>
<span id="cb150-5"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb150-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb150-6"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb150-6" aria-hidden="true" tabindex="-1"></a>hfhs.cage.less <span class="ot">&lt;-</span> <span class="fu">as.factor</span>(hfhs.metadata.less<span class="sc">$</span>DietCage)</span>
<span id="cb150-7"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb150-7" aria-hidden="true" tabindex="-1"></a>hfhs.day.less <span class="ot">&lt;-</span> <span class="fu">as.factor</span>(hfhs.metadata.less<span class="sc">$</span>Day)</span>
<span id="cb150-8"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb150-8" aria-hidden="true" tabindex="-1"></a>hfhs.trt.less <span class="ot">&lt;-</span> <span class="fu">as.factor</span>(hfhs.metadata.less<span class="sc">$</span>Diet)</span>
<span id="cb150-9"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb150-9" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(hfhs.cage.less) <span class="ot">=</span> <span class="fu">names</span>(hfhs.day.less) <span class="ot">=</span> </span>
<span id="cb150-10"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb150-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">names</span>(hfhs.trt.less) <span class="ot">=</span> hfhs.metadata.less<span class="sc">$</span>SampleID</span>
<span id="cb150-11"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb150-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb150-12"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb150-12" aria-hidden="true" tabindex="-1"></a>hfhs.less.pca.before <span class="ot">&lt;-</span> <span class="fu">pca</span>(hfhs.clr.less, <span class="at">ncomp =</span> <span class="dv">3</span>, <span class="at">scale =</span> <span class="cn">TRUE</span>)</span>
<span id="cb150-13"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb150-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb150-14"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb150-14" aria-hidden="true" tabindex="-1"></a>hfhs.less.pca.before.cage.plot <span class="ot">&lt;-</span> </span>
<span id="cb150-15"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb150-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">Scatter_Density</span>(<span class="at">object =</span> hfhs.less.pca.before, </span>
<span id="cb150-16"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb150-16" aria-hidden="true" tabindex="-1"></a>                  <span class="at">batch =</span> hfhs.cage.less, </span>
<span id="cb150-17"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb150-17" aria-hidden="true" tabindex="-1"></a>                  <span class="at">trt =</span> hfhs.trt.less, </span>
<span id="cb150-18"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb150-18" aria-hidden="true" tabindex="-1"></a>                  <span class="at">title =</span> <span class="st">&#39;HFHS data&#39;</span>, </span>
<span id="cb150-19"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb150-19" aria-hidden="true" tabindex="-1"></a>                  <span class="at">trt.legend.title =</span> <span class="st">&#39;Diet&#39;</span>,</span>
<span id="cb150-20"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb150-20" aria-hidden="true" tabindex="-1"></a>                  <span class="at">batch.legend.title =</span> <span class="st">&#39;Cage&#39;</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:HFHSpcaBeforeNew1"></span>
<img src="PLSDAbatch_workflow_files/figure-html/HFHSpcaBeforeNew1-1.png" alt="The PCA sample plot with densities coloured by cages in the reduced HFHS data." width="60%" />
<p class="caption">
Figure 2.12: The PCA sample plot with densities coloured by cages in the reduced HFHS data.
</p>
</div>
<div class="sourceCode" id="cb151"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb151-1"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb151-1" aria-hidden="true" tabindex="-1"></a>hfhs.less.pca.before.day.plot <span class="ot">&lt;-</span> </span>
<span id="cb151-2"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb151-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">Scatter_Density</span>(<span class="at">object =</span> hfhs.less.pca.before, </span>
<span id="cb151-3"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb151-3" aria-hidden="true" tabindex="-1"></a>                  <span class="at">batch =</span> hfhs.day.less, </span>
<span id="cb151-4"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb151-4" aria-hidden="true" tabindex="-1"></a>                  <span class="at">trt =</span> hfhs.trt.less, </span>
<span id="cb151-5"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb151-5" aria-hidden="true" tabindex="-1"></a>                  <span class="at">title =</span> <span class="st">&#39;HFHS data&#39;</span>, </span>
<span id="cb151-6"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb151-6" aria-hidden="true" tabindex="-1"></a>                  <span class="at">trt.legend.title =</span> <span class="st">&#39;Diet&#39;</span>,</span>
<span id="cb151-7"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb151-7" aria-hidden="true" tabindex="-1"></a>                  <span class="at">batch.legend.title =</span> <span class="st">&#39;Day&#39;</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:HFHSpcaBeforeNew2"></span>
<img src="PLSDAbatch_workflow_files/figure-html/HFHSpcaBeforeNew2-1.png" alt="The PCA sample plot with densities coloured by days in the reduced HFHS data." width="60%" />
<p class="caption">
Figure 2.13: The PCA sample plot with densities coloured by days in the reduced HFHS data.
</p>
</div>
<p>We observed a substantial diet variation on component 1, no cage variation, and a difference between samples collected on “Day1” and the other days with the HFHS diet as shown in the above figures.</p>
</div>
<div id="heatmap-2" class="section level4 hasAnchor" number="2.2.2.2">
<h4><span class="header-section-number">2.2.2.2</span> Heatmap<a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#heatmap-2" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>We produce a heatmap using <span style="color: #FF7F00;">pheatmap</span> package. The data first need to be scaled on both OTUs and samples.</p>
<div class="sourceCode" id="cb152"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb152-1"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb152-1" aria-hidden="true" tabindex="-1"></a><span class="co"># scale the clr data on both OTUs and samples</span></span>
<span id="cb152-2"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb152-2" aria-hidden="true" tabindex="-1"></a>hfhs.clr.s <span class="ot">&lt;-</span> <span class="fu">scale</span>(hfhs.clr.less, <span class="at">center =</span> T, <span class="at">scale =</span> T)</span>
<span id="cb152-3"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb152-3" aria-hidden="true" tabindex="-1"></a>hfhs.clr.ss <span class="ot">&lt;-</span> <span class="fu">scale</span>(<span class="fu">t</span>(hfhs.clr.s), <span class="at">center =</span> T, <span class="at">scale =</span> T)</span>
<span id="cb152-4"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb152-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb152-5"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb152-5" aria-hidden="true" tabindex="-1"></a><span class="co"># The cage effect</span></span>
<span id="cb152-6"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb152-6" aria-hidden="true" tabindex="-1"></a>hfhs.anno_col <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">Cage =</span> hfhs.cage.less, </span>
<span id="cb152-7"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb152-7" aria-hidden="true" tabindex="-1"></a>                            <span class="at">Day =</span> hfhs.day.less, </span>
<span id="cb152-8"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb152-8" aria-hidden="true" tabindex="-1"></a>                            <span class="at">Treatment =</span> hfhs.trt.less)</span>
<span id="cb152-9"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb152-9" aria-hidden="true" tabindex="-1"></a>hfhs.anno_colors <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="at">Cage =</span> <span class="fu">color.mixo</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">8</span>), </span>
<span id="cb152-10"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb152-10" aria-hidden="true" tabindex="-1"></a>                         <span class="at">Day =</span> <span class="fu">color.mixo</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>),</span>
<span id="cb152-11"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb152-11" aria-hidden="true" tabindex="-1"></a>                         <span class="at">Treatment =</span> <span class="fu">pb_color</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>))</span>
<span id="cb152-12"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb152-12" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(hfhs.anno_colors<span class="sc">$</span>Cage) <span class="ot">=</span> <span class="fu">levels</span>(hfhs.cage.less)</span>
<span id="cb152-13"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb152-13" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(hfhs.anno_colors<span class="sc">$</span>Day) <span class="ot">=</span> <span class="fu">levels</span>(hfhs.day.less)</span>
<span id="cb152-14"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb152-14" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(hfhs.anno_colors<span class="sc">$</span>Treatment) <span class="ot">=</span> <span class="fu">levels</span>(hfhs.trt.less)</span>
<span id="cb152-15"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb152-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb152-16"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb152-16" aria-hidden="true" tabindex="-1"></a><span class="fu">pheatmap</span>(hfhs.clr.ss, </span>
<span id="cb152-17"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb152-17" aria-hidden="true" tabindex="-1"></a>         <span class="at">cluster_rows =</span> F, </span>
<span id="cb152-18"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb152-18" aria-hidden="true" tabindex="-1"></a>         <span class="at">fontsize_row =</span> <span class="dv">4</span>, </span>
<span id="cb152-19"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb152-19" aria-hidden="true" tabindex="-1"></a>         <span class="at">fontsize_col =</span> <span class="dv">6</span>,</span>
<span id="cb152-20"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb152-20" aria-hidden="true" tabindex="-1"></a>         <span class="at">fontsize =</span> <span class="dv">8</span>,</span>
<span id="cb152-21"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb152-21" aria-hidden="true" tabindex="-1"></a>         <span class="at">clustering_distance_rows =</span> <span class="st">&#39;euclidean&#39;</span>,</span>
<span id="cb152-22"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb152-22" aria-hidden="true" tabindex="-1"></a>         <span class="at">clustering_method =</span> <span class="st">&#39;ward.D&#39;</span>,</span>
<span id="cb152-23"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb152-23" aria-hidden="true" tabindex="-1"></a>         <span class="at">treeheight_row =</span> <span class="dv">30</span>,</span>
<span id="cb152-24"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb152-24" aria-hidden="true" tabindex="-1"></a>         <span class="at">annotation_col =</span> hfhs.anno_col,</span>
<span id="cb152-25"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb152-25" aria-hidden="true" tabindex="-1"></a>         <span class="at">annotation_colors =</span> hfhs.anno_colors,</span>
<span id="cb152-26"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb152-26" aria-hidden="true" tabindex="-1"></a>         <span class="at">border_color =</span> <span class="st">&#39;NA&#39;</span>,</span>
<span id="cb152-27"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb152-27" aria-hidden="true" tabindex="-1"></a>         <span class="at">main =</span> <span class="st">&#39;HFHS data - Scaled&#39;</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:HFHSheatmapS"></span>
<img src="PLSDAbatch_workflow_files/figure-html/HFHSheatmapS-1.png" alt="Hierarchical clustering for samples in the HFHS data." width="90%" />
<p class="caption">
Figure 2.14: Hierarchical clustering for samples in the HFHS data.
</p>
</div>
<p>In the above figure, samples in the <span style="color: #808000;">HFHS data</span> were clustered according to the treatments not cages, indicating no cage effect and a strong treatment effect. Samples from the batch “Day1” were clustered and distinct from the other batches with diet HFHS but not the normal diet, indicating a day effect but very weak.</p>
</div>
<div id="prda-2" class="section level4 hasAnchor" number="2.2.2.3">
<h4><span class="header-section-number">2.2.2.3</span> pRDA<a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#prda-2" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>We apply pRDA with <code>varpart()</code> function from <span style="color: #FF7F00;">vegan</span> R package.</p>
<div class="sourceCode" id="cb153"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb153-1"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb153-1" aria-hidden="true" tabindex="-1"></a>hfhs.factors.df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">trt =</span> hfhs.trt.less, </span>
<span id="cb153-2"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb153-2" aria-hidden="true" tabindex="-1"></a>                              <span class="at">cage =</span> hfhs.cage.less, </span>
<span id="cb153-3"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb153-3" aria-hidden="true" tabindex="-1"></a>                              <span class="at">day =</span> hfhs.day.less)</span>
<span id="cb153-4"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb153-4" aria-hidden="true" tabindex="-1"></a>hfhs.rda.cage.before <span class="ot">&lt;-</span> <span class="fu">varpart</span>(hfhs.clr.less, <span class="sc">~</span> trt, <span class="sc">~</span> cage, </span>
<span id="cb153-5"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb153-5" aria-hidden="true" tabindex="-1"></a>                                <span class="at">data =</span> hfhs.factors.df, <span class="at">scale =</span> T)</span>
<span id="cb153-6"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb153-6" aria-hidden="true" tabindex="-1"></a>hfhs.rda.cage.before<span class="sc">$</span>part<span class="sc">$</span>indfract</span></code></pre></div>
<pre><code>##                 Df R.squared Adj.R.squared Testable
## [a] = X1|X2      0        NA -2.220446e-16    FALSE
## [b] = X2|X1      6        NA  1.277627e-02     TRUE
## [c]              0        NA  2.713111e-01    FALSE
## [d] = Residuals NA        NA  7.159126e-01    FALSE</code></pre>
<p>In the result, <code>X1</code> and <code>X2</code> represent the first and second covariates fitted in the model. <code>[a]</code>, <code>[b]</code> represent the independent proportion of variance explained by <code>X1</code> and <code>X2</code> respectively, and <code>[c]</code> represents the intersection variance shared between <code>X1</code> and <code>X2</code>. In the <span style="color: #808000;">HFHS data</span>, the cage and diet effects have a nested batch x treatment design that is extremely unbalanced. We didn’t detect any treatment variance (indicated in line <code>[a]</code>, Adj.R.squared = 0) but considerable intersection variance (indicated in line <code>[c]</code>, Adj.R.squared = 0.271). Thus, all the treatment variance was explained by the intersection variance shared with batch variance. It means batch and treatment effects are collinear. Therefore, the cage effect in the <span style="color: #808000;">HFHS data</span> can only be accounted for with a linear mixed model, as detailed in section <strong>accounting for batch effects</strong> method “Linear regression”.</p>
<div class="sourceCode" id="cb155"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb155-1"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb155-1" aria-hidden="true" tabindex="-1"></a>hfhs.rda.day.before <span class="ot">&lt;-</span> <span class="fu">varpart</span>(hfhs.clr.less, <span class="sc">~</span> trt, <span class="sc">~</span> day, </span>
<span id="cb155-2"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb155-2" aria-hidden="true" tabindex="-1"></a>                               <span class="at">data =</span> hfhs.factors.df, <span class="at">scale =</span> T)</span>
<span id="cb155-3"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb155-3" aria-hidden="true" tabindex="-1"></a>hfhs.rda.day.before<span class="sc">$</span>part<span class="sc">$</span>indfract</span></code></pre></div>
<pre><code>##                 Df R.squared Adj.R.squared Testable
## [a] = X1|X2      1        NA  0.2721810304     TRUE
## [b] = X2|X1      2        NA  0.0361767632     TRUE
## [c]              0        NA -0.0008699296    FALSE
## [d] = Residuals NA        NA  0.6925121361    FALSE</code></pre>
<p>For the day and diet effects, the batch x treatment design is balanced. There was no intersection variance (indicated in line <code>[c]</code>, Adj.R.squared = 0). The proportion of treatment variance was much higher than batch variance as indicated in lines <code>[a]</code> (Adj.R.squared = 0.272) and <code>[b]</code> (Adj.R.squared = 0.036).</p>
</div>
</div>
<div id="managing-batch-effects-2" class="section level3 hasAnchor" number="2.2.3">
<h3><span class="header-section-number">2.2.3</span> Managing batch effects<a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#managing-batch-effects-2" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="accounting-for-batch-effects-1" class="section level4 hasAnchor" number="2.2.3.1">
<h4><span class="header-section-number">2.2.3.1</span> Accounting for batch effects<a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#accounting-for-batch-effects-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>The methods that we use to account for batch effects include the methods designed for microbiome data: zero-inflated Gaussian (ZIG) mixture model and the methods adapted for microbiome data: linear regression, SVA and RUV4. Among them, SVA and RUV4 are designed for unknown batch effects.</p>
<p><strong>Methods designed for microbiome data</strong></p>
<p><strong>Zero-inflated Gaussian mixture model</strong> To use the ZIG model, we first create a <code>MRexperiment</code> object applying <code>newMRexperiment()</code> (from <span style="color: #FF7F00;">metagenomeSeq</span> package) to microbiome counts and annotated data frames with metadata and taxonomic information generated with <code>AnnotatedDataFrame()</code> from <span style="color: #FF7F00;">Biobase</span> package.</p>
<div class="sourceCode" id="cb157"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb157-1"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb157-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Creating a MRexperiment object (make sure no NA in metadata)</span></span>
<span id="cb157-2"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb157-2" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(HFHS_data<span class="sc">$</span>FullData<span class="sc">$</span>metadata) <span class="ot">&lt;-</span> <span class="fu">rownames</span>(HFHS_data<span class="sc">$</span>FullData<span class="sc">$</span>X.count)</span>
<span id="cb157-3"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb157-3" aria-hidden="true" tabindex="-1"></a>HFHS.phenotypeData <span class="ot">=</span> </span>
<span id="cb157-4"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb157-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">AnnotatedDataFrame</span>(<span class="at">data =</span> HFHS_data<span class="sc">$</span>FullData<span class="sc">$</span>metadata[<span class="sc">-</span>hfhs.dA0.idx,])</span>
<span id="cb157-5"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb157-5" aria-hidden="true" tabindex="-1"></a>HFHS.taxaData <span class="ot">=</span> </span>
<span id="cb157-6"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb157-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">AnnotatedDataFrame</span>(<span class="at">data =</span> <span class="fu">as.data.frame</span>(HFHS_data<span class="sc">$</span>FullData<span class="sc">$</span>taxa))</span>
<span id="cb157-7"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb157-7" aria-hidden="true" tabindex="-1"></a>HFHS.obj <span class="ot">=</span> <span class="fu">newMRexperiment</span>(<span class="at">counts =</span> </span>
<span id="cb157-8"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb157-8" aria-hidden="true" tabindex="-1"></a>                             <span class="fu">t</span>(HFHS_data<span class="sc">$</span>FullData<span class="sc">$</span>X.count[<span class="sc">-</span>hfhs.dA0.idx,]),</span>
<span id="cb157-9"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb157-9" aria-hidden="true" tabindex="-1"></a>                           <span class="at">phenoData =</span> HFHS.phenotypeData, </span>
<span id="cb157-10"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb157-10" aria-hidden="true" tabindex="-1"></a>                           <span class="at">featureData =</span> HFHS.taxaData)</span>
<span id="cb157-11"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb157-11" aria-hidden="true" tabindex="-1"></a>HFHS.obj</span></code></pre></div>
<pre><code>## MRexperiment (storageMode: environment)
## assayData: 4524 features, 149 samples 
##   element names: counts 
## protocolData: none
## phenoData
##   sampleNames: 109M 131M ... 39M (149 total)
##   varLabels: SampleID MouseID ... Replicated (14 total)
##   varMetadata: labelDescription
## featureData
##   featureNames: 4333897 541135 ... 228232 (4524 total)
##   fvarLabels: Kingdom Phylum ... Species (7 total)
##   fvarMetadata: labelDescription
## experimentData: use &#39;experimentData(object)&#39;
## Annotation:</code></pre>
<p>The <span style="color: #808000;">HFHS data</span> are then filtered with <code>filterData()</code> function (from <span style="color: #FF7F00;">metagenomeSeq</span>). We can use <code>MRcounts()</code> to extract the count data from the <code>MRexperiment</code> object.</p>
<div class="sourceCode" id="cb159"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb159-1"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb159-1" aria-hidden="true" tabindex="-1"></a><span class="co"># filtering data to maintain a threshold of minimum depth or OTU presence</span></span>
<span id="cb159-2"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb159-2" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(<span class="fu">MRcounts</span>(HFHS.obj))</span></code></pre></div>
<pre><code>## [1] 4524  149</code></pre>
<div class="sourceCode" id="cb161"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb161-1"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb161-1" aria-hidden="true" tabindex="-1"></a>HFHS.obj <span class="ot">=</span> <span class="fu">filterData</span>(<span class="at">obj =</span> HFHS.obj, <span class="at">present =</span> <span class="dv">20</span>, <span class="at">depth =</span> <span class="dv">5</span>)</span>
<span id="cb161-2"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb161-2" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(<span class="fu">MRcounts</span>(HFHS.obj))</span></code></pre></div>
<pre><code>## [1] 1255  149</code></pre>
<p>After filtering, the <span style="color: #808000;">HFHS data</span> were reduced to 1255 OTUs and 149 samples.</p>
<p>We calculate the percentile for CSS normalisation with <code>cumNormStatFast()</code> function (from <span style="color: #FF7F00;">metagenomeSeq</span> package). The CSS normalisation is applied with <code>cumNorm()</code> and the normalised data can be exported using <code>MRcounts()</code> with <code>norm = TRUE</code>. The normalisation scaling factors for each sample, which are the sum of counts up to the calculated percentile, can be accessed through <code>normFactors()</code>. We calculate the log transformed scaling factors by diving them with their median, which are better than the default scaling factors that are divided by 1000 (<code>log2(normFactors(obj)/1000 + 1)</code>).</p>
<div class="sourceCode" id="cb163"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb163-1"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb163-1" aria-hidden="true" tabindex="-1"></a><span class="co"># calculate the percentile for CSS normalisation</span></span>
<span id="cb163-2"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb163-2" aria-hidden="true" tabindex="-1"></a>HFHS.pctl <span class="ot">=</span> <span class="fu">cumNormStatFast</span>(<span class="at">obj =</span> HFHS.obj)</span>
<span id="cb163-3"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb163-3" aria-hidden="true" tabindex="-1"></a><span class="co"># CSS normalisation</span></span>
<span id="cb163-4"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb163-4" aria-hidden="true" tabindex="-1"></a>HFHS.obj <span class="ot">&lt;-</span> <span class="fu">cumNorm</span>(<span class="at">obj =</span> HFHS.obj, <span class="at">p =</span> HFHS.pctl)</span>
<span id="cb163-5"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb163-5" aria-hidden="true" tabindex="-1"></a><span class="co"># export normalised data</span></span>
<span id="cb163-6"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb163-6" aria-hidden="true" tabindex="-1"></a>HFHS.norm.data <span class="ot">&lt;-</span> <span class="fu">MRcounts</span>(<span class="at">obj =</span> HFHS.obj, <span class="at">norm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb163-7"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb163-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb163-8"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb163-8" aria-hidden="true" tabindex="-1"></a><span class="co"># normalisation scaling factors for each sample </span></span>
<span id="cb163-9"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb163-9" aria-hidden="true" tabindex="-1"></a>HFHS.normFactor <span class="ot">=</span> <span class="fu">normFactors</span>(<span class="at">object =</span> HFHS.obj)</span>
<span id="cb163-10"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb163-10" aria-hidden="true" tabindex="-1"></a>HFHS.normFactor <span class="ot">=</span> <span class="fu">log2</span>(HFHS.normFactor<span class="sc">/</span><span class="fu">median</span>(HFHS.normFactor) <span class="sc">+</span> <span class="dv">1</span>)</span></code></pre></div>
<p>As the ZIG model applies a linear model, the cage effect cannot be fitted. Moreover, the cage effect is very weak which only accounted for 1.3% of the total data variance calculated with pRDA. Thus we only fit and account for the day effect. We create a design matrix with treatment variable (<code>Diet_effect</code>), batch variable (<code>Day_effect</code>) and the log transformed scaling factors by using <code>model.matrix()</code>, and then apply the ZIG model by <code>fitZig()</code> function. We set <code>useCSSoffset = FALSE</code> to avoid using the default scaling factors as we have already included our customised scaling factor (<code>HFHS.normFactor</code>) in the design matrix.</p>
<div class="sourceCode" id="cb164"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb164-1"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb164-1" aria-hidden="true" tabindex="-1"></a><span class="co"># treatment variable</span></span>
<span id="cb164-2"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb164-2" aria-hidden="true" tabindex="-1"></a>Diet_effect <span class="ot">=</span> <span class="fu">pData</span>(<span class="at">object =</span> HFHS.obj)<span class="sc">$</span>Diet</span>
<span id="cb164-3"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb164-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb164-4"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb164-4" aria-hidden="true" tabindex="-1"></a><span class="co"># batch variable</span></span>
<span id="cb164-5"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb164-5" aria-hidden="true" tabindex="-1"></a>Day_effect <span class="ot">=</span> <span class="fu">pData</span>(<span class="at">object =</span> HFHS.obj)<span class="sc">$</span>Day</span>
<span id="cb164-6"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb164-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb164-7"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb164-7" aria-hidden="true" tabindex="-1"></a><span class="co"># build a design matrix</span></span>
<span id="cb164-8"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb164-8" aria-hidden="true" tabindex="-1"></a>HFHS.mod.full <span class="ot">=</span> <span class="fu">model.matrix</span>(<span class="sc">~</span> Diet_effect <span class="sc">+</span> Day_effect <span class="sc">+</span> HFHS.normFactor)</span>
<span id="cb164-9"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb164-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb164-10"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb164-10" aria-hidden="true" tabindex="-1"></a><span class="co"># settings for the fitZig() function</span></span>
<span id="cb164-11"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb164-11" aria-hidden="true" tabindex="-1"></a>HFHS.settings <span class="ot">&lt;-</span> <span class="fu">zigControl</span>(<span class="at">maxit =</span> <span class="dv">10</span>, <span class="at">verbose =</span> <span class="cn">TRUE</span>)</span>
<span id="cb164-12"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb164-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb164-13"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb164-13" aria-hidden="true" tabindex="-1"></a><span class="co"># apply the ZIG model</span></span>
<span id="cb164-14"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb164-14" aria-hidden="true" tabindex="-1"></a>HFHSfit <span class="ot">&lt;-</span> <span class="fu">fitZig</span>(<span class="at">obj =</span> HFHS.obj, <span class="at">mod =</span> HFHS.mod.full, </span>
<span id="cb164-15"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb164-15" aria-hidden="true" tabindex="-1"></a>                  <span class="at">useCSSoffset =</span> <span class="cn">FALSE</span>, <span class="at">control =</span> HFHS.settings)</span></code></pre></div>
<pre><code>## it= 0, nll=236.34, log10(eps+1)=Inf, stillActive=1255
## it= 1, nll=256.75, log10(eps+1)=0.06, stillActive=203
## it= 2, nll=255.94, log10(eps+1)=0.04, stillActive=153
## it= 3, nll=256.56, log10(eps+1)=0.04, stillActive=47
## it= 4, nll=257.05, log10(eps+1)=0.01, stillActive=11
## it= 5, nll=257.28, log10(eps+1)=0.01, stillActive=3
## it= 6, nll=257.39, log10(eps+1)=0.02, stillActive=2
## it= 7, nll=257.46, log10(eps+1)=0.00, stillActive=0</code></pre>
<p>The OTUs with the top 50 smallest p values are extracted using <code>MRcoefs()</code>. We set <code>eff = 0.5</code>, so only the OTUs with at least “0.5” quantile (50%) number of effective samples (positive samples + estimated undersampling zeroes) are extracted.</p>
<div class="sourceCode" id="cb166"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb166-1"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb166-1" aria-hidden="true" tabindex="-1"></a>HFHScoefs <span class="ot">&lt;-</span> <span class="fu">MRcoefs</span>(HFHSfit, <span class="at">coef =</span> <span class="dv">2</span>, <span class="at">group =</span> <span class="dv">3</span>, <span class="at">number =</span> <span class="dv">50</span>, <span class="at">eff =</span> <span class="fl">0.5</span>)</span>
<span id="cb166-2"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb166-2" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(HFHScoefs)</span></code></pre></div>
<pre><code>##         Diet_effectNormal      pvalues   adjPvalues
## 324926          1.0156067 7.360015e-11 9.236819e-08
## 337550          1.1730238 1.110138e-09 4.362904e-07
## 4353745        -0.7058975 1.212492e-09 4.362904e-07
## 357471          0.9636651 1.390567e-09 4.362904e-07
## 313499         -0.6093954 2.160477e-09 5.422797e-07
## 3940440        -0.7199750 5.150634e-09 1.077341e-06</code></pre>
<p><strong>Other methods adapted for microbiome data</strong></p>
<p><strong>Linear regression</strong> Linear regression is conducted with <code>linear_regres()</code> function in <span style="color: #FF7F00;">PLSDAbatch</span>. We integrated the <span style="color: #FF7F00;">performance</span> package that assesses performance of regression models into our function <code>linear_regres()</code>. Therefore, we can apply <code>check_model()</code> from <span style="color: #FF7F00;">performance</span> to the outputs from <code>linear_regres()</code> to diagnose the validity of the model fitted with treatment and batch effects for each variable <span class="citation">(<a href="#ref-daniel2020performance" role="doc-biblioref">LÃŒdecke et al. 2020</a>)</span>. We can extract performance measurements such as adjusted R2, RMSE, RSE, AIC and BIC for the models fitted with and without batch effects, which are also the outputs of <code>linear_regres()</code>.</p>
<p>To deal with the day effect only, we apply <code>type = "linear model"</code> because of the balanced batch x treatment design.</p>
<div class="sourceCode" id="cb168"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb168-1"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb168-1" aria-hidden="true" tabindex="-1"></a>hfhs.lm <span class="ot">&lt;-</span> <span class="fu">linear_regres</span>(<span class="at">data =</span> hfhs.clr.less, </span>
<span id="cb168-2"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb168-2" aria-hidden="true" tabindex="-1"></a>                         <span class="at">trt =</span> hfhs.trt.less, </span>
<span id="cb168-3"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb168-3" aria-hidden="true" tabindex="-1"></a>                         <span class="at">batch.fix =</span> hfhs.day.less, </span>
<span id="cb168-4"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb168-4" aria-hidden="true" tabindex="-1"></a>                         <span class="at">type =</span> <span class="st">&#39;linear model&#39;</span>)</span>
<span id="cb168-5"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb168-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb168-6"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb168-6" aria-hidden="true" tabindex="-1"></a>hfhs.p <span class="ot">&lt;-</span> <span class="fu">sapply</span>(hfhs.lm<span class="sc">$</span>lm.table, <span class="cf">function</span>(x){x<span class="sc">$</span>coefficients[<span class="dv">2</span>,<span class="dv">4</span>]})</span>
<span id="cb168-7"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb168-7" aria-hidden="true" tabindex="-1"></a>hfhs.p.adj <span class="ot">&lt;-</span> <span class="fu">p.adjust</span>(<span class="at">p =</span> hfhs.p, <span class="at">method =</span> <span class="st">&#39;fdr&#39;</span>)</span>
<span id="cb168-8"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb168-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb168-9"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb168-9" aria-hidden="true" tabindex="-1"></a><span class="fu">check_model</span>(hfhs.lm<span class="sc">$</span>model<span class="sc">$</span><span class="st">`</span><span class="at">541135</span><span class="st">`</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:HFHSlm"></span>
<img src="PLSDAbatch_workflow_files/figure-html/HFHSlm-1.png" alt="Diagnostic plots for the model fitted with the day effect of &quot;OTU 541135&quot; in the HFHS data." width="100%" />
<p class="caption">
Figure 2.15: Diagnostic plots for the model fitted with the day effect of “OTU 541135” in the HFHS data.
</p>
</div>
<p>To diagnose the validity of the model fitted with both treatment and batch effects, we use different plots to check the assumptions of each microbial variable. For example, the diagnostic plots of “OTU 541135” are shown in the above figure panel. The simulated data under the fitted model were not very close to the real data (shown as green line), indicating an imperfect fitness (top panel). The linearity (or homoscedasticity) and homogeneity of variance were not satisfied. The correlation between batch (<code>batch.fix</code>) and treatment (<code>trt</code>) effects was very low, indicating a good model with low collinearity. Some samples could be classified as outliers with a Cook’s distance larger than or equal to 0.5, for example, “46”, “35”, “126”, “125” and “16” (middle panel). The distribution of residuals was very close to normal (bottom panel). For the microbial variables with some assumptions not met, we should be careful about their results.</p>
<p>For performance measurements of models fitted with or without batch effects, We show an example of the results for some variables.</p>
<div class="sourceCode" id="cb169"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb169-1"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb169-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(hfhs.lm<span class="sc">$</span>adj.R2)</span></code></pre></div>
<pre><code>##         trt.only trt.batch
## 541135 0.1969688 0.2590190
## 276629 0.6419160 0.6414949
## 196070 0.1618319 0.1530262
## 318732 0.1720195 0.1795344
## 339886 0.1542615 0.1779073
## 337724 0.2343320 0.2445358</code></pre>
<p>If the adjusted <span class="math inline">\(R^2\)</span> of the model with both treatment and batch effects is larger than the model with treatment effects only, e.g., OTU 541135, the model fitted with batch effects explains more data variance, and is thus better than the model without batch effects. Otherwise, the batch effect is not necessary to be added into the linear model.</p>
<p>We next look at the AIC of models fitted with or without batch effects.</p>
<div class="sourceCode" id="cb171"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb171-1"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb171-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(hfhs.lm<span class="sc">$</span>AIC)</span></code></pre></div>
<pre><code>##        trt.only trt.batch
## 541135 462.0799  452.0564
## 276629 294.7536  296.8876
## 196070 504.2549  507.7710
## 318732 462.8639  463.4643
## 339886 248.0616  245.7953
## 337724 424.2142  424.1741</code></pre>
<p>A lower AIC indicates a better fit. Both results were mostly consistent on model selection, except “OTU318732”. Based on adjusted <span class="math inline">\(R^2\)</span>, we needed to include batch effects, while based on AIC, we should not. Therefore, we needed more measurements such as BIC, RSE.</p>
<p>To consider the cage effect only or the cage and day effects together, we apply <code>type = "linear mixed model"</code> to the <span style="color: #808000;">HFHS data</span> because of the nested batch x treatment design for the cage effect.</p>
<div class="sourceCode" id="cb173"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb173-1"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb173-1" aria-hidden="true" tabindex="-1"></a>hfhs.lmm <span class="ot">&lt;-</span> <span class="fu">linear_regres</span>(<span class="at">data =</span> hfhs.clr.less, </span>
<span id="cb173-2"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb173-2" aria-hidden="true" tabindex="-1"></a>                          <span class="at">trt =</span> hfhs.trt.less, </span>
<span id="cb173-3"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb173-3" aria-hidden="true" tabindex="-1"></a>                          <span class="at">batch.fix =</span> hfhs.day.less, </span>
<span id="cb173-4"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb173-4" aria-hidden="true" tabindex="-1"></a>                          <span class="at">batch.random =</span> hfhs.cage.less,</span>
<span id="cb173-5"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb173-5" aria-hidden="true" tabindex="-1"></a>                          <span class="at">type =</span> <span class="st">&#39;linear mixed model&#39;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb174"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb174-1"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb174-1" aria-hidden="true" tabindex="-1"></a>hfhs.p <span class="ot">&lt;-</span> <span class="fu">sapply</span>(hfhs.lmm<span class="sc">$</span>lmm.table, <span class="cf">function</span>(x){x<span class="sc">$</span>coefficients[<span class="dv">2</span>,<span class="dv">5</span>]})</span>
<span id="cb174-2"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb174-2" aria-hidden="true" tabindex="-1"></a>hfhs.p.adj <span class="ot">&lt;-</span> <span class="fu">p.adjust</span>(<span class="at">p =</span> hfhs.p, <span class="at">method =</span> <span class="st">&#39;fdr&#39;</span>)</span>
<span id="cb174-3"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb174-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb174-4"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb174-4" aria-hidden="true" tabindex="-1"></a><span class="fu">check_model</span>(hfhs.lmm<span class="sc">$</span>model<span class="sc">$</span><span class="st">`</span><span class="at">541135</span><span class="st">`</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:HFHSlmm"></span>
<img src="PLSDAbatch_workflow_files/figure-html/HFHSlmm-1.png" alt="Diagnostic plots for the model fitted with the day and cage effects of &quot;OTU541135&quot; in the HFHS data." width="100%" />
<p class="caption">
Figure 2.16: Diagnostic plots for the model fitted with the day and cage effects of “OTU541135” in the HFHS data.
</p>
</div>
<p>According to the diagnostic plots of “OTU541135” as shown in the above figure panel, The simulated data under the fitted model were not very close to the real data (shown as green line), indicating an imperfect fitness (top panel). The linearity (or homoscedasticity) and homogeneity of variance were not satisfied. The correlation between batch (<code>batch.fix</code>) and treatment (<code>trt</code>) effects was very low, indicating a good model with low collinearity. Some samples could be classified as outliers with a Cook’s distance larger than or equal to 0.5, for example, “16”, “35”, “126”, “125” and “136” (middle panel). The distribution of residuals and random effects was very close to normal (bottom panel).</p>
<p>For performance measurements of models fitted with or without batch effects, We show an example of the results for some variables.</p>
<div class="sourceCode" id="cb175"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb175-1"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb175-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(hfhs.lmm<span class="sc">$</span>AIC)</span></code></pre></div>
<pre><code>##        trt.only trt.batch
## 541135 462.0799  461.4074
## 276629 294.7536  310.3455
## 196070 504.2549  515.0180
## 318732 462.8639  472.5556
## 339886 248.0616  260.7115
## 337724 424.2142  432.9937</code></pre>
<p>For some OTUs, the model fitted without batch effects was better with a lower AIC. For example, “OTU276629”, “OTU196070”, “OTU318732”, “OTU339886” and “OTU337724”. “OTU541135” was more appropriate within a model with batch effects.</p>
<p>Since we apply a <code>"linear mixed model"</code> to the <span style="color: #808000;">HFHS data</span>, this type of model can only output conditional <span class="math inline">\(R^2\)</span> that includes the variance of both fixed and random effects (treatment, fixed and random batch effects) and marginal <span class="math inline">\(R^2\)</span> that includes only the variance of fixed effects (treatment and fixed batch effects) <span class="citation">(<a href="#ref-nakagawa2013general" role="doc-biblioref">Nakagawa and Schielzeth 2013</a>)</span>.</p>
<div class="sourceCode" id="cb177"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb177-1"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb177-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(hfhs.lmm<span class="sc">$</span>cond.R2)</span></code></pre></div>
<pre><code>##         trt.only trt.batch
## 541135 0.2023947 0.2748381
## 276629 0.6443355 0.6475581
## 196070 0.1674952 0.1932495
## 318732 0.1776140        NA
## 339886 0.1599760 0.1954862
## 337724 0.2395055 0.2845443</code></pre>
<div class="sourceCode" id="cb179"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb179-1"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb179-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(hfhs.lmm<span class="sc">$</span>marg.R2)</span></code></pre></div>
<pre><code>##         trt.only trt.batch
## 541135 0.2023947 0.2692818
## 276629 0.6443355 0.6435223
## 196070 0.1674952 0.1678637
## 318732 0.1776140 0.1929564
## 339886 0.1599760 0.1916296
## 337724 0.2395055 0.2520680</code></pre>
<p>We observed that some variables resulted in singular fits (<code>error message: Can't compute random effect variances. Some variance components equal zero. Your model may suffer from singulariy.</code>) and the conditional <span class="math inline">\(R^2\)</span>s of these variables were “NA”, which are expected to happen in linear mixed models when covariates are nested. We recommend noting the variables for which this error occurs, as it may lead to unreliable results.</p>
<p><strong>SVA</strong></p>
<p>SVA accounts for unknown batch effects. Here we assume that the batch grouping information in the <span style="color: #808000;">HFHS data</span> is unknown. We first build two design matrices with (<code>hfhs.mod</code>) and without (<code>hfhs.mod0</code>) treatment grouping information generated with <code>model.matrix()</code> function from <span style="color: #FF7F00;">stats</span>. We then use <code>num.sv()</code> from <span style="color: #FF7F00;">sva</span> package to determine the number of batch variables <code>n.sv</code> that will be used to estimate batch effects in function <code>sva()</code>.</p>
<div class="sourceCode" id="cb181"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb181-1"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb181-1" aria-hidden="true" tabindex="-1"></a><span class="co"># estimate batch matrix</span></span>
<span id="cb181-2"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb181-2" aria-hidden="true" tabindex="-1"></a>hfhs.mod <span class="ot">&lt;-</span> <span class="fu">model.matrix</span>( <span class="sc">~</span> hfhs.trt.less)</span>
<span id="cb181-3"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb181-3" aria-hidden="true" tabindex="-1"></a>hfhs.mod0 <span class="ot">&lt;-</span> <span class="fu">model.matrix</span>( <span class="sc">~</span> <span class="dv">1</span>, <span class="at">data =</span> hfhs.trt.less)</span>
<span id="cb181-4"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb181-4" aria-hidden="true" tabindex="-1"></a>hfhs.sva.n <span class="ot">&lt;-</span> <span class="fu">num.sv</span>(<span class="at">dat =</span> <span class="fu">t</span>(hfhs.clr.less), <span class="at">mod =</span> hfhs.mod, </span>
<span id="cb181-5"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb181-5" aria-hidden="true" tabindex="-1"></a>                     <span class="at">method =</span> <span class="st">&#39;leek&#39;</span>)</span>
<span id="cb181-6"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb181-6" aria-hidden="true" tabindex="-1"></a>hfhs.sva <span class="ot">&lt;-</span> <span class="fu">sva</span>(<span class="fu">t</span>(hfhs.clr.less), hfhs.mod, hfhs.mod0, <span class="at">n.sv =</span> hfhs.sva.n)</span></code></pre></div>
<pre><code>## Number of significant surrogate variables is:  1 
## Iteration (out of 5 ):1  2  3  4  5</code></pre>
<p>The estimated batch effects are then applied to <code>f.pvalue()</code> to calculate the P-values of treatment effects. The estimated batch effects in SVA are assumed to be independent of the treatment effects. However, SVA considers some correlation between batch and treatment effects <span class="citation">(<a href="#ref-wang2020managing" role="doc-biblioref">Y. Wang and LêCao 2020</a>)</span>.</p>
<div class="sourceCode" id="cb183"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb183-1"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb183-1" aria-hidden="true" tabindex="-1"></a><span class="co"># include estimated batch effects in the linear model</span></span>
<span id="cb183-2"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb183-2" aria-hidden="true" tabindex="-1"></a>hfhs.mod.batch <span class="ot">&lt;-</span> <span class="fu">cbind</span>(hfhs.mod, hfhs.sva<span class="sc">$</span>sv)</span>
<span id="cb183-3"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb183-3" aria-hidden="true" tabindex="-1"></a>hfhs.mod0.batch <span class="ot">&lt;-</span> <span class="fu">cbind</span>(hfhs.mod0, hfhs.sva<span class="sc">$</span>sv)</span>
<span id="cb183-4"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb183-4" aria-hidden="true" tabindex="-1"></a>hfhs.sva.p <span class="ot">&lt;-</span> <span class="fu">f.pvalue</span>(<span class="fu">t</span>(hfhs.clr.less), hfhs.mod.batch, hfhs.mod0.batch)</span>
<span id="cb183-5"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb183-5" aria-hidden="true" tabindex="-1"></a>hfhs.sva.p.adj <span class="ot">&lt;-</span> <span class="fu">p.adjust</span>(hfhs.sva.p, <span class="at">method =</span> <span class="st">&#39;fdr&#39;</span>)</span></code></pre></div>
<p><strong>RUV4</strong></p>
<p>Before applying RUV4 (<code>RUV4()</code> from <span style="color: #FF7F00;">ruv</span>, we need to specify negative control variables and the number of batch variables to estimate. We can use the empirical negative controls that are not significantly differentially abundant (adjusted P &gt; 0.05) from a linear regression with the treatment information as the only covariate.</p>
<p>We use a loop to fit a linear regression for each microbial variable and adjust P values of treatment effects for multiple comparisons with <code>p.adjust()</code> from <span style="color: #FF7F00;">stats</span>. The empirical negative controls are then extracted according to the adjusted P values.</p>
<div class="sourceCode" id="cb184"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb184-1"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb184-1" aria-hidden="true" tabindex="-1"></a><span class="co"># empirical negative controls</span></span>
<span id="cb184-2"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb184-2" aria-hidden="true" tabindex="-1"></a>hfhs.empir.p <span class="ot">&lt;-</span> <span class="fu">c</span>()</span>
<span id="cb184-3"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb184-3" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(e <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">ncol</span>(hfhs.clr.less)){</span>
<span id="cb184-4"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb184-4" aria-hidden="true" tabindex="-1"></a>  hfhs.empir.lm <span class="ot">&lt;-</span> <span class="fu">lm</span>(hfhs.clr.less[,e] <span class="sc">~</span> hfhs.trt.less)</span>
<span id="cb184-5"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb184-5" aria-hidden="true" tabindex="-1"></a>  hfhs.empir.p[e] <span class="ot">&lt;-</span> <span class="fu">summary</span>(hfhs.empir.lm)<span class="sc">$</span>coefficients[<span class="dv">2</span>,<span class="dv">4</span>]</span>
<span id="cb184-6"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb184-6" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb184-7"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb184-7" aria-hidden="true" tabindex="-1"></a>hfhs.empir.p.adj <span class="ot">&lt;-</span> <span class="fu">p.adjust</span>(<span class="at">p =</span> hfhs.empir.p, <span class="at">method =</span> <span class="st">&#39;fdr&#39;</span>)</span>
<span id="cb184-8"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb184-8" aria-hidden="true" tabindex="-1"></a>hfhs.nc <span class="ot">&lt;-</span> hfhs.empir.p.adj <span class="sc">&gt;</span> <span class="fl">0.05</span></span></code></pre></div>
<p>The number of batch variables <code>k</code> can be determined using <code>getK()</code> function.</p>
<div class="sourceCode" id="cb185"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb185-1"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb185-1" aria-hidden="true" tabindex="-1"></a><span class="co"># estimate k</span></span>
<span id="cb185-2"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb185-2" aria-hidden="true" tabindex="-1"></a>hfhs.k.res <span class="ot">&lt;-</span> <span class="fu">getK</span>(<span class="at">Y =</span> hfhs.clr.less, <span class="at">X =</span> hfhs.trt.less, <span class="at">ctl =</span> hfhs.nc)</span>
<span id="cb185-3"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb185-3" aria-hidden="true" tabindex="-1"></a>hfhs.k <span class="ot">&lt;-</span> hfhs.k.res<span class="sc">$</span>k</span>
<span id="cb185-4"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb185-4" aria-hidden="true" tabindex="-1"></a>hfhs.k <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(hfhs.k <span class="sc">!=</span> <span class="dv">0</span>, hfhs.k, <span class="dv">1</span>)</span></code></pre></div>
<p>We then apply <code>RUV4()</code> with known treatment variables, estimated negative control variables and <code>k</code> batch variables. The calculated P values also need to be adjusted for multiple comparisons.</p>
<div class="sourceCode" id="cb186"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb186-1"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb186-1" aria-hidden="true" tabindex="-1"></a>hfhs.ruv4 <span class="ot">&lt;-</span> <span class="fu">RUV4</span>(<span class="at">Y =</span> hfhs.clr.less, <span class="at">X =</span> hfhs.trt.less, </span>
<span id="cb186-2"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb186-2" aria-hidden="true" tabindex="-1"></a>                  <span class="at">ctl =</span> hfhs.nc, <span class="at">k =</span> hfhs.k) </span>
<span id="cb186-3"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb186-3" aria-hidden="true" tabindex="-1"></a>hfhs.ruv4.p <span class="ot">&lt;-</span> hfhs.ruv4<span class="sc">$</span>p</span>
<span id="cb186-4"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb186-4" aria-hidden="true" tabindex="-1"></a>hfhs.ruv4.p.adj <span class="ot">&lt;-</span> <span class="fu">p.adjust</span>(hfhs.ruv4.p, <span class="at">method =</span> <span class="st">&quot;fdr&quot;</span>)</span></code></pre></div>
<p>Both SVA and RUV4 can account for both the cage and day effects, but not efficient at accounting for the cage effect, as the estimated surrogate batch effect in SVA and negative controls in RUV4 may not capture all the batch variation because of the nested batch x treatment design of the cage effect in the <span style="color: #808000;">HFHS data</span>.</p>
</div>
<div id="correcting-for-batch-effects-1" class="section level4 hasAnchor" number="2.2.3.2">
<h4><span class="header-section-number">2.2.3.2</span> Correcting for batch effects<a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#correcting-for-batch-effects-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>The methods that we use to correct for batch effects include removeBatchEffect, ComBat, PLSDA-batch, sPLSDA-batch, Percentile Normalisation and RUVIII. Among them, RUVIII is designed for unknown batch effects.</p>
<p>Since the cage effect cannot be corrected for because of the collinearity between the cage and treatment effects and the cage effect only accounted for a small amount of the total data variance (1.3% calculated with pRDA), so we only correct for the day effect in the <span style="color: #808000;">HFHS data</span>.</p>
<p><strong>removeBatchEffect</strong></p>
<p>The <code>removeBatchEffect()</code> function is implemented in <span style="color: #FF7F00;">limma</span> package. The design matrix (<code>design</code>) with treatment grouping information can be generated with <code>model.matrix()</code> function from <span style="color: #FF7F00;">stats</span> as shown in section <strong>accounting for batch effects</strong> method “SVA”.</p>
<p>Here we use <code>removeBatchEffect()</code> function with batch grouping information (<code>batch</code>) and treatment design matrix (<code>design</code>) to calculate batch effect corrected data <code>hfhs.rBE</code>.</p>
<div class="sourceCode" id="cb187"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb187-1"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb187-1" aria-hidden="true" tabindex="-1"></a>hfhs.rBE <span class="ot">&lt;-</span> <span class="fu">t</span>(<span class="fu">removeBatchEffect</span>(<span class="fu">t</span>(hfhs.clr.less), <span class="at">batch =</span> hfhs.day.less, </span>
<span id="cb187-2"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb187-2" aria-hidden="true" tabindex="-1"></a>                                <span class="at">design =</span> hfhs.mod))</span></code></pre></div>
<p><strong>ComBat</strong></p>
<p>The <code>ComBat()</code> function (from <span style="color: #FF7F00;">sva</span> package) is implemented as parametric or non-parametric correction with option <code>par.prior</code>. Under a parametric adjustment, we can assess the model’s validity with <code>prior.plots = T</code> <span class="citation">(<a href="#ref-leek2012sva" role="doc-biblioref">Leek et al. 2012</a>)</span>.</p>
<p>Here we use a non-parametric correction (<code>par.prior = F</code>) with batch grouping information (<code>batch</code>) and treatment design matrix (<code>mod</code>) to calculate batch effect corrected data <code>hfhs.ComBat</code>.</p>
<div class="sourceCode" id="cb188"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb188-1"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb188-1" aria-hidden="true" tabindex="-1"></a>hfhs.ComBat <span class="ot">&lt;-</span> <span class="fu">t</span>(<span class="fu">ComBat</span>(<span class="fu">t</span>(hfhs.clr.less), <span class="at">batch =</span> hfhs.day.less, </span>
<span id="cb188-2"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb188-2" aria-hidden="true" tabindex="-1"></a>                        <span class="at">mod =</span> hfhs.mod, <span class="at">par.prior =</span> F))</span></code></pre></div>
<p><strong>PLSDA-batch</strong></p>
<p>The <code>PLSDA_batch()</code> function is implemented in <span style="color: #FF7F00;">PLSDAbatch</span> package. To use this function, we need to specify the optimal number of components related to treatment (<code>ncomp.trt</code>) or batch effects (<code>ncomp.bat</code>).</p>
<p>Here in the <span style="color: #808000;">HFHS data</span>, we use <code>plsda()</code> from <span style="color: #FF7F00;">mixOmics</span> with only treatment grouping information to estimate the optimal number of treatment components to preserve.</p>
<div class="sourceCode" id="cb189"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb189-1"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb189-1" aria-hidden="true" tabindex="-1"></a><span class="co"># estimate the number of treatment components</span></span>
<span id="cb189-2"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb189-2" aria-hidden="true" tabindex="-1"></a>hfhs.trt.tune <span class="ot">&lt;-</span> <span class="fu">plsda</span>(<span class="at">X =</span> hfhs.clr.less, <span class="at">Y =</span> hfhs.trt.less, <span class="at">ncomp =</span> <span class="dv">5</span>)</span>
<span id="cb189-3"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb189-3" aria-hidden="true" tabindex="-1"></a>hfhs.trt.tune<span class="sc">$</span>prop_expl_var <span class="co">#1</span></span></code></pre></div>
<pre><code>## $X
##      comp1      comp2      comp3      comp4      comp5 
## 0.29946816 0.06913457 0.04242662 0.02681544 0.02189936 
## 
## $Y
##      comp1      comp2      comp3      comp4      comp5 
## 1.00000000 0.07124125 0.03858356 0.02564995 0.01555355</code></pre>
<p>We choose the number that explains 100% variance in the outcome matrix <code>Y</code>, thus from the result, 1 component is enough to preserve the treatment information.</p>
<p>We then use <code>PLSDA_batch()</code> function with both treatment and batch grouping information to estimate the optimal number of batch components to remove.</p>
<div class="sourceCode" id="cb191"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb191-1"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb191-1" aria-hidden="true" tabindex="-1"></a><span class="co"># estimate the number of batch components</span></span>
<span id="cb191-2"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb191-2" aria-hidden="true" tabindex="-1"></a>hfhs.batch.tune <span class="ot">&lt;-</span> <span class="fu">PLSDA_batch</span>(<span class="at">X =</span> hfhs.clr.less, </span>
<span id="cb191-3"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb191-3" aria-hidden="true" tabindex="-1"></a>                               <span class="at">Y.trt =</span> hfhs.trt.less, </span>
<span id="cb191-4"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb191-4" aria-hidden="true" tabindex="-1"></a>                               <span class="at">Y.bat =</span> hfhs.day.less,</span>
<span id="cb191-5"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb191-5" aria-hidden="true" tabindex="-1"></a>                               <span class="at">ncomp.trt =</span> <span class="dv">1</span>, <span class="at">ncomp.bat =</span> <span class="dv">10</span>)</span>
<span id="cb191-6"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb191-6" aria-hidden="true" tabindex="-1"></a>hfhs.batch.tune<span class="sc">$</span>explained_variance.bat <span class="co">#2</span></span></code></pre></div>
<pre><code>## $X
##      comp1      comp2      comp3      comp4      comp5      comp6      comp7 
## 0.10753644 0.03608189 0.02988326 0.02634760 0.02728936 0.05015144 0.03120534 
##      comp8      comp9     comp10 
## 0.02908653 0.02430653 0.02103238 
## 
## $Y
##        comp1        comp2        comp3        comp4        comp5        comp6 
## 0.4937952138 0.5062047862 0.5126429685 0.4867993776 0.0005576539 0.5083690943 
##        comp7        comp8        comp9       comp10 
## 0.4789523528 0.0126785529 0.5103228185 0.4734938502</code></pre>
<div class="sourceCode" id="cb193"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb193-1"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb193-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sum</span>(hfhs.batch.tune<span class="sc">$</span>explained_variance.bat<span class="sc">$</span>Y[<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>])</span></code></pre></div>
<pre><code>## [1] 1</code></pre>
<p>Using the same criterion as choosing treatment components, we choose the number of batch components that explains 100% variance in the outcome matrix of batch. According to the result, 2 components are required to remove batch effects.</p>
<p>We then can correct for batch effects applying <code>PLSDA_batch()</code> with treatment, batch grouping information and corresponding optimal number of related components.</p>
<div class="sourceCode" id="cb195"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb195-1"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb195-1" aria-hidden="true" tabindex="-1"></a><span class="do">##########</span></span>
<span id="cb195-2"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb195-2" aria-hidden="true" tabindex="-1"></a>hfhs.PLSDA_batch.res <span class="ot">&lt;-</span> <span class="fu">PLSDA_batch</span>(<span class="at">X =</span> hfhs.clr.less, </span>
<span id="cb195-3"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb195-3" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">Y.trt =</span> hfhs.trt.less, </span>
<span id="cb195-4"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb195-4" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">Y.bat =</span> hfhs.day.less,</span>
<span id="cb195-5"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb195-5" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">ncomp.trt =</span> <span class="dv">1</span>, <span class="at">ncomp.bat =</span> <span class="dv">2</span>)</span>
<span id="cb195-6"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb195-6" aria-hidden="true" tabindex="-1"></a>hfhs.PLSDA_batch <span class="ot">&lt;-</span> hfhs.PLSDA_batch.res<span class="sc">$</span>X.nobatch</span></code></pre></div>
<p><strong>sPLSDA-batch</strong></p>
<p>We apply sPLSDA-batch using the same function <code>PLSDA_batch()</code> but we specify the number of variables to select on each component (usually only treatment-related components <code>keepX.trt</code>). To determine the optimal number of variables to select, we use <code>tune.splsda()</code> function from <span style="color: #FF7F00;">mixOmics</span> package <span class="citation">(<a href="#ref-rohart2017mixomics" role="doc-biblioref">Rohart et al. 2017</a>)</span> with all possible numbers of variables to select for each component (<code>test.keepX</code>).</p>
<div class="sourceCode" id="cb196"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb196-1"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb196-1" aria-hidden="true" tabindex="-1"></a><span class="co"># estimate the number of variables to select per treatment component</span></span>
<span id="cb196-2"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb196-2" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">777</span>)</span>
<span id="cb196-3"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb196-3" aria-hidden="true" tabindex="-1"></a>hfhs.test.keepX <span class="ot">=</span> <span class="fu">c</span>(<span class="fu">seq</span>(<span class="dv">1</span>, <span class="dv">10</span>, <span class="dv">1</span>), <span class="fu">seq</span>(<span class="dv">20</span>, <span class="dv">200</span>, <span class="dv">10</span>), </span>
<span id="cb196-4"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb196-4" aria-hidden="true" tabindex="-1"></a>                    <span class="fu">seq</span>(<span class="dv">250</span>, <span class="dv">500</span>, <span class="dv">50</span>), <span class="dv">515</span>)</span>
<span id="cb196-5"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb196-5" aria-hidden="true" tabindex="-1"></a>hfhs.trt.tune.v <span class="ot">&lt;-</span> <span class="fu">tune.splsda</span>(<span class="at">X =</span> hfhs.clr.less, </span>
<span id="cb196-6"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb196-6" aria-hidden="true" tabindex="-1"></a>                               <span class="at">Y =</span> hfhs.trt.less, </span>
<span id="cb196-7"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb196-7" aria-hidden="true" tabindex="-1"></a>                               <span class="at">ncomp =</span> <span class="dv">1</span>, </span>
<span id="cb196-8"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb196-8" aria-hidden="true" tabindex="-1"></a>                               <span class="at">test.keepX =</span> hfhs.test.keepX, </span>
<span id="cb196-9"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb196-9" aria-hidden="true" tabindex="-1"></a>                               <span class="at">validation =</span> <span class="st">&#39;Mfold&#39;</span>, </span>
<span id="cb196-10"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb196-10" aria-hidden="true" tabindex="-1"></a>                               <span class="at">folds =</span> <span class="dv">4</span>, <span class="at">nrepeat =</span> <span class="dv">50</span>)</span>
<span id="cb196-11"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb196-11" aria-hidden="true" tabindex="-1"></a>hfhs.trt.tune.v<span class="sc">$</span>choice.keepX <span class="co"># 2</span></span></code></pre></div>
<pre><code>## comp1 
##     2</code></pre>
<p>Here the optimal number of variables to select for the treatment component is 2. Since we adjust the amount of treatment variation to preserve, we need to re-choose the optimal number of components related to batch effects using the same criterion mentioned in section <strong>correcting for batch effects</strong> method “PLSDA-batch”.</p>
<div class="sourceCode" id="cb198"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb198-1"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb198-1" aria-hidden="true" tabindex="-1"></a><span class="co"># estimate the number of batch components</span></span>
<span id="cb198-2"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb198-2" aria-hidden="true" tabindex="-1"></a>hfhs.batch.tune <span class="ot">&lt;-</span> <span class="fu">PLSDA_batch</span>(<span class="at">X =</span> hfhs.clr.less, </span>
<span id="cb198-3"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb198-3" aria-hidden="true" tabindex="-1"></a>                               <span class="at">Y.trt =</span> hfhs.trt.less, </span>
<span id="cb198-4"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb198-4" aria-hidden="true" tabindex="-1"></a>                               <span class="at">Y.bat =</span> hfhs.day.less,</span>
<span id="cb198-5"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb198-5" aria-hidden="true" tabindex="-1"></a>                               <span class="at">ncomp.trt =</span> <span class="dv">1</span>, </span>
<span id="cb198-6"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb198-6" aria-hidden="true" tabindex="-1"></a>                               <span class="at">keepX.trt =</span> <span class="dv">2</span>,</span>
<span id="cb198-7"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb198-7" aria-hidden="true" tabindex="-1"></a>                               <span class="at">ncomp.bat =</span> <span class="dv">10</span>)</span>
<span id="cb198-8"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb198-8" aria-hidden="true" tabindex="-1"></a>hfhs.batch.tune<span class="sc">$</span>explained_variance.bat <span class="co">#2</span></span></code></pre></div>
<pre><code>## $X
##      comp1      comp2      comp3      comp4      comp5      comp6      comp7 
## 0.11496719 0.03611134 0.02995074 0.02497055 0.02954931 0.04284935 0.03468263 
##      comp8      comp9     comp10 
## 0.02679416 0.02394709 0.02270728 
## 
## $Y
##       comp1       comp2       comp3       comp4       comp5       comp6 
## 0.493868226 0.506131774 0.511212568 0.484741160 0.004046272 0.512078352 
##       comp7       comp8       comp9      comp10 
## 0.483749135 0.004172513 0.512156744 0.483406313</code></pre>
<div class="sourceCode" id="cb200"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb200-1"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb200-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sum</span>(hfhs.batch.tune<span class="sc">$</span>explained_variance.bat<span class="sc">$</span>Y[<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>])</span></code></pre></div>
<pre><code>## [1] 1</code></pre>
<p>According to the result, we need 2 batch related components to remove batch variance from the data with function <code>PLSDA_batch()</code>.</p>
<div class="sourceCode" id="cb202"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb202-1"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb202-1" aria-hidden="true" tabindex="-1"></a><span class="do">##########</span></span>
<span id="cb202-2"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb202-2" aria-hidden="true" tabindex="-1"></a>hfhs.sPLSDA_batch.res <span class="ot">&lt;-</span> <span class="fu">PLSDA_batch</span>(<span class="at">X =</span> hfhs.clr.less, </span>
<span id="cb202-3"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb202-3" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">Y.trt =</span> hfhs.trt.less, </span>
<span id="cb202-4"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb202-4" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">Y.bat =</span> hfhs.day.less,</span>
<span id="cb202-5"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb202-5" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">ncomp.trt =</span> <span class="dv">1</span>, </span>
<span id="cb202-6"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb202-6" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">keepX.trt =</span> <span class="dv">2</span>,</span>
<span id="cb202-7"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb202-7" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">ncomp.bat =</span> <span class="dv">2</span>)</span>
<span id="cb202-8"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb202-8" aria-hidden="true" tabindex="-1"></a>hfhs.sPLSDA_batch <span class="ot">&lt;-</span> hfhs.sPLSDA_batch.res<span class="sc">$</span>X.nobatch</span></code></pre></div>
<p><strong>Percentile Normalisation</strong></p>
<p>To apply <code>percentile_norm()</code> function from <span style="color: #FF7F00;">PLSDAbatch</span> package, we need to indicate a control group (<code>ctrl.grp</code>).</p>
<div class="sourceCode" id="cb203"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb203-1"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb203-1" aria-hidden="true" tabindex="-1"></a><span class="co"># HFHS data</span></span>
<span id="cb203-2"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb203-2" aria-hidden="true" tabindex="-1"></a>hfhs.PN <span class="ot">&lt;-</span> <span class="fu">percentile_norm</span>(<span class="at">data =</span> hfhs.clr.less, </span>
<span id="cb203-3"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb203-3" aria-hidden="true" tabindex="-1"></a>                           <span class="at">batch =</span> hfhs.day.less, </span>
<span id="cb203-4"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb203-4" aria-hidden="true" tabindex="-1"></a>                           <span class="at">trt =</span> hfhs.trt.less, </span>
<span id="cb203-5"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb203-5" aria-hidden="true" tabindex="-1"></a>                           <span class="at">ctrl.grp =</span> <span class="st">&#39;Normal&#39;</span>)</span></code></pre></div>
<p><strong>RUVIII</strong></p>
<p>The <code>RUVIII()</code> function is from <span style="color: #FF7F00;">ruv</span> package. Similar to <code>RUV4()</code>, we use empirical negative control variables and <code>getK()</code> to determine the number of batch variables (<code>k</code>) to estimate. We also need sample replicates which should be structured into a mapping matrix using <code>replicate.matrix()</code>. We can then obtain batch effect corrected data applying <code>RUVIII()</code> with above elements.</p>
<div class="sourceCode" id="cb204"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb204-1"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb204-1" aria-hidden="true" tabindex="-1"></a>hfhs.replicates <span class="ot">&lt;-</span> hfhs.metadata.less<span class="sc">$</span>MouseID</span>
<span id="cb204-2"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb204-2" aria-hidden="true" tabindex="-1"></a>hfhs.replicates.matrix <span class="ot">&lt;-</span> <span class="fu">replicate.matrix</span>(hfhs.replicates)</span>
<span id="cb204-3"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb204-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb204-4"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb204-4" aria-hidden="true" tabindex="-1"></a>hfhs.RUVIII <span class="ot">&lt;-</span> <span class="fu">RUVIII</span>(<span class="at">Y =</span> hfhs.clr.less, </span>
<span id="cb204-5"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb204-5" aria-hidden="true" tabindex="-1"></a>                      <span class="at">M =</span> hfhs.replicates.matrix, </span>
<span id="cb204-6"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb204-6" aria-hidden="true" tabindex="-1"></a>                      <span class="at">ctl =</span> hfhs.nc, <span class="at">k =</span> hfhs.k)</span>
<span id="cb204-7"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb204-7" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(hfhs.RUVIII) <span class="ot">&lt;-</span> <span class="fu">rownames</span>(hfhs.clr.less)</span></code></pre></div>
<p>Compared to the <span style="color: #0000FF;">AD data</span>, the <span style="color: #808000;">HFHS data</span> have more appropriate sample replicates that fully capture the difference between different time points. The results of RUVIII applied to <span style="color: #808000;">HFHS data</span> would be better than AD data.</p>
</div>
</div>
<div id="assessing-batch-effect-correction-2" class="section level3 hasAnchor" number="2.2.4">
<h3><span class="header-section-number">2.2.4</span> Assessing batch effect correction<a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#assessing-batch-effect-correction-2" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>We apply different visualisation and quantitative methods to assessing batch effect correction.</p>
<div id="methods-that-detect-batch-effects-2" class="section level4 hasAnchor" number="2.2.4.1">
<h4><span class="header-section-number">2.2.4.1</span> Methods that detect batch effects<a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#methods-that-detect-batch-effects-2" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p><strong>PCA</strong></p>
<p>In the <span style="color: #808000;">HFHS data</span>, we compare the PCA sample plots before and after batch effect correction with different methods.</p>
<div class="sourceCode" id="cb205"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb205-1"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb205-1" aria-hidden="true" tabindex="-1"></a>hfhs.pca.before <span class="ot">&lt;-</span> <span class="fu">pca</span>(hfhs.clr.less, <span class="at">ncomp =</span> <span class="dv">3</span>, </span>
<span id="cb205-2"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb205-2" aria-hidden="true" tabindex="-1"></a>                       <span class="at">scale =</span> <span class="cn">TRUE</span>)</span>
<span id="cb205-3"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb205-3" aria-hidden="true" tabindex="-1"></a>hfhs.pca.rBE <span class="ot">&lt;-</span> <span class="fu">pca</span>(hfhs.rBE, <span class="at">ncomp =</span> <span class="dv">3</span>, </span>
<span id="cb205-4"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb205-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">scale =</span> <span class="cn">TRUE</span>)</span>
<span id="cb205-5"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb205-5" aria-hidden="true" tabindex="-1"></a>hfhs.pca.ComBat <span class="ot">&lt;-</span> <span class="fu">pca</span>(hfhs.ComBat, <span class="at">ncomp =</span> <span class="dv">3</span>, </span>
<span id="cb205-6"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb205-6" aria-hidden="true" tabindex="-1"></a>                       <span class="at">scale =</span> <span class="cn">TRUE</span>)</span>
<span id="cb205-7"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb205-7" aria-hidden="true" tabindex="-1"></a>hfhs.pca.PLSDA_batch <span class="ot">&lt;-</span> <span class="fu">pca</span>(hfhs.PLSDA_batch, <span class="at">ncomp =</span> <span class="dv">3</span>, </span>
<span id="cb205-8"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb205-8" aria-hidden="true" tabindex="-1"></a>                            <span class="at">scale =</span> <span class="cn">TRUE</span>)</span>
<span id="cb205-9"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb205-9" aria-hidden="true" tabindex="-1"></a>hfhs.pca.sPLSDA_batch <span class="ot">&lt;-</span> <span class="fu">pca</span>(hfhs.sPLSDA_batch, <span class="at">ncomp =</span> <span class="dv">3</span>, </span>
<span id="cb205-10"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb205-10" aria-hidden="true" tabindex="-1"></a>                             <span class="at">scale =</span> <span class="cn">TRUE</span>)</span>
<span id="cb205-11"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb205-11" aria-hidden="true" tabindex="-1"></a>hfhs.pca.PN <span class="ot">&lt;-</span> <span class="fu">pca</span>(hfhs.PN, <span class="at">ncomp =</span> <span class="dv">3</span>, </span>
<span id="cb205-12"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb205-12" aria-hidden="true" tabindex="-1"></a>                   <span class="at">scale =</span> <span class="cn">TRUE</span>)</span>
<span id="cb205-13"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb205-13" aria-hidden="true" tabindex="-1"></a>hfhs.pca.RUVIII <span class="ot">&lt;-</span> <span class="fu">pca</span>(hfhs.RUVIII, <span class="at">ncomp =</span> <span class="dv">3</span>, </span>
<span id="cb205-14"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb205-14" aria-hidden="true" tabindex="-1"></a>                       <span class="at">scale =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<div class="sourceCode" id="cb206"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb206-1"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb206-1" aria-hidden="true" tabindex="-1"></a>hfhs.pca.before.plot <span class="ot">&lt;-</span> </span>
<span id="cb206-2"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb206-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">Scatter_Density</span>(<span class="at">object =</span> hfhs.pca.before, </span>
<span id="cb206-3"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb206-3" aria-hidden="true" tabindex="-1"></a>                  <span class="at">batch =</span> hfhs.day.less, </span>
<span id="cb206-4"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb206-4" aria-hidden="true" tabindex="-1"></a>                  <span class="at">trt =</span> hfhs.trt.less, </span>
<span id="cb206-5"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb206-5" aria-hidden="true" tabindex="-1"></a>                  <span class="at">title =</span> <span class="st">&#39;Before&#39;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb207"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb207-1"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb207-1" aria-hidden="true" tabindex="-1"></a>hfhs.pca.rBE.plot <span class="ot">&lt;-</span> </span>
<span id="cb207-2"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb207-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">Scatter_Density</span>(<span class="at">object =</span> hfhs.pca.rBE, </span>
<span id="cb207-3"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb207-3" aria-hidden="true" tabindex="-1"></a>                  <span class="at">batch =</span> hfhs.day.less, </span>
<span id="cb207-4"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb207-4" aria-hidden="true" tabindex="-1"></a>                  <span class="at">trt =</span> hfhs.trt.less, </span>
<span id="cb207-5"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb207-5" aria-hidden="true" tabindex="-1"></a>                  <span class="at">title =</span> <span class="st">&#39;removeBatchEffect&#39;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb208"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb208-1"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb208-1" aria-hidden="true" tabindex="-1"></a>hfhs.pca.ComBat.plot <span class="ot">&lt;-</span> </span>
<span id="cb208-2"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb208-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">Scatter_Density</span>(<span class="at">object =</span> hfhs.pca.ComBat, </span>
<span id="cb208-3"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb208-3" aria-hidden="true" tabindex="-1"></a>                  <span class="at">batch =</span> hfhs.day.less, </span>
<span id="cb208-4"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb208-4" aria-hidden="true" tabindex="-1"></a>                  <span class="at">trt =</span> hfhs.trt.less,</span>
<span id="cb208-5"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb208-5" aria-hidden="true" tabindex="-1"></a>                  <span class="at">title =</span> <span class="st">&#39;ComBat&#39;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb209"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb209-1"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb209-1" aria-hidden="true" tabindex="-1"></a>hfhs.pca.PLSDA_batch.plot <span class="ot">&lt;-</span> </span>
<span id="cb209-2"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb209-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">Scatter_Density</span>(<span class="at">object =</span> hfhs.pca.PLSDA_batch, </span>
<span id="cb209-3"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb209-3" aria-hidden="true" tabindex="-1"></a>                  <span class="at">batch =</span> hfhs.day.less, </span>
<span id="cb209-4"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb209-4" aria-hidden="true" tabindex="-1"></a>                  <span class="at">trt =</span> hfhs.trt.less, </span>
<span id="cb209-5"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb209-5" aria-hidden="true" tabindex="-1"></a>                  <span class="at">title =</span> <span class="st">&#39;PLSDA-batch&#39;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb210"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb210-1"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb210-1" aria-hidden="true" tabindex="-1"></a>hfhs.pca.sPLSDA_batch.plot <span class="ot">&lt;-</span> </span>
<span id="cb210-2"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb210-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">Scatter_Density</span>(<span class="at">object =</span> hfhs.pca.sPLSDA_batch, </span>
<span id="cb210-3"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb210-3" aria-hidden="true" tabindex="-1"></a>                  <span class="at">batch =</span> hfhs.day.less, </span>
<span id="cb210-4"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb210-4" aria-hidden="true" tabindex="-1"></a>                  <span class="at">trt =</span> hfhs.trt.less, </span>
<span id="cb210-5"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb210-5" aria-hidden="true" tabindex="-1"></a>                  <span class="at">title =</span> <span class="st">&#39;sPLSDA-batch&#39;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb211"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb211-1"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb211-1" aria-hidden="true" tabindex="-1"></a>hfhs.pca.PN.plot <span class="ot">&lt;-</span> </span>
<span id="cb211-2"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb211-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">Scatter_Density</span>(<span class="at">object =</span> hfhs.pca.PN, </span>
<span id="cb211-3"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb211-3" aria-hidden="true" tabindex="-1"></a>                  <span class="at">batch =</span> hfhs.day.less, </span>
<span id="cb211-4"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb211-4" aria-hidden="true" tabindex="-1"></a>                  <span class="at">trt =</span> hfhs.trt.less, </span>
<span id="cb211-5"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb211-5" aria-hidden="true" tabindex="-1"></a>                  <span class="at">title =</span> <span class="st">&#39;Percentile Normalisation&#39;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb212"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb212-1"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb212-1" aria-hidden="true" tabindex="-1"></a>hfhs.pca.RUVIII.plot <span class="ot">&lt;-</span> </span>
<span id="cb212-2"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb212-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">Scatter_Density</span>(<span class="at">object =</span> hfhs.pca.RUVIII, </span>
<span id="cb212-3"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb212-3" aria-hidden="true" tabindex="-1"></a>                  <span class="at">batch =</span> hfhs.day.less, </span>
<span id="cb212-4"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb212-4" aria-hidden="true" tabindex="-1"></a>                  <span class="at">trt =</span> hfhs.trt.less, </span>
<span id="cb212-5"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb212-5" aria-hidden="true" tabindex="-1"></a>                  <span class="at">title =</span> <span class="st">&#39;RUVIII&#39;</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:HFHSpca"></span>
<img src="PLSDAbatch_workflow_files/figure-html/HFHSpca-1.png" alt="The PCA sample plots with densities before and after batch effect correction in the HFHS data." width="100%" />
<p class="caption">
Figure 2.17: The PCA sample plots with densities before and after batch effect correction in the HFHS data.
</p>
</div>
<p>In the above figures, the differences between the samples from different days were well removed after batch effect correction with PLSDA-batch and RUVIII. We can also compare the boxplots and density plots for key variables identified in PCA driving the major variance or in heatmaps showing obvious patterns before and after batch effect correction (results not shown).</p>
<p><strong>pRDA</strong></p>
<p>We calculate the global explained variance across all microbial variables using pRDA. To achieve this, we create a loop for each variable from the original (uncorrected) and batch effect-corrected data. The final results are then displayed with <code>partVar_plot()</code> from <span style="color: #FF7F00;">PLSDAbatch</span> package.</p>
<div class="sourceCode" id="cb213"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb213-1"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb213-1" aria-hidden="true" tabindex="-1"></a><span class="co"># HFHS data</span></span>
<span id="cb213-2"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb213-2" aria-hidden="true" tabindex="-1"></a>hfhs.corrected.list <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="st">`</span><span class="at">Before correction</span><span class="st">`</span> <span class="ot">=</span> hfhs.clr.less, </span>
<span id="cb213-3"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb213-3" aria-hidden="true" tabindex="-1"></a>                            <span class="at">removeBatchEffect =</span> hfhs.rBE, </span>
<span id="cb213-4"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb213-4" aria-hidden="true" tabindex="-1"></a>                            <span class="at">ComBat =</span> hfhs.ComBat, </span>
<span id="cb213-5"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb213-5" aria-hidden="true" tabindex="-1"></a>                            <span class="st">`</span><span class="at">PLSDA-batch</span><span class="st">`</span> <span class="ot">=</span> hfhs.PLSDA_batch,</span>
<span id="cb213-6"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb213-6" aria-hidden="true" tabindex="-1"></a>                            <span class="st">`</span><span class="at">sPLSDA-batch</span><span class="st">`</span> <span class="ot">=</span> hfhs.sPLSDA_batch, </span>
<span id="cb213-7"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb213-7" aria-hidden="true" tabindex="-1"></a>                            <span class="st">`</span><span class="at">Percentile Normalisation</span><span class="st">`</span> <span class="ot">=</span> hfhs.PN,</span>
<span id="cb213-8"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb213-8" aria-hidden="true" tabindex="-1"></a>                            <span class="at">RUVIII =</span> hfhs.RUVIII)</span>
<span id="cb213-9"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb213-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb213-10"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb213-10" aria-hidden="true" tabindex="-1"></a>hfhs.prop.df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">Treatment =</span> <span class="cn">NA</span>, <span class="at">Batch =</span> <span class="cn">NA</span>, </span>
<span id="cb213-11"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb213-11" aria-hidden="true" tabindex="-1"></a>                           <span class="at">Intersection =</span> <span class="cn">NA</span>, </span>
<span id="cb213-12"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb213-12" aria-hidden="true" tabindex="-1"></a>                           <span class="at">Residuals =</span> <span class="cn">NA</span>) </span>
<span id="cb213-13"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb213-13" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(hfhs.corrected.list)){</span>
<span id="cb213-14"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb213-14" aria-hidden="true" tabindex="-1"></a>  rda.res <span class="ot">=</span> <span class="fu">varpart</span>(hfhs.corrected.list[[i]], <span class="sc">~</span> trt, <span class="sc">~</span> day,</span>
<span id="cb213-15"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb213-15" aria-hidden="true" tabindex="-1"></a>                    <span class="at">data =</span> hfhs.factors.df, <span class="at">scale =</span> T)</span>
<span id="cb213-16"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb213-16" aria-hidden="true" tabindex="-1"></a>  hfhs.prop.df[i, ] <span class="ot">&lt;-</span> rda.res<span class="sc">$</span>part<span class="sc">$</span>indfract<span class="sc">$</span>Adj.R.squared}</span>
<span id="cb213-17"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb213-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb213-18"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb213-18" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(hfhs.prop.df) <span class="ot">=</span> <span class="fu">names</span>(hfhs.corrected.list)</span>
<span id="cb213-19"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb213-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb213-20"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb213-20" aria-hidden="true" tabindex="-1"></a>hfhs.prop.df <span class="ot">&lt;-</span> hfhs.prop.df[, <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">3</span>,<span class="dv">2</span>,<span class="dv">4</span>)]</span>
<span id="cb213-21"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb213-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb213-22"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb213-22" aria-hidden="true" tabindex="-1"></a>hfhs.prop.df[hfhs.prop.df <span class="sc">&lt;</span> <span class="dv">0</span>] <span class="ot">=</span> <span class="dv">0</span></span>
<span id="cb213-23"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb213-23" aria-hidden="true" tabindex="-1"></a>hfhs.prop.df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">t</span>(<span class="fu">apply</span>(hfhs.prop.df, <span class="dv">1</span>, </span>
<span id="cb213-24"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb213-24" aria-hidden="true" tabindex="-1"></a>                                      <span class="cf">function</span>(x){x<span class="sc">/</span><span class="fu">sum</span>(x)})))</span>
<span id="cb213-25"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb213-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb213-26"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb213-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb213-27"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb213-27" aria-hidden="true" tabindex="-1"></a><span class="fu">partVar_plot</span>(<span class="at">prop.df =</span> hfhs.prop.df)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:HFHSprdaSup"></span>
<img src="PLSDAbatch_workflow_files/figure-html/HFHSprdaSup-1.png" alt="Global explained variance before and after batch effect correction for the HFHS data." width="60%" />
<p class="caption">
Figure 2.18: Global explained variance before and after batch effect correction for the HFHS data.
</p>
</div>
<p>In the <span style="color: #808000;">HFHS data</span>, a small amount of batch variance was observed (3.6%) as shown in the above figure. PLSDA-batch achieved the best performance for preserving the largest treatment variance and completely removing batch variance compared to the other methods. The results also indicate that PLSDA-batch is more appropriate for weak batch effects, while sPLSDA-batch is more appropriate for strong batch effects. The RUVIII performed better in the <span style="color: #808000;">HFHS data</span> than the <span style="color: #0000FF;">AD data</span> because the sample replicates may help capturing more batch variation than in the <span style="color: #0000FF;">AD data</span>. Indeed, the sample replicates in <span style="color: #808000;">HFHS data</span> are across different day batches, while the replicates in <span style="color: #0000FF;">AD data</span> do not exist in all batches. Therefore, sample replicates play a critical role in RUVIII.</p>
</div>
<div id="other-methods-2" class="section level4 hasAnchor" number="2.2.4.2">
<h4><span class="header-section-number">2.2.4.2</span> Other methods<a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#other-methods-2" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p><strong><span class="math inline">\(R^2\)</span></strong></p>
<p>The <span class="math inline">\(R^2\)</span> values for each variable are calculated with <code>lm()</code> from <span style="color: #FF7F00;">stats</span> package. To compare the <span class="math inline">\(R^2\)</span> values among variables, we scale the corrected data before <span class="math inline">\(R^2\)</span> calculation. The results are displayed with <code>ggplot()</code> from <span style="color: #FF7F00;">ggplot2</span> R package.</p>
<div class="sourceCode" id="cb214"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb214-1"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb214-1" aria-hidden="true" tabindex="-1"></a><span class="co"># scale</span></span>
<span id="cb214-2"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb214-2" aria-hidden="true" tabindex="-1"></a>hfhs.corr_scale.list <span class="ot">&lt;-</span> <span class="fu">lapply</span>(hfhs.corrected.list, </span>
<span id="cb214-3"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb214-3" aria-hidden="true" tabindex="-1"></a>                               <span class="cf">function</span>(x){<span class="fu">apply</span>(x, <span class="dv">2</span>, scale)})</span>
<span id="cb214-4"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb214-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb214-5"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb214-5" aria-hidden="true" tabindex="-1"></a><span class="co"># HFHS data</span></span>
<span id="cb214-6"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb214-6" aria-hidden="true" tabindex="-1"></a>hfhs.r_values.list <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb214-7"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb214-7" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(hfhs.corr_scale.list)){</span>
<span id="cb214-8"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb214-8" aria-hidden="true" tabindex="-1"></a>  hfhs.r_values <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">trt =</span> <span class="cn">NA</span>, <span class="at">batch =</span> <span class="cn">NA</span>)</span>
<span id="cb214-9"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb214-9" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span>(c <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">ncol</span>(hfhs.corr_scale.list[[i]])){</span>
<span id="cb214-10"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb214-10" aria-hidden="true" tabindex="-1"></a>    hfhs.fit.res.trt <span class="ot">&lt;-</span> <span class="fu">lm</span>(hfhs.corr_scale.list[[i]][,c] <span class="sc">~</span> hfhs.trt.less)</span>
<span id="cb214-11"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb214-11" aria-hidden="true" tabindex="-1"></a>    hfhs.r_values[c,<span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="fu">summary</span>(hfhs.fit.res.trt)<span class="sc">$</span>r.squared</span>
<span id="cb214-12"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb214-12" aria-hidden="true" tabindex="-1"></a>    hfhs.fit.res.batch <span class="ot">&lt;-</span> <span class="fu">lm</span>(hfhs.corr_scale.list[[i]][,c] <span class="sc">~</span> hfhs.day.less)</span>
<span id="cb214-13"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb214-13" aria-hidden="true" tabindex="-1"></a>    hfhs.r_values[c,<span class="dv">2</span>] <span class="ot">&lt;-</span> <span class="fu">summary</span>(hfhs.fit.res.batch)<span class="sc">$</span>r.squared</span>
<span id="cb214-14"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb214-14" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb214-15"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb214-15" aria-hidden="true" tabindex="-1"></a>  hfhs.r_values.list[[i]] <span class="ot">&lt;-</span> hfhs.r_values</span>
<span id="cb214-16"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb214-16" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb214-17"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb214-17" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(hfhs.r_values.list) <span class="ot">&lt;-</span> <span class="fu">names</span>(hfhs.corr_scale.list)</span>
<span id="cb214-18"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb214-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb214-19"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb214-19" aria-hidden="true" tabindex="-1"></a>hfhs.boxp.list <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb214-20"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb214-20" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="fu">seq_len</span>(<span class="fu">length</span>(hfhs.r_values.list))){</span>
<span id="cb214-21"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb214-21" aria-hidden="true" tabindex="-1"></a>  hfhs.boxp.list[[i]] <span class="ot">&lt;-</span> </span>
<span id="cb214-22"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb214-22" aria-hidden="true" tabindex="-1"></a>    <span class="fu">data.frame</span>(<span class="at">r2 =</span> <span class="fu">c</span>(hfhs.r_values.list[[i]][ ,<span class="st">&#39;trt&#39;</span>],</span>
<span id="cb214-23"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb214-23" aria-hidden="true" tabindex="-1"></a>                      hfhs.r_values.list[[i]][ ,<span class="st">&#39;batch&#39;</span>]), </span>
<span id="cb214-24"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb214-24" aria-hidden="true" tabindex="-1"></a>               <span class="at">Effects =</span> <span class="fu">as.factor</span>(<span class="fu">rep</span>(<span class="fu">c</span>(<span class="st">&#39;Treatment&#39;</span>,<span class="st">&#39;Batch&#39;</span>), </span>
<span id="cb214-25"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb214-25" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">each =</span> <span class="dv">515</span>)))</span>
<span id="cb214-26"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb214-26" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb214-27"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb214-27" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(hfhs.boxp.list) <span class="ot">&lt;-</span> <span class="fu">names</span>(hfhs.r_values.list)</span>
<span id="cb214-28"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb214-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb214-29"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb214-29" aria-hidden="true" tabindex="-1"></a>hfhs.r2.boxp <span class="ot">&lt;-</span> <span class="fu">rbind</span>(hfhs.boxp.list<span class="sc">$</span><span class="st">`</span><span class="at">Before correction</span><span class="st">`</span>,</span>
<span id="cb214-30"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb214-30" aria-hidden="true" tabindex="-1"></a>                      hfhs.boxp.list<span class="sc">$</span>removeBatchEffect,</span>
<span id="cb214-31"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb214-31" aria-hidden="true" tabindex="-1"></a>                      hfhs.boxp.list<span class="sc">$</span>ComBat,</span>
<span id="cb214-32"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb214-32" aria-hidden="true" tabindex="-1"></a>                      hfhs.boxp.list<span class="sc">$</span><span class="st">`</span><span class="at">PLSDA-batch</span><span class="st">`</span>,</span>
<span id="cb214-33"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb214-33" aria-hidden="true" tabindex="-1"></a>                      hfhs.boxp.list<span class="sc">$</span><span class="st">`</span><span class="at">sPLSDA-batch</span><span class="st">`</span>,</span>
<span id="cb214-34"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb214-34" aria-hidden="true" tabindex="-1"></a>                      hfhs.boxp.list<span class="sc">$</span><span class="st">`</span><span class="at">Percentile Normalisation</span><span class="st">`</span>,</span>
<span id="cb214-35"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb214-35" aria-hidden="true" tabindex="-1"></a>                      hfhs.boxp.list<span class="sc">$</span>RUVIII)</span>
<span id="cb214-36"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb214-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb214-37"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb214-37" aria-hidden="true" tabindex="-1"></a>hfhs.r2.boxp<span class="sc">$</span>methods <span class="ot">&lt;-</span> <span class="fu">rep</span>(<span class="fu">c</span>(<span class="st">&#39;Before correction&#39;</span>, <span class="st">&#39; removeBatchEffect&#39;</span>, </span>
<span id="cb214-38"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb214-38" aria-hidden="true" tabindex="-1"></a>                              <span class="st">&#39;ComBat&#39;</span>,<span class="st">&#39;PLSDA-batch&#39;</span>, <span class="st">&#39;sPLSDA-batch&#39;</span>,</span>
<span id="cb214-39"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb214-39" aria-hidden="true" tabindex="-1"></a>                              <span class="st">&#39;Percentile Normalisation&#39;</span>, <span class="st">&#39;RUVIII&#39;</span>), </span>
<span id="cb214-40"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb214-40" aria-hidden="true" tabindex="-1"></a>                            <span class="at">each =</span> <span class="dv">1030</span>)</span>
<span id="cb214-41"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb214-41" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb214-42"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb214-42" aria-hidden="true" tabindex="-1"></a>hfhs.r2.boxp<span class="sc">$</span>methods <span class="ot">&lt;-</span> <span class="fu">factor</span>(hfhs.r2.boxp<span class="sc">$</span>methods, </span>
<span id="cb214-43"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb214-43" aria-hidden="true" tabindex="-1"></a>                               <span class="at">levels =</span> <span class="fu">unique</span>(hfhs.r2.boxp<span class="sc">$</span>methods))</span>
<span id="cb214-44"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb214-44" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb214-45"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb214-45" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(hfhs.r2.boxp, <span class="fu">aes</span>(<span class="at">x =</span> Effects, <span class="at">y =</span> r2, <span class="at">fill =</span> Effects)) <span class="sc">+</span></span>
<span id="cb214-46"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb214-46" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>(<span class="at">alpha =</span> <span class="fl">0.80</span>) <span class="sc">+</span></span>
<span id="cb214-47"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb214-47" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span> </span>
<span id="cb214-48"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb214-48" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">18</span>),</span>
<span id="cb214-49"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb214-49" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.title.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb214-50"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb214-50" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.title.y =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb214-51"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb214-51" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">60</span>, <span class="at">hjust =</span> <span class="dv">1</span>, <span class="at">size =</span> <span class="dv">18</span>),</span>
<span id="cb214-52"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb214-52" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.text.y =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">18</span>),</span>
<span id="cb214-53"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb214-53" aria-hidden="true" tabindex="-1"></a>        <span class="at">panel.grid.minor.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb214-54"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb214-54" aria-hidden="true" tabindex="-1"></a>        <span class="at">panel.grid.major.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb214-55"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb214-55" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.position =</span> <span class="st">&quot;right&quot;</span>) <span class="sc">+</span> <span class="fu">facet_grid</span>( <span class="sc">~</span> methods) <span class="sc">+</span> </span>
<span id="cb214-56"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb214-56" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values=</span><span class="fu">pb_color</span>(<span class="fu">c</span>(<span class="dv">12</span>,<span class="dv">14</span>))) </span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:HFHSr21"></span>
<img src="PLSDAbatch_workflow_files/figure-html/HFHSr21-1.png" alt="HFHS study: $R^2$ values for each microbial variable before and after batch effect correction." width="100%" />
<p class="caption">
Figure 2.19: HFHS study: <span class="math inline">\(R^2\)</span> values for each microbial variable before and after batch effect correction.
</p>
</div>
<p>We observed from these plots that the data corrected by sPLSDA-batch, percentile normalisation still included many variables with a large proportion of batch variance.</p>
<div class="sourceCode" id="cb215"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb215-1"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb215-1" aria-hidden="true" tabindex="-1"></a><span class="do">##################################</span></span>
<span id="cb215-2"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb215-2" aria-hidden="true" tabindex="-1"></a>hfhs.barp.list <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb215-3"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb215-3" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="fu">seq_len</span>(<span class="fu">length</span>(hfhs.r_values.list))){</span>
<span id="cb215-4"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb215-4" aria-hidden="true" tabindex="-1"></a>  hfhs.barp.list[[i]] <span class="ot">&lt;-</span> </span>
<span id="cb215-5"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb215-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">data.frame</span>(<span class="at">r2 =</span> <span class="fu">c</span>(<span class="fu">sum</span>(hfhs.r_values.list[[i]][ ,<span class="st">&#39;trt&#39;</span>]),</span>
<span id="cb215-6"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb215-6" aria-hidden="true" tabindex="-1"></a>                      <span class="fu">sum</span>(hfhs.r_values.list[[i]][ ,<span class="st">&#39;batch&#39;</span>])), </span>
<span id="cb215-7"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb215-7" aria-hidden="true" tabindex="-1"></a>               <span class="at">Effects =</span> <span class="fu">c</span>(<span class="st">&#39;Treatment&#39;</span>,<span class="st">&#39;Batch&#39;</span>))</span>
<span id="cb215-8"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb215-8" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb215-9"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb215-9" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(hfhs.barp.list) <span class="ot">&lt;-</span> <span class="fu">names</span>(hfhs.r_values.list)</span>
<span id="cb215-10"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb215-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb215-11"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb215-11" aria-hidden="true" tabindex="-1"></a>hfhs.r2.barp <span class="ot">&lt;-</span> <span class="fu">rbind</span>(hfhs.barp.list<span class="sc">$</span><span class="st">`</span><span class="at">Before correction</span><span class="st">`</span>,</span>
<span id="cb215-12"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb215-12" aria-hidden="true" tabindex="-1"></a>                      hfhs.barp.list<span class="sc">$</span>removeBatchEffect,</span>
<span id="cb215-13"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb215-13" aria-hidden="true" tabindex="-1"></a>                      hfhs.barp.list<span class="sc">$</span>ComBat,</span>
<span id="cb215-14"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb215-14" aria-hidden="true" tabindex="-1"></a>                      hfhs.barp.list<span class="sc">$</span><span class="st">`</span><span class="at">PLSDA-batch</span><span class="st">`</span>,</span>
<span id="cb215-15"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb215-15" aria-hidden="true" tabindex="-1"></a>                      hfhs.barp.list<span class="sc">$</span><span class="st">`</span><span class="at">sPLSDA-batch</span><span class="st">`</span>,</span>
<span id="cb215-16"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb215-16" aria-hidden="true" tabindex="-1"></a>                      hfhs.barp.list<span class="sc">$</span><span class="st">`</span><span class="at">Percentile Normalisation</span><span class="st">`</span>,</span>
<span id="cb215-17"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb215-17" aria-hidden="true" tabindex="-1"></a>                      hfhs.barp.list<span class="sc">$</span>RUVIII)</span>
<span id="cb215-18"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb215-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb215-19"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb215-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb215-20"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb215-20" aria-hidden="true" tabindex="-1"></a>hfhs.r2.barp<span class="sc">$</span>methods <span class="ot">&lt;-</span> <span class="fu">rep</span>(<span class="fu">c</span>(<span class="st">&#39;Before correction&#39;</span>, <span class="st">&#39; removeBatchEffect&#39;</span>, </span>
<span id="cb215-21"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb215-21" aria-hidden="true" tabindex="-1"></a>                              <span class="st">&#39;ComBat&#39;</span>,<span class="st">&#39;PLSDA-batch&#39;</span>, <span class="st">&#39;sPLSDA-batch&#39;</span>,</span>
<span id="cb215-22"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb215-22" aria-hidden="true" tabindex="-1"></a>                              <span class="st">&#39;Percentile Normalisation&#39;</span>, <span class="st">&#39;RUVIII&#39;</span>), <span class="at">each =</span> <span class="dv">2</span>)</span>
<span id="cb215-23"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb215-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb215-24"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb215-24" aria-hidden="true" tabindex="-1"></a>hfhs.r2.barp<span class="sc">$</span>methods <span class="ot">&lt;-</span> <span class="fu">factor</span>(hfhs.r2.barp<span class="sc">$</span>methods, </span>
<span id="cb215-25"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb215-25" aria-hidden="true" tabindex="-1"></a>                               <span class="at">levels =</span> <span class="fu">unique</span>(hfhs.r2.barp<span class="sc">$</span>methods))</span>
<span id="cb215-26"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb215-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb215-27"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb215-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb215-28"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb215-28" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(hfhs.r2.barp, <span class="fu">aes</span>(<span class="at">x =</span> Effects, <span class="at">y =</span> r2, <span class="at">fill =</span> Effects)) <span class="sc">+</span></span>
<span id="cb215-29"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb215-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">stat=</span><span class="st">&quot;identity&quot;</span>) <span class="sc">+</span> </span>
<span id="cb215-30"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb215-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span> </span>
<span id="cb215-31"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb215-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">18</span>),</span>
<span id="cb215-32"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb215-32" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.title.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb215-33"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb215-33" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.title.y =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb215-34"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb215-34" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">60</span>, <span class="at">hjust =</span> <span class="dv">1</span>, <span class="at">size =</span> <span class="dv">18</span>),</span>
<span id="cb215-35"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb215-35" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.text.y =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">18</span>),</span>
<span id="cb215-36"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb215-36" aria-hidden="true" tabindex="-1"></a>        <span class="at">panel.grid.minor.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb215-37"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb215-37" aria-hidden="true" tabindex="-1"></a>        <span class="at">panel.grid.major.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb215-38"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb215-38" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.position =</span> <span class="st">&quot;right&quot;</span>) <span class="sc">+</span> <span class="fu">facet_grid</span>( <span class="sc">~</span> methods) <span class="sc">+</span> </span>
<span id="cb215-39"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb215-39" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values=</span><span class="fu">pb_color</span>(<span class="fu">c</span>(<span class="dv">12</span>,<span class="dv">14</span>)))</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:HFHSr22"></span>
<img src="PLSDAbatch_workflow_files/figure-html/HFHSr22-1.png" alt="HFHS study: the sum of $R^2$ values for each microbial variable before and after batch effect correction." width="100%" />
<p class="caption">
Figure 2.20: HFHS study: the sum of <span class="math inline">\(R^2\)</span> values for each microbial variable before and after batch effect correction.
</p>
</div>
<p>When considering the sum of all variables, the remaining batch variance of corrected data from PLSDA-batch is greater than removeBatchEffect, ComBat and RUVIII. The preserved treatment variance of corrected data from PLSDA-batch is the largest.</p>
<p><strong>Alignment scores</strong></p>
<p>We use the <code>alignment_score()</code> function from <span style="color: #FF7F00;">PLSDAbatch</span>. We need to specify the proportion of data variance to explain (<code>var</code>), the number of nearest neighbours (<code>k</code>) and the number of principal components to estimate (<code>ncomp</code>). We then use <code>ggplot()</code> function from <span style="color: #FF7F00;">ggplot2</span> to visualise the results.</p>
<div class="sourceCode" id="cb216"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb216-1"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb216-1" aria-hidden="true" tabindex="-1"></a><span class="co"># HFHS data</span></span>
<span id="cb216-2"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb216-2" aria-hidden="true" tabindex="-1"></a>hfhs.var <span class="ot">=</span> <span class="fl">0.95</span></span>
<span id="cb216-3"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb216-3" aria-hidden="true" tabindex="-1"></a>hfhs.k <span class="ot">=</span> <span class="dv">15</span></span>
<span id="cb216-4"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb216-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb216-5"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb216-5" aria-hidden="true" tabindex="-1"></a>hfhs.scores <span class="ot">&lt;-</span> <span class="fu">c</span>()</span>
<span id="cb216-6"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb216-6" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(hfhs.corrected.list)){</span>
<span id="cb216-7"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb216-7" aria-hidden="true" tabindex="-1"></a>  res <span class="ot">&lt;-</span> <span class="fu">alignment_score</span>(<span class="at">data =</span> hfhs.corrected.list[[i]], </span>
<span id="cb216-8"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb216-8" aria-hidden="true" tabindex="-1"></a>                         <span class="at">batch =</span> hfhs.day.less, </span>
<span id="cb216-9"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb216-9" aria-hidden="true" tabindex="-1"></a>                         <span class="at">var =</span> hfhs.var, </span>
<span id="cb216-10"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb216-10" aria-hidden="true" tabindex="-1"></a>                         <span class="at">k =</span> hfhs.k, </span>
<span id="cb216-11"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb216-11" aria-hidden="true" tabindex="-1"></a>                         <span class="at">ncomp =</span> <span class="dv">130</span>)</span>
<span id="cb216-12"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb216-12" aria-hidden="true" tabindex="-1"></a>  hfhs.scores <span class="ot">&lt;-</span> <span class="fu">c</span>(hfhs.scores, res)</span>
<span id="cb216-13"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb216-13" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb216-14"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb216-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb216-15"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb216-15" aria-hidden="true" tabindex="-1"></a>hfhs.scores.df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">scores =</span> hfhs.scores, </span>
<span id="cb216-16"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb216-16" aria-hidden="true" tabindex="-1"></a>                             <span class="at">methods =</span> <span class="fu">names</span>(hfhs.corrected.list))</span>
<span id="cb216-17"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb216-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb216-18"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb216-18" aria-hidden="true" tabindex="-1"></a>hfhs.scores.df<span class="sc">$</span>methods <span class="ot">&lt;-</span> <span class="fu">factor</span>(hfhs.scores.df<span class="sc">$</span>methods, </span>
<span id="cb216-19"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb216-19" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">levels =</span> <span class="fu">rev</span>(<span class="fu">names</span>(hfhs.corrected.list)))</span>
<span id="cb216-20"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb216-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb216-21"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb216-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb216-22"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb216-22" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>() <span class="sc">+</span> <span class="fu">geom_col</span>(<span class="fu">aes</span>(<span class="at">x =</span> hfhs.scores.df<span class="sc">$</span>methods, </span>
<span id="cb216-23"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb216-23" aria-hidden="true" tabindex="-1"></a>                        <span class="at">y =</span> hfhs.scores.df<span class="sc">$</span>scores)) <span class="sc">+</span> </span>
<span id="cb216-24"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb216-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_text</span>(<span class="fu">aes</span>(<span class="at">x =</span> hfhs.scores.df<span class="sc">$</span>methods, </span>
<span id="cb216-25"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb216-25" aria-hidden="true" tabindex="-1"></a>                <span class="at">y =</span> hfhs.scores.df<span class="sc">$</span>scores<span class="sc">/</span><span class="dv">2</span>, </span>
<span id="cb216-26"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb216-26" aria-hidden="true" tabindex="-1"></a>                <span class="at">label =</span> <span class="fu">round</span>(hfhs.scores.df<span class="sc">$</span>scores, <span class="dv">3</span>)), </span>
<span id="cb216-27"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb216-27" aria-hidden="true" tabindex="-1"></a>            <span class="at">size =</span> <span class="dv">3</span>, <span class="at">col =</span> <span class="st">&#39;white&#39;</span>) <span class="sc">+</span> </span>
<span id="cb216-28"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb216-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_flip</span>() <span class="sc">+</span> <span class="fu">theme_bw</span>() <span class="sc">+</span> <span class="fu">ylab</span>(<span class="st">&#39;Alignment Scores&#39;</span>) <span class="sc">+</span> </span>
<span id="cb216-29"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb216-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">&#39;&#39;</span>) <span class="sc">+</span> <span class="fu">ylim</span>(<span class="dv">0</span>,<span class="fl">0.85</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:HFHSalignment"></span>
<img src="PLSDAbatch_workflow_files/figure-html/HFHSalignment-1.png" alt="Comparison of alignment scores before and after batch effect correction using different methods for the HFHS data." width="60%" />
<p class="caption">
Figure 2.21: Comparison of alignment scores before and after batch effect correction using different methods for the HFHS data.
</p>
</div>
<p>The alignment scores complement the PCA results, especially when batch effect removal is difficult to assess on PCA sample plots. For example in the PCA plots for the <span style="color: #808000;">HFHS data</span>, we observed that the samples across different batches were better mixed after batch effect correction with PLSDA-batch and RUVIII than before, while the performance of these two methods was difficult to compare. Since a higher alignment score indicates that samples are better mixed, as shown in the above figure, RUVIII gave a superior performance compared to the other methods.</p>
<p>We recommend the sparse version of PLSDA-batch for a very strong batch effect, for example, in the <span style="color: #0000FF;">AD data</span>. However, for a weak batch effect, we do not recommend removing batch effects in the <span style="color: #808000;">HFHS data</span>, but if we have to, we recommend using PLSDA-batch. It is easier to lose treatment variation when the batch variation is very small. Therefore PLSDA-batch can better ensure the complete preservation of treatment variation than a sparse version.</p>
<p><strong>Variable selection</strong></p>
<p>We use <code>splsda()</code> from <span style="color: #FF7F00;">mixOmics</span> to select the top 50 microbial variables that, in combination, discriminate the different treatment groups in the <span style="color: #808000;">HFHS data</span>. We apply <code>splsda()</code> to the different batch effect corrected data from all methods. Then we use <code>upset()</code> from <span style="color: #FF7F00;">UpSetR</span> package <span class="citation">(<a href="#ref-lex2014upset" role="doc-biblioref">Lex et al. 2014</a>)</span> to visualise the concordance of variables selected.</p>
<p>In the code below, we first need to convert the list of variables selected from different method-corrected data into a data frame compatible with <code>upset()</code> using <code>fromList()</code>. We then assign different colour schemes for each variable selection.</p>
<div class="sourceCode" id="cb217"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb217-1"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb217-1" aria-hidden="true" tabindex="-1"></a>hfhs.splsda.select <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb217-2"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb217-2" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(hfhs.corrected.list)){</span>
<span id="cb217-3"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb217-3" aria-hidden="true" tabindex="-1"></a>  splsda.res <span class="ot">&lt;-</span> <span class="fu">splsda</span>(<span class="at">X =</span> hfhs.corrected.list[[i]],</span>
<span id="cb217-4"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb217-4" aria-hidden="true" tabindex="-1"></a>                       <span class="at">Y =</span> hfhs.trt.less, </span>
<span id="cb217-5"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb217-5" aria-hidden="true" tabindex="-1"></a>                       <span class="at">ncomp =</span> <span class="dv">3</span>, <span class="at">keepX =</span> <span class="fu">rep</span>(<span class="dv">50</span>,<span class="dv">3</span>))</span>
<span id="cb217-6"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb217-6" aria-hidden="true" tabindex="-1"></a>  select.res <span class="ot">&lt;-</span> <span class="fu">selectVar</span>(splsda.res, <span class="at">comp =</span> <span class="dv">1</span>)<span class="sc">$</span>name</span>
<span id="cb217-7"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb217-7" aria-hidden="true" tabindex="-1"></a>  hfhs.splsda.select[[i]] <span class="ot">&lt;-</span> select.res</span>
<span id="cb217-8"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb217-8" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb217-9"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb217-9" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(hfhs.splsda.select) <span class="ot">&lt;-</span> <span class="fu">names</span>(hfhs.corrected.list)</span>
<span id="cb217-10"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb217-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb217-11"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb217-11" aria-hidden="true" tabindex="-1"></a><span class="co"># can only visualise 5 methods</span></span>
<span id="cb217-12"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb217-12" aria-hidden="true" tabindex="-1"></a>hfhs.splsda.select <span class="ot">&lt;-</span> hfhs.splsda.select[<span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">4</span>,<span class="dv">7</span>)]</span>
<span id="cb217-13"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb217-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb217-14"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb217-14" aria-hidden="true" tabindex="-1"></a>hfhs.splsda.upsetR <span class="ot">&lt;-</span> <span class="fu">fromList</span>(hfhs.splsda.select)</span>
<span id="cb217-15"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb217-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb217-16"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb217-16" aria-hidden="true" tabindex="-1"></a><span class="fu">upset</span>(hfhs.splsda.upsetR, <span class="at">main.bar.color =</span> <span class="st">&#39;gray36&#39;</span>,</span>
<span id="cb217-17"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb217-17" aria-hidden="true" tabindex="-1"></a>      <span class="at">sets.bar.color =</span> <span class="fu">pb_color</span>(<span class="fu">c</span>(<span class="dv">25</span><span class="sc">:</span><span class="dv">22</span>,<span class="dv">20</span>)), <span class="at">matrix.color =</span> <span class="st">&#39;gray36&#39;</span>,</span>
<span id="cb217-18"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb217-18" aria-hidden="true" tabindex="-1"></a>      <span class="at">order.by =</span> <span class="st">&#39;freq&#39;</span>, <span class="at">empty.intersections =</span> <span class="st">&#39;on&#39;</span>,</span>
<span id="cb217-19"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb217-19" aria-hidden="true" tabindex="-1"></a>      <span class="at">queries =</span> </span>
<span id="cb217-20"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb217-20" aria-hidden="true" tabindex="-1"></a>        <span class="fu">list</span>(<span class="fu">list</span>(<span class="at">query =</span> intersects, <span class="at">params =</span> <span class="fu">list</span>(<span class="st">&#39;Before correction&#39;</span>), </span>
<span id="cb217-21"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb217-21" aria-hidden="true" tabindex="-1"></a>                  <span class="at">color =</span> <span class="fu">pb_color</span>(<span class="dv">20</span>), <span class="at">active =</span> T),</span>
<span id="cb217-22"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb217-22" aria-hidden="true" tabindex="-1"></a>             <span class="fu">list</span>(<span class="at">query =</span> intersects, <span class="at">params =</span> <span class="fu">list</span>(<span class="st">&#39;removeBatchEffect&#39;</span>), </span>
<span id="cb217-23"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb217-23" aria-hidden="true" tabindex="-1"></a>                  <span class="at">color =</span> <span class="fu">pb_color</span>(<span class="dv">22</span>), <span class="at">active =</span> T),</span>
<span id="cb217-24"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb217-24" aria-hidden="true" tabindex="-1"></a>             <span class="fu">list</span>(<span class="at">query =</span> intersects, <span class="at">params =</span> <span class="fu">list</span>(<span class="st">&#39;ComBat&#39;</span>), </span>
<span id="cb217-25"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb217-25" aria-hidden="true" tabindex="-1"></a>                  <span class="at">color =</span> <span class="fu">pb_color</span>(<span class="dv">23</span>), <span class="at">active =</span> T),</span>
<span id="cb217-26"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb217-26" aria-hidden="true" tabindex="-1"></a>             <span class="fu">list</span>(<span class="at">query =</span> intersects, <span class="at">params =</span> <span class="fu">list</span>(<span class="st">&#39;PLSDA-batch&#39;</span>), </span>
<span id="cb217-27"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb217-27" aria-hidden="true" tabindex="-1"></a>                  <span class="at">color =</span> <span class="fu">pb_color</span>(<span class="dv">24</span>), <span class="at">active =</span> T),</span>
<span id="cb217-28"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb217-28" aria-hidden="true" tabindex="-1"></a>             <span class="fu">list</span>(<span class="at">query =</span> intersects, <span class="at">params =</span> <span class="fu">list</span>(<span class="st">&#39;RUVIII&#39;</span>), </span>
<span id="cb217-29"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb217-29" aria-hidden="true" tabindex="-1"></a>                  <span class="at">color =</span> <span class="fu">pb_color</span>(<span class="dv">25</span>), <span class="at">active =</span> T)))</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:HFHSupsetR"></span>
<img src="PLSDAbatch_workflow_files/figure-html/HFHSupsetR-1.png" alt="UpSet plot showing overlap between variables selected from different corrected data for the HFHS study." width="100%" />
<p class="caption">
Figure 2.22: UpSet plot showing overlap between variables selected from different corrected data for the HFHS study.
</p>
</div>
<p>In the above UpSet plot, the left bars indicate the number of variables selected from each data corrected with different methods. The right bar plot combined with the scatterplot show different intersection and their aggregates. We obtained an overlap of 39 out of 50 selected variables between different corrected and uncorrected data. However, the data from each method still included unique variables that were not selected in the other corrected data. As <code>upset()</code> can only include five datasets at once, we only display the uncorrected data and four corrected data that have been efficiently corrected for batch effects from our previous assessments.</p>
<div class="sourceCode" id="cb218"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb218-1"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb218-1" aria-hidden="true" tabindex="-1"></a>hfhs.splsda.select.overlap <span class="ot">&lt;-</span> <span class="fu">venn</span>(hfhs.splsda.select, <span class="at">show.plot =</span> F)</span>
<span id="cb218-2"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb218-2" aria-hidden="true" tabindex="-1"></a>hfhs.inters.splsda <span class="ot">&lt;-</span> <span class="fu">attr</span>(hfhs.splsda.select.overlap, <span class="st">&#39;intersections&#39;</span>)</span>
<span id="cb218-3"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb218-3" aria-hidden="true" tabindex="-1"></a>hfhs.inters.splsda.taxa <span class="ot">&lt;-</span> </span>
<span id="cb218-4"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb218-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">lapply</span>(hfhs.inters.splsda, </span>
<span id="cb218-5"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb218-5" aria-hidden="true" tabindex="-1"></a>         <span class="at">FUN =</span> <span class="cf">function</span>(x){<span class="fu">as.data.frame</span>(HFHS_data<span class="sc">$</span>FullData<span class="sc">$</span>taxa[x, ])})</span>
<span id="cb218-6"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb218-6" aria-hidden="true" tabindex="-1"></a><span class="fu">capture.output</span>(hfhs.inters.splsda.taxa, </span>
<span id="cb218-7"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb218-7" aria-hidden="true" tabindex="-1"></a>               <span class="at">file =</span> <span class="st">&quot;GeneratedData/HFHSselected_50_splsda.txt&quot;</span>)</span></code></pre></div>
</div>
</div>
</div>
<div id="hd-data" class="section level2 hasAnchor" number="2.3">
<h2><span class="header-section-number">2.3</span> HD data<a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#hd-data" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="data-pre-processing-3" class="section level3 hasAnchor" number="2.3.1">
<h3><span class="header-section-number">2.3.1</span> Data pre-processing<a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#data-pre-processing-3" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="prefiltering-1" class="section level4 hasAnchor" number="2.3.1.1">
<h4><span class="header-section-number">2.3.1.1</span> Prefiltering<a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#prefiltering-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>We load the <span style="color: #8601AF;">HD data</span> stored internally with function <code>data()</code>.</p>
<div class="sourceCode" id="cb219"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb219-1"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb219-1" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="at">file =</span> <span class="st">&#39;./data/HD_data.rda&#39;</span>)</span>
<span id="cb219-2"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb219-2" aria-hidden="true" tabindex="-1"></a>hd.count <span class="ot">&lt;-</span> HD_data<span class="sc">$</span>FullData<span class="sc">$</span>X.count</span>
<span id="cb219-3"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb219-3" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(hd.count)</span></code></pre></div>
<pre><code>## [1]  30 368</code></pre>
<div class="sourceCode" id="cb221"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb221-1"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb221-1" aria-hidden="true" tabindex="-1"></a>hd.filter.res <span class="ot">&lt;-</span> <span class="fu">PreFL</span>(<span class="at">data =</span> hd.count)</span>
<span id="cb221-2"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb221-2" aria-hidden="true" tabindex="-1"></a>hd.filter <span class="ot">&lt;-</span> hd.filter.res<span class="sc">$</span>data.filter</span>
<span id="cb221-3"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb221-3" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(hd.filter)</span></code></pre></div>
<pre><code>## [1]  30 269</code></pre>
<div class="sourceCode" id="cb223"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb223-1"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb223-1" aria-hidden="true" tabindex="-1"></a><span class="co"># zero proportion before filtering</span></span>
<span id="cb223-2"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb223-2" aria-hidden="true" tabindex="-1"></a>hd.filter.res<span class="sc">$</span>zero.prob</span></code></pre></div>
<pre><code>## [1] 0.4509058</code></pre>
<div class="sourceCode" id="cb225"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb225-1"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb225-1" aria-hidden="true" tabindex="-1"></a><span class="co"># zero proportion after filtering</span></span>
<span id="cb225-2"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb225-2" aria-hidden="true" tabindex="-1"></a><span class="fu">sum</span>(hd.filter <span class="sc">==</span> <span class="dv">0</span>)<span class="sc">/</span>(<span class="fu">nrow</span>(hd.filter) <span class="sc">*</span> <span class="fu">ncol</span>(hd.filter))</span></code></pre></div>
<pre><code>## [1] 0.3343247</code></pre>
<p>The raw <span style="color: #8601AF;">HD data</span> include 368 OTUs and 30 samples. We use the function <code>PreFL()</code> from our <span style="color: #FF7F00;">PLSDAbatch</span> R package to filter the data. After filtering, the <span style="color: #8601AF;">HD data</span> were reduced to 269 OTUs and 30 samples. The proportion of zeroes was reduced from 45% to 33%.</p>
</div>
<div id="transformation-3" class="section level4 hasAnchor" number="2.3.1.2">
<h4><span class="header-section-number">2.3.1.2</span> Transformation<a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#transformation-3" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Prior to CLR transformation, we add 1 as the offset for the <span style="color: #8601AF;">HD data</span> - that are raw count data. We use <code>logratio.transfo()</code> function in <span style="color: #FF7F00;">mixOmics</span> package to CLR transform the data.</p>
<div class="sourceCode" id="cb227"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb227-1"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb227-1" aria-hidden="true" tabindex="-1"></a>hd.clr <span class="ot">&lt;-</span> <span class="fu">logratio.transfo</span>(<span class="at">X =</span> hd.filter, <span class="at">logratio =</span> <span class="st">&#39;CLR&#39;</span>, <span class="at">offset =</span> <span class="dv">1</span>)</span>
<span id="cb227-2"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb227-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb227-3"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb227-3" aria-hidden="true" tabindex="-1"></a><span class="fu">class</span>(hd.clr) <span class="ot">&lt;-</span> <span class="st">&#39;matrix&#39;</span></span></code></pre></div>
</div>
</div>
<div id="batch-effect-detection-3" class="section level3 hasAnchor" number="2.3.2">
<h3><span class="header-section-number">2.3.2</span> Batch effect detection<a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#batch-effect-detection-3" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="pca-3" class="section level4 hasAnchor" number="2.3.2.1">
<h4><span class="header-section-number">2.3.2.1</span> PCA<a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#pca-3" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>We apply <code>pca()</code> function from <span style="color: #FF7F00;">mixOmics</span> package to the <span style="color: #8601AF;">HD data</span> and use <code>Scatter_Density()</code> function from <span style="color: #FF7F00;">PLSDAbatch</span> to represent the PCA sample plot with densities.</p>
<div class="sourceCode" id="cb228"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb228-1"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb228-1" aria-hidden="true" tabindex="-1"></a><span class="co"># HD data</span></span>
<span id="cb228-2"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb228-2" aria-hidden="true" tabindex="-1"></a>hd.pca.before <span class="ot">&lt;-</span> <span class="fu">pca</span>(hd.clr, <span class="at">ncomp =</span> <span class="dv">3</span>, <span class="at">scale =</span> <span class="cn">TRUE</span>)</span>
<span id="cb228-3"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb228-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb228-4"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb228-4" aria-hidden="true" tabindex="-1"></a>hd.metadata <span class="ot">&lt;-</span> HD_data<span class="sc">$</span>FullData<span class="sc">$</span>metadata</span>
<span id="cb228-5"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb228-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb228-6"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb228-6" aria-hidden="true" tabindex="-1"></a>hd.batch <span class="ot">&lt;-</span> <span class="fu">as.factor</span>(hd.metadata<span class="sc">$</span>Cage)</span>
<span id="cb228-7"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb228-7" aria-hidden="true" tabindex="-1"></a>hd.trt <span class="ot">&lt;-</span> <span class="fu">as.factor</span>(hd.metadata<span class="sc">$</span>Genotype)</span>
<span id="cb228-8"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb228-8" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(hd.batch) <span class="ot">=</span> <span class="fu">names</span>(hd.trt) <span class="ot">=</span> <span class="fu">rownames</span>(hd.metadata) </span>
<span id="cb228-9"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb228-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb228-10"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb228-10" aria-hidden="true" tabindex="-1"></a><span class="fu">Scatter_Density</span>(<span class="at">object =</span> hd.pca.before, </span>
<span id="cb228-11"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb228-11" aria-hidden="true" tabindex="-1"></a>                <span class="at">batch =</span> hd.batch, </span>
<span id="cb228-12"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb228-12" aria-hidden="true" tabindex="-1"></a>                <span class="at">trt =</span> hd.trt, </span>
<span id="cb228-13"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb228-13" aria-hidden="true" tabindex="-1"></a>                <span class="at">title =</span> <span class="st">&#39;HD data&#39;</span>, </span>
<span id="cb228-14"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb228-14" aria-hidden="true" tabindex="-1"></a>                <span class="at">trt.legend.title =</span> <span class="st">&#39;Genotype&#39;</span>, </span>
<span id="cb228-15"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb228-15" aria-hidden="true" tabindex="-1"></a>                <span class="at">batch.legend.title =</span> <span class="st">&#39;Cage&#39;</span>, <span class="at">legend.cex =</span> <span class="fl">0.6</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:HDpcaBefore"></span>
<img src="PLSDAbatch_workflow_files/figure-html/HDpcaBefore-1.png" alt="The PCA sample plot with densities in the HD data." width="60%" />
<p class="caption">
Figure 2.23: The PCA sample plot with densities in the HD data.
</p>
</div>
<p>In the above figure, we observed a clear difference between genotypes but vague separation between cages on the PCA plot.</p>
</div>
<div id="heatmap-3" class="section level4 hasAnchor" number="2.3.2.2">
<h4><span class="header-section-number">2.3.2.2</span> Heatmap<a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#heatmap-3" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>We produce a heatmap using <span style="color: #FF7F00;">pheatmap</span> package. The data first need to be scaled on both OTUs and samples.</p>
<div class="sourceCode" id="cb229"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb229-1"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb229-1" aria-hidden="true" tabindex="-1"></a><span class="co"># scale the clr data on both OTUs and samples</span></span>
<span id="cb229-2"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb229-2" aria-hidden="true" tabindex="-1"></a>hd.clr.s <span class="ot">&lt;-</span> <span class="fu">scale</span>(hd.clr, <span class="at">center =</span> T, <span class="at">scale =</span> T)</span>
<span id="cb229-3"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb229-3" aria-hidden="true" tabindex="-1"></a>hd.clr.ss <span class="ot">&lt;-</span> <span class="fu">scale</span>(<span class="fu">t</span>(hd.clr.s), <span class="at">center =</span> T, <span class="at">scale =</span> T)</span>
<span id="cb229-4"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb229-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb229-5"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb229-5" aria-hidden="true" tabindex="-1"></a>hd.anno_col <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">Cage =</span> hd.batch, <span class="at">Genotype =</span> hd.trt)</span>
<span id="cb229-6"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb229-6" aria-hidden="true" tabindex="-1"></a>hd.anno_colors <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="at">Cage =</span> <span class="fu">color.mixo</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>), </span>
<span id="cb229-7"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb229-7" aria-hidden="true" tabindex="-1"></a>                       <span class="at">Genotype =</span> <span class="fu">pb_color</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>))</span>
<span id="cb229-8"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb229-8" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(hd.anno_colors<span class="sc">$</span>Cage) <span class="ot">=</span> <span class="fu">levels</span>(hd.batch)</span>
<span id="cb229-9"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb229-9" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(hd.anno_colors<span class="sc">$</span>Genotype) <span class="ot">=</span> <span class="fu">levels</span>(hd.trt)</span>
<span id="cb229-10"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb229-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb229-11"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb229-11" aria-hidden="true" tabindex="-1"></a><span class="fu">pheatmap</span>(hd.clr.ss, </span>
<span id="cb229-12"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb229-12" aria-hidden="true" tabindex="-1"></a>         <span class="at">cluster_rows =</span> F, </span>
<span id="cb229-13"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb229-13" aria-hidden="true" tabindex="-1"></a>         <span class="at">fontsize_row =</span> <span class="dv">4</span>, </span>
<span id="cb229-14"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb229-14" aria-hidden="true" tabindex="-1"></a>         <span class="at">fontsize_col =</span> <span class="dv">6</span>,</span>
<span id="cb229-15"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb229-15" aria-hidden="true" tabindex="-1"></a>         <span class="at">fontsize =</span> <span class="dv">8</span>,</span>
<span id="cb229-16"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb229-16" aria-hidden="true" tabindex="-1"></a>         <span class="at">clustering_distance_rows =</span> <span class="st">&#39;euclidean&#39;</span>,</span>
<span id="cb229-17"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb229-17" aria-hidden="true" tabindex="-1"></a>         <span class="at">clustering_method =</span> <span class="st">&#39;ward.D&#39;</span>,</span>
<span id="cb229-18"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb229-18" aria-hidden="true" tabindex="-1"></a>         <span class="at">treeheight_row =</span> <span class="dv">30</span>,</span>
<span id="cb229-19"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb229-19" aria-hidden="true" tabindex="-1"></a>         <span class="at">annotation_col =</span> hd.anno_col,</span>
<span id="cb229-20"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb229-20" aria-hidden="true" tabindex="-1"></a>         <span class="at">annotation_colors =</span> hd.anno_colors,</span>
<span id="cb229-21"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb229-21" aria-hidden="true" tabindex="-1"></a>         <span class="at">border_color =</span> <span class="st">&#39;NA&#39;</span>,</span>
<span id="cb229-22"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb229-22" aria-hidden="true" tabindex="-1"></a>         <span class="at">main =</span> <span class="st">&#39;HD data - Scaled&#39;</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:HDheatmap"></span>
<img src="PLSDAbatch_workflow_files/figure-html/HDheatmap-1.png" alt="Hierarchical clustering for samples in the HD data." width="90%" />
<p class="caption">
Figure 2.24: Hierarchical clustering for samples in the HD data.
</p>
</div>
<p>In the heatmap, samples in the <span style="color: #8601AF;">HD data</span> were mostly grouped by genotypes instead of cages, indicating a weak cage effect.</p>
</div>
<div id="prda-3" class="section level4 hasAnchor" number="2.3.2.3">
<h4><span class="header-section-number">2.3.2.3</span> pRDA<a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#prda-3" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>We apply pRDA with <code>varpart()</code> function from <span style="color: #FF7F00;">vegan</span> R package.</p>
<div class="sourceCode" id="cb230"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb230-1"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb230-1" aria-hidden="true" tabindex="-1"></a>hd.factors.df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">trt =</span> hd.trt, <span class="at">batch =</span> hd.batch)</span>
<span id="cb230-2"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb230-2" aria-hidden="true" tabindex="-1"></a>hd.rda.before <span class="ot">&lt;-</span> <span class="fu">varpart</span>(hd.clr, <span class="sc">~</span> trt, <span class="sc">~</span> batch, </span>
<span id="cb230-3"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb230-3" aria-hidden="true" tabindex="-1"></a>                         <span class="at">data =</span> hd.factors.df, <span class="at">scale =</span> T)</span>
<span id="cb230-4"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb230-4" aria-hidden="true" tabindex="-1"></a>hd.rda.before<span class="sc">$</span>part<span class="sc">$</span>indfract</span></code></pre></div>
<pre><code>##                 Df R.squared Adj.R.squared Testable
## [a] = X1|X2      0        NA  1.110223e-16    FALSE
## [b] = X2|X1      8        NA  1.608205e-01     TRUE
## [c]              0        NA  9.730583e-02    FALSE
## [d] = Residuals NA        NA  7.418737e-01    FALSE</code></pre>
<p>In the result, <code>X1</code> and <code>X2</code> represent the first and second covariates fitted in the model. <code>[a]</code>, <code>[b]</code> represent the independent proportion of variance explained by <code>X1</code> and <code>X2</code> respectively, and <code>[c]</code> represents the intersection variance shared between <code>X1</code> and <code>X2</code>. Collinearity was detected in the <span style="color: #8601AF;">HD data</span> between treatment and batch as indicated by lines <code>[a]</code> and <code>[c]</code> that all treatment variance (<code>[a]</code>: Adj.R.squared = 0, <code>[c]</code>: Adj.R.squared = 0.0973) was explained as intersection variance. The batch x treatment design in the <span style="color: #8601AF;">HD data</span> is nested, in such a design the intersection variance is usually considerable. As the intersection is shared between batch and treatment effects, the batch variance in the <span style="color: #8601AF;">HD data</span> is larger than the treatment variance.</p>
</div>
</div>
<div id="managing-batch-effects-3" class="section level3 hasAnchor" number="2.3.3">
<h3><span class="header-section-number">2.3.3</span> Managing batch effects<a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#managing-batch-effects-3" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Because of the nested batch x treatment design in the <span style="color: #8601AF;">HD data</span>, the only way to account for the batch effect (i.e., the cage effect) is to apply linear mixed model and include the batch effect as a random effect.</p>
<div id="accounting-for-batch-effects-2" class="section level4 hasAnchor" number="2.3.3.1">
<h4><span class="header-section-number">2.3.3.1</span> Accounting for batch effects<a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#accounting-for-batch-effects-2" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p><strong>Linear regression</strong></p>
<p>Linear regression is conducted with <code>linear_regres()</code> function in <span style="color: #FF7F00;">PLSDAbatch</span>. We integrated the <span style="color: #FF7F00;">performance</span> package that assesses performance of regression models into our function <code>linear_regres()</code>. Therefore, we can apply <code>check_model()</code> from <span style="color: #FF7F00;">performance</span> to the outputs from <code>linear_regres()</code> to diagnose the validity of the model fitted with treatment and batch effects for each variable <span class="citation">(<a href="#ref-daniel2020performance" role="doc-biblioref">LÃŒdecke et al. 2020</a>)</span>. We can extract performance measurements such as adjusted R2, RMSE, RSE, AIC and BIC for the models fitted with and without batch effects, which are also the outputs of <code>linear_regres()</code>.</p>
<p>We apply a <code>linear mixed model</code> and fit the batch effect as a random effect because the design of the <span style="color: #8601AF;">HD data</span> is nested. In such a design, the number of batches should be larger than treatment groups; otherwise, the linear mixed model cannot address the batch effect. Note: we assume that there is no interaction between batch and treatment effects on any microbial variable.</p>
<div class="sourceCode" id="cb232"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb232-1"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb232-1" aria-hidden="true" tabindex="-1"></a><span class="co"># HD data</span></span>
<span id="cb232-2"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb232-2" aria-hidden="true" tabindex="-1"></a>hd.clr <span class="ot">&lt;-</span> hd.clr[<span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(hd.clr), <span class="dv">1</span><span class="sc">:</span><span class="fu">ncol</span>(hd.clr)]</span>
<span id="cb232-3"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb232-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb232-4"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb232-4" aria-hidden="true" tabindex="-1"></a>hd.lmm <span class="ot">&lt;-</span> <span class="fu">linear_regres</span>(<span class="at">data =</span> hd.clr, <span class="at">trt =</span> hd.trt, </span>
<span id="cb232-5"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb232-5" aria-hidden="true" tabindex="-1"></a>                        <span class="at">batch.random =</span> hd.batch, </span>
<span id="cb232-6"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb232-6" aria-hidden="true" tabindex="-1"></a>                        <span class="at">type =</span> <span class="st">&#39;linear mixed model&#39;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb233"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb233-1"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb233-1" aria-hidden="true" tabindex="-1"></a>hd.p <span class="ot">&lt;-</span> <span class="fu">sapply</span>(hd.lmm<span class="sc">$</span>lmm.table, <span class="cf">function</span>(x){x<span class="sc">$</span>coefficients[<span class="dv">2</span>,<span class="dv">5</span>]})</span>
<span id="cb233-2"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb233-2" aria-hidden="true" tabindex="-1"></a>hd.p.adj <span class="ot">&lt;-</span> <span class="fu">p.adjust</span>(<span class="at">p =</span> hd.p, <span class="at">method =</span> <span class="st">&#39;fdr&#39;</span>)</span>
<span id="cb233-3"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb233-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb233-4"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb233-4" aria-hidden="true" tabindex="-1"></a><span class="fu">check_model</span>(hd.lmm<span class="sc">$</span>model<span class="sc">$</span>OTU1)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:HDlmmS"></span>
<img src="PLSDAbatch_workflow_files/figure-html/HDlmmS-1.png" alt="Diagnostic plots for the model fitted with the cage effect of &quot;OTU1&quot; in the HD data." width="100%" />
<p class="caption">
Figure 2.25: Diagnostic plots for the model fitted with the cage effect of “OTU1” in the HD data.
</p>
</div>
<p>According to the diagnostic plots of “OTU1” as shown in the above figure panel, the simulated data under the fitted model were not very close to the real data (shown as green line), indicating an imperfect fitness (top panel). The linearity (or homoscedasticity) and homogeneity of variance were not satisfied. Some samples could be classified as outliers with a Cook’s distance larger than or equal to 0.5, for example, “12”, “21”, “20”, “11” and “6” (middle panel). The distribution of residuals was not normal, while the one of random effects was very close to normal (bottom panel).</p>
<p>We then look at the AIC of models fitted with or without batch effects.</p>
<div class="sourceCode" id="cb234"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb234-1"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb234-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(hd.lmm<span class="sc">$</span>AIC)</span></code></pre></div>
<pre><code>##       trt.only trt.batch
## OTU1  60.00637  65.46410
## OTU2 130.14167 130.96344
## OTU3  71.76985  75.58915
## OTU4 134.59992 134.04812
## OTU5 150.25059 148.30644
## OTU6  99.92864  99.05391</code></pre>
<p>For some OTUs, the model fitted without batch effects is better with a lower AIC. For example, “OTU1” and “OTU3”. “OTU5” is more appropriate within a model with batch effects. The others are similar within either model.</p>
<p>Since we apply a <code>"linear mixed model"</code> to the <span style="color: #8601AF;">HD data</span>, this type of model can only output conditional <span class="math inline">\(R^2\)</span> that includes the variance of both fixed and random effects (treatment, fixed and random batch effects) and marginal <span class="math inline">\(R^2\)</span> that includes only the variance of fixed effects (treatment and fixed batch effects) <span class="citation">(<a href="#ref-nakagawa2013general" role="doc-biblioref">Nakagawa and Schielzeth 2013</a>)</span>. The marginal <span class="math inline">\(R^2\)</span>s of the models with and without the batch effect should be the same theoretically, as marginal <span class="math inline">\(R^2\)</span> only includes fixed effects and the treatment effect is the only fixed effect here. In the actual calculation, the results of these two models were not the same due to different ways of model fitting.</p>
<div class="sourceCode" id="cb236"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb236-1"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb236-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(hd.lmm<span class="sc">$</span>cond.R2)</span></code></pre></div>
<pre><code>##        trt.only trt.batch
## OTU1 0.47917775 0.5161317
## OTU2 0.39223824 0.4729848
## OTU3 0.02091376 0.2636432
## OTU4 0.37200835 0.5672392
## OTU5 0.07858373 0.2938212
## OTU6 0.41388250 0.6143134</code></pre>
<div class="sourceCode" id="cb238"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb238-1"><a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#cb238-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(hd.lmm<span class="sc">$</span>marg.R2)</span></code></pre></div>
<pre><code>##        trt.only  trt.batch
## OTU1 0.47917775 0.46735273
## OTU2 0.39223824 0.38459672
## OTU3 0.02091376 0.01489608
## OTU4 0.37200835 0.36122754
## OTU5 0.07858373 0.06052938
## OTU6 0.41388250 0.40749149</code></pre>
<p>We observed that some variables resulted in singular fits (<code>error message: Can't compute random effect variances. Some variance components equal zero. Your model may suffer from singulariy.</code>), which are expected to happen in linear mixed models when covariates are nested. We recommend noting the variables for which this error occurs, as it may lead to unreliable results.</p>
</div>
</div>
</div>
<div id="references-1" class="section level2 hasAnchor" number="2.4">
<h2><span class="header-section-number">2.4</span> References<a href="supplementary-materials-for-batch-effects-management-in-case-studies.html#references-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>

</div>
</div>
<h3>References<a href="references-4.html#references-4" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-daniel2020performance" class="csl-entry">
LÃŒdecke, Daniel, Dominique Makowski, Philip Waggoner, and Indrajeet Patil. 2020. <span>“Performance: Assessment of Regression Models Performance.”</span> <em>CRAN</em>. <a href="https://doi.org/10.5281/zenodo.3952174">https://doi.org/10.5281/zenodo.3952174</a>.
</div>
<div id="ref-leek2012sva" class="csl-entry">
Leek, Jeffrey T, W Evan Johnson, Hilary S Parker, Andrew E Jaffe, and John D Storey. 2012. <span>“The Sva Package for Removing Batch Effects and Other Unwanted Variation in High-Throughput Experiments.”</span> <em>Bioinformatics</em> 28 (6): 882–83.
</div>
<div id="ref-lex2014upset" class="csl-entry">
Lex, Alexander, Nils Gehlenborg, Hendrik Strobelt, Romain Vuillemot, and Hanspeter Pfister. 2014. <span>“UpSet: Visualization of Intersecting Sets.”</span> <em>IEEE Transactions on Visualization and Computer Graphics</em> 20 (12): 1983–92.
</div>
<div id="ref-moskovicz2020skin" class="csl-entry">
Moskovicz, Veronica, Rina Ben-El, Guy Horev, and Boaz Mizrahi. 2020. <span>“Skin Microbiota Dynamics Following b. Subtilis Formulation Challenge.”</span>
</div>
<div id="ref-nakagawa2013general" class="csl-entry">
Nakagawa, Shinichi, and Holger Schielzeth. 2013. <span>“A General and Simple Method for Obtaining R2 from Generalized Linear Mixed-Effects Models.”</span> <em>Methods in Ecology and Evolution</em> 4 (2): 133–42.
</div>
<div id="ref-rohart2017mixomics" class="csl-entry">
Rohart, Florian, Benoı̂t Gautier, Amrit Singh, and Kim-Anh Lê Cao. 2017. <span>“mixOmics: An r Package for ‘Omics Feature Selection and Multiple Data Integration.”</span> <em>PLoS Computational Biology</em> 13 (11): e1005752.
</div>
<div id="ref-wang2020managing" class="csl-entry">
Wang, Yiwen, and Kim-Anh LêCao. 2020. <span>“Managing Batch Effects in Microbiome Data.”</span> <em>Briefings in Bioinformatics</em> 21 (6): 1954–70.
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="index.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="batch-effects-management-in-simulation-studies-gaussian-distribution.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["PLSDAbatch_workflow.pdf", "PLSDAbatch_workflow.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
