<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>PLSDA-batch Vignette • PLSDAbatch</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="PLSDA-batch Vignette">
<meta property="og:description" content="PLSDAbatch">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">PLSDAbatch</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.2.3</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/brief_vignette.html">PLSDA-batch Vignette</a>
    </li>
    <li>
      <a href="../articles/case_studies.html">Batch Effects Management in Case Studies</a>
    </li>
    <li>
      <a href="../articles/simulation_study_Gaussian.html">Batch Effects Management in Simulation Studies (Gaussian Distribution)</a>
    </li>
    <li>
      <a href="../articles/simulation_study_negative_binomial.html">Batch Effects Management in Simulation Studies (Negative Binomial Distribution)</a>
    </li>
    <li>
      <a href="../articles/supp_case_studies.html">Supplementary Materials for Batch Effects Management in Case Studies</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>PLSDA-batch Vignette</h1>
                        <h4 data-toc-skip class="author">Yiwen (Eva)
Wang</h4>
                        <h4 data-toc-skip class="author">Agricultural
Genomics Institute at Shenzhen, Chinese Academy of Agricultural
Sciences, Shenzhen, China</h4>
                        <h4 data-toc-skip class="author"><a href="mailto:wangyiwen@caas.cn" class="email">wangyiwen@caas.cn</a></h4>
            
            <h4 data-toc-skip class="date">09 November, 2022</h4>
      
      
      <div class="hidden name"><code>brief_vignette.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="brief-introduction">Brief Introduction<a class="anchor" aria-label="anchor" href="#brief-introduction"></a>
</h2>
<p><strong>PLSDA-batch</strong> is a new batch effect correction method
based on Projection to Latent Structures Discriminant Analysis to
correct data prior to any downstream analysis. It estimates latent
components related to treatment and batch effects to remove batch
variation. The method is multivariate, non-parametric and performs
dimension reduction. Combined with centered log ratio transformation for
addressing uneven library sizes and compositional structure, PLSDA-batch
addresses all characteristics of microbiome data that existing
correction methods have ignored so far.</p>
<p>Apart from the main method, the R package also includes two variants
called 1/ <em>weighted PLSDA-batch</em> for unbalanced batch x treatment
designs that are commonly encountered in studies with small sample size,
and 2/ <em>sparse PLSDA-batch</em> for selection of discriminative
variables to avoid overfitting in classification problems. These two
variants have widened the scope of applicability of PLSDA-batch to
different data settings <span class="citation">(<a href="#ref-wang2020multivariate" role="doc-biblioref">Wang and Lê Cao
2020</a>)</span>.</p>
<p>This vignette only includes the necessary pre-processing steps
(pre-filtering and transformation) for microbiome data and the usage of
PLSDA-batch series methods. See “<em>Batch Effects Management in Case
Studies</em>” for the complete process including data pre-processing,
batch effect detection and visualisation, choices of methods to account
for or correct for batch effects, assessment of batch effect removal and
variable selection after batch effect correction.</p>
</div>
<div class="section level2">
<h2 id="case-study-description">Case study description<a class="anchor" aria-label="anchor" href="#case-study-description"></a>
</h2>
<p>We considered a case study to illustrate the application of
PLSDA-batch. This study is described as follows:</p>
<p><strong><span style="color: #0000FF;">Anaerobic
digestion</span>.</strong> This study explored the microbial indicators
that could improve the efficacy of anaerobic digestion (AD) bioprocess
and prevent its failure <span class="citation">(<a href="#ref-chapleur2016increasing" role="doc-biblioref">Chapleur et al.
2016</a>)</span>. This data include 75 samples and 567 microbial
variables. The samples were treated with two different ranges of phenol
concentration (effect of interest) and processed at five different dates
(batch effect). This study includes a clear and strong batch effect with
an approx. balanced batch x treatment design.</p>
</div>
<div class="section level2">
<h2 id="packages-installation-and-loading">Packages installation and loading<a class="anchor" aria-label="anchor" href="#packages-installation-and-loading"></a>
</h2>
<p>First, we load the packages necessary for analysis, and check the
version of each package.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Bioconductor</span></span>
<span><span class="va">bioc.pkgs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'mixOmics'</span>, <span class="st">'Biobase'</span><span class="op">)</span></span>
<span><span class="co"># GitHub</span></span>
<span><span class="va">github.pkg</span> <span class="op">&lt;-</span> <span class="st">'PLSDAbatch'</span> </span>
<span><span class="co"># devtools::install_github("https://github.com/EvaYiwenWang/PLSDAbatch")</span></span>
<span></span>
<span><span class="co"># load packages </span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">bioc.pkgs</span>, <span class="va">github.pkg</span><span class="op">)</span>, <span class="va">require</span>, character.only <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##   mixOmics    Biobase PLSDAbatch </span></span>
<span><span class="co">##       TRUE       TRUE       TRUE</span></span></code></pre>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># print package versions</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">bioc.pkgs</span>, <span class="va">github.pkg</span><span class="op">)</span>, <span class="va">package.version</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##   mixOmics    Biobase PLSDAbatch </span></span>
<span><span class="co">##   "6.20.0"   "2.56.0"    "0.2.3"</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="data-pre-processing">Data pre-processing<a class="anchor" aria-label="anchor" href="#data-pre-processing"></a>
</h2>
<div class="section level3">
<h3 id="pre-filtering">Pre-filtering<a class="anchor" aria-label="anchor" href="#pre-filtering"></a>
</h3>
<p>We load the <span style="color: #0000FF;">AD data</span> stored
internally with function <code><a href="https://rdrr.io/r/utils/data.html" class="external-link">data()</a></code>, and then extract the batch
and treatment information out.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># AD data</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">'AD_data'</span><span class="op">)</span> </span>
<span><span class="va">ad.count</span> <span class="op">&lt;-</span> <span class="va">AD_data</span><span class="op">$</span><span class="va">FullData</span><span class="op">$</span><span class="va">X.count</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">ad.count</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1]  75 567</span></span></code></pre>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ad.metadata</span> <span class="op">&lt;-</span> <span class="va">AD_data</span><span class="op">$</span><span class="va">FullData</span><span class="op">$</span><span class="va">metadata</span></span>
<span><span class="va">ad.batch</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="va">ad.metadata</span><span class="op">$</span><span class="va">sequencing_run_date</span>, </span>
<span>                  levels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">ad.metadata</span><span class="op">$</span><span class="va">sequencing_run_date</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">ad.trt</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">as.factor</a></span><span class="op">(</span><span class="va">ad.metadata</span><span class="op">$</span><span class="va">initial_phenol_concentration.regroup</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">ad.batch</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">ad.trt</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">ad.metadata</span><span class="op">)</span></span></code></pre></div>
<p>The raw <span style="color: #0000FF;">AD data</span> include 567 OTUs
and 75 samples. We then use the function <code><a href="../reference/PreFL.html">PreFL()</a></code> from our
<span style="color: #FF7F00;">PLSDAbatch</span> R package to filter the
data.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ad.filter.res</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/PreFL.html">PreFL</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">ad.count</span><span class="op">)</span></span>
<span><span class="va">ad.filter</span> <span class="op">&lt;-</span> <span class="va">ad.filter.res</span><span class="op">$</span><span class="va">data.filter</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">ad.filter</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1]  75 231</span></span></code></pre>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># zero proportion before filtering</span></span>
<span><span class="va">ad.filter.res</span><span class="op">$</span><span class="va">zero.prob</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 0.6328042</span></span></code></pre>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># zero proportion after filtering</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="va">ad.filter</span> <span class="op">==</span> <span class="fl">0</span><span class="op">)</span><span class="op">/</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">ad.filter</span><span class="op">)</span> <span class="op">*</span> <span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">ncol</a></span><span class="op">(</span><span class="va">ad.filter</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 0.3806638</span></span></code></pre>
<p>After filtering, 231 OTUs remained, and the proportion of zeroes
decreased from 63% to 38%.</p>
<p>Note: The <code><a href="../reference/PreFL.html">PreFL()</a></code> function is only dedicated to raw
counts, rather than relative abundance data. We also recommend to start
the pre-filtering on raw counts, rather than relative abundance data to
mitigate the compositionality issue.</p>
</div>
<div class="section level3">
<h3 id="transformation">Transformation<a class="anchor" aria-label="anchor" href="#transformation"></a>
</h3>
<p>Prior to CLR transformation, we recommend adding 1 as the offset for
the data (e.g., <span style="color: #0000FF;">AD data</span>) that are
raw count data, and 0.01 as the offset for the data that are relative
abundance data. We use <code><a href="https://rdrr.io/pkg/mixOmics/man/logratio-transformations.html" class="external-link">logratio.transfo()</a></code> function in <span style="color: #FF7F00;">mixOmics</span> package to CLR transform the
data.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ad.clr</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/mixOmics/man/logratio-transformations.html" class="external-link">logratio.transfo</a></span><span class="op">(</span>X <span class="op">=</span> <span class="va">ad.filter</span>, logratio <span class="op">=</span> <span class="st">'CLR'</span>, offset <span class="op">=</span> <span class="fl">1</span><span class="op">)</span> </span>
<span><span class="fu"><a href="https://rdrr.io/r/base/class.html" class="external-link">class</a></span><span class="op">(</span><span class="va">ad.clr</span><span class="op">)</span> <span class="op">=</span> <span class="st">'matrix'</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="batch-effect-correction">Batch effect correction<a class="anchor" aria-label="anchor" href="#batch-effect-correction"></a>
</h2>
<div class="section level3">
<h3 id="plsda-batch">PLSDA-batch<a class="anchor" aria-label="anchor" href="#plsda-batch"></a>
</h3>
<p>The <code><a href="../reference/PLSDA_batch.html">PLSDA_batch()</a></code> function is implemented in <span style="color: #FF7F00;">PLSDAbatch</span> package. To use this function,
we need to specify the optimal number of components related to treatment
(<code>ncomp.trt</code>) or batch effects (<code>ncomp.bat</code>).</p>
<p>Here in the <span style="color: #0000FF;">AD data</span>, we use
<code><a href="https://rdrr.io/pkg/mixOmics/man/plsda.html" class="external-link">plsda()</a></code> from <span style="color: #FF7F00;">mixOmics</span>
with only treatment grouping information to estimate the optimal number
of treatment components to preserve.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># estimate the number of treatment components</span></span>
<span><span class="va">ad.trt.tune</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/mixOmics/man/plsda.html" class="external-link">plsda</a></span><span class="op">(</span>X <span class="op">=</span> <span class="va">ad.clr</span>, Y <span class="op">=</span> <span class="va">ad.trt</span>, ncomp <span class="op">=</span> <span class="fl">5</span><span class="op">)</span></span>
<span><span class="va">ad.trt.tune</span><span class="op">$</span><span class="va">prop_expl_var</span> <span class="co">#1</span></span></code></pre></div>
<pre><code><span><span class="co">## $X</span></span>
<span><span class="co">##      comp1      comp2      comp3      comp4      comp5 </span></span>
<span><span class="co">## 0.18619506 0.07873817 0.08257396 0.09263497 0.06594757 </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $Y</span></span>
<span><span class="co">##      comp1      comp2      comp3      comp4      comp5 </span></span>
<span><span class="co">## 1.00000000 0.33857374 0.17315267 0.10551296 0.08185822</span></span></code></pre>
<p>We choose the number that explains 100% variance in the outcome
matrix <code>Y</code>, thus from the result, 1 component was enough to
preserve the treatment information.</p>
<p>We then use <code><a href="../reference/PLSDA_batch.html">PLSDA_batch()</a></code> function with both treatment
and batch grouping information to estimate the optimal number of batch
components to remove.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># estimate the number of batch components</span></span>
<span><span class="va">ad.batch.tune</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/PLSDA_batch.html">PLSDA_batch</a></span><span class="op">(</span>X <span class="op">=</span> <span class="va">ad.clr</span>, </span>
<span>                             Y.trt <span class="op">=</span> <span class="va">ad.trt</span>, Y.bat <span class="op">=</span> <span class="va">ad.batch</span>,</span>
<span>                             ncomp.trt <span class="op">=</span> <span class="fl">1</span>, ncomp.bat <span class="op">=</span> <span class="fl">10</span><span class="op">)</span></span>
<span><span class="va">ad.batch.tune</span><span class="op">$</span><span class="va">explained_variance.bat</span> <span class="co">#4</span></span></code></pre></div>
<pre><code><span><span class="co">## $X</span></span>
<span><span class="co">##      comp1      comp2      comp3      comp4      comp5      comp6      comp7 </span></span>
<span><span class="co">## 0.17470922 0.11481264 0.10122717 0.07507395 0.04019490 0.03361579 0.03520149 </span></span>
<span><span class="co">##      comp8      comp9     comp10 </span></span>
<span><span class="co">## 0.02804330 0.02085228 0.01126863 </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $Y</span></span>
<span><span class="co">##     comp1     comp2     comp3     comp4     comp5     comp6     comp7     comp8 </span></span>
<span><span class="co">## 0.2474654 0.2615741 0.2301382 0.2608223 0.2301996 0.2461276 0.2521642 0.1216486 </span></span>
<span><span class="co">##     comp9    comp10 </span></span>
<span><span class="co">## 0.1498601 0.2298161</span></span></code></pre>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="va">ad.batch.tune</span><span class="op">$</span><span class="va">explained_variance.bat</span><span class="op">$</span><span class="va">Y</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_len</a></span><span class="op">(</span><span class="fl">4</span><span class="op">)</span><span class="op">]</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 1</span></span></code></pre>
<p>Using the same criterion as choosing treatment components, we choose
the number of batch components that explains 100% variance in the
outcome matrix of batch. According to the result, 4 components were
required to remove batch effects.</p>
<p>We then can correct for batch effects applying
<code><a href="../reference/PLSDA_batch.html">PLSDA_batch()</a></code> with treatment, batch grouping information
and corresponding optimal number of related components.</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ad.PLSDA_batch.res</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/PLSDA_batch.html">PLSDA_batch</a></span><span class="op">(</span>X <span class="op">=</span> <span class="va">ad.clr</span>, </span>
<span>                                  Y.trt <span class="op">=</span> <span class="va">ad.trt</span>, Y.bat <span class="op">=</span> <span class="va">ad.batch</span>,</span>
<span>                                  ncomp.trt <span class="op">=</span> <span class="fl">1</span>, ncomp.bat <span class="op">=</span> <span class="fl">4</span><span class="op">)</span></span>
<span><span class="va">ad.PLSDA_batch</span> <span class="op">&lt;-</span> <span class="va">ad.PLSDA_batch.res</span><span class="op">$</span><span class="va">X.nobatch</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="splsda-batch">sPLSDA-batch<a class="anchor" aria-label="anchor" href="#splsda-batch"></a>
</h3>
<p>We apply sPLSDA-batch using the same function
<code><a href="../reference/PLSDA_batch.html">PLSDA_batch()</a></code>, but we specify the number of variables to
select on each component (usually only treatment-related components
<code>keepX.trt</code>). To determine the optimal number of variables to
select, we use <code><a href="https://rdrr.io/pkg/mixOmics/man/tune.splsda.html" class="external-link">tune.splsda()</a></code> function from <span style="color: #FF7F00;">mixOmics</span> package <span class="citation">(<a href="#ref-rohart2017mixomics" role="doc-biblioref">Rohart et al. 2017</a>)</span> with all possible
numbers of variables to select for each component
(<code>test.keepX</code>).</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># estimate the number of variables to select per treatment component</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">777</span><span class="op">)</span></span>
<span><span class="va">ad.test.keepX</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">10</span>, <span class="fl">1</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">20</span>, <span class="fl">100</span>, <span class="fl">10</span><span class="op">)</span>, </span>
<span>                  <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">150</span>, <span class="fl">231</span>, <span class="fl">50</span><span class="op">)</span>, <span class="fl">231</span><span class="op">)</span></span>
<span><span class="va">ad.trt.tune.v</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/mixOmics/man/tune.splsda.html" class="external-link">tune.splsda</a></span><span class="op">(</span>X <span class="op">=</span> <span class="va">ad.clr</span>, Y <span class="op">=</span> <span class="va">ad.trt</span>, </span>
<span>                             ncomp <span class="op">=</span> <span class="fl">1</span>, test.keepX <span class="op">=</span> <span class="va">ad.test.keepX</span>, </span>
<span>                             validation <span class="op">=</span> <span class="st">'Mfold'</span>, folds <span class="op">=</span> <span class="fl">4</span>, </span>
<span>                             nrepeat <span class="op">=</span> <span class="fl">50</span><span class="op">)</span></span>
<span><span class="va">ad.trt.tune.v</span><span class="op">$</span><span class="va">choice.keepX</span> <span class="co">#100</span></span></code></pre></div>
<p>Here the optimal number of variables to select for the treatment
component was 100. Since we have adjusted the amount of treatment
variation to preserve, we need to re-choose the optimal number of
components related to batch effects using the same criterion mentioned
in section <em>PLSDA-batch</em>.</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># estimate the number of batch components</span></span>
<span><span class="va">ad.batch.tune</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/PLSDA_batch.html">PLSDA_batch</a></span><span class="op">(</span>X <span class="op">=</span> <span class="va">ad.clr</span>, </span>
<span>                             Y.trt <span class="op">=</span> <span class="va">ad.trt</span>, Y.bat <span class="op">=</span> <span class="va">ad.batch</span>,</span>
<span>                             ncomp.trt <span class="op">=</span> <span class="fl">1</span>, keepX.trt <span class="op">=</span> <span class="fl">100</span>,</span>
<span>                             ncomp.bat <span class="op">=</span> <span class="fl">10</span><span class="op">)</span></span>
<span><span class="va">ad.batch.tune</span><span class="op">$</span><span class="va">explained_variance.bat</span> <span class="co">#4</span></span></code></pre></div>
<pre><code><span><span class="co">## $X</span></span>
<span><span class="co">##      comp1      comp2      comp3      comp4      comp5      comp6      comp7 </span></span>
<span><span class="co">## 0.17420018 0.11477097 0.09813477 0.07894965 0.02918587 0.03720219 0.03784008 </span></span>
<span><span class="co">##      comp8      comp9     comp10 </span></span>
<span><span class="co">## 0.03502515 0.01443753 0.01679660 </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $Y</span></span>
<span><span class="co">##      comp1      comp2      comp3      comp4      comp5      comp6      comp7 </span></span>
<span><span class="co">## 0.24747749 0.26067155 0.23010809 0.26174286 0.26414702 0.25973899 0.24436777 </span></span>
<span><span class="co">##      comp8      comp9     comp10 </span></span>
<span><span class="co">## 0.22873622 0.00300999 0.26885333</span></span></code></pre>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="va">ad.batch.tune</span><span class="op">$</span><span class="va">explained_variance.bat</span><span class="op">$</span><span class="va">Y</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_len</a></span><span class="op">(</span><span class="fl">4</span><span class="op">)</span><span class="op">]</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 1</span></span></code></pre>
<p>According to the result, we needed 4 batch related components to
remove batch variance from the data with function
<code><a href="../reference/PLSDA_batch.html">PLSDA_batch()</a></code>.</p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ad.sPLSDA_batch.res</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/PLSDA_batch.html">PLSDA_batch</a></span><span class="op">(</span>X <span class="op">=</span> <span class="va">ad.clr</span>, </span>
<span>                                   Y.trt <span class="op">=</span> <span class="va">ad.trt</span>, Y.bat <span class="op">=</span> <span class="va">ad.batch</span>,</span>
<span>                                   ncomp.trt <span class="op">=</span> <span class="fl">1</span>, keepX.trt <span class="op">=</span> <span class="fl">100</span>,</span>
<span>                                   ncomp.bat <span class="op">=</span> <span class="fl">4</span><span class="op">)</span></span>
<span><span class="va">ad.sPLSDA_batch</span> <span class="op">&lt;-</span> <span class="va">ad.sPLSDA_batch.res</span><span class="op">$</span><span class="va">X.nobatch</span></span></code></pre></div>
<p>Note: for unbalanced batch x treatment design (with the exception of
the nested design), we can specify <code>balance = FALSE</code> in
<code><a href="../reference/PLSDA_batch.html">PLSDA_batch()</a></code> function to apply weighted PLSDA-batch.</p>
</div>
</div>
<div class="section level2">
<h2 id="variable-selection">Variable selection<a class="anchor" aria-label="anchor" href="#variable-selection"></a>
</h2>
<p>After batch effect correction, we can select discriminative variables
against different treatments. Here, we use <code><a href="https://rdrr.io/pkg/mixOmics/man/splsda.html" class="external-link">splsda()</a></code> from
<span style="color: #FF7F00;">mixOmics</span> to select the top 50
microbial variables that, in combination, discriminate the different
treatment groups in the <span style="color: #0000FF;">AD data</span>.
For the details to apply sPLS-DA, see <a href="http://mixomics.org/methods/spls-da/" class="external-link">mixOmics</a>.</p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">splsda.plsda_batch</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/mixOmics/man/splsda.html" class="external-link">splsda</a></span><span class="op">(</span>X <span class="op">=</span> <span class="va">ad.PLSDA_batch</span>, Y <span class="op">=</span> <span class="va">ad.trt</span>, </span>
<span>                       ncomp <span class="op">=</span> <span class="fl">3</span>, keepX <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="fl">50</span>,<span class="fl">3</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">select.plsda_batch</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/mixOmics/man/selectVar.html" class="external-link">selectVar</a></span><span class="op">(</span><span class="va">splsda.plsda_batch</span>, comp <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">select.plsda_batch</span><span class="op">$</span><span class="va">value</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##        value.var</span></span>
<span><span class="co">## OTU46  0.3084712</span></span>
<span><span class="co">## OTU24  0.2833650</span></span>
<span><span class="co">## OTU28  0.2807199</span></span>
<span><span class="co">## OTU35 -0.2669300</span></span>
<span><span class="co">## OTU68  0.2610368</span></span>
<span><span class="co">## OTU61  0.2558267</span></span></code></pre>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">splsda.splsda_batch</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/mixOmics/man/splsda.html" class="external-link">splsda</a></span><span class="op">(</span>X <span class="op">=</span> <span class="va">ad.sPLSDA_batch</span>, Y <span class="op">=</span> <span class="va">ad.trt</span>, </span>
<span>                       ncomp <span class="op">=</span> <span class="fl">3</span>, keepX <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="fl">50</span>,<span class="fl">3</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">select.splsda_batch</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/mixOmics/man/selectVar.html" class="external-link">selectVar</a></span><span class="op">(</span><span class="va">splsda.splsda_batch</span>, comp <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">select.splsda_batch</span><span class="op">$</span><span class="va">value</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##        value.var</span></span>
<span><span class="co">## OTU46  0.3016966</span></span>
<span><span class="co">## OTU24  0.2783030</span></span>
<span><span class="co">## OTU28  0.2727259</span></span>
<span><span class="co">## OTU35 -0.2650401</span></span>
<span><span class="co">## OTU61  0.2566640</span></span>
<span><span class="co">## OTU68  0.2554550</span></span></code></pre>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/sets.html" class="external-link">intersect</a></span><span class="op">(</span><span class="va">select.plsda_batch</span><span class="op">$</span><span class="va">name</span>, <span class="va">select.splsda_batch</span><span class="op">$</span><span class="va">name</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 43</span></span></code></pre>
<p>The discriminative variables were selected and listed according to
their contributions against sample groups treated with different ranges
of phenol concentration (0-0.5 vs. 1-2 g/L). The overlap between
selections from the data corrected with PLSDA-batch and sPLSDA-batch is
high (43 out of 50), but there still exist different variables between
different selections.</p>
</div>
<div class="section level2">
<h2 class="unnumbered" id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-chapleur2016increasing" class="csl-entry">
Chapleur, Olivier, Céline Madigou, Raphaël Civade, Yohan Rodolphe,
Laurent Mazéas, and Théodore Bouchez. 2016. <span>“Increasing
Concentrations of Phenol Progressively Affect Anaerobic Digestion of
Cellulose and Associated Microbial Communities.”</span>
<em>Biodegradation</em> 27 (1): 15–27.
</div>
<div id="ref-rohart2017mixomics" class="csl-entry">
Rohart, Florian, Benoı̂t Gautier, Amrit Singh, and Kim-Anh Lê Cao. 2017.
<span>“mixOmics: An r Package for ‘Omics Feature Selection and Multiple
Data Integration.”</span> <em>PLoS Computational Biology</em> 13 (11):
e1005752.
</div>
<div id="ref-wang2020multivariate" class="csl-entry">
Wang, Yiwen, and Kim-Anh Lê Cao. 2020. <span>“A Multivariate Method to
Correct for Batch Effects in Microbiome Data.”</span> <em>bioRxiv</em>.
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Yiwen Wang.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
