<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>PLSDA-batch Vignette • PLSDAbatch</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="PLSDA-batch Vignette">
<meta property="og:description" content="PLSDAbatch">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">PLSDAbatch</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.2.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/case_studies.html">PLSDA-batch Vignette</a>
    </li>
    <li>
      <a href="../articles/simulation_study_Gaussian.html">Simulation Study (Gaussian distribution)</a>
    </li>
    <li>
      <a href="../articles/simulation_study_negative_binomial.html">Simulation Study (Negative Binomial)</a>
    </li>
    <li>
      <a href="../articles/supp_case_studies.html">Supplementary Materials for PLSDA-batch Vignette</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>PLSDA-batch Vignette</h1>
                        <h4 data-toc-skip class="author">Yiwen (Eva)
Wang</h4>
                        <h4 data-toc-skip class="author">Agricultural
Genomics Institute at Shenzhen, Chinese Academy of Agricultural
Sciences, Shenzhen, China</h4>
                        <h4 data-toc-skip class="author"><a href="mailto:wangyiwen@caas.cn" class="email">wangyiwen@caas.cn</a></h4>
            
            <h4 data-toc-skip class="date">16 September, 2022</h4>
      
      
      <div class="hidden name"><code>case_studies.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>Investigating the link between microbial composition and phenotypes
has become the limelight of research in recent years, as microorganisms
play a key role in extensive fields including agriculture, healthcare,
food production, industry and climate change <span class="citation">(<a href="#ref-wang2020characterizing" role="doc-biblioref">C. H. Wang et
al. 2020</a>; <a href="#ref-ray2020microbe" role="doc-biblioref">Ray et
al. 2020</a>; <a href="#ref-fan2020gut" role="doc-biblioref">Fan and
Pedersen 2020</a>; <a href="#ref-poirier2020integrating" role="doc-biblioref">Poirier et al. 2020</a>)</span>. The collection of
microorganisms and their genomes within a specific environment is
referred to as the <em>microbiome</em> <span class="citation">(<a href="#ref-marchesi2015vocabulary" role="doc-biblioref">Marchesi and
Ravel 2015</a>)</span>. The microbiome can be profiled using 16S rRNA
gene sequencing or whole-genome shotgun sequencing. The microbiome data
is displayed as an abundance table of counts per sample for each taxon.
This type of data have their inherent characteristics, including zero
inflation, uneven library sizes, compositional structure, and
multivariate nature, which limit statistical analysis.</p>
<p>Microbiome research faces the challenge of results reproducibility
across studies. The potential reasons are poor experimental design and
lack of rigorous operating procedures, which introduce variation in the
data (batch effects) that obscure the effect of interest. Microbiome
data are highly susceptible to batch effects because of the dynamic
nature of microbial communities <span class="citation">(<a href="#ref-wang2020managing" role="doc-biblioref">Y. Wang and LêCao
2020</a>)</span>. Numerous studies have reported batch effects
introduced by sequencing batches <span class="citation">(<a href="#ref-hieken2016microbiome" role="doc-biblioref">Hieken et al.
2016</a>)</span>, the inclusion of independent studies <span class="citation">(<a href="#ref-duvallet2017meta" role="doc-biblioref">Duvallet et al. 2017</a>)</span>, geography, age,
sex, health status, stress and diet <span class="citation">(<a href="#ref-gibson2004dietary" role="doc-biblioref">Gibson et al.
2004</a>; <a href="#ref-lozupone2013meta" role="doc-biblioref">Lozupone
et al. 2013</a>; <a href="#ref-haro2016intestinal" role="doc-biblioref">Haro et al. 2016</a>; <a href="#ref-kim2017optimizing" role="doc-biblioref">Kim et al.
2017</a>)</span>. Attempts have been made to alleviate batch effects
through standardised designs, but batch effects remain unavoidable in
practice. For example, cage effects in murine experiments, age, sex and
diet effects in human experiments, sequencing batches are ubiquitous in
microbiome studies. So far, methods to manage batch effects in
microbiome studies have been lacking, thus limiting microbiome
researchers in their ability to analyse their data.</p>
<p>Several methods to handle batch effects have been proposed. However,
these methods 1/ were primarily developed for gene expression data, thus
do not address the inherent characteristics of microbiome data or 2/ are
limited to differential abundance analysis, thus limiting the breadth of
statistical analysis that can be performed to answer biological
questions for microbiome researchers.</p>
<p>I developed a new batch effect correction method based on Projection
to Latent Structures Discriminant Analysis named “PLSDA-batch” to
correct data prior to any downstream analysis. PLSDA-batch estimates
latent components related to treatment and batch effects to remove batch
variation. The method is multivariate, non-parametric and performs
dimension reduction. Combined with centered log ratio transformation for
addressing uneven library sizes and compositional structure, PLSDA-batch
addresses all characteristics of microbiome data that existing
correction methods have ignored so far. I also developed two variants
for 1/ unbalanced batch x treatment designs that are commonly
encountered in studies with small sample size, and for 2/ selection of
discriminative variables to avoid overfitting in classification
problems. These two variants have widened the scope of applicability of
PLSDA-batch to different data settings <span class="citation">(<a href="#ref-wang2020multivariate" role="doc-biblioref">Y. Wang and Lê Cao
2020</a>)</span>.</p>
<p>My package includes both the new method for batch effect correction,
along with a comprehensive standardised framework for batch effect
management including the application of existing methods ranging from
accounting for batch effects (e.g. with linear models) to correcting for
batch effects (e.g removeBatchEffect from the <em>limma</em> package,
and ComBat from <em>sva</em>) and my proposed method for microbiome
data. My package will benefit researchers who have already generated
their data - despite a poor experimental design, to analyse their data
appropriately. My package will also benefit well-designed studies to
detect batch effects as a quality check to ensure reliable downstream
results.</p>
<p>The framework includes microbiome data pre-filtering, transformation
and batch effect detection, visualisation, accounting for or correcting
for batch effects and assessing batch effect removal and variable
selection after batch effect correction. We illustrated our framework
with data sequenced using 16S rRNA gene sequencing, but shotgun
sequencing data can also be analysed. Our framework guides data analysts
in choosing the appropriate analytic method to either account for or
correct for batch effects. In addition, batch effects can vary in
sources, designs and scale of influence on microbial variables, and
different methods for batch effect management have different
assumptions. We have illustrated each step with exemplar studies and
provided all the functions and packages for reproducibility (see Table
2). The R packages are from the open-source CRAN, Bioconductor projects
or Github repositories, therefore facilitates the analysis of any
microbial studies. The visualisation can be customised to apply to
different results from multiple methods.</p>
</div>
<div class="section level2">
<h2 id="case-studies-description">Case studies description<a class="anchor" aria-label="anchor" href="#case-studies-description"></a>
</h2>
<p>We considered four case studies with different data settings. We
illustrated the complete analysis of only one study, and specific
analysis steps for the other studies. The complete analyses of the other
studies can be found in my <a href="https://github.com/EvaYiwenWang/PLSDAbatch" class="external-link">GitHub</a>.</p>
<p>The datasets from four case studies are stored internally in our R
package <span style="color: #FF7F00;">PLSDAbatch</span>. Their basic
information can be found in Table 1.</p>
<p><strong><span style="color: #0000FF;">Anaerobic
digestion</span>.</strong> This study explored the microbial indicators
that could improve the efficacy of anaerobic digestion (AD) bioprocess
and prevent its failure <span class="citation">(<a href="#ref-chapleur2016increasing" role="doc-biblioref">Chapleur et al.
2016</a>)</span>. The samples were treated with two different ranges of
phenol concentration (effect of interest) and processed at five
different dates (batch effect). This study included a clear and strong
batch effect with an approx. balanced batch x treatment design. Using
this study, we illustrated the complete analysis of batch effect
management.</p>
<p><strong><span style="color: #964B00;">Sponge</span> <em><span style="color: #964B00;">A. aerophoba</span></em>.</strong> This study
investigated the relationship between metabolite concentration and
microbial abundance of specific sponge tissues <span class="citation">(<a href="#ref-sacristan2011exploring" role="doc-biblioref">Sacristán-Soriano et al. 2011</a>)</span>. The
samples were collected from two types of tissues (Ectosome
vs. Choanosome) and processed on two separate denaturing gradient gels
in electrophoresis. This study included relative abundance data only and
a completely balanced batch x treatment design.</p>
<p><strong><span style="color: #808000;">High fat high sugar
diet</span>.</strong> This study aimed to investigate the effect of high
fat high sugar (HFHS) diet on the mouse microbiome <span class="citation">(<a href="#ref-susin2020variable" role="doc-biblioref">Susin et al. 2020</a>)</span>. The samples were
collected at different days from the mice treated with two types of
diets (HFHS vs. normal) and housed in different cages. This study
included a extremely unbalanced (nested) batch x treatment design
between cages and diets and an approx. balanced design between days and
diets. With this study we illustrated how to deal with weak batch
effects.</p>
<p><strong><span style="color: #8601AF;">Mice models with Huntington’s
disease</span>.</strong> This study explored differences in microbial
composition between Huntington’s disease (HD) and wild-type (WT) mice
<span class="citation">(<a href="#ref-kong2020microbiome" role="doc-biblioref">Kong et al. 2020</a>)</span>. The samples were
collected from the mice with different genotypes and housed in different
cages. We illustrated how to manage batch effects with a nested batch x
treatment design.</p>
<p><img src="figures/data_description.png"><strong>Table 1:
Overview of case studies with batch effects and their experimental
designs.</strong> We considered microbial studies for Anaerobic
Digestion (<span style="color: #0000FF;">AD data</span>), sponge (<span style="color: #964B00;">sponge data</span>), mice models with High Fat
High Sugar diets (<span style="color: #808000;">HFHS data</span>) and
Huntington’s Disease (<span style="color: #8601AF;">HD data</span>).</p>
</div>
<div class="section level2">
<h2 id="packages-installation-and-loading">Packages installation and loading<a class="anchor" aria-label="anchor" href="#packages-installation-and-loading"></a>
</h2>
<p>First, we load the packages necessary for analysis, and check the
version of each package.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># CRAN</span></span>
<span><span class="va">cran.pkgs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'pheatmap'</span>, <span class="st">'vegan'</span>, <span class="st">'ruv'</span>, <span class="st">'UpSetR'</span>, <span class="st">'gplots'</span>, </span>
<span>               <span class="st">'ggplot2'</span>, <span class="st">'gridExtra'</span>, <span class="st">'performance'</span><span class="op">)</span></span>
<span><span class="co"># Bioconductor</span></span>
<span><span class="va">bioc.pkgs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'mixOmics'</span>, <span class="st">'sva'</span>, <span class="st">'limma'</span>, <span class="st">'Biobase'</span>, <span class="st">'metagenomeSeq'</span><span class="op">)</span></span>
<span><span class="co"># GitHub</span></span>
<span><span class="va">github.pkg</span> <span class="op">&lt;-</span> <span class="st">'PLSDAbatch'</span> </span>
<span><span class="co"># devtools::install_github("https://github.com/EvaYiwenWang/PLSDAbatch")</span></span>
<span></span>
<span><span class="co"># load packages </span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">cran.pkgs</span>, <span class="va">bioc.pkgs</span>, <span class="va">github.pkg</span><span class="op">)</span>, <span class="va">require</span>, character.only <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##      pheatmap         vegan           ruv        UpSetR        gplots </span></span>
<span><span class="co">##          TRUE          TRUE          TRUE          TRUE          TRUE </span></span>
<span><span class="co">##       ggplot2     gridExtra   performance      mixOmics           sva </span></span>
<span><span class="co">##          TRUE          TRUE          TRUE          TRUE          TRUE </span></span>
<span><span class="co">##         limma       Biobase metagenomeSeq    PLSDAbatch </span></span>
<span><span class="co">##          TRUE          TRUE          TRUE          TRUE</span></span></code></pre>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># print package versions</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">cran.pkgs</span>, <span class="va">bioc.pkgs</span>, <span class="va">github.pkg</span><span class="op">)</span>, <span class="va">package.version</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##      pheatmap         vegan           ruv        UpSetR        gplots </span></span>
<span><span class="co">##      "1.0.12"       "2.6-2"     "0.9.7.1"       "1.4.0"       "3.1.3" </span></span>
<span><span class="co">##       ggplot2     gridExtra   performance      mixOmics           sva </span></span>
<span><span class="co">##       "3.3.6"         "2.3"       "0.9.2"      "6.20.0"      "3.44.0" </span></span>
<span><span class="co">##         limma       Biobase metagenomeSeq    PLSDAbatch </span></span>
<span><span class="co">##      "3.52.2"      "2.56.0"      "1.38.0"       "0.2.2"</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="data-pre-processing">Data pre-processing<a class="anchor" aria-label="anchor" href="#data-pre-processing"></a>
</h2>
<div class="section level3">
<h3 id="pre-filtering">Pre-filtering<a class="anchor" aria-label="anchor" href="#pre-filtering"></a>
</h3>
<p>We load the <span style="color: #0000FF;">AD data</span> stored
internally with function <code><a href="https://rdrr.io/r/utils/data.html" class="external-link">data()</a></code>.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># AD data</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">'AD_data'</span><span class="op">)</span> </span>
<span><span class="va">ad.count</span> <span class="op">&lt;-</span> <span class="va">AD_data</span><span class="op">$</span><span class="va">FullData</span><span class="op">$</span><span class="va">X.count</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">ad.count</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1]  75 567</span></span></code></pre>
<p>The raw <span style="color: #0000FF;">AD data</span> include 567 OTUs
and 75 samples. We then use the function <code><a href="../reference/PreFL.html">PreFL()</a></code> from our
<span style="color: #FF7F00;">PLSDAbatch</span> R package to filter the
data.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ad.filter.res</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/PreFL.html">PreFL</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">ad.count</span><span class="op">)</span></span>
<span><span class="va">ad.filter</span> <span class="op">&lt;-</span> <span class="va">ad.filter.res</span><span class="op">$</span><span class="va">data.filter</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">ad.filter</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1]  75 231</span></span></code></pre>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># zero proportion before filtering</span></span>
<span><span class="va">ad.filter.res</span><span class="op">$</span><span class="va">zero.prob</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 0.6328042</span></span></code></pre>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># zero proportion after filtering</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="va">ad.filter</span> <span class="op">==</span> <span class="fl">0</span><span class="op">)</span><span class="op">/</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">ad.filter</span><span class="op">)</span> <span class="op">*</span> <span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">ncol</a></span><span class="op">(</span><span class="va">ad.filter</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 0.3806638</span></span></code></pre>
<p>After filtering, 231 OTUs remained, and the proportion of zeroes
decreased from 63% to 38%.</p>
<p>We can also load the other data, such as the <span style="color: #964B00;">sponge data</span>.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Sponge data</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">'sponge_data'</span><span class="op">)</span> </span>
<span><span class="va">sponge.tss</span> <span class="op">&lt;-</span> <span class="va">sponge_data</span><span class="op">$</span><span class="va">X.tss</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">sponge.tss</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 32 24</span></span></code></pre>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># zero proportion</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="va">sponge.tss</span> <span class="op">==</span> <span class="fl">0</span><span class="op">)</span><span class="op">/</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">sponge.tss</span><span class="op">)</span> <span class="op">*</span> <span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">ncol</a></span><span class="op">(</span><span class="va">sponge.tss</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 0.5455729</span></span></code></pre>
<p>The <span style="color: #964B00;">sponge data</span> include the
relative abundance of 24 OTUs and 32 samples. Given the small number of
OTUs, we advise not to pre-filter the data.</p>
<p>Note: The <code><a href="../reference/PreFL.html">PreFL()</a></code> function is only dedicated to raw
counts, rather than relative abundance data. We also recommend to start
the pre-filtering on raw counts, rather than relative abundance data to
mitigate the compositionality issue.</p>
</div>
<div class="section level3">
<h3 id="transformation">Transformation<a class="anchor" aria-label="anchor" href="#transformation"></a>
</h3>
<p>Prior to CLR transformation, we recommend adding 1 as the offset for
the <span style="color: #0000FF;">AD data</span> - that are raw count
data, and 0.01 as the offset for the <span style="color: #964B00;">sponge data</span> - that are relative abundance
data. We use <code><a href="https://rdrr.io/pkg/mixOmics/man/logratio-transformations.html" class="external-link">logratio.transfo()</a></code> function in <span style="color: #FF7F00;">mixOmics</span> package to CLR transform the
data.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ad.clr</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/mixOmics/man/logratio-transformations.html" class="external-link">logratio.transfo</a></span><span class="op">(</span>X <span class="op">=</span> <span class="va">ad.filter</span>, logratio <span class="op">=</span> <span class="st">'CLR'</span>, offset <span class="op">=</span> <span class="fl">1</span><span class="op">)</span> </span>
<span><span class="fu"><a href="https://rdrr.io/r/base/class.html" class="external-link">class</a></span><span class="op">(</span><span class="va">ad.clr</span><span class="op">)</span> <span class="op">=</span> <span class="st">'matrix'</span></span>
<span></span>
<span><span class="va">sponge.clr</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/mixOmics/man/logratio-transformations.html" class="external-link">logratio.transfo</a></span><span class="op">(</span>X <span class="op">=</span> <span class="va">sponge.tss</span>, logratio <span class="op">=</span> <span class="st">'CLR'</span>, offset <span class="op">=</span> <span class="fl">0.01</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/class.html" class="external-link">class</a></span><span class="op">(</span><span class="va">sponge.clr</span><span class="op">)</span> <span class="op">=</span> <span class="st">'matrix'</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="batch-effect-detection">Batch effect detection<a class="anchor" aria-label="anchor" href="#batch-effect-detection"></a>
</h2>
<div class="section level3">
<h3 id="pca">PCA<a class="anchor" aria-label="anchor" href="#pca"></a>
</h3>
<p>We apply <code><a href="https://rdrr.io/pkg/mixOmics/man/pca.html" class="external-link">pca()</a></code> function from <span style="color: #FF7F00;">mixOmics</span> package to the <span style="color: #0000FF;">AD data</span> and
<code><a href="../reference/Scatter_Density.html">Scatter_Density()</a></code> function from <span style="color: #FF7F00;">PLSDAbatch</span> to represent the PCA sample
plot with densities.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># AD data</span></span>
<span><span class="va">ad.pca.before</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/mixOmics/man/pca.html" class="external-link">pca</a></span><span class="op">(</span><span class="va">ad.clr</span>, ncomp <span class="op">=</span> <span class="fl">3</span>, scale <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span><span class="va">ad.metadata</span> <span class="op">&lt;-</span> <span class="va">AD_data</span><span class="op">$</span><span class="va">FullData</span><span class="op">$</span><span class="va">metadata</span></span>
<span><span class="va">ad.batch</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="va">ad.metadata</span><span class="op">$</span><span class="va">sequencing_run_date</span>, </span>
<span>                  levels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">ad.metadata</span><span class="op">$</span><span class="va">sequencing_run_date</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">ad.trt</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">as.factor</a></span><span class="op">(</span><span class="va">ad.metadata</span><span class="op">$</span><span class="va">initial_phenol_concentration.regroup</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">ad.batch</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">ad.trt</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">ad.metadata</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="../reference/Scatter_Density.html">Scatter_Density</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">ad.pca.before</span>, batch <span class="op">=</span> <span class="va">ad.batch</span>, trt <span class="op">=</span> <span class="va">ad.trt</span>, </span>
<span>                title <span class="op">=</span> <span class="st">'AD data'</span>, trt.legend.title <span class="op">=</span> <span class="st">'Phenol conc.'</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="case_studies_files/figure-html/ADpcaBefore-1.png" alt="Figure 1: The PCA sample plot with densities in the AD data." width="60%"><p class="caption">
Figure 1: The PCA sample plot with densities in the AD data.
</p>
</div>
<p>In the above figure, we observed 1) the distinction between samples
treated with different phenol concentrations and 2) the differences
between samples sequenced at “14/04/2016”, “21/09/2017” and the other
dates. Therefore, the batch effect related to dates needs to be
removed.</p>
</div>
<div class="section level3">
<h3 id="boxplots-and-density-plots">Boxplots and density plots<a class="anchor" aria-label="anchor" href="#boxplots-and-density-plots"></a>
</h3>
<p>We first identify the top OTU driving the major variance in PCA using
<code><a href="https://rdrr.io/pkg/mixOmics/man/selectVar.html" class="external-link">selectVar()</a></code> in <span style="color: #FF7F00;">mixOmics</span> package. Each identified OTU can
then be plotted as boxplots and density plots using
<code><a href="../reference/box_plot.html">box_plot()</a></code> and <code><a href="../reference/density_plot.html">density_plot()</a></code> in <span style="color: #FF7F00;">PLSDAbatch</span>.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ad.OTU.name</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/mixOmics/man/selectVar.html" class="external-link">selectVar</a></span><span class="op">(</span><span class="va">ad.pca.before</span>, comp <span class="op">=</span> <span class="fl">1</span><span class="op">)</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span></span>
<span><span class="va">ad.OTU_batch</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>value <span class="op">=</span> <span class="va">ad.clr</span><span class="op">[</span>,<span class="va">ad.OTU.name</span><span class="op">]</span>, batch <span class="op">=</span> <span class="va">ad.batch</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/box_plot.html">box_plot</a></span><span class="op">(</span>df <span class="op">=</span> <span class="va">ad.OTU_batch</span>, title <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="va">ad.OTU.name</span>, <span class="st">'(AD data)'</span><span class="op">)</span>, </span>
<span>         x.angle <span class="op">=</span> <span class="fl">30</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="case_studies_files/figure-html/ADboxBefore-1.png" alt='Figure 2: Boxplots of sample values in "OTU28" before batch effect correction in the AD data.' width="60%"><p class="caption">
Figure 2: Boxplots of sample values in “OTU28” before batch effect
correction in the AD data.
</p>
</div>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/density_plot.html">density_plot</a></span><span class="op">(</span>df <span class="op">=</span> <span class="va">ad.OTU_batch</span>, title <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="va">ad.OTU.name</span>, <span class="st">'(AD data)'</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="case_studies_files/figure-html/ADdensityBefore-1.png" alt='Figure 3: Density plots of sample values in "OTU28" before batch effect correction in the AD data.' width="60%"><p class="caption">
Figure 3: Density plots of sample values in “OTU28” before batch effect
correction in the AD data.
</p>
</div>
<p>The boxplot and density plot indicated a strong date batch effect
because of the differences between “14/04/2016”, “21/09/2017” and the
other dates in the “OTU28”.</p>
<p>We also apply a linear regression model to the “OTU28” using
<code><a href="../reference/linear_regres.html">linear_regres()</a></code> from <span style="color: #FF7F00;">PLSDAbatch</span> with batch and treatment
effects as covariates. We set “14/04/2016” and “21/09/2017” as the
reference batch respectively with <code><a href="https://rdrr.io/r/stats/relevel.html" class="external-link">relevel()</a></code> from <span style="color: #FF7F00;">stats</span>.</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># reference batch: 14/04/2016</span></span>
<span><span class="va">ad.batch</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/relevel.html" class="external-link">relevel</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">ad.batch</span>, ref <span class="op">=</span> <span class="st">'14/04/2016'</span><span class="op">)</span></span>
<span></span>
<span><span class="va">ad.OTU.lm</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/linear_regres.html">linear_regres</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">ad.clr</span><span class="op">[</span>,<span class="va">ad.OTU.name</span><span class="op">]</span>, </span>
<span>                           trt <span class="op">=</span> <span class="va">ad.trt</span>, batch.fix <span class="op">=</span> <span class="va">ad.batch</span>, </span>
<span>                           type <span class="op">=</span> <span class="st">'linear model'</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">ad.OTU.lm</span><span class="op">$</span><span class="va">model</span><span class="op">$</span><span class="va">data</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Call:</span></span>
<span><span class="co">## lm(formula = data[, i] ~ trt + batch.fix)</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Residuals:</span></span>
<span><span class="co">##     Min      1Q  Median      3Q     Max </span></span>
<span><span class="co">## -1.9384 -0.3279  0.1635  0.3849  0.9887 </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Coefficients:</span></span>
<span><span class="co">##                     Estimate Std. Error t value Pr(&gt;|t|)    </span></span>
<span><span class="co">## (Intercept)           0.8501     0.2196   3.871 0.000243 ***</span></span>
<span><span class="co">## trt1-2               -1.6871     0.1754  -9.617 2.27e-14 ***</span></span>
<span><span class="co">## batch.fix09/04/2015   1.5963     0.2950   5.410 8.55e-07 ***</span></span>
<span><span class="co">## batch.fix01/07/2016   2.0839     0.2345   8.886 4.82e-13 ***</span></span>
<span><span class="co">## batch.fix14/11/2016   1.7405     0.2480   7.018 1.24e-09 ***</span></span>
<span><span class="co">## batch.fix21/09/2017   1.2646     0.2690   4.701 1.28e-05 ***</span></span>
<span><span class="co">## ---</span></span>
<span><span class="co">## Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Residual standard error: 0.7033 on 69 degrees of freedom</span></span>
<span><span class="co">## Multiple R-squared:  0.7546, Adjusted R-squared:  0.7368 </span></span>
<span><span class="co">## F-statistic: 42.44 on 5 and 69 DF,  p-value: &lt; 2.2e-16</span></span></code></pre>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># reference batch: 21/09/2017</span></span>
<span><span class="va">ad.batch</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/relevel.html" class="external-link">relevel</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">ad.batch</span>, ref <span class="op">=</span> <span class="st">'21/09/2017'</span><span class="op">)</span></span>
<span></span>
<span><span class="va">ad.OTU.lm</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/linear_regres.html">linear_regres</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">ad.clr</span><span class="op">[</span>,<span class="va">ad.OTU.name</span><span class="op">]</span>, </span>
<span>                           trt <span class="op">=</span> <span class="va">ad.trt</span>, batch.fix <span class="op">=</span> <span class="va">ad.batch</span>, </span>
<span>                           type <span class="op">=</span> <span class="st">'linear model'</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">ad.OTU.lm</span><span class="op">$</span><span class="va">model</span><span class="op">$</span><span class="va">data</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Call:</span></span>
<span><span class="co">## lm(formula = data[, i] ~ trt + batch.fix)</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Residuals:</span></span>
<span><span class="co">##     Min      1Q  Median      3Q     Max </span></span>
<span><span class="co">## -1.9384 -0.3279  0.1635  0.3849  0.9887 </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Coefficients:</span></span>
<span><span class="co">##                     Estimate Std. Error t value Pr(&gt;|t|)    </span></span>
<span><span class="co">## (Intercept)           2.1147     0.2502   8.453 2.97e-12 ***</span></span>
<span><span class="co">## trt1-2               -1.6871     0.1754  -9.617 2.27e-14 ***</span></span>
<span><span class="co">## batch.fix14/04/2016  -1.2646     0.2690  -4.701 1.28e-05 ***</span></span>
<span><span class="co">## batch.fix09/04/2015   0.3317     0.3139   1.056  0.29446    </span></span>
<span><span class="co">## batch.fix01/07/2016   0.8193     0.2573   3.185  0.00218 ** </span></span>
<span><span class="co">## batch.fix14/11/2016   0.4759     0.2705   1.760  0.08292 .  </span></span>
<span><span class="co">## ---</span></span>
<span><span class="co">## Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Residual standard error: 0.7033 on 69 degrees of freedom</span></span>
<span><span class="co">## Multiple R-squared:  0.7546, Adjusted R-squared:  0.7368 </span></span>
<span><span class="co">## F-statistic: 42.44 on 5 and 69 DF,  p-value: &lt; 2.2e-16</span></span></code></pre>
<p>From the results of linear regression, we observed P &lt; 0.001 for
the regression coefficients associated with all the other batches when
the reference batch was “14/04/2016”, which confirmed the difference
between the samples from batch “14/04/2016” and the other samples as
observed from previous plots. When the reference batch was “21/09/2017”,
we also observed significant differences between batch “21/09/2017” and
“14/04/2016”, between “21/09/2017” and “01/07/2016”. Therefore, the
batch effect because of “21/09/2017” also exists.</p>
</div>
<div class="section level3">
<h3 id="heatmap">Heatmap<a class="anchor" aria-label="anchor" href="#heatmap"></a>
</h3>
<p>We produce a heatmap using <span style="color: #FF7F00;">pheatmap</span> package. The data first need to
be scaled on both OTUs and samples.</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># scale the clr data on both OTUs and samples</span></span>
<span><span class="va">ad.clr.s</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/scale.html" class="external-link">scale</a></span><span class="op">(</span><span class="va">ad.clr</span>, center <span class="op">=</span> <span class="cn">TRUE</span>, scale <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">ad.clr.ss</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/scale.html" class="external-link">scale</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/t.html" class="external-link">t</a></span><span class="op">(</span><span class="va">ad.clr.s</span><span class="op">)</span>, center <span class="op">=</span> <span class="cn">TRUE</span>, scale <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span><span class="va">ad.anno_col</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>Batch <span class="op">=</span> <span class="va">ad.batch</span>, Treatment <span class="op">=</span> <span class="va">ad.trt</span><span class="op">)</span></span>
<span><span class="va">ad.anno_colors</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>Batch <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/mixOmics/man/colors.html" class="external-link">color.mixo</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_len</a></span><span class="op">(</span><span class="fl">5</span><span class="op">)</span><span class="op">)</span>, </span>
<span>                       Treatment <span class="op">=</span> <span class="fu"><a href="../reference/pb_color.html">pb_color</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_len</a></span><span class="op">(</span><span class="fl">2</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">ad.anno_colors</span><span class="op">$</span><span class="va">Batch</span><span class="op">)</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/levels.html" class="external-link">levels</a></span><span class="op">(</span><span class="va">ad.batch</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">ad.anno_colors</span><span class="op">$</span><span class="va">Treatment</span><span class="op">)</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/levels.html" class="external-link">levels</a></span><span class="op">(</span><span class="va">ad.trt</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">pheatmap</span><span class="op">(</span><span class="va">ad.clr.ss</span>, </span>
<span>         cluster_rows <span class="op">=</span> <span class="cn">FALSE</span>, </span>
<span>         fontsize_row <span class="op">=</span> <span class="fl">4</span>, </span>
<span>         fontsize_col <span class="op">=</span> <span class="fl">6</span>,</span>
<span>         fontsize <span class="op">=</span> <span class="fl">8</span>,</span>
<span>         clustering_distance_rows <span class="op">=</span> <span class="st">'euclidean'</span>,</span>
<span>         clustering_method <span class="op">=</span> <span class="st">'ward.D'</span>,</span>
<span>         treeheight_row <span class="op">=</span> <span class="fl">30</span>,</span>
<span>         annotation_col <span class="op">=</span> <span class="va">ad.anno_col</span>,</span>
<span>         annotation_colors <span class="op">=</span> <span class="va">ad.anno_colors</span>,</span>
<span>         border_color <span class="op">=</span> <span class="st">'NA'</span>,</span>
<span>         main <span class="op">=</span> <span class="st">'AD data - Scaled'</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="case_studies_files/figure-html/ADheatmap-1.png" alt="Figure 4: Hierarchical clustering for samples in the AD data." width="90%"><p class="caption">
Figure 4: Hierarchical clustering for samples in the AD data.
</p>
</div>
<p>In the heatmap, samples in the <span style="color: #0000FF;">AD
data</span> from batch dated “14/04/2016” were clustered and distinct
from other samples, indicating a batch effect.</p>
</div>
<div class="section level3">
<h3 id="prda">pRDA<a class="anchor" aria-label="anchor" href="#prda"></a>
</h3>
<p>We apply pRDA with <code><a href="https://rdrr.io/pkg/vegan/man/varpart.html" class="external-link">varpart()</a></code> function from <span style="color: #FF7F00;">vegan</span> R package.</p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># AD data</span></span>
<span><span class="va">ad.factors.df</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>trt <span class="op">=</span> <span class="va">ad.trt</span>, batch <span class="op">=</span> <span class="va">ad.batch</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/class.html" class="external-link">class</a></span><span class="op">(</span><span class="va">ad.clr</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="st">'matrix'</span></span>
<span><span class="va">ad.rda.before</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/vegan/man/varpart.html" class="external-link">varpart</a></span><span class="op">(</span><span class="va">ad.clr</span>, <span class="op">~</span> <span class="va">trt</span>, <span class="op">~</span> <span class="va">batch</span>, </span>
<span>                         data <span class="op">=</span> <span class="va">ad.factors.df</span>, scale <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">ad.rda.before</span><span class="op">$</span><span class="va">part</span><span class="op">$</span><span class="va">indfract</span></span></code></pre></div>
<pre><code><span><span class="co">##                 Df R.squared Adj.R.squared Testable</span></span>
<span><span class="co">## [a] = X1|X2      1        NA    0.08943682     TRUE</span></span>
<span><span class="co">## [b]              0        NA    0.01296248    FALSE</span></span>
<span><span class="co">## [c] = X2|X1      4        NA    0.26604420     TRUE</span></span>
<span><span class="co">## [d] = Residuals NA        NA    0.63155651    FALSE</span></span></code></pre>
<p>In the result, <code>X1</code> and <code>X2</code> represent the
first and second covariates fitted in the model. <code>[a]</code>,
<code>[c]</code> represent the independent proportion of variance
explained by <code>X1</code> and <code>X2</code> respectively, and
<code>[b]</code> represents the intersection variance shared between
<code>X1</code> and <code>X2</code>. In the <span style="color: #0000FF;">AD data</span>, batch variance (<code>X2</code>)
was larger than treatment variance (<code>X1</code>) with some
interaction proportion (indicated in line <code>[b]</code>,
Adj.R.squared = 0.013). The greater the intersection variance, the more
unbalanced batch x treatment design is. In this study, we considered the
design as approx. balanced.</p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Sponge data</span></span>
<span><span class="va">sponge.batch</span> <span class="op">&lt;-</span> <span class="va">sponge_data</span><span class="op">$</span><span class="va">Y.bat</span></span>
<span><span class="va">sponge.trt</span> <span class="op">&lt;-</span> <span class="va">sponge_data</span><span class="op">$</span><span class="va">Y.trt</span></span>
<span></span>
<span><span class="va">sponge.factors.df</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>trt <span class="op">=</span> <span class="va">sponge.trt</span>, batch <span class="op">=</span> <span class="va">sponge.batch</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/class.html" class="external-link">class</a></span><span class="op">(</span><span class="va">sponge.clr</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="st">'matrix'</span></span>
<span><span class="va">sponge.rda.before</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/vegan/man/varpart.html" class="external-link">varpart</a></span><span class="op">(</span><span class="va">sponge.clr</span>, <span class="op">~</span> <span class="va">trt</span>, <span class="op">~</span> <span class="va">batch</span>, </span>
<span>                             data <span class="op">=</span> <span class="va">sponge.factors.df</span>, scale <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">sponge.rda.before</span><span class="op">$</span><span class="va">part</span><span class="op">$</span><span class="va">indfract</span></span></code></pre></div>
<pre><code><span><span class="co">##                 Df R.squared Adj.R.squared Testable</span></span>
<span><span class="co">## [a] = X1|X2      1        NA    0.16572246     TRUE</span></span>
<span><span class="co">## [b]              0        NA   -0.01063501    FALSE</span></span>
<span><span class="co">## [c] = X2|X1      1        NA    0.16396277     TRUE</span></span>
<span><span class="co">## [d] = Residuals NA        NA    0.68094977    FALSE</span></span></code></pre>
<p>The <span style="color: #964B00;">sponge data</span> have a
completely balanced batch x treatment design, so there was no
intersection variance (indicated in line <code>[b]</code>, Adj.R.squared
= -0.01) detected. The proportion of treatment and batch variance is
nearly equal as indicated in lines <code>[a]</code> and
<code>[c]</code>.</p>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># HD data</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">'HD_data'</span><span class="op">)</span></span>
<span><span class="va">hd.clr</span> <span class="op">&lt;-</span> <span class="va">HD_data</span><span class="op">$</span><span class="va">EgData</span><span class="op">$</span><span class="va">X.clr</span></span>
<span><span class="va">hd.trt</span> <span class="op">&lt;-</span> <span class="va">HD_data</span><span class="op">$</span><span class="va">EgData</span><span class="op">$</span><span class="va">Y.trt</span></span>
<span><span class="va">hd.batch</span> <span class="op">&lt;-</span> <span class="va">HD_data</span><span class="op">$</span><span class="va">EgData</span><span class="op">$</span><span class="va">Y.bat</span></span>
<span></span>
<span><span class="va">hd.factors.df</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>trt <span class="op">=</span> <span class="va">hd.trt</span>, batch <span class="op">=</span> <span class="va">hd.batch</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/class.html" class="external-link">class</a></span><span class="op">(</span><span class="va">hd.clr</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="st">'matrix'</span></span>
<span><span class="va">hd.rda.before</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/vegan/man/varpart.html" class="external-link">varpart</a></span><span class="op">(</span><span class="va">hd.clr</span>, <span class="op">~</span> <span class="va">trt</span>, <span class="op">~</span> <span class="va">batch</span>, </span>
<span>                         data <span class="op">=</span> <span class="va">hd.factors.df</span>, scale <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">hd.rda.before</span><span class="op">$</span><span class="va">part</span><span class="op">$</span><span class="va">indfract</span></span></code></pre></div>
<pre><code><span><span class="co">##                 Df R.squared Adj.R.squared Testable</span></span>
<span><span class="co">## [a] = X1|X2      0        NA -2.220446e-16    FALSE</span></span>
<span><span class="co">## [b]              0        NA  9.730583e-02    FALSE</span></span>
<span><span class="co">## [c] = X2|X1      8        NA  1.608205e-01     TRUE</span></span>
<span><span class="co">## [d] = Residuals NA        NA  7.418737e-01    FALSE</span></span></code></pre>
<p>Collinearity was detected in the <span style="color: #8601AF;">HD
data</span> between treatment and batch as indicated by lines
<code>[a]</code> (Adj.R.squared = 0) and <code>[b]</code> (Adj.R.squared
= 0.097) that all treatment variance was explained as intersection
variance, because the batch x treatment design is nested. The
intersection variance in such a design is usually considerable. As the
intersection is shared between batch and treatment effects, the batch
variance in the <span style="color: #8601AF;">HD data</span> was larger
than the treatment variance.</p>
</div>
</div>
<div class="section level2">
<h2 id="managing-batch-effects">Managing batch effects<a class="anchor" aria-label="anchor" href="#managing-batch-effects"></a>
</h2>
<div class="section level3">
<h3 id="accnt">Accounting for batch effects<a class="anchor" aria-label="anchor" href="#accnt"></a>
</h3>
<p>The methods that we use to account for batch effects include the
methods designed for microbiome data: zero-inflated Gaussian (ZIG)
mixture model and the methods adapted for microbiome data: linear
regression, SVA and RUV4. Among them, SVA and RUV4 were designed for
unknown batch effects.</p>
<div class="section level4">
<h4 id="methods-designed-for-microbiome-data">Methods designed for microbiome data<a class="anchor" aria-label="anchor" href="#methods-designed-for-microbiome-data"></a>
</h4>
<p><strong>Zero-inflated Gaussian mixture model</strong> To use the ZIG
model, we first create a <code>MRexperiment</code> object applying
<code>newMRexperiment()</code> (from <span style="color: #FF7F00;">metagenomeSeq</span> package) to microbiome
counts and annotated data frames with metadata and taxonomic information
generated with <code>AnnotatedDataFrame()</code> from <span style="color: #FF7F00;">Biobase</span> package.</p>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Creating a MRexperiment object (make sure no NA in metadata)</span></span>
<span><span class="va">AD.phenotypeData</span> <span class="op">=</span> <span class="fu">AnnotatedDataFrame</span><span class="op">(</span>data <span class="op">=</span> <span class="va">AD_data</span><span class="op">$</span><span class="va">FullData</span><span class="op">$</span><span class="va">metadata</span><span class="op">)</span></span>
<span><span class="va">AD.taxaData</span> <span class="op">=</span> <span class="fu">AnnotatedDataFrame</span><span class="op">(</span>data <span class="op">=</span> <span class="va">AD_data</span><span class="op">$</span><span class="va">FullData</span><span class="op">$</span><span class="va">taxa</span><span class="op">)</span></span>
<span><span class="va">AD.obj</span> <span class="op">=</span> <span class="fu">newMRexperiment</span><span class="op">(</span>counts <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/t.html" class="external-link">t</a></span><span class="op">(</span><span class="va">AD_data</span><span class="op">$</span><span class="va">FullData</span><span class="op">$</span><span class="va">X.count</span><span class="op">)</span>, </span>
<span>                         phenoData <span class="op">=</span> <span class="va">AD.phenotypeData</span>, </span>
<span>                         featureData <span class="op">=</span> <span class="va">AD.taxaData</span><span class="op">)</span></span>
<span><span class="va">AD.obj</span></span></code></pre></div>
<pre><code><span><span class="co">## MRexperiment (storageMode: environment)</span></span>
<span><span class="co">## assayData: 567 features, 75 samples </span></span>
<span><span class="co">##   element names: counts </span></span>
<span><span class="co">## protocolData: none</span></span>
<span><span class="co">## phenoData</span></span>
<span><span class="co">##   sampleNames: E1aJ16_A E5aJ16_A ... E8cJ96_E (75 total)</span></span>
<span><span class="co">##   varLabels: sample_name.data.extraction analysis_name ...</span></span>
<span><span class="co">##     initial_phenol_concentration.regroup (7 total)</span></span>
<span><span class="co">##   varMetadata: labelDescription</span></span>
<span><span class="co">## featureData</span></span>
<span><span class="co">##   featureNames: OTU12 OTU29 ... OTU17710 (567 total)</span></span>
<span><span class="co">##   fvarLabels: Kingdom Phylum ... Species (7 total)</span></span>
<span><span class="co">##   fvarMetadata: labelDescription</span></span>
<span><span class="co">## experimentData: use 'experimentData(object)'</span></span>
<span><span class="co">## Annotation:</span></span></code></pre>
<p>The <span style="color: #0000FF;">AD count data</span> are then
filtered with <code>filterData()</code> function (from <span style="color: #FF7F00;">metagenomeSeq</span>). We can use
<code>MRcounts()</code> to extract the count data from the
<code>MRexperiment</code> object.</p>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># filtering data to maintain a threshold of minimum depth or OTU presence</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="fu">MRcounts</span><span class="op">(</span><span class="va">AD.obj</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 567  75</span></span></code></pre>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">AD.obj</span> <span class="op">=</span> <span class="fu">filterData</span><span class="op">(</span>obj <span class="op">=</span> <span class="va">AD.obj</span>, present <span class="op">=</span> <span class="fl">20</span>, depth <span class="op">=</span> <span class="fl">5</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="fu">MRcounts</span><span class="op">(</span><span class="va">AD.obj</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 289  75</span></span></code></pre>
<p>After filtering, the <span style="color: #0000FF;">AD count
data</span> were reduced to 289 OTUs and 75 samples.</p>
<p>We calculate the percentile for CSS normalisation with
<code>cumNormStatFast()</code> function (from <span style="color: #FF7F00;">metagenomeSeq</span> package). The CSS
normalisation is applied with <code>cumNorm()</code> and the normalised
data can be exported using <code>MRcounts()</code> with
<code>norm = TRUE</code>. The normalisation scaling factors for each
sample, which are the sum of counts up to the calculated percentile, can
be accessed through <code>normFactors()</code>. We calculate the log
transfomed scaling factors by diving them with their median, which are
better than the default scaling factors that are divided by 1000
(<code>log2(normFactors(obj)/1000 + 1)</code>).</p>
<div class="sourceCode" id="cb38"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># calculate the percentile for CSS normalisation</span></span>
<span><span class="va">AD.pctl</span> <span class="op">=</span> <span class="fu">cumNormStatFast</span><span class="op">(</span>obj <span class="op">=</span> <span class="va">AD.obj</span><span class="op">)</span></span>
<span><span class="co"># CSS normalisation</span></span>
<span><span class="va">AD.obj</span> <span class="op">&lt;-</span> <span class="fu">cumNorm</span><span class="op">(</span>obj <span class="op">=</span> <span class="va">AD.obj</span>, p <span class="op">=</span> <span class="va">AD.pctl</span><span class="op">)</span></span>
<span><span class="co"># export normalised data</span></span>
<span><span class="va">AD.norm.data</span> <span class="op">&lt;-</span> <span class="fu">MRcounts</span><span class="op">(</span>obj <span class="op">=</span> <span class="va">AD.obj</span>, norm <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># normalisation scaling factors for each sample </span></span>
<span><span class="va">AD.normFactor</span> <span class="op">=</span> <span class="fu">normFactors</span><span class="op">(</span>object <span class="op">=</span> <span class="va">AD.obj</span><span class="op">)</span></span>
<span><span class="va">AD.normFactor</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log2</a></span><span class="op">(</span><span class="va">AD.normFactor</span><span class="op">/</span><span class="fu"><a href="https://rdrr.io/r/stats/median.html" class="external-link">median</a></span><span class="op">(</span><span class="va">AD.normFactor</span><span class="op">)</span> <span class="op">+</span> <span class="fl">1</span><span class="op">)</span></span></code></pre></div>
<p>We create a design matrix with treatment variable
(<code>phenol_conc</code>), batch variable (<code>seq_run</code>) and
the log transformed scaling factors using <code><a href="https://rdrr.io/r/stats/model.matrix.html" class="external-link">model.matrix()</a></code>,
and then apply the ZIG model by <code>fitZig()</code> function. We set
<code>useCSSoffset = FALSE</code> to avoid using the default scaling
factors as we have already included our customised scaling factor
(<code>AD.normFactor</code>) in the design matrix.</p>
<div class="sourceCode" id="cb39"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># treatment variable</span></span>
<span><span class="va">phenol_conc</span> <span class="op">=</span> <span class="fu">pData</span><span class="op">(</span>object <span class="op">=</span> <span class="va">AD.obj</span><span class="op">)</span><span class="op">$</span><span class="va">initial_phenol_concentration.regroup</span></span>
<span><span class="co"># batch variable</span></span>
<span><span class="va">seq_run</span> <span class="op">=</span> <span class="fu">pData</span><span class="op">(</span>object <span class="op">=</span> <span class="va">AD.obj</span><span class="op">)</span><span class="op">$</span><span class="va">sequencing_run_date</span></span>
<span></span>
<span><span class="co"># build a design matrix</span></span>
<span><span class="va">AD.mod.full</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/model.matrix.html" class="external-link">model.matrix</a></span><span class="op">(</span><span class="op">~</span> <span class="va">phenol_conc</span> <span class="op">+</span> <span class="va">seq_run</span> <span class="op">+</span> <span class="va">AD.normFactor</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># settings for the fitZig() function</span></span>
<span><span class="va">AD.settings</span> <span class="op">&lt;-</span> <span class="fu">zigControl</span><span class="op">(</span>maxit <span class="op">=</span> <span class="fl">10</span>, verbose <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># apply the ZIG model</span></span>
<span><span class="va">ADfit</span> <span class="op">&lt;-</span> <span class="fu">fitZig</span><span class="op">(</span>obj <span class="op">=</span> <span class="va">AD.obj</span>, mod <span class="op">=</span> <span class="va">AD.mod.full</span>, </span>
<span>                useCSSoffset <span class="op">=</span> <span class="cn">FALSE</span>, control <span class="op">=</span> <span class="va">AD.settings</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## it= 0, nll=123.44, log10(eps+1)=Inf, stillActive=289</span></span>
<span><span class="co">## it= 1, nll=134.33, log10(eps+1)=0.04, stillActive=10</span></span>
<span><span class="co">## it= 2, nll=134.64, log10(eps+1)=0.01, stillActive=1</span></span>
<span><span class="co">## it= 3, nll=134.80, log10(eps+1)=0.00, stillActive=0</span></span></code></pre>
<p>The OTUs with the top 50 smallest p values are extracted using
<code>MRcoefs()</code>. We set <code>eff = 0.5</code>, so only the OTUs
with at least “0.5” quantile (50%) number of effective samples (positive
samples + estimated undersampling zeroes) are extracted.</p>
<div class="sourceCode" id="cb41"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ADcoefs</span> <span class="op">&lt;-</span> <span class="fu">MRcoefs</span><span class="op">(</span><span class="va">ADfit</span>, coef <span class="op">=</span> <span class="fl">2</span>, group <span class="op">=</span> <span class="fl">3</span>, number <span class="op">=</span> <span class="fl">50</span>, eff <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">ADcoefs</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##       phenol_conc1-2      pvalues   adjPvalues</span></span>
<span><span class="co">## OTU68      -2.901041 2.467451e-15 4.081959e-13</span></span>
<span><span class="co">## OTU46      -2.847212 2.824885e-15 4.081959e-13</span></span>
<span><span class="co">## OTU28      -2.449290 4.012023e-14 3.864915e-12</span></span>
<span><span class="co">## OTU24      -2.846452 6.397959e-14 4.622526e-12</span></span>
<span><span class="co">## OTU59      -1.815250 3.784889e-13 2.187666e-11</span></span>
<span><span class="co">## OTU65      -2.141183 3.342621e-11 1.610029e-09</span></span></code></pre>
</div>
<div class="section level4">
<h4 id="other-methods-adapted-for-microbiome-data">Other methods adapted for microbiome data<a class="anchor" aria-label="anchor" href="#other-methods-adapted-for-microbiome-data"></a>
</h4>
<p><strong>Linear regression</strong> Linear regression is conducted
with <code><a href="../reference/linear_regres.html">linear_regres()</a></code> function in <span style="color: #FF7F00;">PLSDAbatch</span>. We integrated the <span style="color: #FF7F00;">performance</span> package that assesses
performance of regression models into our function
<code><a href="../reference/linear_regres.html">linear_regres()</a></code>. Therefore, we can apply
<code>check_model()</code> from <span style="color: #FF7F00;">performance</span> to the outputs from
<code><a href="../reference/linear_regres.html">linear_regres()</a></code> to diagnose the validity of the model
fitted with treatment and batch effects for each variable <span class="citation">(<a href="#ref-daniel2020performance" role="doc-biblioref">LÃŒdecke et al. 2020</a>)</span>. We can extract
performance measurements such as adjusted R2, RMSE, RSE, AIC and BIC for
the models fitted with and without batch effects, which are also the
outputs of <code><a href="../reference/linear_regres.html">linear_regres()</a></code>.</p>
<p>We apply <code>type = "linear model"</code> to the <span style="color: #0000FF;">AD data</span> because of the balanced batch x
treatment design.</p>
<div class="sourceCode" id="cb43"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># AD data</span></span>
<span><span class="va">ad.clr</span> <span class="op">&lt;-</span> <span class="va">ad.clr</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_len</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">ad.clr</span><span class="op">)</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_len</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">ncol</a></span><span class="op">(</span><span class="va">ad.clr</span><span class="op">)</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="va">ad.lm</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/linear_regres.html">linear_regres</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">ad.clr</span>, trt <span class="op">=</span> <span class="va">ad.trt</span>, </span>
<span>                       batch.fix <span class="op">=</span> <span class="va">ad.batch</span>, type <span class="op">=</span> <span class="st">'linear model'</span><span class="op">)</span></span>
<span></span>
<span><span class="va">ad.p</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">ad.lm</span><span class="op">$</span><span class="va">lm.table</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">{</span><span class="va">x</span><span class="op">$</span><span class="va">coefficients</span><span class="op">[</span><span class="fl">2</span>,<span class="fl">4</span><span class="op">]</span><span class="op">}</span><span class="op">)</span></span>
<span><span class="va">ad.p.adj</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/p.adjust.html" class="external-link">p.adjust</a></span><span class="op">(</span>p <span class="op">=</span> <span class="va">ad.p</span>, method <span class="op">=</span> <span class="st">'fdr'</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">check_model</span><span class="op">(</span><span class="va">ad.lm</span><span class="op">$</span><span class="va">model</span><span class="op">$</span><span class="va">OTU12</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="case_studies_files/figure-html/ADlm-1.png" alt='Figure 5: Diagnostic plots for the model fitted with batch effects of "OTU12" in the AD data.' width="100%"><p class="caption">
Figure 5: Diagnostic plots for the model fitted with batch effects of
“OTU12” in the AD data.
</p>
</div>
<p>To diagnose the validity of the model fitted with both treatment and
batch effects, we use different plots to check the assumptions of each
microbial variable. For example, the diagnostic plots of “OTU12” are
shown in the above figure panel. The simulated data under the fitted
model were close to the real data (shown as green line), indicating good
model fitness (top panel). The linearity (or homoscedasticity) and
homogeneity of variance were not satisfied. The correlation between
batch (<code>batch.fix</code>) and treatment (<code>trt</code>) effects
was very low, indicating a good model with low collinearity. Some
samples could be classified as outliers with a Cook’s distance larger
than or equal to 0.5, for example, “57”, “39”, “47”, “44” and “16”
(middle panel). The distribution of residuals was very close to normal
(bottom panel). For the microbial variables with some assumptions not
met, we should be careful about their results.</p>
<p>For performance measurements of models fitted with or without batch
effects, We show an example of the results for some variables.</p>
<div class="sourceCode" id="cb44"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">ad.lm</span><span class="op">$</span><span class="va">adj.R2</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##            trt.only trt.batch</span></span>
<span><span class="co">## OTU12   0.008445545 0.4785532</span></span>
<span><span class="co">## OTU29  -0.012966020 0.7808945</span></span>
<span><span class="co">## OTU77  -0.013343146 0.4779754</span></span>
<span><span class="co">## OTU86   0.183743803 0.2850153</span></span>
<span><span class="co">## OTU97  -0.008989284 0.4502563</span></span>
<span><span class="co">## OTU106  0.072963581 0.8745676</span></span></code></pre>
<p>The adjusted <span class="math inline">\(R^2\)</span> of the model
with both treatment and batch effects for all the OTUs listed was larger
than the model with treatment effects only, suggesting that the model
fitted with batch effects explained more data variance, and was thus
better than the model without batch effects.</p>
<p>We next look at the AIC of models fitted with or without batch
effects.</p>
<div class="sourceCode" id="cb46"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">ad.lm</span><span class="op">$</span><span class="va">AIC</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##        trt.only trt.batch</span></span>
<span><span class="co">## OTU12  208.5626 164.13617</span></span>
<span><span class="co">## OTU29  266.0485 154.99072</span></span>
<span><span class="co">## OTU77  180.7800 134.80638</span></span>
<span><span class="co">## OTU86  212.2987 206.13722</span></span>
<span><span class="co">## OTU97  161.6715 119.90108</span></span>
<span><span class="co">## OTU106 205.4135  59.17005</span></span></code></pre>
<p>A lower AIC indicates a better fit, here for a model fitted with
batch effects for all the OTUs.</p>
<p>Both results strongly indicated that a batch effect should be fitted
in the linear model.</p>
<p>We apply a <code>"linear mixed model"</code> to the <span style="color: #8601AF;">HD data</span> because of the nested batch x
treatment design.</p>
<div class="sourceCode" id="cb48"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># HD data</span></span>
<span><span class="va">hd.lmm</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/linear_regres.html">linear_regres</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">hd.clr</span>, trt <span class="op">=</span> <span class="va">hd.trt</span>, </span>
<span>                        batch.random <span class="op">=</span> <span class="va">hd.batch</span>, </span>
<span>                        type <span class="op">=</span> <span class="st">'linear mixed model'</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb49"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hd.p</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">hd.lmm</span><span class="op">$</span><span class="va">lmm.table</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">{</span><span class="va">x</span><span class="op">$</span><span class="va">coefficients</span><span class="op">[</span><span class="fl">2</span>,<span class="fl">5</span><span class="op">]</span><span class="op">}</span><span class="op">)</span></span>
<span><span class="va">hd.p.adj</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/p.adjust.html" class="external-link">p.adjust</a></span><span class="op">(</span>p <span class="op">=</span> <span class="va">hd.p</span>, method <span class="op">=</span> <span class="st">'fdr'</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">check_model</span><span class="op">(</span><span class="va">hd.lmm</span><span class="op">$</span><span class="va">model</span><span class="op">$</span><span class="va">OTU1</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="case_studies_files/figure-html/HDlmm-1.png" alt='Figure 6: Diagnostic plots for the model fitted with batch effects of "OTU1" in the HD data.' width="100%"><p class="caption">
Figure 6: Diagnostic plots for the model fitted with batch effects of
“OTU1” in the HD data.
</p>
</div>
<p>According to the diagnostic plots of “OTU1” as shown in the above
figure panel, the simulated data under the fitted model were not very
close to the real data (shown as green line), indicating an imperfect
fitness (top panel). The linearity (or homoscedasticity) and homogeneity
of variance were not satisfied. Some samples could be classified as
outliers with a Cook’s distance larger than or equal to 0.5, for
example, “12”, “21”, “20”, “11” and “6” (middle panel). The distribution
of residuals was not normal, while the one of random effects was very
close to normal (bottom panel).</p>
<p>We then look at the AIC of models fitted with or without batch
effects.</p>
<div class="sourceCode" id="cb50"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">hd.lmm</span><span class="op">$</span><span class="va">AIC</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##       trt.only trt.batch</span></span>
<span><span class="co">## OTU1  60.00637  65.46410</span></span>
<span><span class="co">## OTU2 130.14167 130.96344</span></span>
<span><span class="co">## OTU3  71.76985  75.58915</span></span>
<span><span class="co">## OTU4 134.59992 134.04812</span></span>
<span><span class="co">## OTU5 150.25059 148.30644</span></span>
<span><span class="co">## OTU6  99.92864  99.05391</span></span></code></pre>
<p>For some OTUs, the model fitted without batch effects was better with
a lower AIC, e.g., “OTU1”, “OTU2” and “OTU3”. “OTU4”, “OTU5” and “OTU6”
were more appropriate within a model with batch effects.</p>
<p>Since we apply a <code>"linear mixed model"</code> to the <span style="color: #8601AF;">HD data</span>, this type of model can only
output conditional <span class="math inline">\(R^2\)</span> that
includes the variance of both fixed and random effects (treatment, fixed
and random batch effects) and marginal <span class="math inline">\(R^2\)</span> that includes only the variance of
fixed effects (treatment and fixed batch effects) <span class="citation">(<a href="#ref-nakagawa2013general" role="doc-biblioref">Nakagawa and Schielzeth 2013</a>)</span>. The
marginal <span class="math inline">\(R^2\)</span>s of the models with
and without the batch effect in the <span style="color: #8601AF;">HD
data</span> should be the same theoretically, as marginal <span class="math inline">\(R^2\)</span> only includes fixed effects and the
treatment effect was the only fixed effect here. In the actual
calculation, the results of these two models were not the same, because
part of the variance was explained by the random effect in the model
fitted with the batch effect.</p>
<div class="sourceCode" id="cb52"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">hd.lmm</span><span class="op">$</span><span class="va">cond.R2</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##        trt.only trt.batch</span></span>
<span><span class="co">## OTU1 0.47917775 0.5161317</span></span>
<span><span class="co">## OTU2 0.39223824 0.4729848</span></span>
<span><span class="co">## OTU3 0.02091376 0.2636432</span></span>
<span><span class="co">## OTU4 0.37200835 0.5672392</span></span>
<span><span class="co">## OTU5 0.07858373 0.2938212</span></span>
<span><span class="co">## OTU6 0.41388250 0.6143134</span></span></code></pre>
<div class="sourceCode" id="cb54"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">hd.lmm</span><span class="op">$</span><span class="va">marg.R2</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##        trt.only  trt.batch</span></span>
<span><span class="co">## OTU1 0.47917775 0.46735273</span></span>
<span><span class="co">## OTU2 0.39223824 0.38459672</span></span>
<span><span class="co">## OTU3 0.02091376 0.01489608</span></span>
<span><span class="co">## OTU4 0.37200835 0.36122754</span></span>
<span><span class="co">## OTU5 0.07858373 0.06052938</span></span>
<span><span class="co">## OTU6 0.41388250 0.40749149</span></span></code></pre>
<p>We observed that some variables resulted in singular fits
(<code>error message: Can't compute random effect variances. Some variance components equal zero. Your model may suffer from singulariy.</code>),
which are expected to happen in linear mixed models when covariates are
nested. We recommend noting the variables for which this error occurs,
as it may lead to unreliable results.</p>
<p><strong>SVA</strong> accounts for unknown batch effects. Here we
assume that the batch grouping information in the <span style="color: #0000FF;">AD data</span> is unknown. We first build two
design matrices with (<code>ad.mod</code>) and without
(<code>ad.mod0</code>) treatment grouping information generated with
<code><a href="https://rdrr.io/r/stats/model.matrix.html" class="external-link">model.matrix()</a></code> function from <span style="color: #FF7F00;">stats</span>. We then use <code>num.sv()</code>
from <span style="color: #FF7F00;">sva</span> package to determine the
number of batch variables <code>n.sv</code> that is used to estimate
batch effects in function <code>sva()</code>.</p>
<div class="sourceCode" id="cb56"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># estimate batch effects</span></span>
<span><span class="va">ad.mod</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/model.matrix.html" class="external-link">model.matrix</a></span><span class="op">(</span> <span class="op">~</span> <span class="va">ad.trt</span><span class="op">)</span></span>
<span><span class="va">ad.mod0</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/model.matrix.html" class="external-link">model.matrix</a></span><span class="op">(</span> <span class="op">~</span> <span class="fl">1</span>, data <span class="op">=</span> <span class="va">ad.trt</span><span class="op">)</span></span>
<span><span class="va">ad.sva.n</span> <span class="op">&lt;-</span> <span class="fu">num.sv</span><span class="op">(</span>dat <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/t.html" class="external-link">t</a></span><span class="op">(</span><span class="va">ad.clr</span><span class="op">)</span>, mod <span class="op">=</span> <span class="va">ad.mod</span>, method <span class="op">=</span> <span class="st">'leek'</span><span class="op">)</span></span>
<span><span class="va">ad.sva</span> <span class="op">&lt;-</span> <span class="fu">sva</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/t.html" class="external-link">t</a></span><span class="op">(</span><span class="va">ad.clr</span><span class="op">)</span>, <span class="va">ad.mod</span>, <span class="va">ad.mod0</span>, n.sv <span class="op">=</span> <span class="va">ad.sva.n</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Number of significant surrogate variables is:  4 </span></span>
<span><span class="co">## Iteration (out of 5 ):1  2  3  4  5</span></span></code></pre>
<p>The estimated batch effects are then applied to
<code>f.pvalue()</code> to calculate the P-values of treatment effects.
The estimated batch effects in SVA are assumed to be independent of the
treatment effects. However, SVA considers some correlation between batch
and treatment effects <span class="citation">(<a href="#ref-wang2020managing" role="doc-biblioref">Y. Wang and LêCao
2020</a>)</span>.</p>
<div class="sourceCode" id="cb58"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># include estimated batch effects in the linear model</span></span>
<span><span class="va">ad.mod.batch</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">cbind</a></span><span class="op">(</span><span class="va">ad.mod</span>, <span class="va">ad.sva</span><span class="op">$</span><span class="va">sv</span><span class="op">)</span></span>
<span><span class="va">ad.mod0.batch</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">cbind</a></span><span class="op">(</span><span class="va">ad.mod0</span>, <span class="va">ad.sva</span><span class="op">$</span><span class="va">sv</span><span class="op">)</span></span>
<span><span class="va">ad.sva.p</span> <span class="op">&lt;-</span> <span class="fu">f.pvalue</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/t.html" class="external-link">t</a></span><span class="op">(</span><span class="va">ad.clr</span><span class="op">)</span>, <span class="va">ad.mod.batch</span>, <span class="va">ad.mod0.batch</span><span class="op">)</span></span>
<span><span class="va">ad.sva.p.adj</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/p.adjust.html" class="external-link">p.adjust</a></span><span class="op">(</span><span class="va">ad.sva.p</span>, method <span class="op">=</span> <span class="st">'fdr'</span><span class="op">)</span></span></code></pre></div>
<p><strong>RUV4</strong> Before applying RUV4 (<code>RUV4()</code> from
<span style="color: #FF7F00;">ruv</span> package), we need to specify
negative control variables and the number of batch variables to
estimate. We can use the empirical negative controls that are not
significantly differentially abundant (adjusted P &gt; 0.05) from a
linear regression with the treatment information as the only
covariate.</p>
<p>We use a loop to fit a linear regression for each microbial variable
and adjust P values of treatment effects for multiple comparisons with
<code><a href="https://rdrr.io/r/stats/p.adjust.html" class="external-link">p.adjust()</a></code> from <span style="color: #FF7F00;">stats</span>.
The empirical negative controls are then extracted according to the
adjusted P values.</p>
<div class="sourceCode" id="cb59"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># empirical negative controls</span></span>
<span><span class="va">ad.empir.p</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="kw">for</span><span class="op">(</span><span class="va">e</span> <span class="kw">in</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_len</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">ncol</a></span><span class="op">(</span><span class="va">ad.clr</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="va">ad.empir.lm</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/lm.html" class="external-link">lm</a></span><span class="op">(</span><span class="va">ad.clr</span><span class="op">[</span>,<span class="va">e</span><span class="op">]</span> <span class="op">~</span> <span class="va">ad.trt</span><span class="op">)</span></span>
<span>  <span class="va">ad.empir.p</span><span class="op">[</span><span class="va">e</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">ad.empir.lm</span><span class="op">)</span><span class="op">$</span><span class="va">coefficients</span><span class="op">[</span><span class="fl">2</span>,<span class="fl">4</span><span class="op">]</span></span>
<span><span class="op">}</span></span>
<span><span class="va">ad.empir.p.adj</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/p.adjust.html" class="external-link">p.adjust</a></span><span class="op">(</span>p <span class="op">=</span> <span class="va">ad.empir.p</span>, method <span class="op">=</span> <span class="st">'fdr'</span><span class="op">)</span></span>
<span><span class="va">ad.nc</span> <span class="op">&lt;-</span> <span class="va">ad.empir.p.adj</span> <span class="op">&gt;</span> <span class="fl">0.05</span></span></code></pre></div>
<p>The number of batch variables <code>k</code> can be determined using
<code>getK()</code> function.</p>
<div class="sourceCode" id="cb60"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># estimate k</span></span>
<span><span class="va">ad.k.res</span> <span class="op">&lt;-</span> <span class="fu">getK</span><span class="op">(</span>Y <span class="op">=</span> <span class="va">ad.clr</span>, X <span class="op">=</span> <span class="va">ad.trt</span>, ctl <span class="op">=</span> <span class="va">ad.nc</span><span class="op">)</span></span>
<span><span class="va">ad.k</span> <span class="op">&lt;-</span> <span class="va">ad.k.res</span><span class="op">$</span><span class="va">k</span></span></code></pre></div>
<p>We then apply <code>RUV4()</code> with known treatment variables,
estimated negative control variables and <code>k</code> batch variables.
The calculated P values also need to be adjusted for multiple
comparisons.</p>
<div class="sourceCode" id="cb61"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># RUV4</span></span>
<span><span class="va">ad.ruv4</span> <span class="op">&lt;-</span> <span class="fu">RUV4</span><span class="op">(</span>Y <span class="op">=</span> <span class="va">ad.clr</span>, X <span class="op">=</span> <span class="va">ad.trt</span>, ctl <span class="op">=</span> <span class="va">ad.nc</span>, k <span class="op">=</span> <span class="va">ad.k</span><span class="op">)</span> </span>
<span><span class="va">ad.ruv4.p</span> <span class="op">&lt;-</span> <span class="va">ad.ruv4</span><span class="op">$</span><span class="va">p</span></span>
<span><span class="va">ad.ruv4.p.adj</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/p.adjust.html" class="external-link">p.adjust</a></span><span class="op">(</span><span class="va">ad.ruv4.p</span>, method <span class="op">=</span> <span class="st">"fdr"</span><span class="op">)</span></span></code></pre></div>
<p>Note: A package named <span style="color: #FF7F00;">RUVSeq</span> has
been developed for count data. It provides <code>RUVg()</code> using
negative control variables, and also other functions <code>RUVs()</code>
and <code>RUVr()</code> using sample replicates <span class="citation">(<a href="#ref-moskovicz2020skin" role="doc-biblioref">Moskovicz et al. 2020</a>)</span> or residuals from
the regression on treatment effects to estimate and then account for
latent batch effects. However, for CLR-transformed data, we still
recommend the standard <span style="color: #FF7F00;">ruv</span>
package.</p>
</div>
</div>
<div class="section level3">
<h3 id="corrct">Correcting for batch effects<a class="anchor" aria-label="anchor" href="#corrct"></a>
</h3>
<p>The methods that we use to correct for batch effects include
removeBatchEffect, ComBat, PLSDA-batch, sPLSDA-batch, Percentile
Normalisation and RUVIII. Among them, RUVIII was designed for unknown
batch effects. They were all applied to and illustrated with the <span style="color: #0000FF;">AD data</span>.</p>
<p><strong>removeBatchEffect</strong></p>
<p>The <code>removeBatchEffect()</code> function is implemented in <span style="color: #FF7F00;">limma</span> package. The design matrix
(<code>design</code>) with treatment grouping information can be
generated with <code><a href="https://rdrr.io/r/stats/model.matrix.html" class="external-link">model.matrix()</a></code> function from <span style="color: #FF7F00;">stats</span> as shown in section
<strong>accounting for batch effects</strong> method “SVA”.</p>
<p>Here we use <code>removeBatchEffect()</code> function with batch
grouping information (<code>batch</code>) and treatment design matrix
(<code>design</code>) to calculate batch effect corrected data
<code>ad.rBE</code>.</p>
<div class="sourceCode" id="cb62"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ad.rBE</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/t.html" class="external-link">t</a></span><span class="op">(</span><span class="fu">removeBatchEffect</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/t.html" class="external-link">t</a></span><span class="op">(</span><span class="va">ad.clr</span><span class="op">)</span>, batch <span class="op">=</span> <span class="va">ad.batch</span>, </span>
<span>                              design <span class="op">=</span> <span class="va">ad.mod</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><strong>ComBat</strong></p>
<p>The <code>ComBat()</code> function (from <span style="color: #FF7F00;">sva</span> package) is implemented as parametric
or non-parametric correction with option <code>par.prior</code>. Under a
parametric adjustment, we can assess the model’s validity with
<code>prior.plots = T</code> <span class="citation">(<a href="#ref-leek2012sva" role="doc-biblioref">Leek et al.
2012</a>)</span>.</p>
<p>Here we use a non-parametric correction
(<code>par.prior = FALSE</code>) with batch grouping information
(<code>batch</code>) and treatment design matrix (<code>mod</code>) to
calculate batch effect corrected data <code>ad.ComBat</code>.</p>
<div class="sourceCode" id="cb63"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ad.ComBat</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/t.html" class="external-link">t</a></span><span class="op">(</span><span class="fu">ComBat</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/t.html" class="external-link">t</a></span><span class="op">(</span><span class="va">ad.clr</span><span class="op">)</span>, batch <span class="op">=</span> <span class="va">ad.batch</span>, </span>
<span>                      mod <span class="op">=</span> <span class="va">ad.mod</span>, par.prior <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><strong>PLSDA-batch</strong></p>
<p>The <code><a href="../reference/PLSDA_batch.html">PLSDA_batch()</a></code> function is implemented in <span style="color: #FF7F00;">PLSDAbatch</span> package. To use this function,
we need to specify the optimal number of components related to treatment
(<code>ncomp.trt</code>) or batch effects (<code>ncomp.bat</code>).</p>
<p>Here in the <span style="color: #0000FF;">AD data</span>, we use
<code><a href="https://rdrr.io/pkg/mixOmics/man/plsda.html" class="external-link">plsda()</a></code> from <span style="color: #FF7F00;">mixOmics</span>
with only treatment grouping information to estimate the optimal number
of treatment components to preserve.</p>
<div class="sourceCode" id="cb64"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># estimate the number of treatment components</span></span>
<span><span class="va">ad.trt.tune</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/mixOmics/man/plsda.html" class="external-link">plsda</a></span><span class="op">(</span>X <span class="op">=</span> <span class="va">ad.clr</span>, Y <span class="op">=</span> <span class="va">ad.trt</span>, ncomp <span class="op">=</span> <span class="fl">5</span><span class="op">)</span></span>
<span><span class="va">ad.trt.tune</span><span class="op">$</span><span class="va">prop_expl_var</span> <span class="co">#1</span></span></code></pre></div>
<pre><code><span><span class="co">## $X</span></span>
<span><span class="co">##      comp1      comp2      comp3      comp4      comp5 </span></span>
<span><span class="co">## 0.18619506 0.07873817 0.08257396 0.09263497 0.06594757 </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $Y</span></span>
<span><span class="co">##      comp1      comp2      comp3      comp4      comp5 </span></span>
<span><span class="co">## 1.00000000 0.33857374 0.17315267 0.10551296 0.08185822</span></span></code></pre>
<p>We choose the number that explains 100% variance in the outcome
matrix <code>Y</code>, thus from the result, 1 component was enough to
preserve the treatment information.</p>
<p>We then use <code><a href="../reference/PLSDA_batch.html">PLSDA_batch()</a></code> function with both treatment
and batch grouping information to estimate the optimal number of batch
components to remove.</p>
<div class="sourceCode" id="cb66"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># estimate the number of batch components</span></span>
<span><span class="va">ad.batch.tune</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/PLSDA_batch.html">PLSDA_batch</a></span><span class="op">(</span>X <span class="op">=</span> <span class="va">ad.clr</span>, </span>
<span>                             Y.trt <span class="op">=</span> <span class="va">ad.trt</span>, Y.bat <span class="op">=</span> <span class="va">ad.batch</span>,</span>
<span>                             ncomp.trt <span class="op">=</span> <span class="fl">1</span>, ncomp.bat <span class="op">=</span> <span class="fl">10</span><span class="op">)</span></span>
<span><span class="va">ad.batch.tune</span><span class="op">$</span><span class="va">explained_variance.bat</span> <span class="co">#4</span></span></code></pre></div>
<pre><code><span><span class="co">## $X</span></span>
<span><span class="co">##      comp1      comp2      comp3      comp4      comp5      comp6      comp7 </span></span>
<span><span class="co">## 0.17470922 0.11481264 0.10122717 0.07507395 0.03946605 0.03399731 0.03784224 </span></span>
<span><span class="co">##      comp8      comp9     comp10 </span></span>
<span><span class="co">## 0.02313026 0.02395840 0.01389336 </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $Y</span></span>
<span><span class="co">##       comp1       comp2       comp3       comp4       comp5       comp6 </span></span>
<span><span class="co">## 0.247465374 0.261574081 0.230138238 0.260822307 0.230056326 0.246345583 </span></span>
<span><span class="co">##       comp7       comp8       comp9      comp10 </span></span>
<span><span class="co">## 0.267963500 0.252208822 0.003425768 0.240095146</span></span></code></pre>
<div class="sourceCode" id="cb68"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="va">ad.batch.tune</span><span class="op">$</span><span class="va">explained_variance.bat</span><span class="op">$</span><span class="va">Y</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_len</a></span><span class="op">(</span><span class="fl">4</span><span class="op">)</span><span class="op">]</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 1</span></span></code></pre>
<p>Using the same criterion as choosing treatment components, we choose
the number of batch components that explains 100% variance in the
outcome matrix of batch. According to the result, 4 components were
required to remove batch effects.</p>
<p>We then can correct for batch effects applying
<code><a href="../reference/PLSDA_batch.html">PLSDA_batch()</a></code> with treatment, batch grouping information
and corresponding optimal number of related components.</p>
<div class="sourceCode" id="cb70"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ad.PLSDA_batch.res</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/PLSDA_batch.html">PLSDA_batch</a></span><span class="op">(</span>X <span class="op">=</span> <span class="va">ad.clr</span>, </span>
<span>                                  Y.trt <span class="op">=</span> <span class="va">ad.trt</span>, Y.bat <span class="op">=</span> <span class="va">ad.batch</span>,</span>
<span>                                  ncomp.trt <span class="op">=</span> <span class="fl">1</span>, ncomp.bat <span class="op">=</span> <span class="fl">4</span><span class="op">)</span></span>
<span><span class="va">ad.PLSDA_batch</span> <span class="op">&lt;-</span> <span class="va">ad.PLSDA_batch.res</span><span class="op">$</span><span class="va">X.nobatch</span></span></code></pre></div>
<p><strong>sPLSDA-batch</strong></p>
<p>We apply sPLSDA-batch using the same function
<code><a href="../reference/PLSDA_batch.html">PLSDA_batch()</a></code>, but we specify the number of variables to
select on each component (usually only treatment-related components
<code>keepX.trt</code>). To determine the optimal number of variables to
select, we use <code><a href="https://rdrr.io/pkg/mixOmics/man/tune.splsda.html" class="external-link">tune.splsda()</a></code> function from <span style="color: #FF7F00;">mixOmics</span> package <span class="citation">(<a href="#ref-rohart2017mixomics" role="doc-biblioref">Rohart et al. 2017</a>)</span> with all possible
numbers of variables to select for each component
(<code>test.keepX</code>).</p>
<div class="sourceCode" id="cb71"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># estimate the number of variables to select per treatment component</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">777</span><span class="op">)</span></span>
<span><span class="va">ad.test.keepX</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">10</span>, <span class="fl">1</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">20</span>, <span class="fl">100</span>, <span class="fl">10</span><span class="op">)</span>, </span>
<span>                  <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">150</span>, <span class="fl">231</span>, <span class="fl">50</span><span class="op">)</span>, <span class="fl">231</span><span class="op">)</span></span>
<span><span class="va">ad.trt.tune.v</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/mixOmics/man/tune.splsda.html" class="external-link">tune.splsda</a></span><span class="op">(</span>X <span class="op">=</span> <span class="va">ad.clr</span>, Y <span class="op">=</span> <span class="va">ad.trt</span>, </span>
<span>                             ncomp <span class="op">=</span> <span class="fl">1</span>, test.keepX <span class="op">=</span> <span class="va">ad.test.keepX</span>, </span>
<span>                             validation <span class="op">=</span> <span class="st">'Mfold'</span>, folds <span class="op">=</span> <span class="fl">4</span>, </span>
<span>                             nrepeat <span class="op">=</span> <span class="fl">50</span><span class="op">)</span></span>
<span><span class="va">ad.trt.tune.v</span><span class="op">$</span><span class="va">choice.keepX</span> <span class="co">#100</span></span></code></pre></div>
<p>Here the optimal number of variables to select for the treatment
component was 100. Since we have adjusted the amount of treatment
variation to preserve, we need to re-choose the optimal number of
components related to batch effects using the same criterion mentioned
in section <strong>correcting for batch effects</strong> method
“PLSDA-batch”.</p>
<div class="sourceCode" id="cb72"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># estimate the number of batch components</span></span>
<span><span class="va">ad.batch.tune</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/PLSDA_batch.html">PLSDA_batch</a></span><span class="op">(</span>X <span class="op">=</span> <span class="va">ad.clr</span>, </span>
<span>                             Y.trt <span class="op">=</span> <span class="va">ad.trt</span>, Y.bat <span class="op">=</span> <span class="va">ad.batch</span>,</span>
<span>                             ncomp.trt <span class="op">=</span> <span class="fl">1</span>, keepX.trt <span class="op">=</span> <span class="fl">100</span>,</span>
<span>                             ncomp.bat <span class="op">=</span> <span class="fl">10</span><span class="op">)</span></span>
<span><span class="va">ad.batch.tune</span><span class="op">$</span><span class="va">explained_variance.bat</span> <span class="co">#4</span></span></code></pre></div>
<pre><code><span><span class="co">## $X</span></span>
<span><span class="co">##      comp1      comp2      comp3      comp4      comp5      comp6      comp7 </span></span>
<span><span class="co">## 0.17420018 0.11477097 0.09813477 0.07894965 0.02946813 0.03702255 0.03195744 </span></span>
<span><span class="co">##      comp8      comp9     comp10 </span></span>
<span><span class="co">## 0.04098959 0.01559967 0.01584227 </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $Y</span></span>
<span><span class="co">##      comp1      comp2      comp3      comp4      comp5      comp6      comp7 </span></span>
<span><span class="co">## 0.24747749 0.26067155 0.23010809 0.26174286 0.26062893 0.26282635 0.24125867 </span></span>
<span><span class="co">##      comp8      comp9     comp10 </span></span>
<span><span class="co">## 0.22162763 0.01365842 0.25965197</span></span></code></pre>
<div class="sourceCode" id="cb74"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="va">ad.batch.tune</span><span class="op">$</span><span class="va">explained_variance.bat</span><span class="op">$</span><span class="va">Y</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_len</a></span><span class="op">(</span><span class="fl">4</span><span class="op">)</span><span class="op">]</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 1</span></span></code></pre>
<p>According to the result, we needed 4 batch related components to
remove batch variance from the data with function
<code><a href="../reference/PLSDA_batch.html">PLSDA_batch()</a></code>.</p>
<div class="sourceCode" id="cb76"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ad.sPLSDA_batch.res</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/PLSDA_batch.html">PLSDA_batch</a></span><span class="op">(</span>X <span class="op">=</span> <span class="va">ad.clr</span>, </span>
<span>                                   Y.trt <span class="op">=</span> <span class="va">ad.trt</span>, Y.bat <span class="op">=</span> <span class="va">ad.batch</span>,</span>
<span>                                   ncomp.trt <span class="op">=</span> <span class="fl">1</span>, keepX.trt <span class="op">=</span> <span class="fl">100</span>,</span>
<span>                                   ncomp.bat <span class="op">=</span> <span class="fl">4</span><span class="op">)</span></span>
<span><span class="va">ad.sPLSDA_batch</span> <span class="op">&lt;-</span> <span class="va">ad.sPLSDA_batch.res</span><span class="op">$</span><span class="va">X.nobatch</span></span></code></pre></div>
<p>Note: for unbalanced batch x treatment design (with the exception of
the nested design), we can specify <code>balance = FALSE</code> in
<code><a href="../reference/PLSDA_batch.html">PLSDA_batch()</a></code> function to apply weighted PLSDA-batch.</p>
<p><strong>Percentile Normalisation</strong></p>
<p>To apply <code><a href="../reference/percentile_norm.html">percentile_norm()</a></code> function from <span style="color: #FF7F00;">PLSDAbatch</span> package, we need to indicate a
control group (<code>ctrl.grp</code>).</p>
<div class="sourceCode" id="cb77"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ad.PN</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/percentile_norm.html">percentile_norm</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">ad.clr</span>, batch <span class="op">=</span> <span class="va">ad.batch</span>, </span>
<span>                         trt <span class="op">=</span> <span class="va">ad.trt</span>, ctrl.grp <span class="op">=</span> <span class="st">'0-0.5'</span><span class="op">)</span></span></code></pre></div>
<p><strong>RUVIII</strong></p>
<p>The <code>RUVIII()</code> function is from <span style="color: #FF7F00;">ruv</span> package. Similar to
<code>RUV4()</code>, we use empirical negative control variables and
<code>getK()</code> to determine the number of batch variables
(<code>k</code>) to estimate. We also need sample replicates which
should be structured into a mapping matrix using
<code>replicate.matrix()</code>. We can then obtain the batch effect
corrected data applying <code>RUVIII()</code> with above elements.</p>
<div class="sourceCode" id="cb78"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ad.replicates</span> <span class="op">&lt;-</span> <span class="va">ad.metadata</span><span class="op">$</span><span class="va">sample_name.data.extraction</span></span>
<span><span class="va">ad.replicates.matrix</span> <span class="op">&lt;-</span> <span class="fu">replicate.matrix</span><span class="op">(</span><span class="va">ad.replicates</span><span class="op">)</span></span>
<span></span>
<span><span class="va">ad.RUVIII</span> <span class="op">&lt;-</span> <span class="fu">RUVIII</span><span class="op">(</span>Y <span class="op">=</span> <span class="va">ad.clr</span>, M <span class="op">=</span> <span class="va">ad.replicates.matrix</span>, </span>
<span>                    ctl <span class="op">=</span> <span class="va">ad.nc</span>, k <span class="op">=</span> <span class="va">ad.k</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">ad.RUVIII</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">ad.clr</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="assessing-batch-effect-correction">Assessing batch effect correction<a class="anchor" aria-label="anchor" href="#assessing-batch-effect-correction"></a>
</h2>
<p>We apply different visualisation and quantitative methods to
assessing batch effect correction.</p>
<div class="section level3">
<h3 id="methods-that-detect-batch-effects">Methods that detect batch effects<a class="anchor" aria-label="anchor" href="#methods-that-detect-batch-effects"></a>
</h3>
<p><strong>PCA</strong></p>
<p>In the <span style="color: #0000FF;">AD data</span>, we compared the
PCA sample plots before and after batch effect correction with different
methods.</p>
<div class="sourceCode" id="cb79"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ad.pca.before</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/mixOmics/man/pca.html" class="external-link">pca</a></span><span class="op">(</span><span class="va">ad.clr</span>, ncomp <span class="op">=</span> <span class="fl">3</span>, scale <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">ad.pca.rBE</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/mixOmics/man/pca.html" class="external-link">pca</a></span><span class="op">(</span><span class="va">ad.rBE</span>, ncomp <span class="op">=</span> <span class="fl">3</span>, scale <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">ad.pca.ComBat</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/mixOmics/man/pca.html" class="external-link">pca</a></span><span class="op">(</span><span class="va">ad.ComBat</span>, ncomp <span class="op">=</span> <span class="fl">3</span>, scale <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">ad.pca.PLSDA_batch</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/mixOmics/man/pca.html" class="external-link">pca</a></span><span class="op">(</span><span class="va">ad.PLSDA_batch</span>, ncomp <span class="op">=</span> <span class="fl">3</span>, scale <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">ad.pca.sPLSDA_batch</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/mixOmics/man/pca.html" class="external-link">pca</a></span><span class="op">(</span><span class="va">ad.sPLSDA_batch</span>, ncomp <span class="op">=</span> <span class="fl">3</span>, scale <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">ad.pca.PN</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/mixOmics/man/pca.html" class="external-link">pca</a></span><span class="op">(</span><span class="va">ad.PN</span>, ncomp <span class="op">=</span> <span class="fl">3</span>, scale <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">ad.pca.RUVIII</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/mixOmics/man/pca.html" class="external-link">pca</a></span><span class="op">(</span><span class="va">ad.RUVIII</span>, ncomp <span class="op">=</span> <span class="fl">3</span>, scale <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb80"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># order batches</span></span>
<span><span class="va">ad.batch</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="va">ad.metadata</span><span class="op">$</span><span class="va">sequencing_run_date</span>, </span>
<span>                  levels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">ad.metadata</span><span class="op">$</span><span class="va">sequencing_run_date</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">ad.pca.before.plot</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/Scatter_Density.html">Scatter_Density</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">ad.pca.before</span>, </span>
<span>                                      batch <span class="op">=</span> <span class="va">ad.batch</span>, </span>
<span>                                      trt <span class="op">=</span> <span class="va">ad.trt</span>, </span>
<span>                                      title <span class="op">=</span> <span class="st">'Before correction'</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb81"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ad.pca.rBE.plot</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/Scatter_Density.html">Scatter_Density</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">ad.pca.rBE</span>, </span>
<span>                                   batch <span class="op">=</span> <span class="va">ad.batch</span>, </span>
<span>                                   trt <span class="op">=</span> <span class="va">ad.trt</span>, </span>
<span>                                   title <span class="op">=</span> <span class="st">'removeBatchEffect'</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb82"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ad.pca.ComBat.plot</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/Scatter_Density.html">Scatter_Density</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">ad.pca.ComBat</span>, </span>
<span>                                      batch <span class="op">=</span> <span class="va">ad.batch</span>, </span>
<span>                                      trt <span class="op">=</span> <span class="va">ad.trt</span>, </span>
<span>                                      title <span class="op">=</span> <span class="st">'ComBat'</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb83"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ad.pca.PLSDA_batch.plot</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/Scatter_Density.html">Scatter_Density</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">ad.pca.PLSDA_batch</span>, </span>
<span>                                           batch <span class="op">=</span> <span class="va">ad.batch</span>, </span>
<span>                                           trt <span class="op">=</span> <span class="va">ad.trt</span>, </span>
<span>                                           title <span class="op">=</span> <span class="st">'PLSDA-batch'</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb84"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ad.pca.sPLSDA_batch.plot</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/Scatter_Density.html">Scatter_Density</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">ad.pca.sPLSDA_batch</span>, </span>
<span>                                            batch <span class="op">=</span> <span class="va">ad.batch</span>, </span>
<span>                                            trt <span class="op">=</span> <span class="va">ad.trt</span>, </span>
<span>                                            title <span class="op">=</span> <span class="st">'sPLSDA-batch'</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb85"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ad.pca.PN.plot</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/Scatter_Density.html">Scatter_Density</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">ad.pca.PN</span>, </span>
<span>                                  batch <span class="op">=</span> <span class="va">ad.batch</span>, </span>
<span>                                  trt <span class="op">=</span> <span class="va">ad.trt</span>, </span>
<span>                                  title <span class="op">=</span> <span class="st">'Percentile Normalisation'</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb86"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ad.pca.RUVIII.plot</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/Scatter_Density.html">Scatter_Density</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">ad.pca.RUVIII</span>, </span>
<span>                                      batch <span class="op">=</span> <span class="va">ad.batch</span>, </span>
<span>                                      trt <span class="op">=</span> <span class="va">ad.trt</span>, </span>
<span>                                      title <span class="op">=</span> <span class="st">'RUVIII'</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="case_studies_files/figure-html/ADpca-1.png" alt="Figure 7: The PCA sample plots with densities before and after batch effect correction in the AD data." width="100%"><p class="caption">
Figure 7: The PCA sample plots with densities before and after batch
effect correction in the AD data.
</p>
</div>
<p>As shown in the PCA sample plots, the differences between the samples
sequenced at “14/04/2016”, “21/09/2017” and the other dates were removed
after batch effect correction with most methods except percentile
normalisation. The data corrected with PLSDA-batch included more
treatment variation mostly on the first PC than other method-corrected
data, as indicated on the x-axis label (26%). We can also compare the
boxplots and density plots for key variables identified in PCA driving
the major variance or heatmaps showing obvious patterns before and after
batch effect correction (results not shown).</p>
<p><strong>pRDA</strong></p>
<p>We calculate the global explained variance across all microbial
variables using pRDA. To achieve this, we create a loop for each
variable from the original (uncorrected) and batch effect-corrected
data. The final results are then displayed with
<code><a href="../reference/partVar_plot.html">partVar_plot()</a></code> from <span style="color: #FF7F00;">PLSDAbatch</span> package.</p>
<div class="sourceCode" id="cb87"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># AD data</span></span>
<span><span class="va">ad.corrected.list</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>`Before correction` <span class="op">=</span> <span class="va">ad.clr</span>, </span>
<span>                          removeBatchEffect <span class="op">=</span> <span class="va">ad.rBE</span>, </span>
<span>                          ComBat <span class="op">=</span> <span class="va">ad.ComBat</span>, </span>
<span>                          `PLSDA-batch` <span class="op">=</span> <span class="va">ad.PLSDA_batch</span>, </span>
<span>                          `sPLSDA-batch` <span class="op">=</span> <span class="va">ad.sPLSDA_batch</span>, </span>
<span>                          `Percentile Normalisation` <span class="op">=</span> <span class="va">ad.PN</span>,</span>
<span>                          RUVIII <span class="op">=</span> <span class="va">ad.RUVIII</span><span class="op">)</span></span>
<span></span>
<span><span class="va">ad.prop.df</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>Treatment <span class="op">=</span> <span class="cn">NA</span>, Intersection <span class="op">=</span> <span class="cn">NA</span>, </span>
<span>                         Batch <span class="op">=</span> <span class="cn">NA</span>, Residuals <span class="op">=</span> <span class="cn">NA</span><span class="op">)</span> </span>
<span><span class="kw">for</span><span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_len</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">ad.corrected.list</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="va">rda.res</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/vegan/man/varpart.html" class="external-link">varpart</a></span><span class="op">(</span><span class="va">ad.corrected.list</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span>, <span class="op">~</span> <span class="va">trt</span>, <span class="op">~</span> <span class="va">batch</span>,</span>
<span>                    data <span class="op">=</span> <span class="va">ad.factors.df</span>, scale <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span>  <span class="va">ad.prop.df</span><span class="op">[</span><span class="va">i</span>, <span class="op">]</span> <span class="op">&lt;-</span> <span class="va">rda.res</span><span class="op">$</span><span class="va">part</span><span class="op">$</span><span class="va">indfract</span><span class="op">$</span><span class="va">Adj.R.squared</span><span class="op">}</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">ad.prop.df</span><span class="op">)</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">ad.corrected.list</span><span class="op">)</span></span>
<span></span>
<span><span class="va">ad.prop.df</span><span class="op">[</span><span class="va">ad.prop.df</span> <span class="op">&lt;</span> <span class="fl">0</span><span class="op">]</span> <span class="op">=</span> <span class="fl">0</span></span>
<span><span class="va">ad.prop.df</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/t.html" class="external-link">t</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/apply.html" class="external-link">apply</a></span><span class="op">(</span><span class="va">ad.prop.df</span>, <span class="fl">1</span>, </span>
<span>                                    <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">{</span><span class="va">x</span><span class="op">/</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">}</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="../reference/partVar_plot.html">partVar_plot</a></span><span class="op">(</span>prop.df <span class="op">=</span> <span class="va">ad.prop.df</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="case_studies_files/figure-html/ADprda-1.png" alt="Figure 8: Global explained variance before and after batch effect correction for the AD data." width="60%"><p class="caption">
Figure 8: Global explained variance before and after batch effect
correction for the AD data.
</p>
</div>
<p>As shown in the above figure, the intersection between batch and
treatment variance was small (1.3%) for the <span style="color: #0000FF;">AD data</span>, which implies that the batch x
treatment design is not highly unbalanced. Thus the unweighted
PLSDA-batch and sPLSDA-batch were still applicable, and thus the
weighted versions were not used. sPLSDA-batch corrected data led to the
best performance with a slightly higher proportion of treatment variance
explained and undetectable batch and intersection variance compared to
the other methods.</p>
<div class="sourceCode" id="cb88"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># HFHS data</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">'HFHS_data'</span><span class="op">)</span></span>
<span><span class="va">hfhs.corrected.list</span> <span class="op">&lt;-</span> <span class="va">HFHS_data</span><span class="op">$</span><span class="va">CorrectData</span><span class="op">$</span><span class="va">data</span></span>
<span></span>
<span><span class="va">hfhs.trt</span> <span class="op">&lt;-</span> <span class="va">HFHS_data</span><span class="op">$</span><span class="va">CorrectData</span><span class="op">$</span><span class="va">Y.trt</span></span>
<span><span class="va">hfhs.batch</span> <span class="op">&lt;-</span> <span class="va">HFHS_data</span><span class="op">$</span><span class="va">CorrectData</span><span class="op">$</span><span class="va">Y.bat</span></span>
<span><span class="va">hfhs.factors.df</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>trt <span class="op">=</span> <span class="va">hfhs.trt</span>, batch <span class="op">=</span> <span class="va">hfhs.batch</span><span class="op">)</span></span>
<span></span>
<span><span class="va">hfhs.prop.df</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>Treatment <span class="op">=</span> <span class="cn">NA</span>, Intersection <span class="op">=</span> <span class="cn">NA</span>, </span>
<span>                           Batch <span class="op">=</span> <span class="cn">NA</span>, Residuals <span class="op">=</span> <span class="cn">NA</span><span class="op">)</span> </span>
<span><span class="kw">for</span><span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_len</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">hfhs.corrected.list</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="va">rda.res</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/vegan/man/varpart.html" class="external-link">varpart</a></span><span class="op">(</span><span class="va">hfhs.corrected.list</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span>, <span class="op">~</span> <span class="va">trt</span>, <span class="op">~</span> <span class="va">batch</span>,</span>
<span>                    data <span class="op">=</span> <span class="va">hfhs.factors.df</span>, scale <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span>  <span class="va">hfhs.prop.df</span><span class="op">[</span><span class="va">i</span>, <span class="op">]</span> <span class="op">&lt;-</span> <span class="va">rda.res</span><span class="op">$</span><span class="va">part</span><span class="op">$</span><span class="va">indfract</span><span class="op">$</span><span class="va">Adj.R.squared</span><span class="op">}</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">hfhs.prop.df</span><span class="op">)</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">hfhs.corrected.list</span><span class="op">)</span></span>
<span></span>
<span><span class="va">hfhs.prop.df</span><span class="op">[</span><span class="va">hfhs.prop.df</span> <span class="op">&lt;</span> <span class="fl">0</span><span class="op">]</span> <span class="op">=</span> <span class="fl">0</span></span>
<span><span class="va">hfhs.prop.df</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/t.html" class="external-link">t</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/apply.html" class="external-link">apply</a></span><span class="op">(</span><span class="va">hfhs.prop.df</span>, <span class="fl">1</span>, </span>
<span>                                      <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">{</span><span class="va">x</span><span class="op">/</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">}</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span></span>
<span><span class="fu"><a href="../reference/partVar_plot.html">partVar_plot</a></span><span class="op">(</span>prop.df <span class="op">=</span> <span class="va">hfhs.prop.df</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="case_studies_files/figure-html/HFHSprda-1.png" alt="Figure 9: Global explained variance before and after batch effect correction for the HFHS data." width="60%"><p class="caption">
Figure 9: Global explained variance before and after batch effect
correction for the HFHS data.
</p>
</div>
<p>As shown in the above figure, a small amount of batch variance was
observed (3.6%) for the <span style="color: #808000;">HFHS data</span>.
PLSDA-batch achieved the best performance for preserving the largest
treatment variance and completely removing batch variance compared to
the other methods. The results also indicate that PLSDA-batch is more
appropriate for weak batch effects, while sPLSDA-batch is more
appropriate for strong batch effects. The RUVIII performed better in the
<span style="color: #808000;">HFHS data</span> than the <span style="color: #0000FF;">AD data</span> because the sample replicates
might capture more batch variation than in the <span style="color: #0000FF;">AD data</span>. Indeed, the sample replicates in
<span style="color: #808000;">HFHS data</span> are across different day
batches, while the replicates in <span style="color: #0000FF;">AD
data</span> do not exist in all batches. Therefore, sample replicates
play a critical role in RUVIII.</p>
</div>
<div class="section level3">
<h3 id="other-methods-1">Other methods<a class="anchor" aria-label="anchor" href="#other-methods-1"></a>
</h3>
<p><span class="math inline">\(\mathbf{R^2}\)</span></p>
<p>The <span class="math inline">\(R^2\)</span> values for each variable
are calculated with <code><a href="https://rdrr.io/r/stats/lm.html" class="external-link">lm()</a></code> from <span style="color: #FF7F00;">stats</span> package. To compare the <span class="math inline">\(R^2\)</span> values among variables, we scale the
corrected data before <span class="math inline">\(R^2\)</span>
calculation. The results are displayed with <code><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot()</a></code> from
<span style="color: #FF7F00;">ggplot2</span> R package.</p>
<div class="sourceCode" id="cb89"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># AD data</span></span>
<span><span class="co"># scale</span></span>
<span><span class="va">ad.corr_scale.list</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">ad.corrected.list</span>, </span>
<span>                             <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">{</span><span class="fu"><a href="https://rdrr.io/r/base/apply.html" class="external-link">apply</a></span><span class="op">(</span><span class="va">x</span>, <span class="fl">2</span>, <span class="va">scale</span><span class="op">)</span><span class="op">}</span><span class="op">)</span></span>
<span></span>
<span><span class="va">ad.r_values.list</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="kw">for</span><span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_len</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">ad.corr_scale.list</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="va">ad.r_values</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>trt <span class="op">=</span> <span class="cn">NA</span>, batch <span class="op">=</span> <span class="cn">NA</span><span class="op">)</span></span>
<span>  <span class="kw">for</span><span class="op">(</span><span class="va">c</span> <span class="kw">in</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_len</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">ncol</a></span><span class="op">(</span><span class="va">ad.corr_scale.list</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">{</span></span>
<span>    <span class="va">ad.fit.res.trt</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/lm.html" class="external-link">lm</a></span><span class="op">(</span><span class="va">ad.corr_scale.list</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span><span class="op">[</span>,<span class="va">c</span><span class="op">]</span> <span class="op">~</span> <span class="va">ad.trt</span><span class="op">)</span></span>
<span>    <span class="va">ad.r_values</span><span class="op">[</span><span class="va">c</span>,<span class="fl">1</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">ad.fit.res.trt</span><span class="op">)</span><span class="op">$</span><span class="va">r.squared</span></span>
<span>    <span class="va">ad.fit.res.batch</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/lm.html" class="external-link">lm</a></span><span class="op">(</span><span class="va">ad.corr_scale.list</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span><span class="op">[</span>,<span class="va">c</span><span class="op">]</span> <span class="op">~</span> <span class="va">ad.batch</span><span class="op">)</span></span>
<span>    <span class="va">ad.r_values</span><span class="op">[</span><span class="va">c</span>,<span class="fl">2</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">ad.fit.res.batch</span><span class="op">)</span><span class="op">$</span><span class="va">r.squared</span></span>
<span>  <span class="op">}</span></span>
<span>  <span class="va">ad.r_values.list</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="va">ad.r_values</span></span>
<span><span class="op">}</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">ad.r_values.list</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">ad.corr_scale.list</span><span class="op">)</span></span>
<span></span>
<span><span class="va">ad.boxp.list</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="kw">for</span><span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_len</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">ad.r_values.list</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="va">ad.boxp.list</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> </span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>r2 <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">ad.r_values.list</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span><span class="op">[</span> ,<span class="st">'trt'</span><span class="op">]</span>,</span>
<span>                      <span class="va">ad.r_values.list</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span><span class="op">[</span> ,<span class="st">'batch'</span><span class="op">]</span><span class="op">)</span>, </span>
<span>               Effects <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">as.factor</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'Treatment'</span>,<span class="st">'Batch'</span><span class="op">)</span>, </span>
<span>                                       each <span class="op">=</span> <span class="fl">231</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">ad.boxp.list</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">ad.r_values.list</span><span class="op">)</span></span>
<span></span>
<span><span class="va">ad.r2.boxp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">rbind</a></span><span class="op">(</span><span class="va">ad.boxp.list</span><span class="op">$</span><span class="va">`Before correction`</span>,</span>
<span>                    <span class="va">ad.boxp.list</span><span class="op">$</span><span class="va">removeBatchEffect</span>,</span>
<span>                    <span class="va">ad.boxp.list</span><span class="op">$</span><span class="va">ComBat</span>,</span>
<span>                    <span class="va">ad.boxp.list</span><span class="op">$</span><span class="va">`PLSDA-batch`</span>,</span>
<span>                    <span class="va">ad.boxp.list</span><span class="op">$</span><span class="va">`sPLSDA-batch`</span>,</span>
<span>                    <span class="va">ad.boxp.list</span><span class="op">$</span><span class="va">`Percentile Normalisation`</span>,</span>
<span>                    <span class="va">ad.boxp.list</span><span class="op">$</span><span class="va">RUVIII</span><span class="op">)</span></span>
<span></span>
<span><span class="va">ad.r2.boxp</span><span class="op">$</span><span class="va">methods</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'Before correction'</span>, <span class="st">' removeBatchEffect'</span>, </span>
<span>                            <span class="st">'ComBat'</span>,<span class="st">'PLSDA-batch'</span>, <span class="st">'sPLSDA-batch'</span>,</span>
<span>                            <span class="st">'Percentile Normalisation'</span>, <span class="st">'RUVIII'</span><span class="op">)</span>, each <span class="op">=</span> <span class="fl">462</span><span class="op">)</span></span>
<span></span>
<span><span class="va">ad.r2.boxp</span><span class="op">$</span><span class="va">methods</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="va">ad.r2.boxp</span><span class="op">$</span><span class="va">methods</span>, </span>
<span>                             levels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">ad.r2.boxp</span><span class="op">$</span><span class="va">methods</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">ad.r2.boxp</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">Effects</span>, y <span class="op">=</span> <span class="va">r2</span>, fill <span class="op">=</span> <span class="va">Effects</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_boxplot.html" class="external-link">geom_boxplot</a></span><span class="op">(</span>alpha <span class="op">=</span> <span class="fl">0.80</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_bw</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>text <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">18</span><span class="op">)</span>,</span>
<span>        axis.title.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>        axis.title.y <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>        axis.text.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">60</span>, hjust <span class="op">=</span> <span class="fl">1</span>, size <span class="op">=</span> <span class="fl">18</span><span class="op">)</span>,</span>
<span>        axis.text.y <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">18</span><span class="op">)</span>,</span>
<span>        panel.grid.minor.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>        panel.grid.major.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>        legend.position <span class="op">=</span> <span class="st">"right"</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_grid.html" class="external-link">facet_grid</a></span><span class="op">(</span> <span class="op">~</span> <span class="va">methods</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html" class="external-link">scale_fill_manual</a></span><span class="op">(</span>values<span class="op">=</span><span class="fu"><a href="../reference/pb_color.html">pb_color</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">12</span>,<span class="fl">14</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> </span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="case_studies_files/figure-html/ADr2_1-1.png" alt="Figure 10: AD study: $R^2$ values for each microbial variable before and after batch effect correction." width="100%"><p class="caption">
Figure 10: AD study: <span class="math inline">\(R^2\)</span> values for
each microbial variable before and after batch effect correction.
</p>
</div>
<p>The corrected data from ComBat still included a few variables with a
large proportion of batch variance. A large number of variables from the
data corrected by percentile normalisation and RUVIII still included
considerable batch variance, especially RUVIII, which resulting data
contained more proportion of batch variance than treatment variance.</p>
<div class="sourceCode" id="cb90"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">##################################</span></span>
<span><span class="va">ad.barp.list</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="kw">for</span><span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_len</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">ad.r_values.list</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="va">ad.barp.list</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>r2 <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="va">ad.r_values.list</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span><span class="op">[</span> ,<span class="st">'trt'</span><span class="op">]</span><span class="op">)</span>,</span>
<span>                                         <span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="va">ad.r_values.list</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span><span class="op">[</span> ,<span class="st">'batch'</span><span class="op">]</span><span class="op">)</span><span class="op">)</span>, </span>
<span>                                  Effects <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'Treatment'</span>,<span class="st">'Batch'</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">ad.barp.list</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">ad.r_values.list</span><span class="op">)</span></span>
<span></span>
<span><span class="va">ad.r2.barp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">rbind</a></span><span class="op">(</span><span class="va">ad.barp.list</span><span class="op">$</span><span class="va">`Before correction`</span>,</span>
<span>                    <span class="va">ad.barp.list</span><span class="op">$</span><span class="va">removeBatchEffect</span>,</span>
<span>                    <span class="va">ad.barp.list</span><span class="op">$</span><span class="va">ComBat</span>,</span>
<span>                    <span class="va">ad.barp.list</span><span class="op">$</span><span class="va">`PLSDA-batch`</span>,</span>
<span>                    <span class="va">ad.barp.list</span><span class="op">$</span><span class="va">`sPLSDA-batch`</span>,</span>
<span>                    <span class="va">ad.barp.list</span><span class="op">$</span><span class="va">`Percentile Normalisation`</span>,</span>
<span>                    <span class="va">ad.barp.list</span><span class="op">$</span><span class="va">RUVIII</span><span class="op">)</span></span>
<span></span>
<span></span>
<span><span class="va">ad.r2.barp</span><span class="op">$</span><span class="va">methods</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'Before correction'</span>, <span class="st">' removeBatchEffect'</span>, </span>
<span>                            <span class="st">'ComBat'</span>,<span class="st">'PLSDA-batch'</span>, <span class="st">'sPLSDA-batch'</span>,</span>
<span>                            <span class="st">'Percentile Normalisation'</span>, <span class="st">'RUVIII'</span><span class="op">)</span>, each <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span>
<span></span>
<span><span class="va">ad.r2.barp</span><span class="op">$</span><span class="va">methods</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="va">ad.r2.barp</span><span class="op">$</span><span class="va">methods</span>, </span>
<span>                             levels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">ad.r2.barp</span><span class="op">$</span><span class="va">methods</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span></span>
<span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">ad.r2.barp</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">Effects</span>, y <span class="op">=</span> <span class="va">r2</span>, fill <span class="op">=</span> <span class="va">Effects</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_bar.html" class="external-link">geom_bar</a></span><span class="op">(</span>stat<span class="op">=</span><span class="st">"identity"</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_bw</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>text <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">18</span><span class="op">)</span>,</span>
<span>        axis.title.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>        axis.title.y <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>        axis.text.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">60</span>, hjust <span class="op">=</span> <span class="fl">1</span>, size <span class="op">=</span> <span class="fl">18</span><span class="op">)</span>,</span>
<span>        axis.text.y <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">18</span><span class="op">)</span>,</span>
<span>        panel.grid.minor.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>        panel.grid.major.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>        legend.position <span class="op">=</span> <span class="st">"right"</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_grid.html" class="external-link">facet_grid</a></span><span class="op">(</span> <span class="op">~</span> <span class="va">methods</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html" class="external-link">scale_fill_manual</a></span><span class="op">(</span>values<span class="op">=</span><span class="fu"><a href="../reference/pb_color.html">pb_color</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">12</span>,<span class="fl">14</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="case_studies_files/figure-html/ADr2_2-1.png" alt="Figure 11: AD study: the sum of $R^2$ values for each microbial variable before and after batch effect correction." width="100%"><p class="caption">
Figure 11: AD study: the sum of <span class="math inline">\(R^2\)</span>
values for each microbial variable before and after batch effect
correction.
</p>
</div>
<p>The overall sum of <span class="math inline">\(R^2\)</span> values
indicated that removeBatchEffect removed slightly more batch variance
(removeBatchEffect: 1.70, PLSDA-batch: 12.40, sPLSDA-batch: 9.25) but
preserved less treatment variance (removeBatchEffect: 31.75,
PLSDA-batch: 40.00, sPLSDA-batch: 36.22) than our proposed
approaches</p>
<p><strong>Alignment scores</strong></p>
<p>To use the <code><a href="../reference/alignment_score.html">alignment_score()</a></code> function from <span style="color: #FF7F00;">PLSDAbatch</span>, we need to specify the
proportion of data variance to explain (<code>var</code>), the number of
nearest neighbours (<code>k</code>) and the number of principal
components to estimate (<code>ncomp</code>). We then use
<code><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot()</a></code> function from <span style="color: #FF7F00;">ggplot2</span> to visualise the results.</p>
<div class="sourceCode" id="cb91"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># AD data</span></span>
<span><span class="va">ad.scores</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">ad.batch</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">ad.clr</span><span class="op">)</span></span>
<span><span class="kw">for</span><span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_len</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">ad.corrected.list</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="va">res</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/alignment_score.html">alignment_score</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">ad.corrected.list</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span>, </span>
<span>                         batch <span class="op">=</span> <span class="va">ad.batch</span>, </span>
<span>                         var <span class="op">=</span> <span class="fl">0.95</span>, </span>
<span>                         k <span class="op">=</span> <span class="fl">8</span>, </span>
<span>                         ncomp <span class="op">=</span> <span class="fl">50</span><span class="op">)</span></span>
<span>  <span class="va">ad.scores</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">ad.scores</span>, <span class="va">res</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="va">ad.scores.df</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>scores <span class="op">=</span> <span class="va">ad.scores</span>, </span>
<span>                           methods <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">ad.corrected.list</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">ad.scores.df</span><span class="op">$</span><span class="va">methods</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="va">ad.scores.df</span><span class="op">$</span><span class="va">methods</span>, </span>
<span>                               levels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/rev.html" class="external-link">rev</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">ad.corrected.list</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span></span>
<span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_bar.html" class="external-link">geom_col</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">ad.scores.df</span><span class="op">$</span><span class="va">methods</span>, </span>
<span>                        y <span class="op">=</span> <span class="va">ad.scores.df</span><span class="op">$</span><span class="va">scores</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_text.html" class="external-link">geom_text</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">ad.scores.df</span><span class="op">$</span><span class="va">methods</span>, </span>
<span>                y <span class="op">=</span> <span class="va">ad.scores.df</span><span class="op">$</span><span class="va">scores</span><span class="op">/</span><span class="fl">2</span>, </span>
<span>                label <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Round.html" class="external-link">round</a></span><span class="op">(</span><span class="va">ad.scores.df</span><span class="op">$</span><span class="va">scores</span>, <span class="fl">3</span><span class="op">)</span><span class="op">)</span>, </span>
<span>            size <span class="op">=</span> <span class="fl">3</span>, col <span class="op">=</span> <span class="st">'white'</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/coord_flip.html" class="external-link">coord_flip</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_bw</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ylab</a></span><span class="op">(</span><span class="st">'Alignment Scores'</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">xlab</a></span><span class="op">(</span><span class="st">''</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/lims.html" class="external-link">ylim</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0.85</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="case_studies_files/figure-html/ADalignment-1.png" alt="Figure 12: Comparison of alignment scores before and after batch effect correction using different methods for the AD data." width="60%"><p class="caption">
Figure 12: Comparison of alignment scores before and after batch effect
correction using different methods for the AD data.
</p>
</div>
<p>The alignment scores complement the PCA results, especially when
batch effect removal is difficult to assess on PCA sample plots. For
example in previous PCA sample plots (<strong>Figure 7</strong>), we
observed that the samples across different batches were better mixed
after batch effect correction with different methods than before,
whereas the performance of difference methods was difficult to compare.
Since a higher alignment score indicates that samples are better mixed,
as shown in the above bar plot, Combat gave a superior performance
compared to the other methods. However, with the <span class="math inline">\(R^2\)</span> values the ComBat corrected data
still included a few variables with a large proportion of batch variance
(see section <strong>other methods</strong> method “<span class="math inline">\(R^2\)</span>”). Therefore, it is important to
compare different techniques and outputs for an unbiased assessment. In
this example, the lower alignment scores of PLSDA-batch and sPLSDA-batch
corrected data might result from the difference in the PCA sample
projections of the batch effect corrected matrices. The data corrected
with removeBatchEffect and ComBat had a large variance in their PCA
projection, while PLSDA-batch and sPLSDA-batch corrected data had a
small variance. A small variance projection results in a small alignment
score, as it is easy to locate the samples from the same batch as
nearest neighbours. In fact, pRDA presented above (<strong>Figure
8</strong>) quantitatively confirmed that both PLSDA-batch and
sPLSDA-batch entirely removed the batch variance <span class="citation">(<a href="#ref-wang2020multivariate" role="doc-biblioref">Y. Wang and Lê Cao 2020</a>)</span>.</p>
</div>
</div>
<div class="section level2">
<h2 id="variable-selection">Variable selection<a class="anchor" aria-label="anchor" href="#variable-selection"></a>
</h2>
<p>We use <code><a href="https://rdrr.io/pkg/mixOmics/man/splsda.html" class="external-link">splsda()</a></code> from <span style="color: #FF7F00;">mixOmics</span> to select the top 50 microbial
variables that, in combination, discriminate the different treatment
groups in the <span style="color: #0000FF;">AD data</span>. We apply
<code><a href="https://rdrr.io/pkg/mixOmics/man/splsda.html" class="external-link">splsda()</a></code> to the different batch effect corrected data from
all methods. Then we use <code>upset()</code> from <span style="color: #FF7F00;">UpSetR</span> package <span class="citation">(<a href="#ref-lex2014upset" role="doc-biblioref">Lex et al.
2014</a>)</span> to visualise the concordance of variables selected.</p>
<p>In the code below, we first need to convert the list of variables
selected from different method-corrected data into a data frame
compatible with <code>upset()</code> using <code>fromList()</code>. We
then assign different colour schemes for each variable selection.</p>
<div class="sourceCode" id="cb92"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ad.splsda.select</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="kw">for</span><span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_len</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">ad.corrected.list</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="va">splsda.res</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/mixOmics/man/splsda.html" class="external-link">splsda</a></span><span class="op">(</span>X <span class="op">=</span> <span class="va">ad.corrected.list</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span>, Y <span class="op">=</span> <span class="va">ad.trt</span>, </span>
<span>                       ncomp <span class="op">=</span> <span class="fl">3</span>, keepX <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="fl">50</span>,<span class="fl">3</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="va">select.res</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/mixOmics/man/selectVar.html" class="external-link">selectVar</a></span><span class="op">(</span><span class="va">splsda.res</span>, comp <span class="op">=</span> <span class="fl">1</span><span class="op">)</span><span class="op">$</span><span class="va">name</span></span>
<span>  <span class="va">ad.splsda.select</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="va">select.res</span></span>
<span><span class="op">}</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">ad.splsda.select</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">ad.corrected.list</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># can only visualise 5 methods</span></span>
<span><span class="va">ad.splsda.select</span> <span class="op">&lt;-</span> <span class="va">ad.splsda.select</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_len</a></span><span class="op">(</span><span class="fl">5</span><span class="op">)</span><span class="op">]</span></span>
<span></span>
<span><span class="va">ad.splsda.upsetR</span> <span class="op">&lt;-</span> <span class="fu">fromList</span><span class="op">(</span><span class="va">ad.splsda.select</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">upset</span><span class="op">(</span><span class="va">ad.splsda.upsetR</span>, main.bar.color <span class="op">=</span> <span class="st">'gray36'</span>,</span>
<span>      sets.bar.color <span class="op">=</span> <span class="fu"><a href="../reference/pb_color.html">pb_color</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">25</span><span class="op">:</span><span class="fl">22</span>,<span class="fl">20</span><span class="op">)</span><span class="op">)</span>, matrix.color <span class="op">=</span> <span class="st">'gray36'</span>,</span>
<span>      order.by <span class="op">=</span> <span class="st">'freq'</span>, empty.intersections <span class="op">=</span> <span class="st">'on'</span>,</span>
<span>      queries <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>query <span class="op">=</span> <span class="va">intersects</span>, </span>
<span>                          params <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="st">'Before correction'</span><span class="op">)</span>, </span>
<span>                          color <span class="op">=</span> <span class="fu"><a href="../reference/pb_color.html">pb_color</a></span><span class="op">(</span><span class="fl">20</span><span class="op">)</span>, active <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>,</span>
<span>                     <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>query <span class="op">=</span> <span class="va">intersects</span>, </span>
<span>                          params <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="st">'removeBatchEffect'</span><span class="op">)</span>, </span>
<span>                          color <span class="op">=</span> <span class="fu"><a href="../reference/pb_color.html">pb_color</a></span><span class="op">(</span><span class="fl">22</span><span class="op">)</span>, active <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>,</span>
<span>                     <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>query <span class="op">=</span> <span class="va">intersects</span>, </span>
<span>                          params <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="st">'ComBat'</span><span class="op">)</span>, </span>
<span>                          color <span class="op">=</span> <span class="fu"><a href="../reference/pb_color.html">pb_color</a></span><span class="op">(</span><span class="fl">23</span><span class="op">)</span>, active <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>,</span>
<span>                     <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>query <span class="op">=</span> <span class="va">intersects</span>, </span>
<span>                          params <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="st">'PLSDA-batch'</span><span class="op">)</span>, </span>
<span>                          color <span class="op">=</span> <span class="fu"><a href="../reference/pb_color.html">pb_color</a></span><span class="op">(</span><span class="fl">24</span><span class="op">)</span>, active <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>,</span>
<span>                     <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>query <span class="op">=</span> <span class="va">intersects</span>, </span>
<span>                          params <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="st">'sPLSDA-batch'</span><span class="op">)</span>, </span>
<span>                          color <span class="op">=</span> <span class="fu"><a href="../reference/pb_color.html">pb_color</a></span><span class="op">(</span><span class="fl">25</span><span class="op">)</span>, active <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="case_studies_files/figure-html/ADupsetR-1.png" alt="Figure 13: UpSet plot showing overlap between variables selected from different corrected data for the AD study." width="100%"><p class="caption">
Figure 13: UpSet plot showing overlap between variables selected from
different corrected data for the AD study.
</p>
</div>
<p>In the above UpSet plot, the left bars indicate the number of
variables selected from each data corrected with different methods. The
right bar plot combined with the scatterplot show different intersection
and their aggregates. We obtained a high overlap of 36 out of 50
selected variables between different corrected and uncorrected data.
However, the data from each method still included unique variables that
were not selected in the other corrected data, e.g., PLSDA-batch and
sPLSDA-batch. As <code>upset()</code> can only include five datasets at
once, we only displayed the uncorrected data and four corrected data
that had been more efficiently corrected for batch effects from our
previous assessments compared to the other datasets.</p>
<p>We then extract each intersection of selected variables between these
five corrected data with <code>venn()</code> function from <span style="color: #FF7F00;">gplots</span>. We can also save the taxonomic
information of these OTUs into a text file for further
interpretation.</p>
<div class="sourceCode" id="cb93"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ad.splsda.select.overlap</span> <span class="op">&lt;-</span> <span class="fu">venn</span><span class="op">(</span><span class="va">ad.splsda.select</span>, show.plot <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="va">ad.inters.splsda</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/attr.html" class="external-link">attr</a></span><span class="op">(</span><span class="va">ad.splsda.select.overlap</span>, <span class="st">'intersections'</span><span class="op">)</span></span>
<span><span class="va">ad.inters.splsda.taxa</span> <span class="op">&lt;-</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">ad.inters.splsda</span>, </span>
<span>         FUN <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">{</span><span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="va">AD_data</span><span class="op">$</span><span class="va">FullData</span><span class="op">$</span><span class="va">taxa</span><span class="op">[</span><span class="va">x</span>, <span class="op">]</span><span class="op">)</span><span class="op">}</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/capture.output.html" class="external-link">capture.output</a></span><span class="op">(</span><span class="va">ad.inters.splsda.taxa</span>, </span>
<span>               file <span class="op">=</span> <span class="st">"GeneratedData/ADselected_50_splsda.txt"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="summary">Summary<a class="anchor" aria-label="anchor" href="#summary"></a>
</h2>
<p>This vignette presents a complete framework for batch effect
management in microbiome data. The first step includes pre-processing,
with pre-filtering to remove low-count samples and variables, and data
transformation to handle compositional data characteristics. For this
step, we advise against using relative abundance data that are
compositional, if possible. Visual tools, such as PCA, boxplots, density
plots and heatmap, and quantitative approach pRDA are then applied to
detect batch effects. When the batch effect is very weak, for example as
informed by a very small proportion of variance explained by batch with
pRDA, or difficult to visualise with PCA, then there may be no need to
manage batch effects. However, when the batch effect is strong, two
solutions are to either account for batch effects, or to correct for
batch effects from the original data. For either solution, the batch x
treatment design matters. If nested, we can only account for batch
effects with a linear mixed model. If unbalanced but not nested, we can
account for batch effects with any linear models or remove them using
weighted PLSDA-batch. Regarding different batch effects, our proposed
method PLSDA-batch is more appropriate for a weak batch effect, while
sPLSDA-batch for a strong batch effect.</p>
<p>We usually assume batch grouping information is known. When the batch
information is unknown, methods such as SVA estimate batch effects from
the variables least affected by treatment effects, while RUV4 and RUVIII
estimate batch effects from negative control variables or/and sample
replicates. For the latter, negative control variables and sample
replicates should capture the complete batch variation, if not, there
may be a risk that batch effects cannot be completely considered or
removed. We emphasised on this point in the <span style="color: #0000FF;">AD</span> and the <span style="color: #808000;">HFHS</span> studies. For these methods that can
estimate batch effects, the estimated batch effects are independent of
treatment effects. If a correlation between batch and treatment exists,
this correlation cannot be estimated. Spurious correlation between batch
and treatment effects may also be present, for example SVA, which
estimates batch effects from the variables that may still include more
or less treatment effects.</p>
<p>In addition, most methods assume systematic batch effects across the
whole dataset. This is the case of ComBat, where we recommend assessing
the model’s validity first.</p>
<p>The next critical step is to assess the efficacy of batch effect
management for a given method. Such assessment is often implicit in
methods that account for batch effects. In contrast, the methods that
remove batch effects can be assessed by comparing the data before and
after correction. All methods we have presented for batch effect
detection can be used for assessing batch effect correction. We can also
calculate the explained variance <span class="math inline">\(R^2\)</span> of each microbial variable for batch
and treatment covariate respectively, as well as alignment scores
measuring the performance of mixing samples across batches. We also have
noted that some methods may not be very sensitive. For example, PCA
plots rely on somewhat subjective evaluation of batch and treatment
variation visualisation, and alignment scores only measure the
performance of mixing samples. Therefore, a reliable conclusion should
be made based on multiple assessment methods.</p>
<p>Once batch effects are removed, downstream analysis such as
multivariate discriminant analysis, or univariate differential analysis
can be performed. Multivariate methods may be more suitable for
microbiome data analysis since microbial variables are naturally
correlated because of mutual interactions.</p>
<p><img src="figures/function_description.png"><strong>Table 2: A
summary of all functions in each section.</strong></p>
</div>
<div class="section level2">
<h2 class="unnumbered" id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-chapleur2016increasing" class="csl-entry">
Chapleur, Olivier, Céline Madigou, Raphaël Civade, Yohan Rodolphe,
Laurent Mazéas, and Théodore Bouchez. 2016. <span>“Increasing
Concentrations of Phenol Progressively Affect Anaerobic Digestion of
Cellulose and Associated Microbial Communities.”</span>
<em>Biodegradation</em> 27 (1): 15–27.
</div>
<div id="ref-duvallet2017meta" class="csl-entry">
Duvallet, Claire, Sean M Gibbons, Thomas Gurry, Rafael A Irizarry, and
Eric J Alm. 2017. <span>“Meta-Analysis of Gut Microbiome Studies
Identifies Disease-Specific and Shared Responses.”</span> <em>Nature
Communications</em> 8 (1): 1–10.
</div>
<div id="ref-fan2020gut" class="csl-entry">
Fan, Yong, and Oluf Pedersen. 2020. <span>“Gut Microbiota in Human
Metabolic Health and Disease.”</span> <em>Nature Reviews
Microbiology</em>, 1–17.
</div>
<div id="ref-gibson2004dietary" class="csl-entry">
Gibson, Glenn R, Hollie M Probert, Jan Van Loo, Robert A Rastall, and
Marcel B Roberfroid. 2004. <span>“Dietary Modulation of the Human
Colonic Microbiota: Updating the Concept of Prebiotics.”</span>
<em>Nutrition Research Reviews</em> 17 (2): 259–75.
</div>
<div id="ref-haro2016intestinal" class="csl-entry">
Haro, Carmen, Oriol A Rangel-Zúñiga, Juan F Alcalá-Dı́az, Francisco
Gómez-Delgado, Pablo Pérez-Martı́nez, Javier Delgado-Lista, Gracia M
Quintana-Navarro, et al. 2016. <span>“Intestinal Microbiota Is
Influenced by Gender and Body Mass Index.”</span> <em>PloS One</em> 11
(5): e0154090.
</div>
<div id="ref-hieken2016microbiome" class="csl-entry">
Hieken, Tina J, Jun Chen, Tanya L Hoskin, Marina Walther-Antonio,
Stephen Johnson, Sheri Ramaker, Jian Xiao, et al. 2016. <span>“The
Microbiome of Aseptically Collected Human Breast Tissue in Benign and
Malignant Disease.”</span> <em>Scientific Reports</em> 6: 30751.
</div>
<div id="ref-kim2017optimizing" class="csl-entry">
Kim, Dorothy, Casey E Hofstaedter, Chunyu Zhao, Lisa Mattei, Ceylan
Tanes, Erik Clarke, Abigail Lauder, et al. 2017. <span>“Optimizing
Methods and Dodging Pitfalls in Microbiome Research.”</span>
<em>Microbiome</em> 5 (1): 52.
</div>
<div id="ref-kong2020microbiome" class="csl-entry">
Kong, Geraldine, Kim-Anh Lê Cao, Louise M Judd, ShanShan Li, Thibault
Renoir, and Anthony J Hannan. 2020. <span>“Microbiome Profiling Reveals
Gut Dysbiosis in a Transgenic Mouse Model of Huntington’s
Disease.”</span> <em>Neurobiology of Disease</em> 135: 104268.
</div>
<div id="ref-daniel2020performance" class="csl-entry">
LÃŒdecke, Daniel, Dominique Makowski, Philip Waggoner, and Indrajeet
Patil. 2020. <span>“Performance: Assessment of Regression Models
Performance.”</span> <em>CRAN</em>. <a href="https://doi.org/10.5281/zenodo.3952174" class="external-link">https://doi.org/10.5281/zenodo.3952174</a>.
</div>
<div id="ref-leek2012sva" class="csl-entry">
Leek, Jeffrey T, W Evan Johnson, Hilary S Parker, Andrew E Jaffe, and
John D Storey. 2012. <span>“The Sva Package for Removing Batch Effects
and Other Unwanted Variation in High-Throughput Experiments.”</span>
<em>Bioinformatics</em> 28 (6): 882–83.
</div>
<div id="ref-lex2014upset" class="csl-entry">
Lex, Alexander, Nils Gehlenborg, Hendrik Strobelt, Romain Vuillemot, and
Hanspeter Pfister. 2014. <span>“UpSet: Visualization of Intersecting
Sets.”</span> <em>IEEE Transactions on Visualization and Computer
Graphics</em> 20 (12): 1983–92.
</div>
<div id="ref-lozupone2013meta" class="csl-entry">
Lozupone, Catherine, Jesse Stombaugh, Antonio Gonzalez, Gail Ackermann,
Doug Wendel, Yoshiki Vázquez-Baeza, Janet K Jansson, Jeffrey I Gordon,
and Rob Knight. 2013. <span>“Meta-Analyses of Studies of the Human
Microbiota.”</span> <em>Genome Research</em>, gr–151803.
</div>
<div id="ref-marchesi2015vocabulary" class="csl-entry">
Marchesi, Julian R, and Jacques Ravel. 2015. <span>“The Vocabulary of
Microbiome Research: A Proposal.”</span> Springer.
</div>
<div id="ref-moskovicz2020skin" class="csl-entry">
Moskovicz, Veronica, Rina Ben-El, Guy Horev, and Boaz Mizrahi. 2020.
<span>“Skin Microbiota Dynamics Following b. Subtilis Formulation
Challenge.”</span>
</div>
<div id="ref-nakagawa2013general" class="csl-entry">
Nakagawa, Shinichi, and Holger Schielzeth. 2013. <span>“A General and
Simple Method for Obtaining R2 from Generalized Linear Mixed-Effects
Models.”</span> <em>Methods in Ecology and Evolution</em> 4 (2): 133–42.
</div>
<div id="ref-poirier2020integrating" class="csl-entry">
Poirier, Simon, Sébastien Déjean, Cédric Midoux, Kim-Anh Lê Cao, and
Olivier Chapleur. 2020. <span>“Integrating Independent Microbial Studies
to Build Predictive Models of Anaerobic Digestion Inhibition by Ammonia
and Phenol.”</span> <em>Bioresource Technology</em> 316: 123952.
</div>
<div id="ref-ray2020microbe" class="csl-entry">
Ray, Prasun, Venkatachalam Lakshmanan, Jessy L Labbé, and Kelly D
Craven. 2020. <span>“Microbe to Microbiome: A Paradigm Shift in the
Application of Microorganisms for Sustainable Agriculture.”</span>
<em>Frontiers in Microbiology</em> 11: 3323.
</div>
<div id="ref-rohart2017mixomics" class="csl-entry">
Rohart, Florian, Benoı̂t Gautier, Amrit Singh, and Kim-Anh Lê Cao. 2017.
<span>“mixOmics: An r Package for ‘Omics Feature Selection and Multiple
Data Integration.”</span> <em>PLoS Computational Biology</em> 13 (11):
e1005752.
</div>
<div id="ref-sacristan2011exploring" class="csl-entry">
Sacristán-Soriano, Oriol, Bernard Banaigs, Emilio O Casamayor, and Mikel
A Becerro. 2011. <span>“Exploring the Links Between Natural Products and
Bacterial Assemblages in the Sponge Aplysina Aerophoba.”</span>
<em>Appl. Environ. Microbiol.</em> 77 (3): 862–70.
</div>
<div id="ref-susin2020variable" class="csl-entry">
Susin, Antoni, Yiwen Wang, Kim-Anh Lê Cao, and M Luz Calle. 2020.
<span>“Variable Selection in Microbiome Compositional Data
Analysis.”</span> <em>NAR Genomics and Bioinformatics</em> 2 (2):
lqaa029.
</div>
<div id="ref-wang2020characterizing" class="csl-entry">
Wang, Charlotte H, Linda Wu, Zengyan Wang, Magdy S Alabady, Daniel
Parson, Zainab Molumo, and Sarah C Fankhauser. 2020.
<span>“Characterizing Changes in Soil Microbiome Abundance and Diversity
Due to Different Cover Crop Techniques.”</span> <em>PloS One</em> 15
(5): e0232453.
</div>
<div id="ref-wang2020multivariate" class="csl-entry">
Wang, Yiwen, and Kim-Anh Lê Cao. 2020. <span>“A Multivariate Method to
Correct for Batch Effects in Microbiome Data.”</span> <em>bioRxiv</em>.
</div>
<div id="ref-wang2020managing" class="csl-entry">
Wang, Yiwen, and Kim-Anh LêCao. 2020. <span>“Managing Batch Effects in
Microbiome Data.”</span> <em>Briefings in Bioinformatics</em> 21 (6):
1954–70.
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Yiwen Wang.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
