<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Supplementary Materials for Batch Effects Management in Case Studies • PLSDAbatch</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Supplementary Materials for Batch Effects Management in Case Studies">
<meta property="og:description" content="PLSDAbatch">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">PLSDAbatch</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.2.3</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/brief_vignette.html">PLSDA-batch Vignette</a>
    </li>
    <li>
      <a href="../articles/case_studies.html">Batch Effects Management in Case Studies</a>
    </li>
    <li>
      <a href="../articles/simulation_study_Gaussian.html">Batch Effects Management in Simulation Studies (Gaussian Distribution)</a>
    </li>
    <li>
      <a href="../articles/simulation_study_negative_binomial.html">Batch Effects Management in Simulation Studies (Negative Binomial Distribution)</a>
    </li>
    <li>
      <a href="../articles/supp_case_studies.html">Supplementary Materials for Batch Effects Management in Case Studies</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Supplementary Materials for Batch Effects
Management in Case Studies</h1>
                        <h4 data-toc-skip class="author">Yiwen (Eva)
Wang</h4>
                        <h4 data-toc-skip class="author">Agricultural
Genomics Institute at Shenzhen, Chinese Academy of Agricultural
Sciences, Shenzhen, China</h4>
                        <h4 data-toc-skip class="author"><a href="mailto:wangyiwen@caas.cn" class="email">wangyiwen@caas.cn</a></h4>
            
            <h4 data-toc-skip class="date">10 November, 2022</h4>
      
      
      <div class="hidden name"><code>supp_case_studies.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="packages-installation-and-loading">Packages installation and loading<a class="anchor" aria-label="anchor" href="#packages-installation-and-loading"></a>
</h2>
<p>First, we load the packages necessary for analysis, and check the
version of each package.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># CRAN</span></span>
<span><span class="va">cran.pkgs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'pheatmap'</span>, <span class="st">'vegan'</span>, <span class="st">'ruv'</span>, <span class="st">'UpSetR'</span>, <span class="st">'gplots'</span>, </span>
<span>               <span class="st">'ggplot2'</span>, <span class="st">'gridExtra'</span>, <span class="st">'performance'</span><span class="op">)</span></span>
<span><span class="co"># Bioconductor</span></span>
<span><span class="va">bioc.pkgs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'mixOmics'</span>, <span class="st">'sva'</span>, <span class="st">'limma'</span>, <span class="st">'Biobase'</span>, <span class="st">'metagenomeSeq'</span><span class="op">)</span></span>
<span><span class="co"># GitHub</span></span>
<span><span class="va">github.pkg</span> <span class="op">&lt;-</span> <span class="st">'PLSDAbatch'</span> </span>
<span><span class="co"># devtools::install_github("https://github.com/EvaYiwenWang/PLSDAbatch")</span></span>
<span></span>
<span><span class="co"># load packages </span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">cran.pkgs</span>, <span class="va">bioc.pkgs</span>, <span class="va">github.pkg</span><span class="op">)</span>, <span class="va">require</span>, character.only <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##      pheatmap         vegan           ruv        UpSetR        gplots </span></span>
<span><span class="co">##          TRUE          TRUE          TRUE          TRUE          TRUE </span></span>
<span><span class="co">##       ggplot2     gridExtra   performance      mixOmics           sva </span></span>
<span><span class="co">##          TRUE          TRUE          TRUE          TRUE          TRUE </span></span>
<span><span class="co">##         limma       Biobase metagenomeSeq    PLSDAbatch </span></span>
<span><span class="co">##          TRUE          TRUE          TRUE          TRUE</span></span></code></pre>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># print package versions</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">cran.pkgs</span>, <span class="va">bioc.pkgs</span>, <span class="va">github.pkg</span><span class="op">)</span>, <span class="va">package.version</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##      pheatmap         vegan           ruv        UpSetR        gplots </span></span>
<span><span class="co">##      "1.0.12"       "2.6-4"     "0.9.7.1"       "1.4.0"       "3.1.3" </span></span>
<span><span class="co">##       ggplot2     gridExtra   performance      mixOmics           sva </span></span>
<span><span class="co">##       "3.3.6"         "2.3"       "0.9.2"      "6.20.0"      "3.44.0" </span></span>
<span><span class="co">##         limma       Biobase metagenomeSeq    PLSDAbatch </span></span>
<span><span class="co">##      "3.52.2"      "2.56.0"      "1.38.0"       "0.2.3"</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="sponge-data">Sponge data<a class="anchor" aria-label="anchor" href="#sponge-data"></a>
</h2>
<div class="section level3">
<h3 id="data-pre-processing">Data pre-processing<a class="anchor" aria-label="anchor" href="#data-pre-processing"></a>
</h3>
<div class="section level4">
<h4 id="pre-filtering">Pre-filtering<a class="anchor" aria-label="anchor" href="#pre-filtering"></a>
</h4>
<p>We load the <span style="color: #964B00;">sponge data</span> stored
internally with function <code><a href="https://rdrr.io/r/utils/data.html" class="external-link">data()</a></code>.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Sponge data</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">'sponge_data'</span><span class="op">)</span> </span>
<span><span class="va">sponge.tss</span> <span class="op">&lt;-</span> <span class="va">sponge_data</span><span class="op">$</span><span class="va">X.tss</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">sponge.tss</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 32 24</span></span></code></pre>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># zero proportion</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/mean.html" class="external-link">mean</a></span><span class="op">(</span><span class="va">sponge.tss</span> <span class="op">==</span> <span class="fl">0</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 0.5455729</span></span></code></pre>
<p>The <span style="color: #964B00;">sponge data</span> include the
relative abundance of 24 OTUs and 32 samples. Given the small number of
OTUs we advise not to pre-filter the data.</p>
</div>
<div class="section level4">
<h4 id="transformation">Transformation<a class="anchor" aria-label="anchor" href="#transformation"></a>
</h4>
<p>Prior to CLR transformation, we recommend adding 0.01 as the offset
for the <span style="color: #964B00;">sponge data</span> - that are
relative abundance data. We use <code><a href="https://rdrr.io/pkg/mixOmics/man/logratio-transformations.html" class="external-link">logratio.transfo()</a></code> function
in <span style="color: #FF7F00;">mixOmics</span> package to CLR
transform the data.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sponge.clr</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/mixOmics/man/logratio-transformations.html" class="external-link">logratio.transfo</a></span><span class="op">(</span>X <span class="op">=</span> <span class="va">sponge.tss</span>, logratio <span class="op">=</span> <span class="st">'CLR'</span>, offset <span class="op">=</span> <span class="fl">0.01</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/class.html" class="external-link">class</a></span><span class="op">(</span><span class="va">sponge.clr</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="st">'matrix'</span></span></code></pre></div>
</div>
</div>
<div class="section level3">
<h3 id="batch-effect-detection">Batch effect detection<a class="anchor" aria-label="anchor" href="#batch-effect-detection"></a>
</h3>
<div class="section level4">
<h4 id="pca">PCA<a class="anchor" aria-label="anchor" href="#pca"></a>
</h4>
<p>We apply <code><a href="https://rdrr.io/pkg/mixOmics/man/pca.html" class="external-link">pca()</a></code> function from <span style="color: #FF7F00;">mixOmics</span> package to the <span style="color: #964B00;">sponge data</span> and
<code><a href="../reference/Scatter_Density.html">Scatter_Density()</a></code> function from <span style="color: #FF7F00;">PLSDAbatch</span> to represent the PCA sample
plot with densities.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sponge.pca.before</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/mixOmics/man/pca.html" class="external-link">pca</a></span><span class="op">(</span><span class="va">sponge.clr</span>, ncomp <span class="op">=</span> <span class="fl">3</span>, scale <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span><span class="va">sponge.batch</span> <span class="op">&lt;-</span> <span class="va">sponge_data</span><span class="op">$</span><span class="va">Y.bat</span></span>
<span><span class="va">sponge.trt</span> <span class="op">&lt;-</span> <span class="va">sponge_data</span><span class="op">$</span><span class="va">Y.trt</span></span>
<span></span>
<span><span class="fu"><a href="../reference/Scatter_Density.html">Scatter_Density</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">sponge.pca.before</span>, </span>
<span>                batch <span class="op">=</span> <span class="va">sponge.batch</span>, </span>
<span>                trt <span class="op">=</span> <span class="va">sponge.trt</span>, title <span class="op">=</span> <span class="st">'Sponge data'</span>, </span>
<span>                trt.legend.title <span class="op">=</span> <span class="st">'Tissue types'</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="supp_case_studies_files/figure-html/SpongepcaBefore-1.png" alt="Figure 1: The PCA sample plot with densities in the sponge data." width="60%"><p class="caption">
Figure 1: The PCA sample plot with densities in the sponge data.
</p>
</div>
<p>In the above figure, the tissue variation (effects of interest)
accounted for 24% of data variance on component 1, while the gel
variation (batch effects) for 21% on component 2. Therefore, the batch
effect is slightly weaker than the treatment effect in the <span style="color: #964B00;">sponge data</span>.</p>
</div>
<div class="section level4">
<h4 id="heatmap">Heatmap<a class="anchor" aria-label="anchor" href="#heatmap"></a>
</h4>
<p>We produce a heatmap using <span style="color: #FF7F00;">pheatmap</span> package. The data first need to
be scaled on both OTUs and samples.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># scale the clr data on both OTUs and samples</span></span>
<span><span class="va">sponge.clr.s</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/scale.html" class="external-link">scale</a></span><span class="op">(</span><span class="va">sponge.clr</span>, center <span class="op">=</span> <span class="cn">T</span>, scale <span class="op">=</span> <span class="cn">T</span><span class="op">)</span></span>
<span><span class="va">sponge.clr.ss</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/scale.html" class="external-link">scale</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/t.html" class="external-link">t</a></span><span class="op">(</span><span class="va">sponge.clr.s</span><span class="op">)</span>, center <span class="op">=</span> <span class="cn">T</span>, scale <span class="op">=</span> <span class="cn">T</span><span class="op">)</span></span>
<span></span>
<span><span class="va">sponge.anno_col</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>Batch <span class="op">=</span> <span class="va">sponge.batch</span>, Tissue <span class="op">=</span> <span class="va">sponge.trt</span><span class="op">)</span></span>
<span><span class="va">sponge.anno_colors</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>Batch <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/mixOmics/man/colors.html" class="external-link">color.mixo</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">2</span><span class="op">)</span>, </span>
<span>                           Tissue <span class="op">=</span> <span class="fu"><a href="../reference/pb_color.html">pb_color</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">2</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">sponge.anno_colors</span><span class="op">$</span><span class="va">Batch</span><span class="op">)</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/levels.html" class="external-link">levels</a></span><span class="op">(</span><span class="va">sponge.batch</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">sponge.anno_colors</span><span class="op">$</span><span class="va">Tissue</span><span class="op">)</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/levels.html" class="external-link">levels</a></span><span class="op">(</span><span class="va">sponge.trt</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">pheatmap</span><span class="op">(</span><span class="va">sponge.clr.ss</span>, </span>
<span>         cluster_rows <span class="op">=</span> <span class="cn">F</span>, </span>
<span>         fontsize_row <span class="op">=</span> <span class="fl">4</span>, </span>
<span>         fontsize_col <span class="op">=</span> <span class="fl">6</span>,</span>
<span>         fontsize <span class="op">=</span> <span class="fl">8</span>,</span>
<span>         clustering_distance_rows <span class="op">=</span> <span class="st">'euclidean'</span>,</span>
<span>         clustering_method <span class="op">=</span> <span class="st">'ward.D'</span>,</span>
<span>         treeheight_row <span class="op">=</span> <span class="fl">30</span>,</span>
<span>         annotation_col <span class="op">=</span> <span class="va">sponge.anno_col</span>,</span>
<span>         annotation_colors <span class="op">=</span> <span class="va">sponge.anno_colors</span>,</span>
<span>         border_color <span class="op">=</span> <span class="st">'NA'</span>,</span>
<span>         main <span class="op">=</span> <span class="st">'Sponge data - Scaled'</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="supp_case_studies_files/figure-html/Spongeheatmap-1.png" alt="Figure 2: Hierarchical clustering for samples in the sponge data." width="90%"><p class="caption">
Figure 2: Hierarchical clustering for samples in the sponge data.
</p>
</div>
<p>In the heatmap, samples in the <span style="color: #964B00;">sponge
data</span> were clustered or grouped by batches instead of tissues,
indicating a batch effect.</p>
</div>
<div class="section level4">
<h4 id="prda">pRDA<a class="anchor" aria-label="anchor" href="#prda"></a>
</h4>
<p>We apply pRDA with <code><a href="https://rdrr.io/pkg/vegan/man/varpart.html" class="external-link">varpart()</a></code> function from <span style="color: #FF7F00;">vegan</span> R package.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sponge.factors.df</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>trt <span class="op">=</span> <span class="va">sponge.trt</span>, </span>
<span>                                batch <span class="op">=</span> <span class="va">sponge.batch</span><span class="op">)</span></span>
<span><span class="va">sponge.rda.before</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/vegan/man/varpart.html" class="external-link">varpart</a></span><span class="op">(</span><span class="va">sponge.clr</span>, <span class="op">~</span> <span class="va">trt</span>, <span class="op">~</span> <span class="va">batch</span>, </span>
<span>                             data <span class="op">=</span> <span class="va">sponge.factors.df</span>, </span>
<span>                             scale <span class="op">=</span> <span class="cn">T</span><span class="op">)</span></span>
<span><span class="va">sponge.rda.before</span><span class="op">$</span><span class="va">part</span><span class="op">$</span><span class="va">indfract</span></span></code></pre></div>
<pre><code><span><span class="co">##                 Df R.squared Adj.R.squared Testable</span></span>
<span><span class="co">## [a] = X1|X2      1        NA    0.16572246     TRUE</span></span>
<span><span class="co">## [b] = X2|X1      1        NA    0.16396277     TRUE</span></span>
<span><span class="co">## [c]              0        NA   -0.01063501    FALSE</span></span>
<span><span class="co">## [d] = Residuals NA        NA    0.68094977    FALSE</span></span></code></pre>
<p>In the result, <code>X1</code> and <code>X2</code> represent the
first and second covariates fitted in the model. <code>[a]</code>,
<code>[b]</code> represent the independent proportion of variance
explained by <code>X1</code> and <code>X2</code> respectively, and
<code>[c]</code> represents the intersection variance shared between
<code>X1</code> and <code>X2</code>. The <span style="color: #964B00;">sponge data</span> have a completely balanced
batch x treatment design, so there was no intersection variance
(indicated in line <code>[c]</code>, Adj.R.squared = -0.01) detected.
The proportion of treatment and batch variance was nearly equal as
indicated in lines <code>[a]</code> and <code>[b]</code>.</p>
</div>
</div>
<div class="section level3">
<h3 id="managing-batch-effects">Managing batch effects<a class="anchor" aria-label="anchor" href="#managing-batch-effects"></a>
</h3>
<div class="section level4">
<h4 id="accounting-for-batch-effects">Accounting for batch effects<a class="anchor" aria-label="anchor" href="#accounting-for-batch-effects"></a>
</h4>
<p>The methods that we use to account for batch effects include the
method designed for microbiome data: zero-inflated Gaussian (ZIG)
mixture model and the methods adapted for microbiome data: linear
regression, SVA and RUV4. Among them, SVA and RUV4 are designed for
unknown batch effects.</p>
<p>As the ZIG model is applicable to counts data, the <span style="color: #964B00;">sponge data</span> are not qualified for this
model. We therefore only apply the methods that can be adapted for
microbiome data.</p>
<p><strong>Linear regression</strong></p>
<p>Linear regression is conducted with <code><a href="../reference/linear_regres.html">linear_regres()</a></code>
function in <span style="color: #FF7F00;">PLSDAbatch</span>. We
integrated the <span style="color: #FF7F00;">performance</span> package
that assesses performance of regression models into our function
<code><a href="../reference/linear_regres.html">linear_regres()</a></code>. Therefore, we can apply
<code>check_model()</code> from <span style="color: #FF7F00;">performance</span> to the outputs from
<code><a href="../reference/linear_regres.html">linear_regres()</a></code> to diagnose the validity of the model
fitted with treatment and batch effects for each variable <span class="citation">(<a href="#ref-daniel2020performance" role="doc-biblioref">LÃŒdecke et al. 2020</a>)</span>. We can extract
performance measurements such as adjusted R2, RMSE, RSE, AIC and BIC for
the models fitted with and without batch effects, which are also the
outputs of <code><a href="../reference/linear_regres.html">linear_regres()</a></code>.</p>
<p>We apply <code>type = "linear model"</code> to the <span style="color: #964B00;">sponge data</span> because of the balanced batch
x treatment design.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sponge.clr</span> <span class="op">&lt;-</span> <span class="va">sponge.clr</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">sponge.clr</span><span class="op">)</span>, <span class="fl">1</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">ncol</a></span><span class="op">(</span><span class="va">sponge.clr</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="va">sponge.lm</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/linear_regres.html">linear_regres</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">sponge.clr</span>, </span>
<span>                           trt <span class="op">=</span> <span class="va">sponge.trt</span>, </span>
<span>                           batch.fix <span class="op">=</span> <span class="va">sponge.batch</span>, </span>
<span>                           type <span class="op">=</span> <span class="st">'linear model'</span><span class="op">)</span></span>
<span></span>
<span><span class="va">sponge.p</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">sponge.lm</span><span class="op">$</span><span class="va">lm.table</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">{</span><span class="va">x</span><span class="op">$</span><span class="va">coefficients</span><span class="op">[</span><span class="fl">2</span>,<span class="fl">4</span><span class="op">]</span><span class="op">}</span><span class="op">)</span></span>
<span><span class="va">sponge.p.adj</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/p.adjust.html" class="external-link">p.adjust</a></span><span class="op">(</span>p <span class="op">=</span> <span class="va">sponge.p</span>, method <span class="op">=</span> <span class="st">'fdr'</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">check_model</span><span class="op">(</span><span class="va">sponge.lm</span><span class="op">$</span><span class="va">model</span><span class="op">$</span><span class="va">OTU2</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="supp_case_studies_files/figure-html/Spongelm-1.png" alt='Figure 3: Diagnostic plots for the model fitted with batch effects of "OTU2" in the sponge data.' width="100%"><p class="caption">
Figure 3: Diagnostic plots for the model fitted with batch effects of
“OTU2” in the sponge data.
</p>
</div>
<p>To diagnose the validity of the model fitted with both treatment and
batch effects, we use different plots to check the assumptions of each
microbial variable. For example, the diagnostic plots of “OTU2” are
shown in the above figure panel. The simulated data under the fitted
model were not very close to the real data (shown as green line),
indicating an imperfect fitness (top panel). The linearity (or
homoscedasticity) and homogeneity of variance were not satisfied. The
correlation between batch (<code>batch.fix</code>) and treatment
(<code>trt</code>) effects was very low, indicating a good model with
low collinearity. Some samples could be classified as outliers with a
Cook’s distance larger than or equal to 0.5, for example, “21”, “15”,
“7”, “1” and “13” (middle panel). The distribution of residuals was not
close to normal (bottom panel). For the microbial variables with some
assumptions not met, we should be careful about their results.</p>
<p>For performance measurements of models fitted with or without batch
effects, we show an example of the results for some variables.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">sponge.lm</span><span class="op">$</span><span class="va">adj.R2</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##         trt.only  trt.batch</span></span>
<span><span class="co">## OTU1  0.06423228 0.05701977</span></span>
<span><span class="co">## OTU2  0.60492094 0.67117014</span></span>
<span><span class="co">## OTU3  0.19003076 0.18125188</span></span>
<span><span class="co">## OTU4 -0.03327738 0.23731780</span></span>
<span><span class="co">## OTU5  0.97483298 0.97643741</span></span>
<span><span class="co">## OTU6  0.97496003 0.97634939</span></span></code></pre>
<p>If the adjusted <span class="math inline">\(R^2\)</span> of the model
with both treatment and batch effects is larger than the model with
treatment effects only, e.g., OTU2, OTU4, OTU5 and OTU6, the model
fitted with batch effects explains more data variance, and is thus
better than the model without batch effects. Otherwise, the batch effect
is not necessary to be added into the linear model.</p>
<p>We next look at the AIC of models fitted with or without batch
effects.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">sponge.lm</span><span class="op">$</span><span class="va">AIC</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##       trt.only  trt.batch</span></span>
<span><span class="co">## OTU1 47.901506 49.0623531</span></span>
<span><span class="co">## OTU2 74.001621 69.0433180</span></span>
<span><span class="co">## OTU3 -5.630456 -4.3703385</span></span>
<span><span class="co">## OTU4 90.699895 81.8982606</span></span>
<span><span class="co">## OTU5 21.727333 20.5345128</span></span>
<span><span class="co">## OTU6  1.196912  0.2853681</span></span></code></pre>
<p>A lower AIC indicates a better fit. Both results of the adjusted
<span class="math inline">\(R^2\)</span> and AIC were consistent on
model selection for the listed OTUs.</p>
<p><strong>SVA</strong></p>
<p>SVA accounts for unknown batch effects. Here we assume that the batch
grouping information in the <span style="color: #964B00;">sponge
data</span> is unknown. We first build two design matrices with
(<code>sponge.mod</code>) and without (<code>sponge.mod0</code>)
treatment grouping information generated with
<code><a href="https://rdrr.io/r/stats/model.matrix.html" class="external-link">model.matrix()</a></code> function from <span style="color: #FF7F00;">stats</span>. We then use <code><a href="https://rdrr.io/pkg/sva/man/num.sv.html" class="external-link">num.sv()</a></code>
from <span style="color: #FF7F00;">sva</span> package to determine the
number of batch variables <code>n.sv</code> that is used to estimate
batch effects in function <code><a href="https://rdrr.io/pkg/sva/man/sva.html" class="external-link">sva()</a></code>.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># estimate batch matrix</span></span>
<span><span class="va">sponge.mod</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/model.matrix.html" class="external-link">model.matrix</a></span><span class="op">(</span> <span class="op">~</span> <span class="va">sponge.trt</span><span class="op">)</span></span>
<span><span class="va">sponge.mod0</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/model.matrix.html" class="external-link">model.matrix</a></span><span class="op">(</span> <span class="op">~</span> <span class="fl">1</span>, data <span class="op">=</span> <span class="va">sponge.trt</span><span class="op">)</span></span>
<span><span class="va">sponge.sva.n</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/sva/man/num.sv.html" class="external-link">num.sv</a></span><span class="op">(</span>dat <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/t.html" class="external-link">t</a></span><span class="op">(</span><span class="va">sponge.clr</span><span class="op">)</span>, mod <span class="op">=</span> <span class="va">sponge.mod</span>, method <span class="op">=</span> <span class="st">'leek'</span><span class="op">)</span></span>
<span><span class="va">sponge.sva</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/sva/man/sva.html" class="external-link">sva</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/t.html" class="external-link">t</a></span><span class="op">(</span><span class="va">sponge.clr</span><span class="op">)</span>, <span class="va">sponge.mod</span>, <span class="va">sponge.mod0</span>, n.sv <span class="op">=</span> <span class="va">sponge.sva.n</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Number of significant surrogate variables is:  1 </span></span>
<span><span class="co">## Iteration (out of 5 ):1  2  3  4  5</span></span></code></pre>
<p>The estimated batch effects are then applied to
<code><a href="https://rdrr.io/pkg/sva/man/f.pvalue.html" class="external-link">f.pvalue()</a></code> to calculate the P-values of treatment effects.
The estimated batch effects in SVA are assumed to be independent of the
treatment effects. However, SVA considers some correlation between batch
and treatment effects <span class="citation">(<a href="#ref-wang2020managing" role="doc-biblioref">Wang and LêCao
2020</a>)</span>.</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># include estimated batch effects in the linear model</span></span>
<span><span class="va">sponge.mod.batch</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">cbind</a></span><span class="op">(</span><span class="va">sponge.mod</span>, <span class="va">sponge.sva</span><span class="op">$</span><span class="va">sv</span><span class="op">)</span></span>
<span><span class="va">sponge.mod0.batch</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">cbind</a></span><span class="op">(</span><span class="va">sponge.mod0</span>, <span class="va">sponge.sva</span><span class="op">$</span><span class="va">sv</span><span class="op">)</span></span>
<span><span class="va">sponge.sva.p</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/sva/man/f.pvalue.html" class="external-link">f.pvalue</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/t.html" class="external-link">t</a></span><span class="op">(</span><span class="va">sponge.clr</span><span class="op">)</span>, <span class="va">sponge.mod.batch</span>, <span class="va">sponge.mod0.batch</span><span class="op">)</span></span>
<span><span class="va">sponge.sva.p.adj</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/p.adjust.html" class="external-link">p.adjust</a></span><span class="op">(</span><span class="va">sponge.sva.p</span>, method <span class="op">=</span> <span class="st">'fdr'</span><span class="op">)</span></span></code></pre></div>
<p><strong>RUV4</strong></p>
<p>Before applying RUV4 (<code>RUV4()</code> from <span style="color: #FF7F00;">ruv</span> package), we need to specify negative
control variables and the number of batch variables to estimate. We can
use the empirical negative controls that are not significantly
differentially abundant (adjusted P &gt; 0.05) from a linear regression
with the treatment information as the only covariate.</p>
<p>We use a loop to fit a linear regression for each microbial variable
and adjust P values of treatment effects for multiple comparisons with
<code><a href="https://rdrr.io/r/stats/p.adjust.html" class="external-link">p.adjust()</a></code> from <span style="color: #FF7F00;">stats</span>.
The empirical negative controls are then extracted according to the
adjusted P values.</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># empirical negative controls</span></span>
<span><span class="va">sponge.empir.p</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="kw">for</span><span class="op">(</span><span class="va">e</span> <span class="kw">in</span> <span class="fl">1</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">ncol</a></span><span class="op">(</span><span class="va">sponge.clr</span><span class="op">)</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="va">sponge.empir.lm</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/lm.html" class="external-link">lm</a></span><span class="op">(</span><span class="va">sponge.clr</span><span class="op">[</span>,<span class="va">e</span><span class="op">]</span> <span class="op">~</span> <span class="va">sponge.trt</span><span class="op">)</span></span>
<span>  <span class="va">sponge.empir.p</span><span class="op">[</span><span class="va">e</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">sponge.empir.lm</span><span class="op">)</span><span class="op">$</span><span class="va">coefficients</span><span class="op">[</span><span class="fl">2</span>,<span class="fl">4</span><span class="op">]</span></span>
<span><span class="op">}</span></span>
<span><span class="va">sponge.empir.p.adj</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/p.adjust.html" class="external-link">p.adjust</a></span><span class="op">(</span>p <span class="op">=</span> <span class="va">sponge.empir.p</span>, method <span class="op">=</span> <span class="st">'fdr'</span><span class="op">)</span></span>
<span><span class="va">sponge.nc</span> <span class="op">&lt;-</span> <span class="va">sponge.empir.p.adj</span> <span class="op">&gt;</span> <span class="fl">0.05</span></span></code></pre></div>
<p>The number of batch variables <code>k</code> can be determined using
<code>getK()</code> function.</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># estimate k</span></span>
<span><span class="va">sponge.k.res</span> <span class="op">&lt;-</span> <span class="fu">getK</span><span class="op">(</span>Y <span class="op">=</span> <span class="va">sponge.clr</span>, X <span class="op">=</span> <span class="va">sponge.trt</span>, ctl <span class="op">=</span> <span class="va">sponge.nc</span><span class="op">)</span></span>
<span><span class="va">sponge.k</span> <span class="op">&lt;-</span> <span class="va">sponge.k.res</span><span class="op">$</span><span class="va">k</span></span>
<span><span class="va">sponge.k</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html" class="external-link">ifelse</a></span><span class="op">(</span><span class="va">sponge.k</span> <span class="op">!=</span> <span class="fl">0</span>, <span class="va">sponge.k</span>, <span class="fl">1</span><span class="op">)</span></span></code></pre></div>
<p>We then apply <code>RUV4()</code> with known treatment variables,
estimated negative control variables and <code>k</code> batch variables.
The calculated P values also need to be adjusted for multiple
comparisons.</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sponge.ruv4</span> <span class="op">&lt;-</span> <span class="fu">RUV4</span><span class="op">(</span>Y <span class="op">=</span> <span class="va">sponge.clr</span>, X <span class="op">=</span> <span class="va">sponge.trt</span>, </span>
<span>                    ctl <span class="op">=</span> <span class="va">sponge.nc</span>, k <span class="op">=</span> <span class="va">sponge.k</span><span class="op">)</span> </span>
<span><span class="va">sponge.ruv4.p</span> <span class="op">&lt;-</span> <span class="va">sponge.ruv4</span><span class="op">$</span><span class="va">p</span></span>
<span><span class="va">sponge.ruv4.p.adj</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/p.adjust.html" class="external-link">p.adjust</a></span><span class="op">(</span><span class="va">sponge.ruv4.p</span>, method <span class="op">=</span> <span class="st">"fdr"</span><span class="op">)</span></span></code></pre></div>
<p>Note: A package named <span style="color: #FF7F00;">RUVSeq</span> has
been developed for count data. It provides <code>RUVg()</code> using
negative control variables, and also other functions <code>RUVs()</code>
and <code>RUVr()</code> using sample replicates <span class="citation">(<a href="#ref-moskovicz2020skin" role="doc-biblioref">Moskovicz et al. 2020</a>)</span> or residuals from
the regression on treatment effects to estimate and then account for
latent batch effects. However, for CLR-transformed data, we still
recommend the standard <span style="color: #FF7F00;">ruv</span>
package.</p>
</div>
<div class="section level4">
<h4 id="correcting-for-batch-effects">Correcting for batch effects<a class="anchor" aria-label="anchor" href="#correcting-for-batch-effects"></a>
</h4>
<p>The methods that we use to correct for batch effects include
removeBatchEffect, ComBat, PLSDA-batch, sPLSDA-batch, Percentile
Normalisation and RUVIII. Among them, RUVIII is designed for unknown
batch effects. Except RUVIII that requires sample replicates which the
<span style="color: #964B00;">sponge data</span> do not have, these
methods were all applied to and illustrated with the <span style="color: #964B00;">sponge data</span>.</p>
<p><strong>removeBatchEffect</strong></p>
<p>The <code><a href="https://rdrr.io/pkg/limma/man/removeBatchEffect.html" class="external-link">removeBatchEffect()</a></code> function is implemented in <span style="color: #FF7F00;">limma</span> package. The design matrix
(<code>design</code>) with treatment grouping information can be
generated with <code><a href="https://rdrr.io/r/stats/model.matrix.html" class="external-link">model.matrix()</a></code> function from <span style="color: #FF7F00;">stats</span> as shown in section
<strong>accounting for batch effects</strong> method “SVA”.</p>
<p>Here we use <code><a href="https://rdrr.io/pkg/limma/man/removeBatchEffect.html" class="external-link">removeBatchEffect()</a></code> function with batch
grouping information (<code>batch</code>) and treatment design matrix
(<code>design</code>) to calculate batch effect corrected data
<code>sponge.rBE</code>.</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sponge.rBE</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/t.html" class="external-link">t</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/limma/man/removeBatchEffect.html" class="external-link">removeBatchEffect</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/t.html" class="external-link">t</a></span><span class="op">(</span><span class="va">sponge.clr</span><span class="op">)</span>, batch <span class="op">=</span> <span class="va">sponge.batch</span>, </span>
<span>                                  design <span class="op">=</span> <span class="va">sponge.mod</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><strong>ComBat</strong></p>
<p>The <code><a href="https://rdrr.io/pkg/sva/man/ComBat.html" class="external-link">ComBat()</a></code> function (from <span style="color: #FF7F00;">sva</span> package) is implemented as parametric
or non-parametric correction with option <code>par.prior</code>. Under a
parametric adjustment, we can assess the model’s validity with
<code>prior.plots = T</code> <span class="citation">(<a href="#ref-leek2012sva" role="doc-biblioref">Leek et al.
2012</a>)</span>.</p>
<p>Here we use a non-parametric correction
(<code>par.prior = FALSE</code>) with batch grouping information
(<code>batch</code>) and treatment design matrix (<code>mod</code>) to
calculate batch effect corrected data <code>sponge.ComBat</code>.</p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sponge.ComBat</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/t.html" class="external-link">t</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/sva/man/ComBat.html" class="external-link">ComBat</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/t.html" class="external-link">t</a></span><span class="op">(</span><span class="va">sponge.clr</span><span class="op">)</span>, batch <span class="op">=</span> <span class="va">sponge.batch</span>, </span>
<span>                          mod <span class="op">=</span> <span class="va">sponge.mod</span>, par.prior <span class="op">=</span> <span class="cn">F</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><strong>PLSDA-batch</strong></p>
<p>The <code><a href="../reference/PLSDA_batch.html">PLSDA_batch()</a></code> function is implemented in <span style="color: #FF7F00;">PLSDAbatch</span> package. To use this function,
we need to specify the optimal number of components related to treatment
(<code>ncomp.trt</code>) or batch effects (<code>ncomp.bat</code>).</p>
<p>Here in the <span style="color: #964B00;">sponge data</span>, we use
<code><a href="https://rdrr.io/pkg/mixOmics/man/plsda.html" class="external-link">plsda()</a></code> from <span style="color: #FF7F00;">mixOmics</span>
with only treatment grouping information to estimate the optimal number
of treatment components to preserve.</p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># estimate the number of treatment components</span></span>
<span><span class="va">sponge.trt.tune</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/mixOmics/man/plsda.html" class="external-link">plsda</a></span><span class="op">(</span>X <span class="op">=</span> <span class="va">sponge.clr</span>, Y <span class="op">=</span> <span class="va">sponge.trt</span>, ncomp <span class="op">=</span> <span class="fl">5</span><span class="op">)</span></span>
<span><span class="va">sponge.trt.tune</span><span class="op">$</span><span class="va">prop_expl_var</span> <span class="co">#1</span></span></code></pre></div>
<pre><code><span><span class="co">## $X</span></span>
<span><span class="co">##      comp1      comp2      comp3      comp4      comp5 </span></span>
<span><span class="co">## 0.22802218 0.10478658 0.16376238 0.06861581 0.04524292 </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $Y</span></span>
<span><span class="co">##      comp1      comp2      comp3      comp4      comp5 </span></span>
<span><span class="co">## 1.00000000 0.16584332 0.06024061 0.03278432 0.01144732</span></span></code></pre>
<p>We choose the number that explains 100% variance in the outcome
matrix <code>Y</code>, thus from the result, 1 component was enough to
preserve the treatment information.</p>
<p>We then use <code><a href="../reference/PLSDA_batch.html">PLSDA_batch()</a></code> function with both treatment
and batch grouping information to estimate the optimal number of batch
components to remove.</p>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># estimate the number of batch components</span></span>
<span><span class="va">sponge.batch.tune</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/PLSDA_batch.html">PLSDA_batch</a></span><span class="op">(</span>X <span class="op">=</span> <span class="va">sponge.clr</span>, </span>
<span>                                 Y.trt <span class="op">=</span> <span class="va">sponge.trt</span>, </span>
<span>                                 Y.bat <span class="op">=</span> <span class="va">sponge.batch</span>,</span>
<span>                                 ncomp.trt <span class="op">=</span> <span class="fl">1</span>, </span>
<span>                                 ncomp.bat <span class="op">=</span> <span class="fl">5</span><span class="op">)</span></span>
<span><span class="va">sponge.batch.tune</span><span class="op">$</span><span class="va">explained_variance.bat</span> <span class="co">#1</span></span></code></pre></div>
<pre><code><span><span class="co">## $X</span></span>
<span><span class="co">##      comp1      comp2      comp3      comp4      comp5 </span></span>
<span><span class="co">## 0.27660372 0.08286158 0.08187289 0.07026693 0.07081271 </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $Y</span></span>
<span><span class="co">## comp1 comp2 comp3 comp4 comp5 </span></span>
<span><span class="co">##     1     1     1     1     1</span></span></code></pre>
<p>Using the same criterion as choosing treatment components, we choose
the number of batch components that explains 100% variance in the
outcome matrix of batch. According to the result, 1 component was
required to remove batch effects.</p>
<p>We then can correct for batch effects applying
<code><a href="../reference/PLSDA_batch.html">PLSDA_batch()</a></code> with treatment, batch grouping information
and corresponding optimal number of related components.</p>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sponge.PLSDA_batch.res</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/PLSDA_batch.html">PLSDA_batch</a></span><span class="op">(</span>X <span class="op">=</span> <span class="va">sponge.clr</span>, </span>
<span>                                      Y.trt <span class="op">=</span> <span class="va">sponge.trt</span>, </span>
<span>                                      Y.bat <span class="op">=</span> <span class="va">sponge.batch</span>,</span>
<span>                                      ncomp.trt <span class="op">=</span> <span class="fl">1</span>, </span>
<span>                                      ncomp.bat <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span>
<span><span class="va">sponge.PLSDA_batch</span> <span class="op">&lt;-</span> <span class="va">sponge.PLSDA_batch.res</span><span class="op">$</span><span class="va">X.nobatch</span></span></code></pre></div>
<p><strong>sPLSDA-batch</strong></p>
<p>We apply sPLSDA-batch using the same function
<code><a href="../reference/PLSDA_batch.html">PLSDA_batch()</a></code> but we specify the number of variables to
select on each component (usually only treatment-related components
<code>keepX.trt</code>). To determine the optimal number of variables to
select, we use <code><a href="https://rdrr.io/pkg/mixOmics/man/tune.splsda.html" class="external-link">tune.splsda()</a></code> function from <span style="color: #FF7F00;">mixOmics</span> package <span class="citation">(<a href="#ref-rohart2017mixomics" role="doc-biblioref">Rohart et al. 2017</a>)</span> with all possible
numbers of variables to select for each component
(<code>test.keepX</code>).</p>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># estimate the number of variables to select per treatment component</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">777</span><span class="op">)</span></span>
<span><span class="va">sponge.test.keepX</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">24</span>, <span class="fl">1</span><span class="op">)</span></span>
<span><span class="va">sponge.trt.tune.v</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/mixOmics/man/tune.splsda.html" class="external-link">tune.splsda</a></span><span class="op">(</span>X <span class="op">=</span> <span class="va">sponge.clr</span>, </span>
<span>                                 Y <span class="op">=</span> <span class="va">sponge.trt</span>, </span>
<span>                                 ncomp <span class="op">=</span> <span class="fl">1</span>, </span>
<span>                                 test.keepX <span class="op">=</span> <span class="va">sponge.test.keepX</span>, </span>
<span>                                 validation <span class="op">=</span> <span class="st">'Mfold'</span>, </span>
<span>                                 folds <span class="op">=</span> <span class="fl">4</span>, nrepeat <span class="op">=</span> <span class="fl">50</span><span class="op">)</span></span>
<span><span class="va">sponge.trt.tune.v</span><span class="op">$</span><span class="va">choice.keepX</span> <span class="co">#1</span></span></code></pre></div>
<pre><code><span><span class="co">## comp1 </span></span>
<span><span class="co">##     1</span></span></code></pre>
<p>Here the optimal number of variables to select for the treatment
component was 1. Since we have adjusted the amount of treatment
variation to preserve, we need to re-choose the optimal number of
components related to batch effects using the same criterion mentioned
in section <strong>correcting for batch effects</strong> method
“PLSDA-batch”.</p>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># estimate the number of batch components</span></span>
<span><span class="va">sponge.batch.tune</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/PLSDA_batch.html">PLSDA_batch</a></span><span class="op">(</span>X <span class="op">=</span> <span class="va">sponge.clr</span>, </span>
<span>                                 Y.trt <span class="op">=</span> <span class="va">sponge.trt</span>, </span>
<span>                                 Y.bat <span class="op">=</span> <span class="va">sponge.batch</span>,</span>
<span>                                 ncomp.trt <span class="op">=</span> <span class="fl">1</span>, </span>
<span>                                 keepX.trt <span class="op">=</span> <span class="fl">1</span>,</span>
<span>                                 ncomp.bat <span class="op">=</span> <span class="fl">5</span><span class="op">)</span></span>
<span><span class="va">sponge.batch.tune</span><span class="op">$</span><span class="va">explained_variance.bat</span> <span class="co">#1</span></span></code></pre></div>
<pre><code><span><span class="co">## $X</span></span>
<span><span class="co">##      comp1      comp2      comp3      comp4      comp5 </span></span>
<span><span class="co">## 0.25672979 0.07716171 0.08936628 0.08434746 0.09422519 </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $Y</span></span>
<span><span class="co">## comp1 comp2 comp3 comp4 comp5 </span></span>
<span><span class="co">##     1     1     1     1     1</span></span></code></pre>
<p>According to the result, we needed only 1 batch related component to
remove batch variance from the data with function
<code><a href="../reference/PLSDA_batch.html">PLSDA_batch()</a></code>.</p>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sponge.sPLSDA_batch.res</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/PLSDA_batch.html">PLSDA_batch</a></span><span class="op">(</span>X <span class="op">=</span> <span class="va">sponge.clr</span>, </span>
<span>                                       Y.trt <span class="op">=</span> <span class="va">sponge.trt</span>, </span>
<span>                                       Y.bat <span class="op">=</span> <span class="va">sponge.batch</span>,</span>
<span>                                       ncomp.trt <span class="op">=</span> <span class="fl">1</span>, </span>
<span>                                       keepX.trt <span class="op">=</span> <span class="fl">1</span>,</span>
<span>                                       ncomp.bat <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span>
<span><span class="va">sponge.sPLSDA_batch</span> <span class="op">&lt;-</span> <span class="va">sponge.sPLSDA_batch.res</span><span class="op">$</span><span class="va">X.nobatch</span></span></code></pre></div>
<p>Note: for unbalanced batch x treatment design (with the exception of
the nested design), we can specify <code>balance = FALSE</code> in
<code><a href="../reference/PLSDA_batch.html">PLSDA_batch()</a></code> function to apply weighted PLSDA-batch.</p>
<p><strong>Percentile Normalisation</strong></p>
<p>To apply <code><a href="../reference/percentile_norm.html">percentile_norm()</a></code> function from <span style="color: #FF7F00;">PLSDAbatch</span> package, we need to indicate a
control group (<code>ctrl.grp</code>).</p>
<div class="sourceCode" id="cb37"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sponge.PN</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/percentile_norm.html">percentile_norm</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">sponge.clr</span>, </span>
<span>                             batch <span class="op">=</span> <span class="va">sponge.batch</span>, </span>
<span>                             trt <span class="op">=</span> <span class="va">sponge.trt</span>, </span>
<span>                             ctrl.grp <span class="op">=</span> <span class="st">'C'</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level3">
<h3 id="assessing-batch-effect-correction">Assessing batch effect correction<a class="anchor" aria-label="anchor" href="#assessing-batch-effect-correction"></a>
</h3>
<p>We apply different visualisation and quantitative methods to
assessing batch effect correction.</p>
<div class="section level4">
<h4 id="methods-that-detect-batch-effects">Methods that detect batch effects<a class="anchor" aria-label="anchor" href="#methods-that-detect-batch-effects"></a>
</h4>
<p><strong>PCA</strong></p>
<p>In the <span style="color: #964B00;">sponge data</span>, we compared
the PCA sample plots before and after batch effect correction with
different methods.</p>
<div class="sourceCode" id="cb38"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sponge.pca.before</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/mixOmics/man/pca.html" class="external-link">pca</a></span><span class="op">(</span><span class="va">sponge.clr</span>, ncomp <span class="op">=</span> <span class="fl">3</span>, </span>
<span>                         scale <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">sponge.pca.rBE</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/mixOmics/man/pca.html" class="external-link">pca</a></span><span class="op">(</span><span class="va">sponge.rBE</span>, ncomp <span class="op">=</span> <span class="fl">3</span>, </span>
<span>                      scale <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">sponge.pca.ComBat</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/mixOmics/man/pca.html" class="external-link">pca</a></span><span class="op">(</span><span class="va">sponge.ComBat</span>, ncomp <span class="op">=</span> <span class="fl">3</span>, </span>
<span>                         scale <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">sponge.pca.PLSDA_batch</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/mixOmics/man/pca.html" class="external-link">pca</a></span><span class="op">(</span><span class="va">sponge.PLSDA_batch</span>, ncomp <span class="op">=</span> <span class="fl">3</span>, </span>
<span>                              scale <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">sponge.pca.sPLSDA_batch</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/mixOmics/man/pca.html" class="external-link">pca</a></span><span class="op">(</span><span class="va">sponge.sPLSDA_batch</span>, ncomp <span class="op">=</span> <span class="fl">3</span>, </span>
<span>                               scale <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">sponge.pca.PN</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/mixOmics/man/pca.html" class="external-link">pca</a></span><span class="op">(</span><span class="va">sponge.PN</span>, ncomp <span class="op">=</span> <span class="fl">3</span>, </span>
<span>                     scale <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb39"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sponge.pca.before.plot</span> <span class="op">&lt;-</span> </span>
<span>  <span class="fu"><a href="../reference/Scatter_Density.html">Scatter_Density</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">sponge.pca.before</span>, </span>
<span>                  batch <span class="op">=</span> <span class="va">sponge.batch</span>, </span>
<span>                  trt <span class="op">=</span> <span class="va">sponge.trt</span>, </span>
<span>                  title <span class="op">=</span> <span class="st">'Before'</span>, </span>
<span>                  trt.legend.title <span class="op">=</span> <span class="st">'Tissue'</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb40"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sponge.pca.rBE.plot</span> <span class="op">&lt;-</span> </span>
<span>  <span class="fu"><a href="../reference/Scatter_Density.html">Scatter_Density</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">sponge.pca.rBE</span>, </span>
<span>                  batch <span class="op">=</span> <span class="va">sponge.batch</span>, </span>
<span>                  trt <span class="op">=</span> <span class="va">sponge.trt</span>, </span>
<span>                  title <span class="op">=</span> <span class="st">'removeBatchEffect'</span>, </span>
<span>                  trt.legend.title <span class="op">=</span> <span class="st">'Tissue'</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb41"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sponge.pca.ComBat.plot</span> <span class="op">&lt;-</span> </span>
<span>  <span class="fu"><a href="../reference/Scatter_Density.html">Scatter_Density</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">sponge.pca.ComBat</span>, </span>
<span>                  batch <span class="op">=</span> <span class="va">sponge.batch</span>, </span>
<span>                  trt <span class="op">=</span> <span class="va">sponge.trt</span>, </span>
<span>                  title <span class="op">=</span> <span class="st">'ComBat'</span>, </span>
<span>                  trt.legend.title <span class="op">=</span> <span class="st">'Tissue'</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb42"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sponge.pca.PLSDA_batch.plot</span> <span class="op">&lt;-</span> </span>
<span>  <span class="fu"><a href="../reference/Scatter_Density.html">Scatter_Density</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">sponge.pca.PLSDA_batch</span>, </span>
<span>                  batch <span class="op">=</span> <span class="va">sponge.batch</span>, </span>
<span>                  trt <span class="op">=</span> <span class="va">sponge.trt</span>, </span>
<span>                  title <span class="op">=</span> <span class="st">'PLSDA-batch'</span>, </span>
<span>                  trt.legend.title <span class="op">=</span> <span class="st">'Tissue'</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb43"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sponge.pca.sPLSDA_batch.plot</span> <span class="op">&lt;-</span> </span>
<span>  <span class="fu"><a href="../reference/Scatter_Density.html">Scatter_Density</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">sponge.pca.sPLSDA_batch</span>, </span>
<span>                  batch <span class="op">=</span> <span class="va">sponge.batch</span>, </span>
<span>                  trt <span class="op">=</span> <span class="va">sponge.trt</span>, </span>
<span>                  title <span class="op">=</span> <span class="st">'sPLSDA-batch'</span>, </span>
<span>                  trt.legend.title <span class="op">=</span> <span class="st">'Tissue'</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb44"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sponge.pca.PN.plot</span> <span class="op">&lt;-</span> </span>
<span>  <span class="fu"><a href="../reference/Scatter_Density.html">Scatter_Density</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">sponge.pca.PN</span>, </span>
<span>                  batch <span class="op">=</span> <span class="va">sponge.batch</span>, </span>
<span>                  trt <span class="op">=</span> <span class="va">sponge.trt</span>, </span>
<span>                  title <span class="op">=</span> <span class="st">'Percentile Normalisation'</span>, </span>
<span>                  trt.legend.title <span class="op">=</span> <span class="st">'Tissue'</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="supp_case_studies_files/figure-html/Spongepca-1.png" alt="Figure 4: The PCA sample plots with densities before and after batch effect correction in the sponge data." width="100%"><p class="caption">
Figure 4: The PCA sample plots with densities before and after batch
effect correction in the sponge data.
</p>
</div>
<p>As shown in the PCA sample plots, the difference between the samples
from two different batches was removed after batch effect correction
with most methods except percentile normalisation. In the data corrected
with percentile normalisation, the samples from tissue E still included
a distinction between samples from two batches. We can also compare the
boxplots and density plots for key variables identified in PCA driving
the major variance or heatmaps showing obvious patterns before and after
batch effect correction (results not shown).</p>
<p><strong>pRDA</strong></p>
<p>We calculate the global explained variance across all microbial
variables using pRDA. To achieve this, we create a loop for each
variable from the original (uncorrected) and batch effect-corrected
data. The final results are then displayed with
<code><a href="../reference/partVar_plot.html">partVar_plot()</a></code> from <span style="color: #FF7F00;">PLSDAbatch</span> package.</p>
<div class="sourceCode" id="cb45"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sponge.corrected.list</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>`Before correction` <span class="op">=</span> <span class="va">sponge.clr</span>, </span>
<span>                              removeBatchEffect <span class="op">=</span> <span class="va">sponge.rBE</span>, </span>
<span>                              ComBat <span class="op">=</span> <span class="va">sponge.ComBat</span>, </span>
<span>                              `PLSDA-batch` <span class="op">=</span> <span class="va">sponge.PLSDA_batch</span>, </span>
<span>                              `sPLSDA-batch` <span class="op">=</span> <span class="va">sponge.sPLSDA_batch</span>, </span>
<span>                              `Percentile Normalisation` <span class="op">=</span> <span class="va">sponge.PN</span><span class="op">)</span></span>
<span></span>
<span><span class="va">sponge.prop.df</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>Treatment <span class="op">=</span> <span class="cn">NA</span>, Batch <span class="op">=</span> <span class="cn">NA</span>, </span>
<span>                             Intersection <span class="op">=</span> <span class="cn">NA</span>, </span>
<span>                             Residuals <span class="op">=</span> <span class="cn">NA</span><span class="op">)</span> </span>
<span><span class="kw">for</span><span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="fl">1</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">sponge.corrected.list</span><span class="op">)</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="va">rda.res</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/vegan/man/varpart.html" class="external-link">varpart</a></span><span class="op">(</span><span class="va">sponge.corrected.list</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span>, <span class="op">~</span> <span class="va">trt</span>, <span class="op">~</span> <span class="va">batch</span>,</span>
<span>                    data <span class="op">=</span> <span class="va">sponge.factors.df</span>, scale <span class="op">=</span> <span class="cn">T</span><span class="op">)</span></span>
<span>  <span class="va">sponge.prop.df</span><span class="op">[</span><span class="va">i</span>, <span class="op">]</span> <span class="op">&lt;-</span> <span class="va">rda.res</span><span class="op">$</span><span class="va">part</span><span class="op">$</span><span class="va">indfract</span><span class="op">$</span><span class="va">Adj.R.squared</span><span class="op">}</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">sponge.prop.df</span><span class="op">)</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">sponge.corrected.list</span><span class="op">)</span></span>
<span></span>
<span><span class="va">sponge.prop.df</span> <span class="op">&lt;-</span> <span class="va">sponge.prop.df</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">3</span>,<span class="fl">2</span>,<span class="fl">4</span><span class="op">)</span><span class="op">]</span></span>
<span></span>
<span><span class="va">sponge.prop.df</span><span class="op">[</span><span class="va">sponge.prop.df</span> <span class="op">&lt;</span> <span class="fl">0</span><span class="op">]</span> <span class="op">=</span> <span class="fl">0</span></span>
<span><span class="va">sponge.prop.df</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/t.html" class="external-link">t</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/apply.html" class="external-link">apply</a></span><span class="op">(</span><span class="va">sponge.prop.df</span>, <span class="fl">1</span>, </span>
<span>                                        <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">{</span><span class="va">x</span><span class="op">/</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">}</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span></span>
<span><span class="fu"><a href="../reference/partVar_plot.html">partVar_plot</a></span><span class="op">(</span>prop.df <span class="op">=</span> <span class="va">sponge.prop.df</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="supp_case_studies_files/figure-html/Spongeprda-1.png" alt="Figure 5: Global explained variance before and after batch effect correction for the sponge data." width="60%"><p class="caption">
Figure 5: Global explained variance before and after batch effect
correction for the sponge data.
</p>
</div>
<p>As shown in the above figure, the intersection between batch and
treatment variance was none for the <span style="color: #964B00;">sponge
data</span>, because the batch x treatment design is balanced.
PLSDA-batch and sPLSDA-batch corrected data led to the best performance
with a slightly higher proportion of explained treatment variance and no
batch variance compared to the other methods.</p>
</div>
<div class="section level4">
<h4 id="other-methods">Other methods<a class="anchor" aria-label="anchor" href="#other-methods"></a>
</h4>
<p><span class="math inline">\(\mathbf{R^2}\)</span></p>
<p>The <span class="math inline">\(R^2\)</span> values for each variable
are calculated with <code><a href="https://rdrr.io/r/stats/lm.html" class="external-link">lm()</a></code> from <span style="color: #FF7F00;">stats</span> package. To compare the <span class="math inline">\(R^2\)</span> values among variables, we scale the
corrected data before <span class="math inline">\(R^2\)</span>
calculation. The results are displayed with <code><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot()</a></code> from
<span style="color: #FF7F00;">ggplot2</span> R package.</p>
<div class="sourceCode" id="cb46"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># scale</span></span>
<span><span class="va">sponge.corr_scale.list</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">sponge.corrected.list</span>, </span>
<span>                                 <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">{</span><span class="fu"><a href="https://rdrr.io/r/base/apply.html" class="external-link">apply</a></span><span class="op">(</span><span class="va">x</span>, <span class="fl">2</span>, <span class="va">scale</span><span class="op">)</span><span class="op">}</span><span class="op">)</span></span>
<span></span>
<span><span class="va">sponge.r_values.list</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="kw">for</span><span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="fl">1</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">sponge.corr_scale.list</span><span class="op">)</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="va">sponge.r_values</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>trt <span class="op">=</span> <span class="cn">NA</span>, batch <span class="op">=</span> <span class="cn">NA</span><span class="op">)</span></span>
<span>  <span class="kw">for</span><span class="op">(</span><span class="va">c</span> <span class="kw">in</span> <span class="fl">1</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">ncol</a></span><span class="op">(</span><span class="va">sponge.corr_scale.list</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span><span class="op">)</span><span class="op">)</span><span class="op">{</span></span>
<span>    <span class="va">sponge.fit.res.trt</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/lm.html" class="external-link">lm</a></span><span class="op">(</span><span class="va">sponge.corr_scale.list</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span><span class="op">[</span>,<span class="va">c</span><span class="op">]</span> <span class="op">~</span> <span class="va">sponge.trt</span><span class="op">)</span></span>
<span>    <span class="va">sponge.r_values</span><span class="op">[</span><span class="va">c</span>,<span class="fl">1</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">sponge.fit.res.trt</span><span class="op">)</span><span class="op">$</span><span class="va">r.squared</span></span>
<span>    <span class="va">sponge.fit.res.batch</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/lm.html" class="external-link">lm</a></span><span class="op">(</span><span class="va">sponge.corr_scale.list</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span><span class="op">[</span>,<span class="va">c</span><span class="op">]</span> <span class="op">~</span> <span class="va">sponge.batch</span><span class="op">)</span></span>
<span>    <span class="va">sponge.r_values</span><span class="op">[</span><span class="va">c</span>,<span class="fl">2</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">sponge.fit.res.batch</span><span class="op">)</span><span class="op">$</span><span class="va">r.squared</span></span>
<span>  <span class="op">}</span></span>
<span>  <span class="va">sponge.r_values.list</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="va">sponge.r_values</span></span>
<span><span class="op">}</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">sponge.r_values.list</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">sponge.corr_scale.list</span><span class="op">)</span></span>
<span></span>
<span><span class="va">sponge.boxp.list</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="kw">for</span><span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_len</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">sponge.r_values.list</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="va">sponge.boxp.list</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> </span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>r2 <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">sponge.r_values.list</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span><span class="op">[</span> ,<span class="st">'trt'</span><span class="op">]</span>,</span>
<span>                      <span class="va">sponge.r_values.list</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span><span class="op">[</span> ,<span class="st">'batch'</span><span class="op">]</span><span class="op">)</span>, </span>
<span>               Effects <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">as.factor</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'Treatment'</span>,<span class="st">'Batch'</span><span class="op">)</span>, </span>
<span>                                       each <span class="op">=</span> <span class="fl">24</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">sponge.boxp.list</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">sponge.r_values.list</span><span class="op">)</span></span>
<span></span>
<span><span class="va">sponge.r2.boxp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">rbind</a></span><span class="op">(</span><span class="va">sponge.boxp.list</span><span class="op">$</span><span class="va">`Before correction`</span>,</span>
<span>                        <span class="va">sponge.boxp.list</span><span class="op">$</span><span class="va">removeBatchEffect</span>,</span>
<span>                        <span class="va">sponge.boxp.list</span><span class="op">$</span><span class="va">ComBat</span>,</span>
<span>                        <span class="va">sponge.boxp.list</span><span class="op">$</span><span class="va">`PLSDA-batch`</span>,</span>
<span>                        <span class="va">sponge.boxp.list</span><span class="op">$</span><span class="va">`sPLSDA-batch`</span>,</span>
<span>                        <span class="va">sponge.boxp.list</span><span class="op">$</span><span class="va">`Percentile Normalisation`</span>,</span>
<span>                        <span class="va">sponge.boxp.list</span><span class="op">$</span><span class="va">RUVIII</span><span class="op">)</span></span>
<span></span>
<span><span class="va">sponge.r2.boxp</span><span class="op">$</span><span class="va">methods</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'Before correction'</span>, <span class="st">' removeBatchEffect'</span>, </span>
<span>                                <span class="st">'ComBat'</span>,<span class="st">'PLSDA-batch'</span>, <span class="st">'sPLSDA-batch'</span>,</span>
<span>                                <span class="st">'Percentile Normalisation'</span><span class="op">)</span>, each <span class="op">=</span> <span class="fl">48</span><span class="op">)</span></span>
<span></span>
<span><span class="va">sponge.r2.boxp</span><span class="op">$</span><span class="va">methods</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="va">sponge.r2.boxp</span><span class="op">$</span><span class="va">methods</span>, </span>
<span>                                 levels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">sponge.r2.boxp</span><span class="op">$</span><span class="va">methods</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">sponge.r2.boxp</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">Effects</span>, y <span class="op">=</span> <span class="va">r2</span>, fill <span class="op">=</span> <span class="va">Effects</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_boxplot.html" class="external-link">geom_boxplot</a></span><span class="op">(</span>alpha <span class="op">=</span> <span class="fl">0.80</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_bw</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>text <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">18</span><span class="op">)</span>,</span>
<span>        axis.title.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>        axis.title.y <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>        axis.text.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">60</span>, hjust <span class="op">=</span> <span class="fl">1</span>, size <span class="op">=</span> <span class="fl">18</span><span class="op">)</span>,</span>
<span>        axis.text.y <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">18</span><span class="op">)</span>,</span>
<span>        panel.grid.minor.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>        panel.grid.major.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>        legend.position <span class="op">=</span> <span class="st">"right"</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_grid.html" class="external-link">facet_grid</a></span><span class="op">(</span> <span class="op">~</span> <span class="va">methods</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html" class="external-link">scale_fill_manual</a></span><span class="op">(</span>values<span class="op">=</span><span class="fu"><a href="../reference/pb_color.html">pb_color</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">12</span>,<span class="fl">14</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> </span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="supp_case_studies_files/figure-html/Sponger2_1-1.png" alt="Figure 6: Sponge study: $R^2$ values for each microbial variable before and after batch effect correction." width="100%"><p class="caption">
Figure 6: Sponge study: <span class="math inline">\(R^2\)</span> values
for each microbial variable before and after batch effect correction.
</p>
</div>
<p>We observed that the data corrected by ComBat, percentile
normalisation still included a few variables with a large proportion of
batch variance as shown in the above figures.</p>
<div class="sourceCode" id="cb47"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">##################################</span></span>
<span><span class="va">sponge.barp.list</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="kw">for</span><span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_len</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">sponge.r_values.list</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="va">sponge.barp.list</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> </span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>r2 <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="va">sponge.r_values.list</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span><span class="op">[</span> ,<span class="st">'trt'</span><span class="op">]</span><span class="op">)</span>,</span>
<span>                      <span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="va">sponge.r_values.list</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span><span class="op">[</span> ,<span class="st">'batch'</span><span class="op">]</span><span class="op">)</span><span class="op">)</span>, </span>
<span>               Effects <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'Treatment'</span>,<span class="st">'Batch'</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">sponge.barp.list</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">sponge.r_values.list</span><span class="op">)</span></span>
<span></span>
<span><span class="va">sponge.r2.barp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">rbind</a></span><span class="op">(</span><span class="va">sponge.barp.list</span><span class="op">$</span><span class="va">`Before correction`</span>,</span>
<span>                        <span class="va">sponge.barp.list</span><span class="op">$</span><span class="va">removeBatchEffect</span>,</span>
<span>                        <span class="va">sponge.barp.list</span><span class="op">$</span><span class="va">ComBat</span>,</span>
<span>                        <span class="va">sponge.barp.list</span><span class="op">$</span><span class="va">`PLSDA-batch`</span>,</span>
<span>                        <span class="va">sponge.barp.list</span><span class="op">$</span><span class="va">`sPLSDA-batch`</span>,</span>
<span>                        <span class="va">sponge.barp.list</span><span class="op">$</span><span class="va">`Percentile Normalisation`</span>,</span>
<span>                        <span class="va">sponge.barp.list</span><span class="op">$</span><span class="va">RUVIII</span><span class="op">)</span></span>
<span></span>
<span></span>
<span><span class="va">sponge.r2.barp</span><span class="op">$</span><span class="va">methods</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'Before correction'</span>, <span class="st">' removeBatchEffect'</span>, </span>
<span>                                <span class="st">'ComBat'</span>,<span class="st">'PLSDA-batch'</span>, <span class="st">'sPLSDA-batch'</span>,</span>
<span>                                <span class="st">'Percentile Normalisation'</span><span class="op">)</span>, each <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span>
<span></span>
<span><span class="va">sponge.r2.barp</span><span class="op">$</span><span class="va">methods</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="va">sponge.r2.barp</span><span class="op">$</span><span class="va">methods</span>, </span>
<span>                                 levels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">sponge.r2.barp</span><span class="op">$</span><span class="va">methods</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span></span>
<span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">sponge.r2.barp</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">Effects</span>, y <span class="op">=</span> <span class="va">r2</span>, fill <span class="op">=</span> <span class="va">Effects</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_bar.html" class="external-link">geom_bar</a></span><span class="op">(</span>stat<span class="op">=</span><span class="st">"identity"</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_bw</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>text <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">18</span><span class="op">)</span>,</span>
<span>        axis.title.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>        axis.title.y <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>        axis.text.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">60</span>, hjust <span class="op">=</span> <span class="fl">1</span>, size <span class="op">=</span> <span class="fl">18</span><span class="op">)</span>,</span>
<span>        axis.text.y <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">18</span><span class="op">)</span>,</span>
<span>        panel.grid.minor.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>        panel.grid.major.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>        legend.position <span class="op">=</span> <span class="st">"right"</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_grid.html" class="external-link">facet_grid</a></span><span class="op">(</span> <span class="op">~</span> <span class="va">methods</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html" class="external-link">scale_fill_manual</a></span><span class="op">(</span>values<span class="op">=</span><span class="fu"><a href="../reference/pb_color.html">pb_color</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">12</span>,<span class="fl">14</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="supp_case_studies_files/figure-html/Sponger2_2-1.png" alt="Figure 7: Sponge study: the sum of $R^2$ values for each microbial variable before and after batch effect correction." width="100%"><p class="caption">
Figure 7: Sponge study: the sum of <span class="math inline">\(R^2\)</span> values for each microbial variable
before and after batch effect correction.
</p>
</div>
<p>When considering the sum of all variables, the remaining batch
variance of corrected data from sPLSDA-batch was similar as PLSDA-batch,
which was greater than removeBatchEffect. The preserved treatment
variance of corrected data from sPLSDA-batch was also similar as
PLSDA-batch which was greater than removeBatchEffect.</p>
<p><strong>Alignment scores</strong></p>
<p>To use the <code><a href="../reference/alignment_score.html">alignment_score()</a></code> function from <span style="color: #FF7F00;">PLSDAbatch</span>, we need to specify the
proportion of data variance to explain (<code>var</code>), the number of
nearest neighbours (<code>k</code>) and the number of principal
components to estimate (<code>ncomp</code>). We then use
<code><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot()</a></code> function from <span style="color: #FF7F00;">ggplot2</span> to visualise the results.</p>
<div class="sourceCode" id="cb48"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sponge.var</span> <span class="op">=</span> <span class="fl">0.95</span></span>
<span><span class="va">sponge.k</span> <span class="op">=</span> <span class="fl">3</span></span>
<span></span>
<span><span class="va">sponge.scores</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="kw">for</span><span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="fl">1</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">sponge.corrected.list</span><span class="op">)</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="va">res</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/alignment_score.html">alignment_score</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">sponge.corrected.list</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span>, </span>
<span>                         batch <span class="op">=</span> <span class="va">sponge.batch</span>, </span>
<span>                         var <span class="op">=</span> <span class="va">sponge.var</span>, </span>
<span>                         k <span class="op">=</span> <span class="va">sponge.k</span>, </span>
<span>                         ncomp <span class="op">=</span> <span class="fl">20</span><span class="op">)</span></span>
<span>  <span class="va">sponge.scores</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">sponge.scores</span>, <span class="va">res</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="va">sponge.scores.df</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>scores <span class="op">=</span> <span class="va">sponge.scores</span>, </span>
<span>                               methods <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">sponge.corrected.list</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">sponge.scores.df</span><span class="op">$</span><span class="va">methods</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="va">sponge.scores.df</span><span class="op">$</span><span class="va">methods</span>, </span>
<span>                                   levels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/rev.html" class="external-link">rev</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">sponge.corrected.list</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span></span>
<span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_bar.html" class="external-link">geom_col</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">sponge.scores.df</span><span class="op">$</span><span class="va">methods</span>, </span>
<span>                        y <span class="op">=</span> <span class="va">sponge.scores.df</span><span class="op">$</span><span class="va">scores</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_text.html" class="external-link">geom_text</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">sponge.scores.df</span><span class="op">$</span><span class="va">methods</span>, </span>
<span>                y <span class="op">=</span> <span class="va">sponge.scores.df</span><span class="op">$</span><span class="va">scores</span><span class="op">/</span><span class="fl">2</span>, </span>
<span>                label <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Round.html" class="external-link">round</a></span><span class="op">(</span><span class="va">sponge.scores.df</span><span class="op">$</span><span class="va">scores</span>, <span class="fl">3</span><span class="op">)</span><span class="op">)</span>, </span>
<span>            size <span class="op">=</span> <span class="fl">3</span>, col <span class="op">=</span> <span class="st">'white'</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/coord_flip.html" class="external-link">coord_flip</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_bw</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ylab</a></span><span class="op">(</span><span class="st">'Alignment Scores'</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">xlab</a></span><span class="op">(</span><span class="st">''</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/lims.html" class="external-link">ylim</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0.4</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="supp_case_studies_files/figure-html/Spongealignment-1.png" alt="Figure 8: Comparison of alignment scores before and after batch effect correction using different methods for the sponge data." width="60%"><p class="caption">
Figure 8: Comparison of alignment scores before and after batch effect
correction using different methods for the sponge data.
</p>
</div>
<p>The alignment scores complement the PCA results, especially when
batch effect removal is difficult to assess on PCA sample plots. Since a
higher alignment score indicates that samples are better mixed, as shown
in the above figure, sPLSDA-batch gave a superior performance compared
to the other methods.</p>
<p><strong>Variable selection</strong></p>
<p>We use <code><a href="https://rdrr.io/pkg/mixOmics/man/splsda.html" class="external-link">splsda()</a></code> from <span style="color: #FF7F00;">mixOmics</span> to select the top 5 microbial
variables that, in combination, discriminate the different treatment
groups in the <span style="color: #964B00;">sponge data</span>. We apply
<code><a href="https://rdrr.io/pkg/mixOmics/man/splsda.html" class="external-link">splsda()</a></code> to the different batch effect corrected data from
all methods. Then we use <code>upset()</code> from <span style="color: #FF7F00;">UpSetR</span> package <span class="citation">(<a href="#ref-lex2014upset" role="doc-biblioref">Lex et al.
2014</a>)</span> to visualise the concordance of variables selected.</p>
<p>In the code below, we first need to convert the list of variables
selected from different method-corrected data into a data frame
compatible with <code>upset()</code> using <code>fromList()</code>. We
then assign different colour schemes for each variable selection.</p>
<div class="sourceCode" id="cb49"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sponge.splsda.select</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="kw">for</span><span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="fl">1</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">sponge.corrected.list</span><span class="op">)</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="va">splsda.res</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/mixOmics/man/splsda.html" class="external-link">splsda</a></span><span class="op">(</span>X <span class="op">=</span> <span class="va">sponge.corrected.list</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span>, </span>
<span>                       Y <span class="op">=</span> <span class="va">sponge.trt</span>, </span>
<span>                       ncomp <span class="op">=</span> <span class="fl">3</span>, keepX <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="fl">5</span>,<span class="fl">3</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="va">select.res</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/mixOmics/man/selectVar.html" class="external-link">selectVar</a></span><span class="op">(</span><span class="va">splsda.res</span>, comp <span class="op">=</span> <span class="fl">1</span><span class="op">)</span><span class="op">$</span><span class="va">name</span></span>
<span>  <span class="va">sponge.splsda.select</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="va">select.res</span></span>
<span><span class="op">}</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">sponge.splsda.select</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">sponge.corrected.list</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># can only visualise 5 methods</span></span>
<span><span class="va">sponge.splsda.select</span> <span class="op">&lt;-</span> <span class="va">sponge.splsda.select</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">)</span><span class="op">]</span></span>
<span></span>
<span><span class="va">sponge.splsda.upsetR</span> <span class="op">&lt;-</span> <span class="fu">fromList</span><span class="op">(</span><span class="va">sponge.splsda.select</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">upset</span><span class="op">(</span><span class="va">sponge.splsda.upsetR</span>, main.bar.color <span class="op">=</span> <span class="st">'gray36'</span>,</span>
<span>      sets.bar.color <span class="op">=</span> <span class="fu"><a href="../reference/pb_color.html">pb_color</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">25</span><span class="op">:</span><span class="fl">22</span>,<span class="fl">20</span><span class="op">)</span><span class="op">)</span>, matrix.color <span class="op">=</span> <span class="st">'gray36'</span>,</span>
<span>      order.by <span class="op">=</span> <span class="st">'freq'</span>, empty.intersections <span class="op">=</span> <span class="st">'on'</span>,</span>
<span>      queries <span class="op">=</span> </span>
<span>        <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>query <span class="op">=</span> <span class="va">intersects</span>, params <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="st">'Before correction'</span><span class="op">)</span>, </span>
<span>                  color <span class="op">=</span> <span class="fu"><a href="../reference/pb_color.html">pb_color</a></span><span class="op">(</span><span class="fl">20</span><span class="op">)</span>, active <span class="op">=</span> <span class="cn">T</span><span class="op">)</span>,</span>
<span>             <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>query <span class="op">=</span> <span class="va">intersects</span>, params <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="st">'removeBatchEffect'</span><span class="op">)</span>, </span>
<span>                  color <span class="op">=</span> <span class="fu"><a href="../reference/pb_color.html">pb_color</a></span><span class="op">(</span><span class="fl">22</span><span class="op">)</span>, active <span class="op">=</span> <span class="cn">T</span><span class="op">)</span>,</span>
<span>             <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>query <span class="op">=</span> <span class="va">intersects</span>, params <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="st">'ComBat'</span><span class="op">)</span>, </span>
<span>                  color <span class="op">=</span> <span class="fu"><a href="../reference/pb_color.html">pb_color</a></span><span class="op">(</span><span class="fl">23</span><span class="op">)</span>, active <span class="op">=</span> <span class="cn">T</span><span class="op">)</span>,</span>
<span>             <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>query <span class="op">=</span> <span class="va">intersects</span>, params <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="st">'PLSDA-batch'</span><span class="op">)</span>, </span>
<span>                  color <span class="op">=</span> <span class="fu"><a href="../reference/pb_color.html">pb_color</a></span><span class="op">(</span><span class="fl">24</span><span class="op">)</span>, active <span class="op">=</span> <span class="cn">T</span><span class="op">)</span>,</span>
<span>             <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>query <span class="op">=</span> <span class="va">intersects</span>, params <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="st">'sPLSDA-batch'</span><span class="op">)</span>, </span>
<span>                  color <span class="op">=</span> <span class="fu"><a href="../reference/pb_color.html">pb_color</a></span><span class="op">(</span><span class="fl">25</span><span class="op">)</span>, active <span class="op">=</span> <span class="cn">T</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="supp_case_studies_files/figure-html/SpongeupsetR-1.png" alt="Figure 9: UpSet plot showing overlap between variables selected from different corrected data for the sponge study." width="100%"><p class="caption">
Figure 9: UpSet plot showing overlap between variables selected from
different corrected data for the sponge study.
</p>
</div>
<p>In the above figure, the left bars indicate the number of variables
selected from each data corrected with different methods. The right bar
plot combined with the scatterplot show different intersection and their
aggregates. We obtained an overlap of 4 out of 5 selected variables
between different corrected and uncorrected data. However, the data from
each method still included unique variables that were not selected in
the other corrected data, e.g., ComBat and PLSDA-batch. As
<code>upset()</code> can only include five datasets at once, we only
displayed the uncorrected data and four corrected data that had been
more efficiently corrected for batch effects from our previous
assessments compared to the other datasets.</p>
</div>
</div>
</div>
<div class="section level2">
<h2 id="hfhs-data">HFHS data<a class="anchor" aria-label="anchor" href="#hfhs-data"></a>
</h2>
<div class="section level3">
<h3 id="data-pre-processing-1">Data pre-processing<a class="anchor" aria-label="anchor" href="#data-pre-processing-1"></a>
</h3>
<div class="section level4">
<h4 id="prefiltering">Prefiltering<a class="anchor" aria-label="anchor" href="#prefiltering"></a>
</h4>
<p>We load the <span style="color: #808000;">HFHS data</span> stored
internally with function <code><a href="https://rdrr.io/r/utils/data.html" class="external-link">data()</a></code>.</p>
<div class="sourceCode" id="cb50"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">'HFHS_data'</span><span class="op">)</span> </span>
<span><span class="va">hfhs.count</span> <span class="op">&lt;-</span> <span class="va">HFHS_data</span><span class="op">$</span><span class="va">FullData</span><span class="op">$</span><span class="va">X.count</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">hfhs.count</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1]  250 4524</span></span></code></pre>
<div class="sourceCode" id="cb52"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hfhs.filter.res</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/PreFL.html">PreFL</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">hfhs.count</span><span class="op">)</span></span>
<span><span class="va">hfhs.filter</span> <span class="op">&lt;-</span> <span class="va">hfhs.filter.res</span><span class="op">$</span><span class="va">data.filter</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">hfhs.filter</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 250 515</span></span></code></pre>
<div class="sourceCode" id="cb54"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># zero proportion before filtering</span></span>
<span><span class="va">hfhs.filter.res</span><span class="op">$</span><span class="va">zero.prob</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 0.8490805</span></span></code></pre>
<div class="sourceCode" id="cb56"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># zero proportion after filtering</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="va">hfhs.filter</span> <span class="op">==</span> <span class="fl">0</span><span class="op">)</span><span class="op">/</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">hfhs.filter</span><span class="op">)</span> <span class="op">*</span> <span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">ncol</a></span><span class="op">(</span><span class="va">hfhs.filter</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 0.2955184</span></span></code></pre>
<p>The raw <span style="color: #808000;">HFHS data</span> include 4524
OTUs and 250 samples. We use the function <code><a href="../reference/PreFL.html">PreFL()</a></code> from our
<span style="color: #FF7F00;">PLSDAbatch</span> R package to filter the
data. After filtering, the <span style="color: #808000;">HFHS
data</span> were reduced to 515 OTUs and 250 samples. The proportion of
zeroes was reduced from 85% to 30%.</p>
</div>
<div class="section level4">
<h4 id="transformation-1">Transformation<a class="anchor" aria-label="anchor" href="#transformation-1"></a>
</h4>
<p>Prior to CLR transformation, we recommend adding 1 as the offset for
the <span style="color: #808000;">HFHS data</span> - that are raw count
data. We use <code><a href="https://rdrr.io/pkg/mixOmics/man/logratio-transformations.html" class="external-link">logratio.transfo()</a></code> function in <span style="color: #FF7F00;">mixOmics</span> package to CLR transform the
data.</p>
<div class="sourceCode" id="cb58"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hfhs.clr</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/mixOmics/man/logratio-transformations.html" class="external-link">logratio.transfo</a></span><span class="op">(</span>X <span class="op">=</span> <span class="va">hfhs.filter</span>, logratio <span class="op">=</span> <span class="st">'CLR'</span>, offset <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/class.html" class="external-link">class</a></span><span class="op">(</span><span class="va">hfhs.clr</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="st">'matrix'</span></span></code></pre></div>
</div>
</div>
<div class="section level3">
<h3 id="batch-effect-detection-1">Batch effect detection<a class="anchor" aria-label="anchor" href="#batch-effect-detection-1"></a>
</h3>
<div class="section level4">
<h4 id="pca-1">PCA<a class="anchor" aria-label="anchor" href="#pca-1"></a>
</h4>
<p>We apply <code><a href="https://rdrr.io/pkg/mixOmics/man/pca.html" class="external-link">pca()</a></code> function from <span style="color: #FF7F00;">mixOmics</span> package to the <span style="color: #808000;">HFHS data</span> and use
<code><a href="../reference/Scatter_Density.html">Scatter_Density()</a></code> function from <span style="color: #FF7F00;">PLSDAbatch</span> to represent the PCA sample
plot with densities.</p>
<div class="sourceCode" id="cb59"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hfhs.pca.before</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/mixOmics/man/pca.html" class="external-link">pca</a></span><span class="op">(</span><span class="va">hfhs.clr</span>, ncomp <span class="op">=</span> <span class="fl">3</span>, scale <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span><span class="va">hfhs.metadata</span> <span class="op">&lt;-</span> <span class="va">HFHS_data</span><span class="op">$</span><span class="va">FullData</span><span class="op">$</span><span class="va">metadata</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">hfhs.metadata</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">hfhs.metadata</span><span class="op">$</span><span class="va">SampleID</span></span>
<span><span class="va">hfhs.metadata</span> <span class="op">&lt;-</span> <span class="va">hfhs.metadata</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">hfhs.clr</span><span class="op">)</span>,<span class="op">]</span></span>
<span></span>
<span><span class="va">hfhs.cage</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">as.factor</a></span><span class="op">(</span><span class="va">hfhs.metadata</span><span class="op">$</span><span class="va">DietCage</span><span class="op">)</span></span>
<span><span class="va">hfhs.day</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">as.factor</a></span><span class="op">(</span><span class="va">hfhs.metadata</span><span class="op">$</span><span class="va">Day</span><span class="op">)</span></span>
<span><span class="va">hfhs.trt</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">as.factor</a></span><span class="op">(</span><span class="va">hfhs.metadata</span><span class="op">$</span><span class="va">Diet</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">hfhs.cage</span><span class="op">)</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">hfhs.day</span><span class="op">)</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">hfhs.trt</span><span class="op">)</span> <span class="op">=</span> <span class="va">hfhs.metadata</span><span class="op">$</span><span class="va">SampleID</span></span>
<span></span>
<span><span class="va">hfhs.pca.before.cage.plot</span> <span class="op">&lt;-</span> </span>
<span>  <span class="fu"><a href="../reference/Scatter_Density.html">Scatter_Density</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">hfhs.pca.before</span>, </span>
<span>                  batch <span class="op">=</span> <span class="va">hfhs.cage</span>, </span>
<span>                  trt <span class="op">=</span> <span class="va">hfhs.trt</span>, </span>
<span>                  title <span class="op">=</span> <span class="st">'HFHS data'</span>, </span>
<span>                  trt.legend.title <span class="op">=</span> <span class="st">'Diet'</span>,</span>
<span>                  batch.legend.title <span class="op">=</span> <span class="st">'Cage'</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="supp_case_studies_files/figure-html/HFHSpcaBefore1-1.png" alt="Figure 10: The PCA sample plot with densities coloured by cages in the HFHS data." width="60%"><p class="caption">
Figure 10: The PCA sample plot with densities coloured by cages in the
HFHS data.
</p>
</div>
<p>In the above figure, part of the samples with different diets were
mixed together, while all the samples from different batches (i.e.,
cages) were well mixed and not distinct. This result indicates no cage
effect for the whole data and no treatment effect for part of
samples.</p>
<div class="sourceCode" id="cb60"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hfhs.pca.before.day.plot</span> <span class="op">&lt;-</span> </span>
<span>  <span class="fu"><a href="../reference/Scatter_Density.html">Scatter_Density</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">hfhs.pca.before</span>, </span>
<span>                  batch <span class="op">=</span> <span class="va">hfhs.day</span>, </span>
<span>                  trt <span class="op">=</span> <span class="va">hfhs.trt</span>, </span>
<span>                  title <span class="op">=</span> <span class="st">'HFHS data'</span>, </span>
<span>                  trt.legend.title <span class="op">=</span> <span class="st">'Diet'</span>,</span>
<span>                  batch.legend.title <span class="op">=</span> <span class="st">'Day'</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="supp_case_studies_files/figure-html/HFHSpcaBefore2-1.png" alt="Figure 11: The PCA sample plot with densities coloured by days in the HFHS data." width="60%"><p class="caption">
Figure 11: The PCA sample plot with densities coloured by days in the
HFHS data.
</p>
</div>
<p>As shown in the above figure, we did not observe a difference between
samples with different diets on arrival day “A” and “Day0”. This result
was consistent with the experimental design, as mice were not treated
with different diets on these two days. We thus removed the samples from
these two days.</p>
<div class="sourceCode" id="cb61"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># remove samples from arrival day and day0</span></span>
<span><span class="va">hfhs.dA0.idx</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/which.html" class="external-link">which</a></span><span class="op">(</span><span class="va">hfhs.day</span> <span class="op">==</span> <span class="st">'A'</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/which.html" class="external-link">which</a></span><span class="op">(</span><span class="va">hfhs.day</span> <span class="op">==</span> <span class="st">'Day0'</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">hfhs.clr.less</span> <span class="op">&lt;-</span> <span class="va">hfhs.clr</span><span class="op">[</span><span class="op">-</span><span class="va">hfhs.dA0.idx</span>,<span class="op">]</span></span>
<span><span class="va">hfhs.metadata.less</span> <span class="op">&lt;-</span> <span class="va">hfhs.metadata</span><span class="op">[</span><span class="op">-</span><span class="va">hfhs.dA0.idx</span>,<span class="op">]</span></span>
<span></span>
<span><span class="va">hfhs.cage.less</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">as.factor</a></span><span class="op">(</span><span class="va">hfhs.metadata.less</span><span class="op">$</span><span class="va">DietCage</span><span class="op">)</span></span>
<span><span class="va">hfhs.day.less</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">as.factor</a></span><span class="op">(</span><span class="va">hfhs.metadata.less</span><span class="op">$</span><span class="va">Day</span><span class="op">)</span></span>
<span><span class="va">hfhs.trt.less</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">as.factor</a></span><span class="op">(</span><span class="va">hfhs.metadata.less</span><span class="op">$</span><span class="va">Diet</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">hfhs.cage.less</span><span class="op">)</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">hfhs.day.less</span><span class="op">)</span> <span class="op">=</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">hfhs.trt.less</span><span class="op">)</span> <span class="op">=</span> <span class="va">hfhs.metadata.less</span><span class="op">$</span><span class="va">SampleID</span></span>
<span></span>
<span><span class="va">hfhs.less.pca.before</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/mixOmics/man/pca.html" class="external-link">pca</a></span><span class="op">(</span><span class="va">hfhs.clr.less</span>, ncomp <span class="op">=</span> <span class="fl">3</span>, scale <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span><span class="va">hfhs.less.pca.before.cage.plot</span> <span class="op">&lt;-</span> </span>
<span>  <span class="fu"><a href="../reference/Scatter_Density.html">Scatter_Density</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">hfhs.less.pca.before</span>, </span>
<span>                  batch <span class="op">=</span> <span class="va">hfhs.cage.less</span>, </span>
<span>                  trt <span class="op">=</span> <span class="va">hfhs.trt.less</span>, </span>
<span>                  title <span class="op">=</span> <span class="st">'HFHS data'</span>, </span>
<span>                  trt.legend.title <span class="op">=</span> <span class="st">'Diet'</span>,</span>
<span>                  batch.legend.title <span class="op">=</span> <span class="st">'Cage'</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="supp_case_studies_files/figure-html/HFHSpcaBeforeNew1-1.png" alt="Figure 12: The PCA sample plot with densities coloured by cages in the reduced HFHS data." width="60%"><p class="caption">
Figure 12: The PCA sample plot with densities coloured by cages in the
reduced HFHS data.
</p>
</div>
<div class="sourceCode" id="cb62"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hfhs.less.pca.before.day.plot</span> <span class="op">&lt;-</span> </span>
<span>  <span class="fu"><a href="../reference/Scatter_Density.html">Scatter_Density</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">hfhs.less.pca.before</span>, </span>
<span>                  batch <span class="op">=</span> <span class="va">hfhs.day.less</span>, </span>
<span>                  trt <span class="op">=</span> <span class="va">hfhs.trt.less</span>, </span>
<span>                  title <span class="op">=</span> <span class="st">'HFHS data'</span>, </span>
<span>                  trt.legend.title <span class="op">=</span> <span class="st">'Diet'</span>,</span>
<span>                  batch.legend.title <span class="op">=</span> <span class="st">'Day'</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="supp_case_studies_files/figure-html/HFHSpcaBeforeNew2-1.png" alt="Figure 13: The PCA sample plot with densities coloured by days in the reduced HFHS data." width="60%"><p class="caption">
Figure 13: The PCA sample plot with densities coloured by days in the
reduced HFHS data.
</p>
</div>
<p>We observed a substantial diet variation on component 1, no cage
variation, and a difference between samples collected on “Day1” and the
other days with the HFHS diet as shown in the above figures.</p>
</div>
<div class="section level4">
<h4 id="heatmap-1">Heatmap<a class="anchor" aria-label="anchor" href="#heatmap-1"></a>
</h4>
<p>We produce a heatmap using <span style="color: #FF7F00;">pheatmap</span> package. The data first need to
be scaled on both OTUs and samples.</p>
<div class="sourceCode" id="cb63"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># scale the clr data on both OTUs and samples</span></span>
<span><span class="va">hfhs.clr.s</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/scale.html" class="external-link">scale</a></span><span class="op">(</span><span class="va">hfhs.clr.less</span>, center <span class="op">=</span> <span class="cn">T</span>, scale <span class="op">=</span> <span class="cn">T</span><span class="op">)</span></span>
<span><span class="va">hfhs.clr.ss</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/scale.html" class="external-link">scale</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/t.html" class="external-link">t</a></span><span class="op">(</span><span class="va">hfhs.clr.s</span><span class="op">)</span>, center <span class="op">=</span> <span class="cn">T</span>, scale <span class="op">=</span> <span class="cn">T</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># The cage effect</span></span>
<span><span class="va">hfhs.anno_col</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>Cage <span class="op">=</span> <span class="va">hfhs.cage.less</span>, </span>
<span>                            Day <span class="op">=</span> <span class="va">hfhs.day.less</span>, </span>
<span>                            Treatment <span class="op">=</span> <span class="va">hfhs.trt.less</span><span class="op">)</span></span>
<span><span class="va">hfhs.anno_colors</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>Cage <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/mixOmics/man/colors.html" class="external-link">color.mixo</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">8</span><span class="op">)</span>, </span>
<span>                         Day <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/mixOmics/man/colors.html" class="external-link">color.mixo</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">3</span><span class="op">)</span>,</span>
<span>                         Treatment <span class="op">=</span> <span class="fu"><a href="../reference/pb_color.html">pb_color</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">2</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">hfhs.anno_colors</span><span class="op">$</span><span class="va">Cage</span><span class="op">)</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/levels.html" class="external-link">levels</a></span><span class="op">(</span><span class="va">hfhs.cage.less</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">hfhs.anno_colors</span><span class="op">$</span><span class="va">Day</span><span class="op">)</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/levels.html" class="external-link">levels</a></span><span class="op">(</span><span class="va">hfhs.day.less</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">hfhs.anno_colors</span><span class="op">$</span><span class="va">Treatment</span><span class="op">)</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/levels.html" class="external-link">levels</a></span><span class="op">(</span><span class="va">hfhs.trt.less</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">pheatmap</span><span class="op">(</span><span class="va">hfhs.clr.ss</span>, </span>
<span>         cluster_rows <span class="op">=</span> <span class="cn">F</span>, </span>
<span>         fontsize_row <span class="op">=</span> <span class="fl">4</span>, </span>
<span>         fontsize_col <span class="op">=</span> <span class="fl">6</span>,</span>
<span>         fontsize <span class="op">=</span> <span class="fl">8</span>,</span>
<span>         clustering_distance_rows <span class="op">=</span> <span class="st">'euclidean'</span>,</span>
<span>         clustering_method <span class="op">=</span> <span class="st">'ward.D'</span>,</span>
<span>         treeheight_row <span class="op">=</span> <span class="fl">30</span>,</span>
<span>         annotation_col <span class="op">=</span> <span class="va">hfhs.anno_col</span>,</span>
<span>         annotation_colors <span class="op">=</span> <span class="va">hfhs.anno_colors</span>,</span>
<span>         border_color <span class="op">=</span> <span class="st">'NA'</span>,</span>
<span>         main <span class="op">=</span> <span class="st">'HFHS data - Scaled'</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="supp_case_studies_files/figure-html/HFHSheatmapS-1.png" alt="Figure 14: Hierarchical clustering for samples in the HFHS data." width="90%"><p class="caption">
Figure 14: Hierarchical clustering for samples in the HFHS data.
</p>
</div>
<p>In the above figure, samples in the <span style="color: #808000;">HFHS data</span> were clustered according to the
treatments not cages, indicating no cage effect and a strong treatment
effect. Samples from the batch “Day1” were clustered and distinct from
the other batches with diet HFHS but not the normal diet, indicating a
day effect but very weak.</p>
</div>
<div class="section level4">
<h4 id="prda-1">pRDA<a class="anchor" aria-label="anchor" href="#prda-1"></a>
</h4>
<p>We apply pRDA with <code><a href="https://rdrr.io/pkg/vegan/man/varpart.html" class="external-link">varpart()</a></code> function from <span style="color: #FF7F00;">vegan</span> R package.</p>
<div class="sourceCode" id="cb64"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hfhs.factors.df</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>trt <span class="op">=</span> <span class="va">hfhs.trt.less</span>, </span>
<span>                              cage <span class="op">=</span> <span class="va">hfhs.cage.less</span>, </span>
<span>                              day <span class="op">=</span> <span class="va">hfhs.day.less</span><span class="op">)</span></span>
<span><span class="va">hfhs.rda.cage.before</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/vegan/man/varpart.html" class="external-link">varpart</a></span><span class="op">(</span><span class="va">hfhs.clr.less</span>, <span class="op">~</span> <span class="va">trt</span>, <span class="op">~</span> <span class="va">cage</span>, </span>
<span>                                data <span class="op">=</span> <span class="va">hfhs.factors.df</span>, scale <span class="op">=</span> <span class="cn">T</span><span class="op">)</span></span>
<span><span class="va">hfhs.rda.cage.before</span><span class="op">$</span><span class="va">part</span><span class="op">$</span><span class="va">indfract</span></span></code></pre></div>
<pre><code><span><span class="co">##                 Df R.squared Adj.R.squared Testable</span></span>
<span><span class="co">## [a] = X1|X2      0        NA  7.771561e-16    FALSE</span></span>
<span><span class="co">## [b] = X2|X1      6        NA  1.277627e-02     TRUE</span></span>
<span><span class="co">## [c]              0        NA  2.713111e-01    FALSE</span></span>
<span><span class="co">## [d] = Residuals NA        NA  7.159126e-01    FALSE</span></span></code></pre>
<p>In the result, <code>X1</code> and <code>X2</code> represent the
first and second covariates fitted in the model. <code>[a]</code>,
<code>[b]</code> represent the independent proportion of variance
explained by <code>X1</code> and <code>X2</code> respectively, and
<code>[c]</code> represents the intersection variance shared between
<code>X1</code> and <code>X2</code>. In the <span style="color: #808000;">HFHS data</span>, the cage and diet effects have
a nested batch x treatment design that is extremely unbalanced. We
didn’t detect any treatment variance (indicated in line
<code>[a]</code>, Adj.R.squared = 0) but considerable intersection
variance (indicated in line <code>[c]</code>, Adj.R.squared = 0.271).
Thus, all the treatment variance was explained by the intersection
variance shared with batch variance. It means batch and treatment
effects are collinear. Therefore, the cage effect in the <span style="color: #808000;">HFHS data</span> can only be accounted for with
a linear mixed model, as detailed in section <strong>accounting for
batch effects</strong> method “Linear regression”.</p>
<div class="sourceCode" id="cb66"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hfhs.rda.day.before</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/vegan/man/varpart.html" class="external-link">varpart</a></span><span class="op">(</span><span class="va">hfhs.clr.less</span>, <span class="op">~</span> <span class="va">trt</span>, <span class="op">~</span> <span class="va">day</span>, </span>
<span>                               data <span class="op">=</span> <span class="va">hfhs.factors.df</span>, scale <span class="op">=</span> <span class="cn">T</span><span class="op">)</span></span>
<span><span class="va">hfhs.rda.day.before</span><span class="op">$</span><span class="va">part</span><span class="op">$</span><span class="va">indfract</span></span></code></pre></div>
<pre><code><span><span class="co">##                 Df R.squared Adj.R.squared Testable</span></span>
<span><span class="co">## [a] = X1|X2      1        NA  0.2721810304     TRUE</span></span>
<span><span class="co">## [b] = X2|X1      2        NA  0.0361767632     TRUE</span></span>
<span><span class="co">## [c]              0        NA -0.0008699296    FALSE</span></span>
<span><span class="co">## [d] = Residuals NA        NA  0.6925121361    FALSE</span></span></code></pre>
<p>For the day and diet effects, the batch x treatment design is
balanced. There was no intersection variance (indicated in line
<code>[c]</code>, Adj.R.squared = 0). The proportion of treatment
variance was much higher than batch variance as indicated in lines
<code>[a]</code> (Adj.R.squared = 0.272) and <code>[b]</code>
(Adj.R.squared = 0.036).</p>
</div>
</div>
<div class="section level3">
<h3 id="managing-batch-effects-1">Managing batch effects<a class="anchor" aria-label="anchor" href="#managing-batch-effects-1"></a>
</h3>
<div class="section level4">
<h4 id="accounting-for-batch-effects-1">Accounting for batch effects<a class="anchor" aria-label="anchor" href="#accounting-for-batch-effects-1"></a>
</h4>
<p>The methods that we use to account for batch effects include the
methods designed for microbiome data: zero-inflated Gaussian (ZIG)
mixture model and the methods adapted for microbiome data: linear
regression, SVA and RUV4. Among them, SVA and RUV4 are designed for
unknown batch effects.</p>
<p><strong>Methods designed for microbiome data</strong></p>
<p><strong>Zero-inflated Gaussian mixture model</strong> To use the ZIG
model, we first create a <code>MRexperiment</code> object applying
<code>newMRexperiment()</code> (from <span style="color: #FF7F00;">metagenomeSeq</span> package) to microbiome
counts and annotated data frames with metadata and taxonomic information
generated with <code>AnnotatedDataFrame()</code> from <span style="color: #FF7F00;">Biobase</span> package.</p>
<div class="sourceCode" id="cb68"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Creating a MRexperiment object (make sure no NA in metadata)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">HFHS_data</span><span class="op">$</span><span class="va">FullData</span><span class="op">$</span><span class="va">metadata</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">HFHS_data</span><span class="op">$</span><span class="va">FullData</span><span class="op">$</span><span class="va">X.count</span><span class="op">)</span></span>
<span><span class="va">HFHS.phenotypeData</span> <span class="op">=</span> </span>
<span>  <span class="fu">AnnotatedDataFrame</span><span class="op">(</span>data <span class="op">=</span> <span class="va">HFHS_data</span><span class="op">$</span><span class="va">FullData</span><span class="op">$</span><span class="va">metadata</span><span class="op">[</span><span class="op">-</span><span class="va">hfhs.dA0.idx</span>,<span class="op">]</span><span class="op">)</span></span>
<span><span class="va">HFHS.taxaData</span> <span class="op">=</span> </span>
<span>  <span class="fu">AnnotatedDataFrame</span><span class="op">(</span>data <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="va">HFHS_data</span><span class="op">$</span><span class="va">FullData</span><span class="op">$</span><span class="va">taxa</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">HFHS.obj</span> <span class="op">=</span> <span class="fu">newMRexperiment</span><span class="op">(</span>counts <span class="op">=</span> </span>
<span>                             <span class="fu"><a href="https://rdrr.io/r/base/t.html" class="external-link">t</a></span><span class="op">(</span><span class="va">HFHS_data</span><span class="op">$</span><span class="va">FullData</span><span class="op">$</span><span class="va">X.count</span><span class="op">[</span><span class="op">-</span><span class="va">hfhs.dA0.idx</span>,<span class="op">]</span><span class="op">)</span>,</span>
<span>                           phenoData <span class="op">=</span> <span class="va">HFHS.phenotypeData</span>, </span>
<span>                           featureData <span class="op">=</span> <span class="va">HFHS.taxaData</span><span class="op">)</span></span>
<span><span class="va">HFHS.obj</span></span></code></pre></div>
<pre><code><span><span class="co">## MRexperiment (storageMode: environment)</span></span>
<span><span class="co">## assayData: 4524 features, 149 samples </span></span>
<span><span class="co">##   element names: counts </span></span>
<span><span class="co">## protocolData: none</span></span>
<span><span class="co">## phenoData</span></span>
<span><span class="co">##   sampleNames: 109M 131M ... 39M (149 total)</span></span>
<span><span class="co">##   varLabels: SampleID MouseID ... Replicated (14 total)</span></span>
<span><span class="co">##   varMetadata: labelDescription</span></span>
<span><span class="co">## featureData</span></span>
<span><span class="co">##   featureNames: 4333897 541135 ... 228232 (4524 total)</span></span>
<span><span class="co">##   fvarLabels: Kingdom Phylum ... Species (7 total)</span></span>
<span><span class="co">##   fvarMetadata: labelDescription</span></span>
<span><span class="co">## experimentData: use 'experimentData(object)'</span></span>
<span><span class="co">## Annotation:</span></span></code></pre>
<p>The <span style="color: #808000;">HFHS data</span> are then filtered
with <code>filterData()</code> function (from <span style="color: #FF7F00;">metagenomeSeq</span>). We can use
<code>MRcounts()</code> to extract the count data from the
<code>MRexperiment</code> object.</p>
<div class="sourceCode" id="cb70"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># filtering data to maintain a threshold of minimum depth or OTU presence</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="fu">MRcounts</span><span class="op">(</span><span class="va">HFHS.obj</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 4524  149</span></span></code></pre>
<div class="sourceCode" id="cb72"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">HFHS.obj</span> <span class="op">=</span> <span class="fu">filterData</span><span class="op">(</span>obj <span class="op">=</span> <span class="va">HFHS.obj</span>, present <span class="op">=</span> <span class="fl">20</span>, depth <span class="op">=</span> <span class="fl">5</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="fu">MRcounts</span><span class="op">(</span><span class="va">HFHS.obj</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 1255  149</span></span></code></pre>
<p>After filtering, the <span style="color: #808000;">HFHS data</span>
were reduced to 1255 OTUs and 149 samples.</p>
<p>We calculate the percentile for CSS normalisation with
<code>cumNormStatFast()</code> function (from <span style="color: #FF7F00;">metagenomeSeq</span> package). The CSS
normalisation is applied with <code>cumNorm()</code> and the normalised
data can be exported using <code>MRcounts()</code> with
<code>norm = TRUE</code>. The normalisation scaling factors for each
sample, which are the sum of counts up to the calculated percentile, can
be accessed through <code>normFactors()</code>. We calculate the log
transformed scaling factors by diving them with their median, which are
better than the default scaling factors that are divided by 1000
(<code>log2(normFactors(obj)/1000 + 1)</code>).</p>
<div class="sourceCode" id="cb74"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># calculate the percentile for CSS normalisation</span></span>
<span><span class="va">HFHS.pctl</span> <span class="op">=</span> <span class="fu">cumNormStatFast</span><span class="op">(</span>obj <span class="op">=</span> <span class="va">HFHS.obj</span><span class="op">)</span></span>
<span><span class="co"># CSS normalisation</span></span>
<span><span class="va">HFHS.obj</span> <span class="op">&lt;-</span> <span class="fu">cumNorm</span><span class="op">(</span>obj <span class="op">=</span> <span class="va">HFHS.obj</span>, p <span class="op">=</span> <span class="va">HFHS.pctl</span><span class="op">)</span></span>
<span><span class="co"># export normalised data</span></span>
<span><span class="va">HFHS.norm.data</span> <span class="op">&lt;-</span> <span class="fu">MRcounts</span><span class="op">(</span>obj <span class="op">=</span> <span class="va">HFHS.obj</span>, norm <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># normalisation scaling factors for each sample </span></span>
<span><span class="va">HFHS.normFactor</span> <span class="op">=</span> <span class="fu">normFactors</span><span class="op">(</span>object <span class="op">=</span> <span class="va">HFHS.obj</span><span class="op">)</span></span>
<span><span class="va">HFHS.normFactor</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log2</a></span><span class="op">(</span><span class="va">HFHS.normFactor</span><span class="op">/</span><span class="fu"><a href="https://rdrr.io/r/stats/median.html" class="external-link">median</a></span><span class="op">(</span><span class="va">HFHS.normFactor</span><span class="op">)</span> <span class="op">+</span> <span class="fl">1</span><span class="op">)</span></span></code></pre></div>
<p>As the ZIG model applies a linear model, the cage effect cannot be
fitted. Moreover, the cage effect is very weak which only accounted for
1.3% of the total data variance calculated with pRDA. Thus we only fit
and account for the day effect. We create a design matrix with treatment
variable (<code>Diet_effect</code>), batch variable
(<code>Day_effect</code>) and the log transformed scaling factors by
using <code><a href="https://rdrr.io/r/stats/model.matrix.html" class="external-link">model.matrix()</a></code>, and then apply the ZIG model by
<code>fitZig()</code> function. We set <code>useCSSoffset = FALSE</code>
to avoid using the default scaling factors as we have already included
our customised scaling factor (<code>HFHS.normFactor</code>) in the
design matrix.</p>
<div class="sourceCode" id="cb75"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># treatment variable</span></span>
<span><span class="va">Diet_effect</span> <span class="op">=</span> <span class="fu">pData</span><span class="op">(</span>object <span class="op">=</span> <span class="va">HFHS.obj</span><span class="op">)</span><span class="op">$</span><span class="va">Diet</span></span>
<span></span>
<span><span class="co"># batch variable</span></span>
<span><span class="va">Day_effect</span> <span class="op">=</span> <span class="fu">pData</span><span class="op">(</span>object <span class="op">=</span> <span class="va">HFHS.obj</span><span class="op">)</span><span class="op">$</span><span class="va">Day</span></span>
<span></span>
<span><span class="co"># build a design matrix</span></span>
<span><span class="va">HFHS.mod.full</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/model.matrix.html" class="external-link">model.matrix</a></span><span class="op">(</span><span class="op">~</span> <span class="va">Diet_effect</span> <span class="op">+</span> <span class="va">Day_effect</span> <span class="op">+</span> <span class="va">HFHS.normFactor</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># settings for the fitZig() function</span></span>
<span><span class="va">HFHS.settings</span> <span class="op">&lt;-</span> <span class="fu">zigControl</span><span class="op">(</span>maxit <span class="op">=</span> <span class="fl">10</span>, verbose <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># apply the ZIG model</span></span>
<span><span class="va">HFHSfit</span> <span class="op">&lt;-</span> <span class="fu">fitZig</span><span class="op">(</span>obj <span class="op">=</span> <span class="va">HFHS.obj</span>, mod <span class="op">=</span> <span class="va">HFHS.mod.full</span>, </span>
<span>                  useCSSoffset <span class="op">=</span> <span class="cn">FALSE</span>, control <span class="op">=</span> <span class="va">HFHS.settings</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## it= 0, nll=236.34, log10(eps+1)=Inf, stillActive=1255</span></span>
<span><span class="co">## it= 1, nll=256.75, log10(eps+1)=0.06, stillActive=203</span></span>
<span><span class="co">## it= 2, nll=255.94, log10(eps+1)=0.04, stillActive=153</span></span>
<span><span class="co">## it= 3, nll=256.56, log10(eps+1)=0.04, stillActive=47</span></span>
<span><span class="co">## it= 4, nll=257.05, log10(eps+1)=0.01, stillActive=11</span></span>
<span><span class="co">## it= 5, nll=257.28, log10(eps+1)=0.01, stillActive=3</span></span>
<span><span class="co">## it= 6, nll=257.39, log10(eps+1)=0.02, stillActive=2</span></span>
<span><span class="co">## it= 7, nll=257.46, log10(eps+1)=0.00, stillActive=0</span></span></code></pre>
<p>The OTUs with the top 50 smallest p values are extracted using
<code>MRcoefs()</code>. We set <code>eff = 0.5</code>, so only the OTUs
with at least “0.5” quantile (50%) number of effective samples (positive
samples + estimated undersampling zeroes) are extracted.</p>
<div class="sourceCode" id="cb77"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">HFHScoefs</span> <span class="op">&lt;-</span> <span class="fu">MRcoefs</span><span class="op">(</span><span class="va">HFHSfit</span>, coef <span class="op">=</span> <span class="fl">2</span>, group <span class="op">=</span> <span class="fl">3</span>, number <span class="op">=</span> <span class="fl">50</span>, eff <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">HFHScoefs</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##         Diet_effectNormal      pvalues   adjPvalues</span></span>
<span><span class="co">## 324926          1.0156067 7.360015e-11 9.236819e-08</span></span>
<span><span class="co">## 337550          1.1730238 1.110138e-09 4.362904e-07</span></span>
<span><span class="co">## 4353745        -0.7058975 1.212492e-09 4.362904e-07</span></span>
<span><span class="co">## 357471          0.9636651 1.390567e-09 4.362904e-07</span></span>
<span><span class="co">## 313499         -0.6093954 2.160477e-09 5.422797e-07</span></span>
<span><span class="co">## 3940440        -0.7199750 5.150634e-09 1.077341e-06</span></span></code></pre>
<p><strong>Other methods adapted for microbiome data</strong></p>
<p><strong>Linear regression</strong> Linear regression is conducted
with <code><a href="../reference/linear_regres.html">linear_regres()</a></code> function in <span style="color: #FF7F00;">PLSDAbatch</span>. We integrated the <span style="color: #FF7F00;">performance</span> package that assesses
performance of regression models into our function
<code><a href="../reference/linear_regres.html">linear_regres()</a></code>. Therefore, we can apply
<code>check_model()</code> from <span style="color: #FF7F00;">performance</span> to the outputs from
<code><a href="../reference/linear_regres.html">linear_regres()</a></code> to diagnose the validity of the model
fitted with treatment and batch effects for each variable <span class="citation">(<a href="#ref-daniel2020performance" role="doc-biblioref">LÃŒdecke et al. 2020</a>)</span>. We can extract
performance measurements such as adjusted R2, RMSE, RSE, AIC and BIC for
the models fitted with and without batch effects, which are also the
outputs of <code><a href="../reference/linear_regres.html">linear_regres()</a></code>.</p>
<p>To deal with the day effect only, we apply
<code>type = "linear model"</code> because of the balanced batch x
treatment design.</p>
<div class="sourceCode" id="cb79"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hfhs.lm</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/linear_regres.html">linear_regres</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">hfhs.clr.less</span>, </span>
<span>                         trt <span class="op">=</span> <span class="va">hfhs.trt.less</span>, </span>
<span>                         batch.fix <span class="op">=</span> <span class="va">hfhs.day.less</span>, </span>
<span>                         type <span class="op">=</span> <span class="st">'linear model'</span><span class="op">)</span></span>
<span></span>
<span><span class="va">hfhs.p</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">hfhs.lm</span><span class="op">$</span><span class="va">lm.table</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">{</span><span class="va">x</span><span class="op">$</span><span class="va">coefficients</span><span class="op">[</span><span class="fl">2</span>,<span class="fl">4</span><span class="op">]</span><span class="op">}</span><span class="op">)</span></span>
<span><span class="va">hfhs.p.adj</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/p.adjust.html" class="external-link">p.adjust</a></span><span class="op">(</span>p <span class="op">=</span> <span class="va">hfhs.p</span>, method <span class="op">=</span> <span class="st">'fdr'</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">check_model</span><span class="op">(</span><span class="va">hfhs.lm</span><span class="op">$</span><span class="va">model</span><span class="op">$</span><span class="va">`541135`</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="supp_case_studies_files/figure-html/HFHSlm-1.png" alt='Figure 15: Diagnostic plots for the model fitted with the day effect of "OTU 541135" in the HFHS data.' width="100%"><p class="caption">
Figure 15: Diagnostic plots for the model fitted with the day effect of
“OTU 541135” in the HFHS data.
</p>
</div>
<p>To diagnose the validity of the model fitted with both treatment and
batch effects, we use different plots to check the assumptions of each
microbial variable. For example, the diagnostic plots of “OTU 541135”
are shown in the above figure panel. The simulated data under the fitted
model were not very close to the real data (shown as green line),
indicating an imperfect fitness (top panel). The linearity (or
homoscedasticity) and homogeneity of variance were not satisfied. The
correlation between batch (<code>batch.fix</code>) and treatment
(<code>trt</code>) effects was very low, indicating a good model with
low collinearity. Some samples could be classified as outliers with a
Cook’s distance larger than or equal to 0.5, for example, “46”, “35”,
“126”, “125” and “16” (middle panel). The distribution of residuals was
very close to normal (bottom panel). For the microbial variables with
some assumptions not met, we should be careful about their results.</p>
<p>For performance measurements of models fitted with or without batch
effects, We show an example of the results for some variables.</p>
<div class="sourceCode" id="cb80"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">hfhs.lm</span><span class="op">$</span><span class="va">adj.R2</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##         trt.only trt.batch</span></span>
<span><span class="co">## 541135 0.1969688 0.2590190</span></span>
<span><span class="co">## 276629 0.6419160 0.6414949</span></span>
<span><span class="co">## 196070 0.1618319 0.1530262</span></span>
<span><span class="co">## 318732 0.1720195 0.1795344</span></span>
<span><span class="co">## 339886 0.1542615 0.1779073</span></span>
<span><span class="co">## 337724 0.2343320 0.2445358</span></span></code></pre>
<p>If the adjusted <span class="math inline">\(R^2\)</span> of the model
with both treatment and batch effects is larger than the model with
treatment effects only, e.g., OTU 541135, the model fitted with batch
effects explains more data variance, and is thus better than the model
without batch effects. Otherwise, the batch effect is not necessary to
be added into the linear model.</p>
<p>We next look at the AIC of models fitted with or without batch
effects.</p>
<div class="sourceCode" id="cb82"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">hfhs.lm</span><span class="op">$</span><span class="va">AIC</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##        trt.only trt.batch</span></span>
<span><span class="co">## 541135 462.0799  452.0564</span></span>
<span><span class="co">## 276629 294.7536  296.8876</span></span>
<span><span class="co">## 196070 504.2549  507.7710</span></span>
<span><span class="co">## 318732 462.8639  463.4643</span></span>
<span><span class="co">## 339886 248.0616  245.7953</span></span>
<span><span class="co">## 337724 424.2142  424.1741</span></span></code></pre>
<p>A lower AIC indicates a better fit. Both results were mostly
consistent on model selection, except “OTU318732”. Based on adjusted
<span class="math inline">\(R^2\)</span>, we needed to include batch
effects, while based on AIC, we should not. Therefore, we needed more
measurements such as BIC, RSE.</p>
<p>To consider the cage effect only or the cage and day effects
together, we apply <code>type = "linear mixed model"</code> to the <span style="color: #808000;">HFHS data</span> because of the nested batch x
treatment design for the cage effect.</p>
<div class="sourceCode" id="cb84"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hfhs.lmm</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/linear_regres.html">linear_regres</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">hfhs.clr.less</span>, </span>
<span>                          trt <span class="op">=</span> <span class="va">hfhs.trt.less</span>, </span>
<span>                          batch.fix <span class="op">=</span> <span class="va">hfhs.day.less</span>, </span>
<span>                          batch.random <span class="op">=</span> <span class="va">hfhs.cage.less</span>,</span>
<span>                          type <span class="op">=</span> <span class="st">'linear mixed model'</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb85"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hfhs.p</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">hfhs.lmm</span><span class="op">$</span><span class="va">lmm.table</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">{</span><span class="va">x</span><span class="op">$</span><span class="va">coefficients</span><span class="op">[</span><span class="fl">2</span>,<span class="fl">5</span><span class="op">]</span><span class="op">}</span><span class="op">)</span></span>
<span><span class="va">hfhs.p.adj</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/p.adjust.html" class="external-link">p.adjust</a></span><span class="op">(</span>p <span class="op">=</span> <span class="va">hfhs.p</span>, method <span class="op">=</span> <span class="st">'fdr'</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">check_model</span><span class="op">(</span><span class="va">hfhs.lmm</span><span class="op">$</span><span class="va">model</span><span class="op">$</span><span class="va">`541135`</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="supp_case_studies_files/figure-html/HFHSlmm-1.png" alt='Figure 16: Diagnostic plots for the model fitted with the day and cage effects of "OTU541135" in the HFHS data.' width="100%"><p class="caption">
Figure 16: Diagnostic plots for the model fitted with the day and cage
effects of “OTU541135” in the HFHS data.
</p>
</div>
<p>According to the diagnostic plots of “OTU541135” as shown in the
above figure panel, The simulated data under the fitted model were not
very close to the real data (shown as green line), indicating an
imperfect fitness (top panel). The linearity (or homoscedasticity) and
homogeneity of variance were not satisfied. The correlation between
batch (<code>batch.fix</code>) and treatment (<code>trt</code>) effects
was very low, indicating a good model with low collinearity. Some
samples could be classified as outliers with a Cook’s distance larger
than or equal to 0.5, for example, “16”, “35”, “126”, “125” and “136”
(middle panel). The distribution of residuals and random effects was
very close to normal (bottom panel).</p>
<p>For performance measurements of models fitted with or without batch
effects, We show an example of the results for some variables.</p>
<div class="sourceCode" id="cb86"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">hfhs.lmm</span><span class="op">$</span><span class="va">AIC</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##        trt.only trt.batch</span></span>
<span><span class="co">## 541135 462.0799  461.4074</span></span>
<span><span class="co">## 276629 294.7536  310.3455</span></span>
<span><span class="co">## 196070 504.2549  515.0180</span></span>
<span><span class="co">## 318732 462.8639  472.5556</span></span>
<span><span class="co">## 339886 248.0616  260.7115</span></span>
<span><span class="co">## 337724 424.2142  432.9937</span></span></code></pre>
<p>For some OTUs, the model fitted without batch effects was better with
a lower AIC. For example, “OTU276629”, “OTU196070”, “OTU318732”,
“OTU339886” and “OTU337724”. “OTU541135” was more appropriate within a
model with batch effects.</p>
<p>Since we apply a <code>"linear mixed model"</code> to the <span style="color: #808000;">HFHS data</span>, this type of model can only
output conditional <span class="math inline">\(R^2\)</span> that
includes the variance of both fixed and random effects (treatment, fixed
and random batch effects) and marginal <span class="math inline">\(R^2\)</span> that includes only the variance of
fixed effects (treatment and fixed batch effects) <span class="citation">(<a href="#ref-nakagawa2013general" role="doc-biblioref">Nakagawa and Schielzeth 2013</a>)</span>.</p>
<div class="sourceCode" id="cb88"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">hfhs.lmm</span><span class="op">$</span><span class="va">cond.R2</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##         trt.only trt.batch</span></span>
<span><span class="co">## 541135 0.2023947 0.2748381</span></span>
<span><span class="co">## 276629 0.6443355 0.6475581</span></span>
<span><span class="co">## 196070 0.1674952 0.1932495</span></span>
<span><span class="co">## 318732 0.1776140        NA</span></span>
<span><span class="co">## 339886 0.1599760 0.1954862</span></span>
<span><span class="co">## 337724 0.2395055 0.2845443</span></span></code></pre>
<div class="sourceCode" id="cb90"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">hfhs.lmm</span><span class="op">$</span><span class="va">marg.R2</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##         trt.only trt.batch</span></span>
<span><span class="co">## 541135 0.2023947 0.2692818</span></span>
<span><span class="co">## 276629 0.6443355 0.6435223</span></span>
<span><span class="co">## 196070 0.1674952 0.1678637</span></span>
<span><span class="co">## 318732 0.1776140 0.1929564</span></span>
<span><span class="co">## 339886 0.1599760 0.1916296</span></span>
<span><span class="co">## 337724 0.2395055 0.2520680</span></span></code></pre>
<p>We observed that some variables resulted in singular fits
(<code>error message: Can't compute random effect variances. Some variance components equal zero. Your model may suffer from singulariy.</code>)
and the conditional <span class="math inline">\(R^2\)</span>s of these
variables were “NA”, which are expected to happen in linear mixed models
when covariates are nested. We recommend noting the variables for which
this error occurs, as it may lead to unreliable results.</p>
<p><strong>SVA</strong></p>
<p>SVA accounts for unknown batch effects. Here we assume that the batch
grouping information in the <span style="color: #808000;">HFHS
data</span> is unknown. We first build two design matrices with
(<code>hfhs.mod</code>) and without (<code>hfhs.mod0</code>) treatment
grouping information generated with <code><a href="https://rdrr.io/r/stats/model.matrix.html" class="external-link">model.matrix()</a></code> function
from <span style="color: #FF7F00;">stats</span>. We then use
<code><a href="https://rdrr.io/pkg/sva/man/num.sv.html" class="external-link">num.sv()</a></code> from <span style="color: #FF7F00;">sva</span>
package to determine the number of batch variables <code>n.sv</code>
that will be used to estimate batch effects in function
<code><a href="https://rdrr.io/pkg/sva/man/sva.html" class="external-link">sva()</a></code>.</p>
<div class="sourceCode" id="cb92"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># estimate batch matrix</span></span>
<span><span class="va">hfhs.mod</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/model.matrix.html" class="external-link">model.matrix</a></span><span class="op">(</span> <span class="op">~</span> <span class="va">hfhs.trt.less</span><span class="op">)</span></span>
<span><span class="va">hfhs.mod0</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/model.matrix.html" class="external-link">model.matrix</a></span><span class="op">(</span> <span class="op">~</span> <span class="fl">1</span>, data <span class="op">=</span> <span class="va">hfhs.trt.less</span><span class="op">)</span></span>
<span><span class="va">hfhs.sva.n</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/sva/man/num.sv.html" class="external-link">num.sv</a></span><span class="op">(</span>dat <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/t.html" class="external-link">t</a></span><span class="op">(</span><span class="va">hfhs.clr.less</span><span class="op">)</span>, mod <span class="op">=</span> <span class="va">hfhs.mod</span>, </span>
<span>                     method <span class="op">=</span> <span class="st">'leek'</span><span class="op">)</span></span>
<span><span class="va">hfhs.sva</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/sva/man/sva.html" class="external-link">sva</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/t.html" class="external-link">t</a></span><span class="op">(</span><span class="va">hfhs.clr.less</span><span class="op">)</span>, <span class="va">hfhs.mod</span>, <span class="va">hfhs.mod0</span>, n.sv <span class="op">=</span> <span class="va">hfhs.sva.n</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Number of significant surrogate variables is:  1 </span></span>
<span><span class="co">## Iteration (out of 5 ):1  2  3  4  5</span></span></code></pre>
<p>The estimated batch effects are then applied to
<code><a href="https://rdrr.io/pkg/sva/man/f.pvalue.html" class="external-link">f.pvalue()</a></code> to calculate the P-values of treatment effects.
The estimated batch effects in SVA are assumed to be independent of the
treatment effects. However, SVA considers some correlation between batch
and treatment effects <span class="citation">(<a href="#ref-wang2020managing" role="doc-biblioref">Wang and LêCao
2020</a>)</span>.</p>
<div class="sourceCode" id="cb94"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># include estimated batch effects in the linear model</span></span>
<span><span class="va">hfhs.mod.batch</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">cbind</a></span><span class="op">(</span><span class="va">hfhs.mod</span>, <span class="va">hfhs.sva</span><span class="op">$</span><span class="va">sv</span><span class="op">)</span></span>
<span><span class="va">hfhs.mod0.batch</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">cbind</a></span><span class="op">(</span><span class="va">hfhs.mod0</span>, <span class="va">hfhs.sva</span><span class="op">$</span><span class="va">sv</span><span class="op">)</span></span>
<span><span class="va">hfhs.sva.p</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/sva/man/f.pvalue.html" class="external-link">f.pvalue</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/t.html" class="external-link">t</a></span><span class="op">(</span><span class="va">hfhs.clr.less</span><span class="op">)</span>, <span class="va">hfhs.mod.batch</span>, <span class="va">hfhs.mod0.batch</span><span class="op">)</span></span>
<span><span class="va">hfhs.sva.p.adj</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/p.adjust.html" class="external-link">p.adjust</a></span><span class="op">(</span><span class="va">hfhs.sva.p</span>, method <span class="op">=</span> <span class="st">'fdr'</span><span class="op">)</span></span></code></pre></div>
<p><strong>RUV4</strong></p>
<p>Before applying RUV4 (<code>RUV4()</code> from <span style="color: #FF7F00;">ruv</span>, we need to specify negative control
variables and the number of batch variables to estimate. We can use the
empirical negative controls that are not significantly differentially
abundant (adjusted P &gt; 0.05) from a linear regression with the
treatment information as the only covariate.</p>
<p>We use a loop to fit a linear regression for each microbial variable
and adjust P values of treatment effects for multiple comparisons with
<code><a href="https://rdrr.io/r/stats/p.adjust.html" class="external-link">p.adjust()</a></code> from <span style="color: #FF7F00;">stats</span>.
The empirical negative controls are then extracted according to the
adjusted P values.</p>
<div class="sourceCode" id="cb95"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># empirical negative controls</span></span>
<span><span class="va">hfhs.empir.p</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="kw">for</span><span class="op">(</span><span class="va">e</span> <span class="kw">in</span> <span class="fl">1</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">ncol</a></span><span class="op">(</span><span class="va">hfhs.clr.less</span><span class="op">)</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="va">hfhs.empir.lm</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/lm.html" class="external-link">lm</a></span><span class="op">(</span><span class="va">hfhs.clr.less</span><span class="op">[</span>,<span class="va">e</span><span class="op">]</span> <span class="op">~</span> <span class="va">hfhs.trt.less</span><span class="op">)</span></span>
<span>  <span class="va">hfhs.empir.p</span><span class="op">[</span><span class="va">e</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">hfhs.empir.lm</span><span class="op">)</span><span class="op">$</span><span class="va">coefficients</span><span class="op">[</span><span class="fl">2</span>,<span class="fl">4</span><span class="op">]</span></span>
<span><span class="op">}</span></span>
<span><span class="va">hfhs.empir.p.adj</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/p.adjust.html" class="external-link">p.adjust</a></span><span class="op">(</span>p <span class="op">=</span> <span class="va">hfhs.empir.p</span>, method <span class="op">=</span> <span class="st">'fdr'</span><span class="op">)</span></span>
<span><span class="va">hfhs.nc</span> <span class="op">&lt;-</span> <span class="va">hfhs.empir.p.adj</span> <span class="op">&gt;</span> <span class="fl">0.05</span></span></code></pre></div>
<p>The number of batch variables <code>k</code> can be determined using
<code>getK()</code> function.</p>
<div class="sourceCode" id="cb96"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># estimate k</span></span>
<span><span class="va">hfhs.k.res</span> <span class="op">&lt;-</span> <span class="fu">getK</span><span class="op">(</span>Y <span class="op">=</span> <span class="va">hfhs.clr.less</span>, X <span class="op">=</span> <span class="va">hfhs.trt.less</span>, ctl <span class="op">=</span> <span class="va">hfhs.nc</span><span class="op">)</span></span>
<span><span class="va">hfhs.k</span> <span class="op">&lt;-</span> <span class="va">hfhs.k.res</span><span class="op">$</span><span class="va">k</span></span>
<span><span class="va">hfhs.k</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html" class="external-link">ifelse</a></span><span class="op">(</span><span class="va">hfhs.k</span> <span class="op">!=</span> <span class="fl">0</span>, <span class="va">hfhs.k</span>, <span class="fl">1</span><span class="op">)</span></span></code></pre></div>
<p>We then apply <code>RUV4()</code> with known treatment variables,
estimated negative control variables and <code>k</code> batch variables.
The calculated P values also need to be adjusted for multiple
comparisons.</p>
<div class="sourceCode" id="cb97"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hfhs.ruv4</span> <span class="op">&lt;-</span> <span class="fu">RUV4</span><span class="op">(</span>Y <span class="op">=</span> <span class="va">hfhs.clr.less</span>, X <span class="op">=</span> <span class="va">hfhs.trt.less</span>, </span>
<span>                  ctl <span class="op">=</span> <span class="va">hfhs.nc</span>, k <span class="op">=</span> <span class="va">hfhs.k</span><span class="op">)</span> </span>
<span><span class="va">hfhs.ruv4.p</span> <span class="op">&lt;-</span> <span class="va">hfhs.ruv4</span><span class="op">$</span><span class="va">p</span></span>
<span><span class="va">hfhs.ruv4.p.adj</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/p.adjust.html" class="external-link">p.adjust</a></span><span class="op">(</span><span class="va">hfhs.ruv4.p</span>, method <span class="op">=</span> <span class="st">"fdr"</span><span class="op">)</span></span></code></pre></div>
<p>Both SVA and RUV4 can account for both the cage and day effects, but
not efficient at accounting for the cage effect, as the estimated
surrogate batch effect in SVA and negative controls in RUV4 may not
capture all the batch variation because of the nested batch x treatment
design of the cage effect in the <span style="color: #808000;">HFHS
data</span>.</p>
</div>
<div class="section level4">
<h4 id="correcting-for-batch-effects-1">Correcting for batch effects<a class="anchor" aria-label="anchor" href="#correcting-for-batch-effects-1"></a>
</h4>
<p>The methods that we use to correct for batch effects include
removeBatchEffect, ComBat, PLSDA-batch, sPLSDA-batch, Percentile
Normalisation and RUVIII. Among them, RUVIII is designed for unknown
batch effects.</p>
<p>Since the cage effect cannot be corrected for because of the
collinearity between the cage and treatment effects and the cage effect
only accounted for a small amount of the total data variance (1.3%
calculated with pRDA), so we only correct for the day effect in the
<span style="color: #808000;">HFHS data</span>.</p>
<p><strong>removeBatchEffect</strong></p>
<p>The <code><a href="https://rdrr.io/pkg/limma/man/removeBatchEffect.html" class="external-link">removeBatchEffect()</a></code> function is implemented in <span style="color: #FF7F00;">limma</span> package. The design matrix
(<code>design</code>) with treatment grouping information can be
generated with <code><a href="https://rdrr.io/r/stats/model.matrix.html" class="external-link">model.matrix()</a></code> function from <span style="color: #FF7F00;">stats</span> as shown in section
<strong>accounting for batch effects</strong> method “SVA”.</p>
<p>Here we use <code><a href="https://rdrr.io/pkg/limma/man/removeBatchEffect.html" class="external-link">removeBatchEffect()</a></code> function with batch
grouping information (<code>batch</code>) and treatment design matrix
(<code>design</code>) to calculate batch effect corrected data
<code>hfhs.rBE</code>.</p>
<div class="sourceCode" id="cb98"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hfhs.rBE</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/t.html" class="external-link">t</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/limma/man/removeBatchEffect.html" class="external-link">removeBatchEffect</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/t.html" class="external-link">t</a></span><span class="op">(</span><span class="va">hfhs.clr.less</span><span class="op">)</span>, batch <span class="op">=</span> <span class="va">hfhs.day.less</span>, </span>
<span>                                design <span class="op">=</span> <span class="va">hfhs.mod</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><strong>ComBat</strong></p>
<p>The <code><a href="https://rdrr.io/pkg/sva/man/ComBat.html" class="external-link">ComBat()</a></code> function (from <span style="color: #FF7F00;">sva</span> package) is implemented as parametric
or non-parametric correction with option <code>par.prior</code>. Under a
parametric adjustment, we can assess the model’s validity with
<code>prior.plots = T</code> <span class="citation">(<a href="#ref-leek2012sva" role="doc-biblioref">Leek et al.
2012</a>)</span>.</p>
<p>Here we use a non-parametric correction (<code>par.prior = F</code>)
with batch grouping information (<code>batch</code>) and treatment
design matrix (<code>mod</code>) to calculate batch effect corrected
data <code>hfhs.ComBat</code>.</p>
<div class="sourceCode" id="cb99"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hfhs.ComBat</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/t.html" class="external-link">t</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/sva/man/ComBat.html" class="external-link">ComBat</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/t.html" class="external-link">t</a></span><span class="op">(</span><span class="va">hfhs.clr.less</span><span class="op">)</span>, batch <span class="op">=</span> <span class="va">hfhs.day.less</span>, </span>
<span>                        mod <span class="op">=</span> <span class="va">hfhs.mod</span>, par.prior <span class="op">=</span> <span class="cn">F</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><strong>PLSDA-batch</strong></p>
<p>The <code><a href="../reference/PLSDA_batch.html">PLSDA_batch()</a></code> function is implemented in <span style="color: #FF7F00;">PLSDAbatch</span> package. To use this function,
we need to specify the optimal number of components related to treatment
(<code>ncomp.trt</code>) or batch effects (<code>ncomp.bat</code>).</p>
<p>Here in the <span style="color: #808000;">HFHS data</span>, we use
<code><a href="https://rdrr.io/pkg/mixOmics/man/plsda.html" class="external-link">plsda()</a></code> from <span style="color: #FF7F00;">mixOmics</span>
with only treatment grouping information to estimate the optimal number
of treatment components to preserve.</p>
<div class="sourceCode" id="cb100"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># estimate the number of treatment components</span></span>
<span><span class="va">hfhs.trt.tune</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/mixOmics/man/plsda.html" class="external-link">plsda</a></span><span class="op">(</span>X <span class="op">=</span> <span class="va">hfhs.clr.less</span>, Y <span class="op">=</span> <span class="va">hfhs.trt.less</span>, ncomp <span class="op">=</span> <span class="fl">5</span><span class="op">)</span></span>
<span><span class="va">hfhs.trt.tune</span><span class="op">$</span><span class="va">prop_expl_var</span> <span class="co">#1</span></span></code></pre></div>
<pre><code><span><span class="co">## $X</span></span>
<span><span class="co">##      comp1      comp2      comp3      comp4      comp5 </span></span>
<span><span class="co">## 0.29946816 0.06913457 0.04242662 0.02681544 0.02189936 </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $Y</span></span>
<span><span class="co">##      comp1      comp2      comp3      comp4      comp5 </span></span>
<span><span class="co">## 1.00000000 0.07124125 0.03858356 0.02564995 0.01555355</span></span></code></pre>
<p>We choose the number that explains 100% variance in the outcome
matrix <code>Y</code>, thus from the result, 1 component is enough to
preserve the treatment information.</p>
<p>We then use <code><a href="../reference/PLSDA_batch.html">PLSDA_batch()</a></code> function with both treatment
and batch grouping information to estimate the optimal number of batch
components to remove.</p>
<div class="sourceCode" id="cb102"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># estimate the number of batch components</span></span>
<span><span class="va">hfhs.batch.tune</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/PLSDA_batch.html">PLSDA_batch</a></span><span class="op">(</span>X <span class="op">=</span> <span class="va">hfhs.clr.less</span>, </span>
<span>                               Y.trt <span class="op">=</span> <span class="va">hfhs.trt.less</span>, </span>
<span>                               Y.bat <span class="op">=</span> <span class="va">hfhs.day.less</span>,</span>
<span>                               ncomp.trt <span class="op">=</span> <span class="fl">1</span>, ncomp.bat <span class="op">=</span> <span class="fl">10</span><span class="op">)</span></span>
<span><span class="va">hfhs.batch.tune</span><span class="op">$</span><span class="va">explained_variance.bat</span> <span class="co">#2</span></span></code></pre></div>
<pre><code><span><span class="co">## $X</span></span>
<span><span class="co">##      comp1      comp2      comp3      comp4      comp5      comp6      comp7 </span></span>
<span><span class="co">## 0.10753644 0.03608189 0.02988203 0.02638198 0.04335240 0.02931383 0.02430825 </span></span>
<span><span class="co">##      comp8      comp9     comp10 </span></span>
<span><span class="co">## 0.03177304 0.01553524 0.03029469 </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $Y</span></span>
<span><span class="co">##        comp1        comp2        comp3        comp4        comp5        comp6 </span></span>
<span><span class="co">## 0.4937952138 0.5062047862 0.5128452381 0.4869109170 0.0002438449 0.5069367566 </span></span>
<span><span class="co">##        comp7        comp8        comp9       comp10 </span></span>
<span><span class="co">## 0.4921936988 0.0008695446 0.5082371149 0.4740866865</span></span></code></pre>
<div class="sourceCode" id="cb104"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="va">hfhs.batch.tune</span><span class="op">$</span><span class="va">explained_variance.bat</span><span class="op">$</span><span class="va">Y</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">2</span><span class="op">]</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 1</span></span></code></pre>
<p>Using the same criterion as choosing treatment components, we choose
the number of batch components that explains 100% variance in the
outcome matrix of batch. According to the result, 2 components are
required to remove batch effects.</p>
<p>We then can correct for batch effects applying
<code><a href="../reference/PLSDA_batch.html">PLSDA_batch()</a></code> with treatment, batch grouping information
and corresponding optimal number of related components.</p>
<div class="sourceCode" id="cb106"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">##########</span></span>
<span><span class="va">hfhs.PLSDA_batch.res</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/PLSDA_batch.html">PLSDA_batch</a></span><span class="op">(</span>X <span class="op">=</span> <span class="va">hfhs.clr.less</span>, </span>
<span>                                    Y.trt <span class="op">=</span> <span class="va">hfhs.trt.less</span>, </span>
<span>                                    Y.bat <span class="op">=</span> <span class="va">hfhs.day.less</span>,</span>
<span>                                    ncomp.trt <span class="op">=</span> <span class="fl">1</span>, ncomp.bat <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span>
<span><span class="va">hfhs.PLSDA_batch</span> <span class="op">&lt;-</span> <span class="va">hfhs.PLSDA_batch.res</span><span class="op">$</span><span class="va">X.nobatch</span></span></code></pre></div>
<p><strong>sPLSDA-batch</strong></p>
<p>We apply sPLSDA-batch using the same function
<code><a href="../reference/PLSDA_batch.html">PLSDA_batch()</a></code> but we specify the number of variables to
select on each component (usually only treatment-related components
<code>keepX.trt</code>). To determine the optimal number of variables to
select, we use <code><a href="https://rdrr.io/pkg/mixOmics/man/tune.splsda.html" class="external-link">tune.splsda()</a></code> function from <span style="color: #FF7F00;">mixOmics</span> package <span class="citation">(<a href="#ref-rohart2017mixomics" role="doc-biblioref">Rohart et al. 2017</a>)</span> with all possible
numbers of variables to select for each component
(<code>test.keepX</code>).</p>
<div class="sourceCode" id="cb107"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># estimate the number of variables to select per treatment component</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">777</span><span class="op">)</span></span>
<span><span class="va">hfhs.test.keepX</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">10</span>, <span class="fl">1</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">20</span>, <span class="fl">200</span>, <span class="fl">10</span><span class="op">)</span>, </span>
<span>                    <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">250</span>, <span class="fl">500</span>, <span class="fl">50</span><span class="op">)</span>, <span class="fl">515</span><span class="op">)</span></span>
<span><span class="va">hfhs.trt.tune.v</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/mixOmics/man/tune.splsda.html" class="external-link">tune.splsda</a></span><span class="op">(</span>X <span class="op">=</span> <span class="va">hfhs.clr.less</span>, </span>
<span>                               Y <span class="op">=</span> <span class="va">hfhs.trt.less</span>, </span>
<span>                               ncomp <span class="op">=</span> <span class="fl">1</span>, </span>
<span>                               test.keepX <span class="op">=</span> <span class="va">hfhs.test.keepX</span>, </span>
<span>                               validation <span class="op">=</span> <span class="st">'Mfold'</span>, </span>
<span>                               folds <span class="op">=</span> <span class="fl">4</span>, nrepeat <span class="op">=</span> <span class="fl">50</span><span class="op">)</span></span>
<span><span class="va">hfhs.trt.tune.v</span><span class="op">$</span><span class="va">choice.keepX</span> <span class="co"># 2</span></span></code></pre></div>
<pre><code><span><span class="co">## comp1 </span></span>
<span><span class="co">##     2</span></span></code></pre>
<p>Here the optimal number of variables to select for the treatment
component is 2. Since we adjust the amount of treatment variation to
preserve, we need to re-choose the optimal number of components related
to batch effects using the same criterion mentioned in section
<strong>correcting for batch effects</strong> method “PLSDA-batch”.</p>
<div class="sourceCode" id="cb109"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># estimate the number of batch components</span></span>
<span><span class="va">hfhs.batch.tune</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/PLSDA_batch.html">PLSDA_batch</a></span><span class="op">(</span>X <span class="op">=</span> <span class="va">hfhs.clr.less</span>, </span>
<span>                               Y.trt <span class="op">=</span> <span class="va">hfhs.trt.less</span>, </span>
<span>                               Y.bat <span class="op">=</span> <span class="va">hfhs.day.less</span>,</span>
<span>                               ncomp.trt <span class="op">=</span> <span class="fl">1</span>, </span>
<span>                               keepX.trt <span class="op">=</span> <span class="fl">2</span>,</span>
<span>                               ncomp.bat <span class="op">=</span> <span class="fl">10</span><span class="op">)</span></span>
<span><span class="va">hfhs.batch.tune</span><span class="op">$</span><span class="va">explained_variance.bat</span> <span class="co">#2</span></span></code></pre></div>
<pre><code><span><span class="co">## $X</span></span>
<span><span class="co">##      comp1      comp2      comp3      comp4      comp5      comp6      comp7 </span></span>
<span><span class="co">## 0.11496719 0.03611134 0.02997732 0.02496738 0.03220821 0.03283139 0.04100050 </span></span>
<span><span class="co">##      comp8      comp9     comp10 </span></span>
<span><span class="co">## 0.02775988 0.02420573 0.01955264 </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $Y</span></span>
<span><span class="co">##        comp1        comp2        comp3        comp4        comp5        comp6 </span></span>
<span><span class="co">## 0.4938682259 0.5061317741 0.5120581093 0.4871878556 0.0007540351 0.5090445753 </span></span>
<span><span class="co">##        comp7        comp8        comp9       comp10 </span></span>
<span><span class="co">## 0.4896338693 0.0013215554 0.4942076677 0.4961940448</span></span></code></pre>
<div class="sourceCode" id="cb111"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="va">hfhs.batch.tune</span><span class="op">$</span><span class="va">explained_variance.bat</span><span class="op">$</span><span class="va">Y</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">2</span><span class="op">]</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 1</span></span></code></pre>
<p>According to the result, we need 2 batch related components to remove
batch variance from the data with function
<code><a href="../reference/PLSDA_batch.html">PLSDA_batch()</a></code>.</p>
<div class="sourceCode" id="cb113"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">##########</span></span>
<span><span class="va">hfhs.sPLSDA_batch.res</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/PLSDA_batch.html">PLSDA_batch</a></span><span class="op">(</span>X <span class="op">=</span> <span class="va">hfhs.clr.less</span>, </span>
<span>                                     Y.trt <span class="op">=</span> <span class="va">hfhs.trt.less</span>, </span>
<span>                                     Y.bat <span class="op">=</span> <span class="va">hfhs.day.less</span>,</span>
<span>                                     ncomp.trt <span class="op">=</span> <span class="fl">1</span>, </span>
<span>                                     keepX.trt <span class="op">=</span> <span class="fl">2</span>,</span>
<span>                                     ncomp.bat <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span>
<span><span class="va">hfhs.sPLSDA_batch</span> <span class="op">&lt;-</span> <span class="va">hfhs.sPLSDA_batch.res</span><span class="op">$</span><span class="va">X.nobatch</span></span></code></pre></div>
<p><strong>Percentile Normalisation</strong></p>
<p>To apply <code><a href="../reference/percentile_norm.html">percentile_norm()</a></code> function from <span style="color: #FF7F00;">PLSDAbatch</span> package, we need to indicate a
control group (<code>ctrl.grp</code>).</p>
<div class="sourceCode" id="cb114"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># HFHS data</span></span>
<span><span class="va">hfhs.PN</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/percentile_norm.html">percentile_norm</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">hfhs.clr.less</span>, </span>
<span>                           batch <span class="op">=</span> <span class="va">hfhs.day.less</span>, </span>
<span>                           trt <span class="op">=</span> <span class="va">hfhs.trt.less</span>, </span>
<span>                           ctrl.grp <span class="op">=</span> <span class="st">'Normal'</span><span class="op">)</span></span></code></pre></div>
<p><strong>RUVIII</strong></p>
<p>The <code>RUVIII()</code> function is from <span style="color: #FF7F00;">ruv</span> package. Similar to
<code>RUV4()</code>, we use empirical negative control variables and
<code>getK()</code> to determine the number of batch variables
(<code>k</code>) to estimate. We also need sample replicates which
should be structured into a mapping matrix using
<code>replicate.matrix()</code>. We can then obtain batch effect
corrected data applying <code>RUVIII()</code> with above elements.</p>
<div class="sourceCode" id="cb115"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hfhs.replicates</span> <span class="op">&lt;-</span> <span class="va">hfhs.metadata.less</span><span class="op">$</span><span class="va">MouseID</span></span>
<span><span class="va">hfhs.replicates.matrix</span> <span class="op">&lt;-</span> <span class="fu">replicate.matrix</span><span class="op">(</span><span class="va">hfhs.replicates</span><span class="op">)</span></span>
<span></span>
<span><span class="va">hfhs.RUVIII</span> <span class="op">&lt;-</span> <span class="fu">RUVIII</span><span class="op">(</span>Y <span class="op">=</span> <span class="va">hfhs.clr.less</span>, </span>
<span>                      M <span class="op">=</span> <span class="va">hfhs.replicates.matrix</span>, </span>
<span>                      ctl <span class="op">=</span> <span class="va">hfhs.nc</span>, k <span class="op">=</span> <span class="va">hfhs.k</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">hfhs.RUVIII</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">hfhs.clr.less</span><span class="op">)</span></span></code></pre></div>
<p>Compared to the <span style="color: #0000FF;">AD data</span>, the
<span style="color: #808000;">HFHS data</span> have more appropriate
sample replicates that fully capture the difference between different
time points. The results of RUVIII applied to <span style="color: #808000;">HFHS data</span> would be better than AD
data.</p>
</div>
</div>
<div class="section level3">
<h3 id="assessing-batch-effect-correction-1">Assessing batch effect correction<a class="anchor" aria-label="anchor" href="#assessing-batch-effect-correction-1"></a>
</h3>
<p>We apply different visualisation and quantitative methods to
assessing batch effect correction.</p>
<div class="section level4">
<h4 id="methods-that-detect-batch-effects-1">Methods that detect batch effects<a class="anchor" aria-label="anchor" href="#methods-that-detect-batch-effects-1"></a>
</h4>
<p><strong>PCA</strong></p>
<p>In the <span style="color: #808000;">HFHS data</span>, we compare the
PCA sample plots before and after batch effect correction with different
methods.</p>
<div class="sourceCode" id="cb116"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hfhs.pca.before</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/mixOmics/man/pca.html" class="external-link">pca</a></span><span class="op">(</span><span class="va">hfhs.clr.less</span>, ncomp <span class="op">=</span> <span class="fl">3</span>, </span>
<span>                       scale <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">hfhs.pca.rBE</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/mixOmics/man/pca.html" class="external-link">pca</a></span><span class="op">(</span><span class="va">hfhs.rBE</span>, ncomp <span class="op">=</span> <span class="fl">3</span>, </span>
<span>                    scale <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">hfhs.pca.ComBat</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/mixOmics/man/pca.html" class="external-link">pca</a></span><span class="op">(</span><span class="va">hfhs.ComBat</span>, ncomp <span class="op">=</span> <span class="fl">3</span>, </span>
<span>                       scale <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">hfhs.pca.PLSDA_batch</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/mixOmics/man/pca.html" class="external-link">pca</a></span><span class="op">(</span><span class="va">hfhs.PLSDA_batch</span>, ncomp <span class="op">=</span> <span class="fl">3</span>, </span>
<span>                            scale <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">hfhs.pca.sPLSDA_batch</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/mixOmics/man/pca.html" class="external-link">pca</a></span><span class="op">(</span><span class="va">hfhs.sPLSDA_batch</span>, ncomp <span class="op">=</span> <span class="fl">3</span>, </span>
<span>                             scale <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">hfhs.pca.PN</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/mixOmics/man/pca.html" class="external-link">pca</a></span><span class="op">(</span><span class="va">hfhs.PN</span>, ncomp <span class="op">=</span> <span class="fl">3</span>, </span>
<span>                   scale <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">hfhs.pca.RUVIII</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/mixOmics/man/pca.html" class="external-link">pca</a></span><span class="op">(</span><span class="va">hfhs.RUVIII</span>, ncomp <span class="op">=</span> <span class="fl">3</span>, </span>
<span>                       scale <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb117"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hfhs.pca.before.plot</span> <span class="op">&lt;-</span> </span>
<span>  <span class="fu"><a href="../reference/Scatter_Density.html">Scatter_Density</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">hfhs.pca.before</span>, </span>
<span>                  batch <span class="op">=</span> <span class="va">hfhs.day.less</span>, </span>
<span>                  trt <span class="op">=</span> <span class="va">hfhs.trt.less</span>, </span>
<span>                  title <span class="op">=</span> <span class="st">'Before'</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb118"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hfhs.pca.rBE.plot</span> <span class="op">&lt;-</span> </span>
<span>  <span class="fu"><a href="../reference/Scatter_Density.html">Scatter_Density</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">hfhs.pca.rBE</span>, </span>
<span>                  batch <span class="op">=</span> <span class="va">hfhs.day.less</span>, </span>
<span>                  trt <span class="op">=</span> <span class="va">hfhs.trt.less</span>, </span>
<span>                  title <span class="op">=</span> <span class="st">'removeBatchEffect'</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb119"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hfhs.pca.ComBat.plot</span> <span class="op">&lt;-</span> </span>
<span>  <span class="fu"><a href="../reference/Scatter_Density.html">Scatter_Density</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">hfhs.pca.ComBat</span>, </span>
<span>                  batch <span class="op">=</span> <span class="va">hfhs.day.less</span>, </span>
<span>                  trt <span class="op">=</span> <span class="va">hfhs.trt.less</span>,</span>
<span>                  title <span class="op">=</span> <span class="st">'ComBat'</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb120"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hfhs.pca.PLSDA_batch.plot</span> <span class="op">&lt;-</span> </span>
<span>  <span class="fu"><a href="../reference/Scatter_Density.html">Scatter_Density</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">hfhs.pca.PLSDA_batch</span>, </span>
<span>                  batch <span class="op">=</span> <span class="va">hfhs.day.less</span>, </span>
<span>                  trt <span class="op">=</span> <span class="va">hfhs.trt.less</span>, </span>
<span>                  title <span class="op">=</span> <span class="st">'PLSDA-batch'</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb121"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hfhs.pca.sPLSDA_batch.plot</span> <span class="op">&lt;-</span> </span>
<span>  <span class="fu"><a href="../reference/Scatter_Density.html">Scatter_Density</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">hfhs.pca.sPLSDA_batch</span>, </span>
<span>                  batch <span class="op">=</span> <span class="va">hfhs.day.less</span>, </span>
<span>                  trt <span class="op">=</span> <span class="va">hfhs.trt.less</span>, </span>
<span>                  title <span class="op">=</span> <span class="st">'sPLSDA-batch'</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb122"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hfhs.pca.PN.plot</span> <span class="op">&lt;-</span> </span>
<span>  <span class="fu"><a href="../reference/Scatter_Density.html">Scatter_Density</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">hfhs.pca.PN</span>, </span>
<span>                  batch <span class="op">=</span> <span class="va">hfhs.day.less</span>, </span>
<span>                  trt <span class="op">=</span> <span class="va">hfhs.trt.less</span>, </span>
<span>                  title <span class="op">=</span> <span class="st">'Percentile Normalisation'</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb123"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hfhs.pca.RUVIII.plot</span> <span class="op">&lt;-</span> </span>
<span>  <span class="fu"><a href="../reference/Scatter_Density.html">Scatter_Density</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">hfhs.pca.RUVIII</span>, </span>
<span>                  batch <span class="op">=</span> <span class="va">hfhs.day.less</span>, </span>
<span>                  trt <span class="op">=</span> <span class="va">hfhs.trt.less</span>, </span>
<span>                  title <span class="op">=</span> <span class="st">'RUVIII'</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="supp_case_studies_files/figure-html/HFHSpca-1.png" alt="Figure 17: The PCA sample plots with densities before and after batch effect correction in the HFHS data." width="100%"><p class="caption">
Figure 17: The PCA sample plots with densities before and after batch
effect correction in the HFHS data.
</p>
</div>
<p>In the above figures, the differences between the samples from
different days were well removed after batch effect correction with
PLSDA-batch and RUVIII. We can also compare the boxplots and density
plots for key variables identified in PCA driving the major variance or
in heatmaps showing obvious patterns before and after batch effect
correction (results not shown).</p>
<p><strong>pRDA</strong></p>
<p>We calculate the global explained variance across all microbial
variables using pRDA. To achieve this, we create a loop for each
variable from the original (uncorrected) and batch effect-corrected
data. The final results are then displayed with
<code><a href="../reference/partVar_plot.html">partVar_plot()</a></code> from <span style="color: #FF7F00;">PLSDAbatch</span> package.</p>
<div class="sourceCode" id="cb124"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># HFHS data</span></span>
<span><span class="va">hfhs.corrected.list</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>`Before correction` <span class="op">=</span> <span class="va">hfhs.clr.less</span>, </span>
<span>                            removeBatchEffect <span class="op">=</span> <span class="va">hfhs.rBE</span>, </span>
<span>                            ComBat <span class="op">=</span> <span class="va">hfhs.ComBat</span>, </span>
<span>                            `PLSDA-batch` <span class="op">=</span> <span class="va">hfhs.PLSDA_batch</span>,</span>
<span>                            `sPLSDA-batch` <span class="op">=</span> <span class="va">hfhs.sPLSDA_batch</span>, </span>
<span>                            `Percentile Normalisation` <span class="op">=</span> <span class="va">hfhs.PN</span>,</span>
<span>                            RUVIII <span class="op">=</span> <span class="va">hfhs.RUVIII</span><span class="op">)</span></span>
<span></span>
<span><span class="va">hfhs.prop.df</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>Treatment <span class="op">=</span> <span class="cn">NA</span>, Batch <span class="op">=</span> <span class="cn">NA</span>, </span>
<span>                           Intersection <span class="op">=</span> <span class="cn">NA</span>, </span>
<span>                           Residuals <span class="op">=</span> <span class="cn">NA</span><span class="op">)</span> </span>
<span><span class="kw">for</span><span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="fl">1</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">hfhs.corrected.list</span><span class="op">)</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="va">rda.res</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/vegan/man/varpart.html" class="external-link">varpart</a></span><span class="op">(</span><span class="va">hfhs.corrected.list</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span>, <span class="op">~</span> <span class="va">trt</span>, <span class="op">~</span> <span class="va">day</span>,</span>
<span>                    data <span class="op">=</span> <span class="va">hfhs.factors.df</span>, scale <span class="op">=</span> <span class="cn">T</span><span class="op">)</span></span>
<span>  <span class="va">hfhs.prop.df</span><span class="op">[</span><span class="va">i</span>, <span class="op">]</span> <span class="op">&lt;-</span> <span class="va">rda.res</span><span class="op">$</span><span class="va">part</span><span class="op">$</span><span class="va">indfract</span><span class="op">$</span><span class="va">Adj.R.squared</span><span class="op">}</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">hfhs.prop.df</span><span class="op">)</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">hfhs.corrected.list</span><span class="op">)</span></span>
<span></span>
<span><span class="va">hfhs.prop.df</span> <span class="op">&lt;-</span> <span class="va">hfhs.prop.df</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">3</span>,<span class="fl">2</span>,<span class="fl">4</span><span class="op">)</span><span class="op">]</span></span>
<span></span>
<span><span class="va">hfhs.prop.df</span><span class="op">[</span><span class="va">hfhs.prop.df</span> <span class="op">&lt;</span> <span class="fl">0</span><span class="op">]</span> <span class="op">=</span> <span class="fl">0</span></span>
<span><span class="va">hfhs.prop.df</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/t.html" class="external-link">t</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/apply.html" class="external-link">apply</a></span><span class="op">(</span><span class="va">hfhs.prop.df</span>, <span class="fl">1</span>, </span>
<span>                                      <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">{</span><span class="va">x</span><span class="op">/</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">}</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span></span>
<span><span class="fu"><a href="../reference/partVar_plot.html">partVar_plot</a></span><span class="op">(</span>prop.df <span class="op">=</span> <span class="va">hfhs.prop.df</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="supp_case_studies_files/figure-html/HFHSprda-1.png" alt="Figure 18: Global explained variance before and after batch effect correction for the HFHS data." width="60%"><p class="caption">
Figure 18: Global explained variance before and after batch effect
correction for the HFHS data.
</p>
</div>
<p>In the <span style="color: #808000;">HFHS data</span>, a small amount
of batch variance was observed (3.6%) as shown in the above figure.
PLSDA-batch achieved the best performance for preserving the largest
treatment variance and completely removing batch variance compared to
the other methods. The results also indicate that PLSDA-batch is more
appropriate for weak batch effects, while sPLSDA-batch is more
appropriate for strong batch effects. The RUVIII performed better in the
<span style="color: #808000;">HFHS data</span> than the <span style="color: #0000FF;">AD data</span> because the sample replicates may
help capturing more batch variation than in the <span style="color: #0000FF;">AD data</span>. Indeed, the sample replicates in
<span style="color: #808000;">HFHS data</span> are across different day
batches, while the replicates in <span style="color: #0000FF;">AD
data</span> do not exist in all batches. Therefore, sample replicates
play a critical role in RUVIII.</p>
</div>
<div class="section level4">
<h4 id="other-methods-1">Other methods<a class="anchor" aria-label="anchor" href="#other-methods-1"></a>
</h4>
<p><strong><span class="math inline">\(R^2\)</span></strong></p>
<p>The <span class="math inline">\(R^2\)</span> values for each variable
are calculated with <code><a href="https://rdrr.io/r/stats/lm.html" class="external-link">lm()</a></code> from <span style="color: #FF7F00;">stats</span> package. To compare the <span class="math inline">\(R^2\)</span> values among variables, we scale the
corrected data before <span class="math inline">\(R^2\)</span>
calculation. The results are displayed with <code><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot()</a></code> from
<span style="color: #FF7F00;">ggplot2</span> R package.</p>
<div class="sourceCode" id="cb125"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># scale</span></span>
<span><span class="va">hfhs.corr_scale.list</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">hfhs.corrected.list</span>, </span>
<span>                               <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">{</span><span class="fu"><a href="https://rdrr.io/r/base/apply.html" class="external-link">apply</a></span><span class="op">(</span><span class="va">x</span>, <span class="fl">2</span>, <span class="va">scale</span><span class="op">)</span><span class="op">}</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># HFHS data</span></span>
<span><span class="va">hfhs.r_values.list</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="kw">for</span><span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="fl">1</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">hfhs.corr_scale.list</span><span class="op">)</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="va">hfhs.r_values</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>trt <span class="op">=</span> <span class="cn">NA</span>, batch <span class="op">=</span> <span class="cn">NA</span><span class="op">)</span></span>
<span>  <span class="kw">for</span><span class="op">(</span><span class="va">c</span> <span class="kw">in</span> <span class="fl">1</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">ncol</a></span><span class="op">(</span><span class="va">hfhs.corr_scale.list</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span><span class="op">)</span><span class="op">)</span><span class="op">{</span></span>
<span>    <span class="va">hfhs.fit.res.trt</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/lm.html" class="external-link">lm</a></span><span class="op">(</span><span class="va">hfhs.corr_scale.list</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span><span class="op">[</span>,<span class="va">c</span><span class="op">]</span> <span class="op">~</span> <span class="va">hfhs.trt.less</span><span class="op">)</span></span>
<span>    <span class="va">hfhs.r_values</span><span class="op">[</span><span class="va">c</span>,<span class="fl">1</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">hfhs.fit.res.trt</span><span class="op">)</span><span class="op">$</span><span class="va">r.squared</span></span>
<span>    <span class="va">hfhs.fit.res.batch</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/lm.html" class="external-link">lm</a></span><span class="op">(</span><span class="va">hfhs.corr_scale.list</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span><span class="op">[</span>,<span class="va">c</span><span class="op">]</span> <span class="op">~</span> <span class="va">hfhs.day.less</span><span class="op">)</span></span>
<span>    <span class="va">hfhs.r_values</span><span class="op">[</span><span class="va">c</span>,<span class="fl">2</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">hfhs.fit.res.batch</span><span class="op">)</span><span class="op">$</span><span class="va">r.squared</span></span>
<span>  <span class="op">}</span></span>
<span>  <span class="va">hfhs.r_values.list</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="va">hfhs.r_values</span></span>
<span><span class="op">}</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">hfhs.r_values.list</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">hfhs.corr_scale.list</span><span class="op">)</span></span>
<span></span>
<span><span class="va">hfhs.boxp.list</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="kw">for</span><span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_len</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">hfhs.r_values.list</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="va">hfhs.boxp.list</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> </span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>r2 <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">hfhs.r_values.list</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span><span class="op">[</span> ,<span class="st">'trt'</span><span class="op">]</span>,</span>
<span>                      <span class="va">hfhs.r_values.list</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span><span class="op">[</span> ,<span class="st">'batch'</span><span class="op">]</span><span class="op">)</span>, </span>
<span>               Effects <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">as.factor</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'Treatment'</span>,<span class="st">'Batch'</span><span class="op">)</span>, </span>
<span>                                       each <span class="op">=</span> <span class="fl">515</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">hfhs.boxp.list</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">hfhs.r_values.list</span><span class="op">)</span></span>
<span></span>
<span><span class="va">hfhs.r2.boxp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">rbind</a></span><span class="op">(</span><span class="va">hfhs.boxp.list</span><span class="op">$</span><span class="va">`Before correction`</span>,</span>
<span>                      <span class="va">hfhs.boxp.list</span><span class="op">$</span><span class="va">removeBatchEffect</span>,</span>
<span>                      <span class="va">hfhs.boxp.list</span><span class="op">$</span><span class="va">ComBat</span>,</span>
<span>                      <span class="va">hfhs.boxp.list</span><span class="op">$</span><span class="va">`PLSDA-batch`</span>,</span>
<span>                      <span class="va">hfhs.boxp.list</span><span class="op">$</span><span class="va">`sPLSDA-batch`</span>,</span>
<span>                      <span class="va">hfhs.boxp.list</span><span class="op">$</span><span class="va">`Percentile Normalisation`</span>,</span>
<span>                      <span class="va">hfhs.boxp.list</span><span class="op">$</span><span class="va">RUVIII</span><span class="op">)</span></span>
<span></span>
<span><span class="va">hfhs.r2.boxp</span><span class="op">$</span><span class="va">methods</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'Before correction'</span>, <span class="st">' removeBatchEffect'</span>, </span>
<span>                              <span class="st">'ComBat'</span>,<span class="st">'PLSDA-batch'</span>, <span class="st">'sPLSDA-batch'</span>,</span>
<span>                              <span class="st">'Percentile Normalisation'</span>, <span class="st">'RUVIII'</span><span class="op">)</span>, </span>
<span>                            each <span class="op">=</span> <span class="fl">1030</span><span class="op">)</span></span>
<span></span>
<span><span class="va">hfhs.r2.boxp</span><span class="op">$</span><span class="va">methods</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="va">hfhs.r2.boxp</span><span class="op">$</span><span class="va">methods</span>, </span>
<span>                               levels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">hfhs.r2.boxp</span><span class="op">$</span><span class="va">methods</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">hfhs.r2.boxp</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">Effects</span>, y <span class="op">=</span> <span class="va">r2</span>, fill <span class="op">=</span> <span class="va">Effects</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_boxplot.html" class="external-link">geom_boxplot</a></span><span class="op">(</span>alpha <span class="op">=</span> <span class="fl">0.80</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_bw</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>text <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">18</span><span class="op">)</span>,</span>
<span>        axis.title.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>        axis.title.y <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>        axis.text.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">60</span>, hjust <span class="op">=</span> <span class="fl">1</span>, size <span class="op">=</span> <span class="fl">18</span><span class="op">)</span>,</span>
<span>        axis.text.y <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">18</span><span class="op">)</span>,</span>
<span>        panel.grid.minor.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>        panel.grid.major.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>        legend.position <span class="op">=</span> <span class="st">"right"</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_grid.html" class="external-link">facet_grid</a></span><span class="op">(</span> <span class="op">~</span> <span class="va">methods</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html" class="external-link">scale_fill_manual</a></span><span class="op">(</span>values<span class="op">=</span><span class="fu"><a href="../reference/pb_color.html">pb_color</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">12</span>,<span class="fl">14</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> </span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="supp_case_studies_files/figure-html/HFHSr2_1-1.png" alt="Figure 19: HFHS study: $R^2$ values for each microbial variable before and after batch effect correction." width="100%"><p class="caption">
Figure 19: HFHS study: <span class="math inline">\(R^2\)</span> values
for each microbial variable before and after batch effect correction.
</p>
</div>
<p>We observed from these plots that the data corrected by sPLSDA-batch,
percentile normalisation still included many variables with a large
proportion of batch variance.</p>
<div class="sourceCode" id="cb126"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">##################################</span></span>
<span><span class="va">hfhs.barp.list</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="kw">for</span><span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_len</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">hfhs.r_values.list</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="va">hfhs.barp.list</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> </span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>r2 <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="va">hfhs.r_values.list</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span><span class="op">[</span> ,<span class="st">'trt'</span><span class="op">]</span><span class="op">)</span>,</span>
<span>                      <span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="va">hfhs.r_values.list</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span><span class="op">[</span> ,<span class="st">'batch'</span><span class="op">]</span><span class="op">)</span><span class="op">)</span>, </span>
<span>               Effects <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'Treatment'</span>,<span class="st">'Batch'</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">hfhs.barp.list</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">hfhs.r_values.list</span><span class="op">)</span></span>
<span></span>
<span><span class="va">hfhs.r2.barp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">rbind</a></span><span class="op">(</span><span class="va">hfhs.barp.list</span><span class="op">$</span><span class="va">`Before correction`</span>,</span>
<span>                      <span class="va">hfhs.barp.list</span><span class="op">$</span><span class="va">removeBatchEffect</span>,</span>
<span>                      <span class="va">hfhs.barp.list</span><span class="op">$</span><span class="va">ComBat</span>,</span>
<span>                      <span class="va">hfhs.barp.list</span><span class="op">$</span><span class="va">`PLSDA-batch`</span>,</span>
<span>                      <span class="va">hfhs.barp.list</span><span class="op">$</span><span class="va">`sPLSDA-batch`</span>,</span>
<span>                      <span class="va">hfhs.barp.list</span><span class="op">$</span><span class="va">`Percentile Normalisation`</span>,</span>
<span>                      <span class="va">hfhs.barp.list</span><span class="op">$</span><span class="va">RUVIII</span><span class="op">)</span></span>
<span></span>
<span></span>
<span><span class="va">hfhs.r2.barp</span><span class="op">$</span><span class="va">methods</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'Before correction'</span>, <span class="st">' removeBatchEffect'</span>, </span>
<span>                              <span class="st">'ComBat'</span>,<span class="st">'PLSDA-batch'</span>, <span class="st">'sPLSDA-batch'</span>,</span>
<span>                              <span class="st">'Percentile Normalisation'</span>, <span class="st">'RUVIII'</span><span class="op">)</span>, each <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span>
<span></span>
<span><span class="va">hfhs.r2.barp</span><span class="op">$</span><span class="va">methods</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="va">hfhs.r2.barp</span><span class="op">$</span><span class="va">methods</span>, </span>
<span>                               levels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">hfhs.r2.barp</span><span class="op">$</span><span class="va">methods</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span></span>
<span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">hfhs.r2.barp</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">Effects</span>, y <span class="op">=</span> <span class="va">r2</span>, fill <span class="op">=</span> <span class="va">Effects</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_bar.html" class="external-link">geom_bar</a></span><span class="op">(</span>stat<span class="op">=</span><span class="st">"identity"</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_bw</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>text <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">18</span><span class="op">)</span>,</span>
<span>        axis.title.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>        axis.title.y <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>        axis.text.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">60</span>, hjust <span class="op">=</span> <span class="fl">1</span>, size <span class="op">=</span> <span class="fl">18</span><span class="op">)</span>,</span>
<span>        axis.text.y <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">18</span><span class="op">)</span>,</span>
<span>        panel.grid.minor.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>        panel.grid.major.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>        legend.position <span class="op">=</span> <span class="st">"right"</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_grid.html" class="external-link">facet_grid</a></span><span class="op">(</span> <span class="op">~</span> <span class="va">methods</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html" class="external-link">scale_fill_manual</a></span><span class="op">(</span>values<span class="op">=</span><span class="fu"><a href="../reference/pb_color.html">pb_color</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">12</span>,<span class="fl">14</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="supp_case_studies_files/figure-html/HFHSr2_2-1.png" alt="Figure 20: HFHS study: the sum of $R^2$ values for each microbial variable before and after batch effect correction." width="100%"><p class="caption">
Figure 20: HFHS study: the sum of <span class="math inline">\(R^2\)</span> values for each microbial variable
before and after batch effect correction.
</p>
</div>
<p>When considering the sum of all variables, the remaining batch
variance of corrected data from PLSDA-batch is greater than
removeBatchEffect, ComBat and RUVIII. The preserved treatment variance
of corrected data from PLSDA-batch is the largest.</p>
<p><strong>Alignment scores</strong></p>
<p>We use the <code><a href="../reference/alignment_score.html">alignment_score()</a></code> function from <span style="color: #FF7F00;">PLSDAbatch</span>. We need to specify the
proportion of data variance to explain (<code>var</code>), the number of
nearest neighbours (<code>k</code>) and the number of principal
components to estimate (<code>ncomp</code>). We then use
<code><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot()</a></code> function from <span style="color: #FF7F00;">ggplot2</span> to visualise the results.</p>
<div class="sourceCode" id="cb127"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># HFHS data</span></span>
<span><span class="va">hfhs.var</span> <span class="op">=</span> <span class="fl">0.95</span></span>
<span><span class="va">hfhs.k</span> <span class="op">=</span> <span class="fl">15</span></span>
<span></span>
<span><span class="va">hfhs.scores</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="kw">for</span><span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="fl">1</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">hfhs.corrected.list</span><span class="op">)</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="va">res</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/alignment_score.html">alignment_score</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">hfhs.corrected.list</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span>, </span>
<span>                         batch <span class="op">=</span> <span class="va">hfhs.day.less</span>, </span>
<span>                         var <span class="op">=</span> <span class="va">hfhs.var</span>, </span>
<span>                         k <span class="op">=</span> <span class="va">hfhs.k</span>, </span>
<span>                         ncomp <span class="op">=</span> <span class="fl">130</span><span class="op">)</span></span>
<span>  <span class="va">hfhs.scores</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">hfhs.scores</span>, <span class="va">res</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="va">hfhs.scores.df</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>scores <span class="op">=</span> <span class="va">hfhs.scores</span>, </span>
<span>                             methods <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">hfhs.corrected.list</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">hfhs.scores.df</span><span class="op">$</span><span class="va">methods</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="va">hfhs.scores.df</span><span class="op">$</span><span class="va">methods</span>, </span>
<span>                                 levels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/rev.html" class="external-link">rev</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">hfhs.corrected.list</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span></span>
<span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_bar.html" class="external-link">geom_col</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">hfhs.scores.df</span><span class="op">$</span><span class="va">methods</span>, </span>
<span>                        y <span class="op">=</span> <span class="va">hfhs.scores.df</span><span class="op">$</span><span class="va">scores</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_text.html" class="external-link">geom_text</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">hfhs.scores.df</span><span class="op">$</span><span class="va">methods</span>, </span>
<span>                y <span class="op">=</span> <span class="va">hfhs.scores.df</span><span class="op">$</span><span class="va">scores</span><span class="op">/</span><span class="fl">2</span>, </span>
<span>                label <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Round.html" class="external-link">round</a></span><span class="op">(</span><span class="va">hfhs.scores.df</span><span class="op">$</span><span class="va">scores</span>, <span class="fl">3</span><span class="op">)</span><span class="op">)</span>, </span>
<span>            size <span class="op">=</span> <span class="fl">3</span>, col <span class="op">=</span> <span class="st">'white'</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/coord_flip.html" class="external-link">coord_flip</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_bw</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ylab</a></span><span class="op">(</span><span class="st">'Alignment Scores'</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">xlab</a></span><span class="op">(</span><span class="st">''</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/lims.html" class="external-link">ylim</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0.85</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="supp_case_studies_files/figure-html/HFHSalignment-1.png" alt="Figure 21: Comparison of alignment scores before and after batch effect correction using different methods for the HFHS data." width="60%"><p class="caption">
Figure 21: Comparison of alignment scores before and after batch effect
correction using different methods for the HFHS data.
</p>
</div>
<p>The alignment scores complement the PCA results, especially when
batch effect removal is difficult to assess on PCA sample plots. For
example in the PCA plots for the <span style="color: #808000;">HFHS
data</span>, we observed that the samples across different batches were
better mixed after batch effect correction with PLSDA-batch and RUVIII
than before, while the performance of these two methods was difficult to
compare. Since a higher alignment score indicates that samples are
better mixed, as shown in the above figure, RUVIII gave a superior
performance compared to the other methods.</p>
<p>We recommend the sparse version of PLSDA-batch for a very strong
batch effect, for example, in the <span style="color: #0000FF;">AD
data</span>. However, for a weak batch effect, we do not recommend
removing batch effects in the <span style="color: #808000;">HFHS
data</span>, but if we have to, we recommend using PLSDA-batch. It is
easier to lose treatment variation when the batch variation is very
small. Therefore PLSDA-batch can better ensure the complete preservation
of treatment variation than a sparse version.</p>
<p><strong>Variable selection</strong></p>
<p>We use <code><a href="https://rdrr.io/pkg/mixOmics/man/splsda.html" class="external-link">splsda()</a></code> from <span style="color: #FF7F00;">mixOmics</span> to select the top 50 microbial
variables that, in combination, discriminate the different treatment
groups in the <span style="color: #808000;">HFHS data</span>. We apply
<code><a href="https://rdrr.io/pkg/mixOmics/man/splsda.html" class="external-link">splsda()</a></code> to the different batch effect corrected data from
all methods. Then we use <code>upset()</code> from <span style="color: #FF7F00;">UpSetR</span> package <span class="citation">(<a href="#ref-lex2014upset" role="doc-biblioref">Lex et al.
2014</a>)</span> to visualise the concordance of variables selected.</p>
<p>In the code below, we first need to convert the list of variables
selected from different method-corrected data into a data frame
compatible with <code>upset()</code> using <code>fromList()</code>. We
then assign different colour schemes for each variable selection.</p>
<div class="sourceCode" id="cb128"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hfhs.splsda.select</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="kw">for</span><span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="fl">1</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">hfhs.corrected.list</span><span class="op">)</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="va">splsda.res</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/mixOmics/man/splsda.html" class="external-link">splsda</a></span><span class="op">(</span>X <span class="op">=</span> <span class="va">hfhs.corrected.list</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span>,</span>
<span>                       Y <span class="op">=</span> <span class="va">hfhs.trt.less</span>, </span>
<span>                       ncomp <span class="op">=</span> <span class="fl">3</span>, keepX <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="fl">50</span>,<span class="fl">3</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="va">select.res</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/mixOmics/man/selectVar.html" class="external-link">selectVar</a></span><span class="op">(</span><span class="va">splsda.res</span>, comp <span class="op">=</span> <span class="fl">1</span><span class="op">)</span><span class="op">$</span><span class="va">name</span></span>
<span>  <span class="va">hfhs.splsda.select</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="va">select.res</span></span>
<span><span class="op">}</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">hfhs.splsda.select</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">hfhs.corrected.list</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># can only visualise 5 methods</span></span>
<span><span class="va">hfhs.splsda.select</span> <span class="op">&lt;-</span> <span class="va">hfhs.splsda.select</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">4</span>,<span class="fl">7</span><span class="op">)</span><span class="op">]</span></span>
<span></span>
<span><span class="va">hfhs.splsda.upsetR</span> <span class="op">&lt;-</span> <span class="fu">fromList</span><span class="op">(</span><span class="va">hfhs.splsda.select</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">upset</span><span class="op">(</span><span class="va">hfhs.splsda.upsetR</span>, main.bar.color <span class="op">=</span> <span class="st">'gray36'</span>,</span>
<span>      sets.bar.color <span class="op">=</span> <span class="fu"><a href="../reference/pb_color.html">pb_color</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">25</span><span class="op">:</span><span class="fl">22</span>,<span class="fl">20</span><span class="op">)</span><span class="op">)</span>, matrix.color <span class="op">=</span> <span class="st">'gray36'</span>,</span>
<span>      order.by <span class="op">=</span> <span class="st">'freq'</span>, empty.intersections <span class="op">=</span> <span class="st">'on'</span>,</span>
<span>      queries <span class="op">=</span> </span>
<span>        <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>query <span class="op">=</span> <span class="va">intersects</span>, params <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="st">'Before correction'</span><span class="op">)</span>, </span>
<span>                  color <span class="op">=</span> <span class="fu"><a href="../reference/pb_color.html">pb_color</a></span><span class="op">(</span><span class="fl">20</span><span class="op">)</span>, active <span class="op">=</span> <span class="cn">T</span><span class="op">)</span>,</span>
<span>             <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>query <span class="op">=</span> <span class="va">intersects</span>, params <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="st">'removeBatchEffect'</span><span class="op">)</span>, </span>
<span>                  color <span class="op">=</span> <span class="fu"><a href="../reference/pb_color.html">pb_color</a></span><span class="op">(</span><span class="fl">22</span><span class="op">)</span>, active <span class="op">=</span> <span class="cn">T</span><span class="op">)</span>,</span>
<span>             <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>query <span class="op">=</span> <span class="va">intersects</span>, params <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="st">'ComBat'</span><span class="op">)</span>, </span>
<span>                  color <span class="op">=</span> <span class="fu"><a href="../reference/pb_color.html">pb_color</a></span><span class="op">(</span><span class="fl">23</span><span class="op">)</span>, active <span class="op">=</span> <span class="cn">T</span><span class="op">)</span>,</span>
<span>             <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>query <span class="op">=</span> <span class="va">intersects</span>, params <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="st">'PLSDA-batch'</span><span class="op">)</span>, </span>
<span>                  color <span class="op">=</span> <span class="fu"><a href="../reference/pb_color.html">pb_color</a></span><span class="op">(</span><span class="fl">24</span><span class="op">)</span>, active <span class="op">=</span> <span class="cn">T</span><span class="op">)</span>,</span>
<span>             <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>query <span class="op">=</span> <span class="va">intersects</span>, params <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="st">'RUVIII'</span><span class="op">)</span>, </span>
<span>                  color <span class="op">=</span> <span class="fu"><a href="../reference/pb_color.html">pb_color</a></span><span class="op">(</span><span class="fl">25</span><span class="op">)</span>, active <span class="op">=</span> <span class="cn">T</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="supp_case_studies_files/figure-html/HFHSupsetR-1.png" alt="Figure 22: UpSet plot showing overlap between variables selected from different corrected data for the HFHS study." width="100%"><p class="caption">
Figure 22: UpSet plot showing overlap between variables selected from
different corrected data for the HFHS study.
</p>
</div>
<p>In the above UpSet plot, the left bars indicate the number of
variables selected from each data corrected with different methods. The
right bar plot combined with the scatterplot show different intersection
and their aggregates. We obtained an overlap of 39 out of 50 selected
variables between different corrected and uncorrected data. However, the
data from each method still included unique variables that were not
selected in the other corrected data. As <code>upset()</code> can only
include five datasets at once, we only display the uncorrected data and
four corrected data that have been efficiently corrected for batch
effects from our previous assessments.</p>
<div class="sourceCode" id="cb129"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hfhs.splsda.select.overlap</span> <span class="op">&lt;-</span> <span class="fu">venn</span><span class="op">(</span><span class="va">hfhs.splsda.select</span>, show.plot <span class="op">=</span> <span class="cn">F</span><span class="op">)</span></span>
<span><span class="va">hfhs.inters.splsda</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/attr.html" class="external-link">attr</a></span><span class="op">(</span><span class="va">hfhs.splsda.select.overlap</span>, <span class="st">'intersections'</span><span class="op">)</span></span>
<span><span class="va">hfhs.inters.splsda.taxa</span> <span class="op">&lt;-</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">hfhs.inters.splsda</span>, </span>
<span>         FUN <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">{</span><span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="va">HFHS_data</span><span class="op">$</span><span class="va">FullData</span><span class="op">$</span><span class="va">taxa</span><span class="op">[</span><span class="va">x</span>, <span class="op">]</span><span class="op">)</span><span class="op">}</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/capture.output.html" class="external-link">capture.output</a></span><span class="op">(</span><span class="va">hfhs.inters.splsda.taxa</span>, </span>
<span>               file <span class="op">=</span> <span class="st">"GeneratedData/HFHSselected_50_splsda.txt"</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
</div>
<div class="section level2">
<h2 id="hd-data">HD data<a class="anchor" aria-label="anchor" href="#hd-data"></a>
</h2>
<div class="section level3">
<h3 id="data-pre-processing-2">Data pre-processing<a class="anchor" aria-label="anchor" href="#data-pre-processing-2"></a>
</h3>
<div class="section level4">
<h4 id="prefiltering-1">Prefiltering<a class="anchor" aria-label="anchor" href="#prefiltering-1"></a>
</h4>
<p>We load the <span style="color: #8601AF;">HD data</span> stored
internally with function <code><a href="https://rdrr.io/r/utils/data.html" class="external-link">data()</a></code>.</p>
<div class="sourceCode" id="cb130"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">'HD_data'</span><span class="op">)</span> </span>
<span><span class="va">hd.count</span> <span class="op">&lt;-</span> <span class="va">HD_data</span><span class="op">$</span><span class="va">FullData</span><span class="op">$</span><span class="va">X.count</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">hd.count</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1]  30 368</span></span></code></pre>
<div class="sourceCode" id="cb132"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hd.filter.res</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/PreFL.html">PreFL</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">hd.count</span><span class="op">)</span></span>
<span><span class="va">hd.filter</span> <span class="op">&lt;-</span> <span class="va">hd.filter.res</span><span class="op">$</span><span class="va">data.filter</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">hd.filter</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1]  30 269</span></span></code></pre>
<div class="sourceCode" id="cb134"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># zero proportion before filtering</span></span>
<span><span class="va">hd.filter.res</span><span class="op">$</span><span class="va">zero.prob</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 0.4509058</span></span></code></pre>
<div class="sourceCode" id="cb136"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># zero proportion after filtering</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="va">hd.filter</span> <span class="op">==</span> <span class="fl">0</span><span class="op">)</span><span class="op">/</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">hd.filter</span><span class="op">)</span> <span class="op">*</span> <span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">ncol</a></span><span class="op">(</span><span class="va">hd.filter</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 0.3343247</span></span></code></pre>
<p>The raw <span style="color: #8601AF;">HD data</span> include 368 OTUs
and 30 samples. We use the function <code><a href="../reference/PreFL.html">PreFL()</a></code> from our <span style="color: #FF7F00;">PLSDAbatch</span> R package to filter the data.
After filtering, the <span style="color: #8601AF;">HD data</span> were
reduced to 269 OTUs and 30 samples. The proportion of zeroes was reduced
from 45% to 33%.</p>
</div>
<div class="section level4">
<h4 id="transformation-2">Transformation<a class="anchor" aria-label="anchor" href="#transformation-2"></a>
</h4>
<p>Prior to CLR transformation, we add 1 as the offset for the <span style="color: #8601AF;">HD data</span> - that are raw count data. We use
<code><a href="https://rdrr.io/pkg/mixOmics/man/logratio-transformations.html" class="external-link">logratio.transfo()</a></code> function in <span style="color: #FF7F00;">mixOmics</span> package to CLR transform the
data.</p>
<div class="sourceCode" id="cb138"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hd.clr</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/mixOmics/man/logratio-transformations.html" class="external-link">logratio.transfo</a></span><span class="op">(</span>X <span class="op">=</span> <span class="va">hd.filter</span>, logratio <span class="op">=</span> <span class="st">'CLR'</span>, offset <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/class.html" class="external-link">class</a></span><span class="op">(</span><span class="va">hd.clr</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="st">'matrix'</span></span></code></pre></div>
</div>
</div>
<div class="section level3">
<h3 id="batch-effect-detection-2">Batch effect detection<a class="anchor" aria-label="anchor" href="#batch-effect-detection-2"></a>
</h3>
<div class="section level4">
<h4 id="pca-2">PCA<a class="anchor" aria-label="anchor" href="#pca-2"></a>
</h4>
<p>We apply <code><a href="https://rdrr.io/pkg/mixOmics/man/pca.html" class="external-link">pca()</a></code> function from <span style="color: #FF7F00;">mixOmics</span> package to the <span style="color: #8601AF;">HD data</span> and use
<code><a href="../reference/Scatter_Density.html">Scatter_Density()</a></code> function from <span style="color: #FF7F00;">PLSDAbatch</span> to represent the PCA sample
plot with densities.</p>
<div class="sourceCode" id="cb139"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># HD data</span></span>
<span><span class="va">hd.pca.before</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/mixOmics/man/pca.html" class="external-link">pca</a></span><span class="op">(</span><span class="va">hd.clr</span>, ncomp <span class="op">=</span> <span class="fl">3</span>, scale <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span><span class="va">hd.metadata</span> <span class="op">&lt;-</span> <span class="va">HD_data</span><span class="op">$</span><span class="va">FullData</span><span class="op">$</span><span class="va">metadata</span></span>
<span></span>
<span><span class="va">hd.batch</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">as.factor</a></span><span class="op">(</span><span class="va">hd.metadata</span><span class="op">$</span><span class="va">Cage</span><span class="op">)</span></span>
<span><span class="va">hd.trt</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">as.factor</a></span><span class="op">(</span><span class="va">hd.metadata</span><span class="op">$</span><span class="va">Genotype</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">hd.batch</span><span class="op">)</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">hd.trt</span><span class="op">)</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">hd.metadata</span><span class="op">)</span> </span>
<span></span>
<span><span class="fu"><a href="../reference/Scatter_Density.html">Scatter_Density</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">hd.pca.before</span>, </span>
<span>                batch <span class="op">=</span> <span class="va">hd.batch</span>, </span>
<span>                trt <span class="op">=</span> <span class="va">hd.trt</span>, </span>
<span>                title <span class="op">=</span> <span class="st">'HD data'</span>, </span>
<span>                trt.legend.title <span class="op">=</span> <span class="st">'Genotype'</span>, </span>
<span>                batch.legend.title <span class="op">=</span> <span class="st">'Cage'</span>, legend.cex <span class="op">=</span> <span class="fl">0.6</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="supp_case_studies_files/figure-html/HDpcaBefore-1.png" alt="Figure 23: The PCA sample plot with densities in the HD data." width="60%"><p class="caption">
Figure 23: The PCA sample plot with densities in the HD data.
</p>
</div>
<p>In the above figure, we observed a clear difference between genotypes
but vague separation between cages on the PCA plot.</p>
</div>
<div class="section level4">
<h4 id="heatmap-2">Heatmap<a class="anchor" aria-label="anchor" href="#heatmap-2"></a>
</h4>
<p>We produce a heatmap using <span style="color: #FF7F00;">pheatmap</span> package. The data first need to
be scaled on both OTUs and samples.</p>
<div class="sourceCode" id="cb140"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># scale the clr data on both OTUs and samples</span></span>
<span><span class="va">hd.clr.s</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/scale.html" class="external-link">scale</a></span><span class="op">(</span><span class="va">hd.clr</span>, center <span class="op">=</span> <span class="cn">T</span>, scale <span class="op">=</span> <span class="cn">T</span><span class="op">)</span></span>
<span><span class="va">hd.clr.ss</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/scale.html" class="external-link">scale</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/t.html" class="external-link">t</a></span><span class="op">(</span><span class="va">hd.clr.s</span><span class="op">)</span>, center <span class="op">=</span> <span class="cn">T</span>, scale <span class="op">=</span> <span class="cn">T</span><span class="op">)</span></span>
<span></span>
<span><span class="va">hd.anno_col</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>Cage <span class="op">=</span> <span class="va">hd.batch</span>, Genotype <span class="op">=</span> <span class="va">hd.trt</span><span class="op">)</span></span>
<span><span class="va">hd.anno_colors</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>Cage <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/mixOmics/man/colors.html" class="external-link">color.mixo</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span><span class="op">)</span>, </span>
<span>                       Genotype <span class="op">=</span> <span class="fu"><a href="../reference/pb_color.html">pb_color</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">2</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">hd.anno_colors</span><span class="op">$</span><span class="va">Cage</span><span class="op">)</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/levels.html" class="external-link">levels</a></span><span class="op">(</span><span class="va">hd.batch</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">hd.anno_colors</span><span class="op">$</span><span class="va">Genotype</span><span class="op">)</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/levels.html" class="external-link">levels</a></span><span class="op">(</span><span class="va">hd.trt</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">pheatmap</span><span class="op">(</span><span class="va">hd.clr.ss</span>, </span>
<span>         cluster_rows <span class="op">=</span> <span class="cn">F</span>, </span>
<span>         fontsize_row <span class="op">=</span> <span class="fl">4</span>, </span>
<span>         fontsize_col <span class="op">=</span> <span class="fl">6</span>,</span>
<span>         fontsize <span class="op">=</span> <span class="fl">8</span>,</span>
<span>         clustering_distance_rows <span class="op">=</span> <span class="st">'euclidean'</span>,</span>
<span>         clustering_method <span class="op">=</span> <span class="st">'ward.D'</span>,</span>
<span>         treeheight_row <span class="op">=</span> <span class="fl">30</span>,</span>
<span>         annotation_col <span class="op">=</span> <span class="va">hd.anno_col</span>,</span>
<span>         annotation_colors <span class="op">=</span> <span class="va">hd.anno_colors</span>,</span>
<span>         border_color <span class="op">=</span> <span class="st">'NA'</span>,</span>
<span>         main <span class="op">=</span> <span class="st">'HD data - Scaled'</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="supp_case_studies_files/figure-html/HDheatmap-1.png" alt="Figure 24: Hierarchical clustering for samples in the HD data." width="90%"><p class="caption">
Figure 24: Hierarchical clustering for samples in the HD data.
</p>
</div>
<p>In the heatmap, samples in the <span style="color: #8601AF;">HD
data</span> were mostly grouped by genotypes instead of cages,
indicating a weak cage effect.</p>
</div>
<div class="section level4">
<h4 id="prda-2">pRDA<a class="anchor" aria-label="anchor" href="#prda-2"></a>
</h4>
<p>We apply pRDA with <code><a href="https://rdrr.io/pkg/vegan/man/varpart.html" class="external-link">varpart()</a></code> function from <span style="color: #FF7F00;">vegan</span> R package.</p>
<div class="sourceCode" id="cb141"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hd.factors.df</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>trt <span class="op">=</span> <span class="va">hd.trt</span>, batch <span class="op">=</span> <span class="va">hd.batch</span><span class="op">)</span></span>
<span><span class="va">hd.rda.before</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/vegan/man/varpart.html" class="external-link">varpart</a></span><span class="op">(</span><span class="va">hd.clr</span>, <span class="op">~</span> <span class="va">trt</span>, <span class="op">~</span> <span class="va">batch</span>, </span>
<span>                         data <span class="op">=</span> <span class="va">hd.factors.df</span>, scale <span class="op">=</span> <span class="cn">T</span><span class="op">)</span></span>
<span><span class="va">hd.rda.before</span><span class="op">$</span><span class="va">part</span><span class="op">$</span><span class="va">indfract</span></span></code></pre></div>
<pre><code><span><span class="co">##                 Df R.squared Adj.R.squared Testable</span></span>
<span><span class="co">## [a] = X1|X2      0        NA -2.220446e-16    FALSE</span></span>
<span><span class="co">## [b] = X2|X1      8        NA  1.608205e-01     TRUE</span></span>
<span><span class="co">## [c]              0        NA  9.730583e-02    FALSE</span></span>
<span><span class="co">## [d] = Residuals NA        NA  7.418737e-01    FALSE</span></span></code></pre>
<p>In the result, <code>X1</code> and <code>X2</code> represent the
first and second covariates fitted in the model. <code>[a]</code>,
<code>[b]</code> represent the independent proportion of variance
explained by <code>X1</code> and <code>X2</code> respectively, and
<code>[c]</code> represents the intersection variance shared between
<code>X1</code> and <code>X2</code>. Collinearity was detected in the
<span style="color: #8601AF;">HD data</span> between treatment and batch
as indicated by lines <code>[a]</code> and <code>[c]</code> that all
treatment variance (<code>[a]</code>: Adj.R.squared = 0,
<code>[c]</code>: Adj.R.squared = 0.0973) was explained as intersection
variance. The batch x treatment design in the <span style="color: #8601AF;">HD data</span> is nested, in such a design the
intersection variance is usually considerable. As the intersection is
shared between batch and treatment effects, the batch variance in the
<span style="color: #8601AF;">HD data</span> is larger than the
treatment variance.</p>
</div>
</div>
<div class="section level3">
<h3 id="managing-batch-effects-2">Managing batch effects<a class="anchor" aria-label="anchor" href="#managing-batch-effects-2"></a>
</h3>
<p>Because of the nested batch x treatment design in the <span style="color: #8601AF;">HD data</span>, the only way to account for the
batch effect (i.e., the cage effect) is to apply linear mixed model and
include the batch effect as a random effect.</p>
<div class="section level4">
<h4 id="accounting-for-batch-effects-2">Accounting for batch effects<a class="anchor" aria-label="anchor" href="#accounting-for-batch-effects-2"></a>
</h4>
<p><strong>Linear regression</strong></p>
<p>Linear regression is conducted with <code><a href="../reference/linear_regres.html">linear_regres()</a></code>
function in <span style="color: #FF7F00;">PLSDAbatch</span>. We
integrated the <span style="color: #FF7F00;">performance</span> package
that assesses performance of regression models into our function
<code><a href="../reference/linear_regres.html">linear_regres()</a></code>. Therefore, we can apply
<code>check_model()</code> from <span style="color: #FF7F00;">performance</span> to the outputs from
<code><a href="../reference/linear_regres.html">linear_regres()</a></code> to diagnose the validity of the model
fitted with treatment and batch effects for each variable <span class="citation">(<a href="#ref-daniel2020performance" role="doc-biblioref">LÃŒdecke et al. 2020</a>)</span>. We can extract
performance measurements such as adjusted R2, RMSE, RSE, AIC and BIC for
the models fitted with and without batch effects, which are also the
outputs of <code><a href="../reference/linear_regres.html">linear_regres()</a></code>.</p>
<p>We apply a <code>linear mixed model</code> and fit the batch effect
as a random effect because the design of the <span style="color: #8601AF;">HD data</span> is nested. In such a design, the
number of batches should be larger than treatment groups; otherwise, the
linear mixed model cannot address the batch effect. Note: we assume that
there is no interaction between batch and treatment effects on any
microbial variable.</p>
<div class="sourceCode" id="cb143"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># HD data</span></span>
<span><span class="va">hd.clr</span> <span class="op">&lt;-</span> <span class="va">hd.clr</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">hd.clr</span><span class="op">)</span>, <span class="fl">1</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">ncol</a></span><span class="op">(</span><span class="va">hd.clr</span><span class="op">)</span><span class="op">]</span></span>
<span></span>
<span><span class="va">hd.lmm</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/linear_regres.html">linear_regres</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">hd.clr</span>, trt <span class="op">=</span> <span class="va">hd.trt</span>, </span>
<span>                        batch.random <span class="op">=</span> <span class="va">hd.batch</span>, </span>
<span>                        type <span class="op">=</span> <span class="st">'linear mixed model'</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb144"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hd.p</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">hd.lmm</span><span class="op">$</span><span class="va">lmm.table</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">{</span><span class="va">x</span><span class="op">$</span><span class="va">coefficients</span><span class="op">[</span><span class="fl">2</span>,<span class="fl">5</span><span class="op">]</span><span class="op">}</span><span class="op">)</span></span>
<span><span class="va">hd.p.adj</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/p.adjust.html" class="external-link">p.adjust</a></span><span class="op">(</span>p <span class="op">=</span> <span class="va">hd.p</span>, method <span class="op">=</span> <span class="st">'fdr'</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">check_model</span><span class="op">(</span><span class="va">hd.lmm</span><span class="op">$</span><span class="va">model</span><span class="op">$</span><span class="va">OTU1</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="supp_case_studies_files/figure-html/HDlmmS-1.png" alt='Figure 25: Diagnostic plots for the model fitted with the cage effect of "OTU1" in the HD data.' width="100%"><p class="caption">
Figure 25: Diagnostic plots for the model fitted with the cage effect of
“OTU1” in the HD data.
</p>
</div>
<p>According to the diagnostic plots of “OTU1” as shown in the above
figure panel, the simulated data under the fitted model were not very
close to the real data (shown as green line), indicating an imperfect
fitness (top panel). The linearity (or homoscedasticity) and homogeneity
of variance were not satisfied. Some samples could be classified as
outliers with a Cook’s distance larger than or equal to 0.5, for
example, “12”, “21”, “20”, “11” and “6” (middle panel). The distribution
of residuals was not normal, while the one of random effects was very
close to normal (bottom panel).</p>
<p>We then look at the AIC of models fitted with or without batch
effects.</p>
<div class="sourceCode" id="cb145"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">hd.lmm</span><span class="op">$</span><span class="va">AIC</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##       trt.only trt.batch</span></span>
<span><span class="co">## OTU1  60.00637  65.46410</span></span>
<span><span class="co">## OTU2 130.14167 130.96344</span></span>
<span><span class="co">## OTU3  71.76985  75.58915</span></span>
<span><span class="co">## OTU4 134.59992 134.04812</span></span>
<span><span class="co">## OTU5 150.25059 148.30644</span></span>
<span><span class="co">## OTU6  99.92864  99.05391</span></span></code></pre>
<p>For some OTUs, the model fitted without batch effects is better with
a lower AIC. For example, “OTU1” and “OTU3”. “OTU5” is more appropriate
within a model with batch effects. The others are similar within either
model.</p>
<p>Since we apply a <code>"linear mixed model"</code> to the <span style="color: #8601AF;">HD data</span>, this type of model can only
output conditional <span class="math inline">\(R^2\)</span> that
includes the variance of both fixed and random effects (treatment, fixed
and random batch effects) and marginal <span class="math inline">\(R^2\)</span> that includes only the variance of
fixed effects (treatment and fixed batch effects) <span class="citation">(<a href="#ref-nakagawa2013general" role="doc-biblioref">Nakagawa and Schielzeth 2013</a>)</span>. The
marginal <span class="math inline">\(R^2\)</span>s of the models with
and without the batch effect should be the same theoretically, as
marginal <span class="math inline">\(R^2\)</span> only includes fixed
effects and the treatment effect is the only fixed effect here. In the
actual calculation, the results of these two models were not the same
due to different ways of model fitting.</p>
<div class="sourceCode" id="cb147"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">hd.lmm</span><span class="op">$</span><span class="va">cond.R2</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##        trt.only trt.batch</span></span>
<span><span class="co">## OTU1 0.47917775 0.5161317</span></span>
<span><span class="co">## OTU2 0.39223824 0.4729848</span></span>
<span><span class="co">## OTU3 0.02091376 0.2636432</span></span>
<span><span class="co">## OTU4 0.37200835 0.5672392</span></span>
<span><span class="co">## OTU5 0.07858373 0.2938212</span></span>
<span><span class="co">## OTU6 0.41388250 0.6143134</span></span></code></pre>
<div class="sourceCode" id="cb149"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">hd.lmm</span><span class="op">$</span><span class="va">marg.R2</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##        trt.only  trt.batch</span></span>
<span><span class="co">## OTU1 0.47917775 0.46735273</span></span>
<span><span class="co">## OTU2 0.39223824 0.38459672</span></span>
<span><span class="co">## OTU3 0.02091376 0.01489608</span></span>
<span><span class="co">## OTU4 0.37200835 0.36122754</span></span>
<span><span class="co">## OTU5 0.07858373 0.06052938</span></span>
<span><span class="co">## OTU6 0.41388250 0.40749149</span></span></code></pre>
<p>We observed that some variables resulted in singular fits
(<code>error message: Can't compute random effect variances. Some variance components equal zero. Your model may suffer from singulariy.</code>),
which are expected to happen in linear mixed models when covariates are
nested. We recommend noting the variables for which this error occurs,
as it may lead to unreliable results.</p>
</div>
</div>
</div>
<div class="section level2">
<h2 class="unnumbered" id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-daniel2020performance" class="csl-entry">
LÃŒdecke, Daniel, Dominique Makowski, Philip Waggoner, and Indrajeet
Patil. 2020. <span>“Performance: Assessment of Regression Models
Performance.”</span> <em>CRAN</em>. <a href="https://doi.org/10.5281/zenodo.3952174" class="external-link">https://doi.org/10.5281/zenodo.3952174</a>.
</div>
<div id="ref-leek2012sva" class="csl-entry">
Leek, Jeffrey T, W Evan Johnson, Hilary S Parker, Andrew E Jaffe, and
John D Storey. 2012. <span>“The Sva Package for Removing Batch Effects
and Other Unwanted Variation in High-Throughput Experiments.”</span>
<em>Bioinformatics</em> 28 (6): 882–83.
</div>
<div id="ref-lex2014upset" class="csl-entry">
Lex, Alexander, Nils Gehlenborg, Hendrik Strobelt, Romain Vuillemot, and
Hanspeter Pfister. 2014. <span>“UpSet: Visualization of Intersecting
Sets.”</span> <em>IEEE Transactions on Visualization and Computer
Graphics</em> 20 (12): 1983–92.
</div>
<div id="ref-moskovicz2020skin" class="csl-entry">
Moskovicz, Veronica, Rina Ben-El, Guy Horev, and Boaz Mizrahi. 2020.
<span>“Skin Microbiota Dynamics Following b. Subtilis Formulation
Challenge.”</span>
</div>
<div id="ref-nakagawa2013general" class="csl-entry">
Nakagawa, Shinichi, and Holger Schielzeth. 2013. <span>“A General and
Simple Method for Obtaining R2 from Generalized Linear Mixed-Effects
Models.”</span> <em>Methods in Ecology and Evolution</em> 4 (2): 133–42.
</div>
<div id="ref-rohart2017mixomics" class="csl-entry">
Rohart, Florian, Benoı̂t Gautier, Amrit Singh, and Kim-Anh Lê Cao. 2017.
<span>“mixOmics: An r Package for ‘Omics Feature Selection and Multiple
Data Integration.”</span> <em>PLoS Computational Biology</em> 13 (11):
e1005752.
</div>
<div id="ref-wang2020managing" class="csl-entry">
Wang, Yiwen, and Kim-Anh LêCao. 2020. <span>“Managing Batch Effects in
Microbiome Data.”</span> <em>Briefings in Bioinformatics</em> 21 (6):
1954–70.
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Yiwen Wang.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
